<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>A content in each mapped read</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">“A” content in each mapped read</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>6/11/2018</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-06-11</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/3f145aef444d0beac696febf329fedbc3fb19c79" target="_blank">3f145ae</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  data/18486.genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/gene_cov/
    Untracked:  data/leafcutter/
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/ssFC200.cov.bed
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   code/Snakefile

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/3f145aef444d0beac696febf329fedbc3fb19c79/analysis/a.content.Rmd" target="_blank">3f145ae</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-06-11
</td>
<td style="text-align:left;">
signature of mult nucleotide analysus
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/4ef7d85554ed0db97548af29c6d994675af4837e/docs/a.content.html" target="_blank">4ef7d85</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-06-11
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/03c7b1448ea0c93a1d7a0b2d589e8ef964bf7bba/analysis/a.content.Rmd" target="_blank">03c7b14</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-06-11
</td>
<td style="text-align:left;">
start A analysis
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<p>The goal of this analysis is to start to understand the sequence composition of the three prime seq reads. This may help me detect misspriming at AAAAA rich regions rather than true site usage. The genomic sequence does not carry the polyadenylation signal, this means reads mapping to a genomic AAAAAA region may be false positives. Gruber et al. removed reads that consisted of more than 80% AAAA.</p>
<div id="percent-nucleotide-in-bins" class="section level2">
<h2>Percent nucleotide in bins</h2>
<p>One method is to measure the number of AAAAAs in my bins with bedtools nuc. I willl need a fasta file and a bed file with the bin.</p>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>Loading required package: rmarkdown</code></pre>
<pre><code>This is workflowr version 1.0.1
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(ggplot2)
library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 3.4.4</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(tidyr)
library(reshape2)</code></pre>
<pre><code>Warning: package &#39;reshape2&#39; was built under R version 3.4.3</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=nuc.bin
#SBATCH --time=8:00:00
#SBATCH --output=nuc.bin.out
#SBATCH --error=nuc.bin.err
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --mail-type=END

module load Anaconda3  

source activate three-prime-env


bedtools nuc -s -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed /project2/gilad/briana/genome_anotation_data/an.int.genome_200_strandspec.bed &gt; /project2/gilad/briana/threeprimeseq/data/bin200.nuccov.bed  </code></pre>
<p>I can now pull this file into R.</p>
<pre class="r"><code>bin_nuccov=read.table(&quot;../data/bin200.nuccov.bed&quot;)
names(bin_nuccov)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;bin&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;gene&quot;, &quot;pct_at&quot;, &quot;pct_gc&quot;, &quot;numA&quot;, &quot;numC&quot;, &quot;numG&quot;, &quot;numT&quot;, &quot;numN&quot;, &quot;numOther&quot;, &quot;seqlen&quot;)

perc_A_bin=bin_nuccov %&gt;% select(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;,&quot;bin&quot;, &quot;strand&quot;, &quot;gene&quot;, &quot;numA&quot;) %&gt;% mutate(percA=numA/200)</code></pre>
<pre><code>Warning: package &#39;bindrcpp&#39; was built under R version 3.4.4</code></pre>
<pre class="r"><code>ggplot(perc_A_bin, aes(percA)) + geom_histogram(bins=30) + labs(title=&quot;Percent of As in the bins&quot;)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-4-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/4ef7d85554ed0db97548af29c6d994675af4837e/docs/figure/a.content.Rmd/unnamed-chunk-4-1.png" target="_blank">4ef7d85</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-06-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>I will apply the same filter as I did in the cov.200bp.wind file. I will keep bins with greater than 0 reads in half of the libraries.</p>
<pre class="r"><code>cov_all=read.table(&quot;../data/ssFC200.cov.bed&quot;, header = T, stringsAsFactors = FALSE)
#remember name switch!
names=c(&quot;Geneid&quot;,&quot;Chr&quot;, &quot;Start&quot;, &quot;End&quot;, &quot;Strand&quot;, &quot;Length&quot;, &quot;N_18486&quot;,&quot;T_18486&quot;,&quot;N_18497&quot;,&quot;T_18497&quot;,&quot;N_18500&quot;,&quot;T_18500&quot;,&quot;N_18505&quot;,&#39;T_18505&#39;,&quot;N_18508&quot;,&quot;T_18508&quot;,&quot;N_18853&quot;,&quot;T_18853&quot;,&quot;N_18870&quot;,&quot;T_18870&quot;,&quot;N_19128&quot;,&quot;T_19128&quot;,&quot;N_19141&quot;,&quot;T_19141&quot;,&quot;N_19193&quot;,&quot;T_19193&quot;,&quot;N_19209&quot;,&quot;T_19209&quot;,&quot;N_19223&quot;,&quot;N_19225&quot;,&quot;T_19225&quot;,&quot;T_19223&quot;,&quot;N_19238&quot;,&quot;T_19238&quot;,&quot;N_19239&quot;,&quot;T_19239&quot;,&quot;N_19257&quot;,&quot;T_19257&quot;)

colnames(cov_all)= names

cov_nums_only=cov_all[,7:38]

keep.exprs=rowSums(cov_nums_only&gt;0) &gt;= 16

cov_all_filt=cov_all[keep.exprs,] 

cov_all_filt_bins= cov_all_filt %&gt;% separate(col=Geneid, into=c(&quot;bin&quot;,&quot;gene&quot;), sep=&quot;.E&quot;) %&gt;% select(bin)

cov_all_filt_bins$bin=as.integer(cov_all_filt_bins$bin)</code></pre>
<p>I will intersect the percA file with the bins in the filtered file.</p>
<pre class="r"><code>perc_A_bin_filt= perc_A_bin %&gt;% semi_join(cov_all_filt_bins, by=&quot;bin&quot;)</code></pre>
<p>I can no plot the distribution of percA in this.</p>
<pre class="r"><code>ggplot(perc_A_bin_filt, aes(percA)) + geom_histogram(bins = 30) + labs(title=&quot;Percent of As in the bins after filtering&quot;)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-7-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/4ef7d85554ed0db97548af29c6d994675af4837e/docs/figure/a.content.Rmd/unnamed-chunk-7-1.png" target="_blank">4ef7d85</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-06-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>I will compare this distribution to those for other nucleotides. (C)</p>
<pre class="r"><code>perc_C_bin=bin_nuccov %&gt;% select(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;,&quot;bin&quot;, &quot;strand&quot;, &quot;gene&quot;, &quot;numC&quot;) %&gt;% mutate(percC=numC/200)

ggplot(perc_C_bin, aes(percC)) + geom_histogram(bins=30) + labs(title=&quot;Percent of Cs in the bins&quot;)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>perc_C_bin_filt= perc_C_bin %&gt;% semi_join(cov_all_filt_bins, by=&quot;bin&quot;)
ggplot(perc_C_bin_filt, aes(percC)) + geom_histogram(bins = 30) + labs(title=&quot;Percent of Cs in the bins after filtering&quot;)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /> ’ For G</p>
<pre class="r"><code>perc_G_bin=bin_nuccov %&gt;% select(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;,&quot;bin&quot;, &quot;strand&quot;, &quot;gene&quot;, &quot;numG&quot;) %&gt;% mutate(percG=numG/200)

ggplot(perc_G_bin, aes(percG)) + geom_histogram(bins=30) + labs(title=&quot;Percent of Gs in the bins&quot;)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>perc_G_bin_filt= perc_G_bin %&gt;% semi_join(cov_all_filt_bins, by=&quot;bin&quot;)
ggplot(perc_G_bin_filt, aes(percG)) + geom_histogram(bins = 30) + labs(title=&quot;Percent of Gs in the bins after filtering&quot;)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>for T</p>
<pre class="r"><code>perc_T_bin=bin_nuccov %&gt;% select(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;,&quot;bin&quot;, &quot;strand&quot;, &quot;gene&quot;, &quot;numT&quot;) %&gt;% mutate(percT=numT/200)

ggplot(perc_T_bin, aes(percT)) + geom_histogram(bins=30) + labs(title=&quot;Percent of Ts in the bins&quot;)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>perc_T_bin_filt= perc_T_bin %&gt;% semi_join(cov_all_filt_bins, by=&quot;bin&quot;)
ggplot(perc_T_bin_filt, aes(percT)) + geom_histogram(bins = 30) + labs(title=&quot;Percent of Ts in the bins after filtering&quot;)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now I will join all of the percent usage of each nucleotide in the filtered bins so I can plot them on one plot.</p>
<pre class="r"><code>percNuc= perc_A_bin_filt %&gt;% left_join(perc_T_bin_filt, by=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;bin&quot;, &quot;strand&quot;, &quot;gene&quot;)) %&gt;% left_join(perc_G_bin_filt, by=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;bin&quot;, &quot;strand&quot;, &quot;gene&quot;)) %&gt;% left_join(perc_C_bin_filt, by=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;bin&quot;, &quot;strand&quot;, &quot;gene&quot;)) %&gt;% select(&quot;bin&quot;, &quot;percA&quot;, &quot;percT&quot;, &quot;percG&quot;, &quot;percC&quot;)

percNuc_melt=melt(percNuc, id.vars = &quot;bin&quot;)

ggplot(percNuc_melt, aes(value)) + geom_histogram(bins = 30) + facet_wrap(~variable) + labs(title=&quot;Percent each nucleotide in filtered bins&quot;)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="strech-of-nucleotides" class="section level2">
<h2>Strech of Nucleotides</h2>
<p>Next check is if the bins have 5 As in a row. I can do this using bedtools nuc as well.<br />
###Five A’s</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=nucA
#SBATCH --time=8:00:00
#SBATCH --output=nucA.out
#SBATCH --error=nucA.err
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --mail-type=END

module load Anaconda3  

source activate three-prime-env


bedtools nuc -s -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed /project2/gilad/briana/genome_anotation_data/an.int.genome_200_strandspec.bed -pattern &quot;AAAAA&quot; &gt; /project2/gilad/briana/threeprimeseq/data/bin200.5.A.nuccov.bed  </code></pre>
<pre class="r"><code>bin_Anuccov=read.table(&quot;../data/bin200.Anuccov.bed&quot;)
names(bin_Anuccov)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;bin&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;gene&quot;, &quot;pct_at&quot;, &quot;pct_gc&quot;, &quot;numA&quot;, &quot;numC&quot;, &quot;numG&quot;, &quot;numT&quot;, &quot;numN&quot;, &quot;numOther&quot;, &quot;seqlen&quot;, &quot;fiveA&quot;)

hist(bin_Anuccov$fiveA)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>I will filter this the same way I filtered the other file.</p>
<pre class="r"><code> bin_Anuccov_filt = bin_Anuccov %&gt;% semi_join(cov_all_filt_bins, by=&quot;bin&quot;) %&gt;% select( bin,  gene,  fiveA)

hist(bin_Anuccov_filt$fiveA)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(bin_Anuccov_filt$fiveA)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  0.0000  0.4404  0.0000 13.0000 </code></pre>
<p>Count the number of bins with each value for number of 5 AAAAA</p>
<pre class="r"><code>countA_regions= bin_Anuccov_filt %&gt;% group_by(fiveA) %&gt;% count(fiveA)</code></pre>
<div id="five-ts" class="section level3">
<h3>Five T’s</h3>
<p>I will compare this to regions with 5 T’s<br />
(This isnt the best comparison because the probability of each of these stretches genome wide may be different)</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=nucT
#SBATCH --time=8:00:00
#SBATCH --output=nucT.out
#SBATCH --error=nucT.err
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --mail-type=END

module load Anaconda3  

source activate three-prime-env


bedtools nuc -s -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed /project2/gilad/briana/genome_anotation_data/an.int.genome_200_strandspec.bed -pattern &quot;TTTTT&quot; &gt; /project2/gilad/briana/threeprimeseq/data/bin200.5.T.nuccov.bed  </code></pre>
<pre class="r"><code>bin_Tnuccov=read.table(&quot;../data/bin200.5.T.nuccov.bed&quot;)
names(bin_Tnuccov)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;bin&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;gene&quot;, &quot;pct_at&quot;, &quot;pct_gc&quot;, &quot;numA&quot;, &quot;numC&quot;, &quot;numG&quot;, &quot;numT&quot;, &quot;numN&quot;, &quot;numOther&quot;, &quot;seqlen&quot;, &quot;fiveT&quot;)

summary(bin_Tnuccov$fiveT)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  0.0000  0.4191  0.0000 14.0000 </code></pre>
<pre class="r"><code>bin_Tnuccov_filt = bin_Tnuccov %&gt;% semi_join(cov_all_filt_bins, by=&quot;bin&quot;) %&gt;% select( bin,  gene,  fiveT)

hist(bin_Tnuccov_filt$fiveT)</code></pre>
<p><img src="figure/a.content.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(bin_Tnuccov_filt$fiveT)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  0.0000  0.5163  1.0000  9.0000 </code></pre>
<pre class="r"><code>countT_regions= bin_Tnuccov_filt %&gt;% group_by(fiveT) %&gt;% count(fiveT)</code></pre>
<p>The numbers are not super different.</p>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2.2  reshape2_1.4.3  tidyr_0.7.2     dplyr_0.7.5    
[5] ggplot2_2.2.1   workflowr_1.0.1 rmarkdown_1.8.5

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.17      compiler_3.4.2    pillar_1.1.0     
 [4] git2r_0.21.0      plyr_1.8.4        bindr_0.1.1      
 [7] R.methodsS3_1.7.1 R.utils_2.6.0     tools_3.4.2      
[10] digest_0.6.15     evaluate_0.10.1   tibble_1.4.2     
[13] gtable_0.2.0      pkgconfig_2.0.1   rlang_0.2.1      
[16] yaml_2.1.19       stringr_1.3.1     knitr_1.18       
[19] rprojroot_1.3-2   grid_3.4.2        tidyselect_0.2.4 
[22] glue_1.2.0        R6_2.2.2          purrr_0.2.5      
[25] magrittr_1.5      whisker_0.3-2     backports_1.1.2  
[28] scales_0.5.0      htmltools_0.3.6   assertthat_0.2.0 
[31] colorspace_1.3-2  labeling_0.3      stringi_1.2.2    
[34] lazyeval_0.2.1    munsell_0.4.3     R.oo_1.22.0      </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.0.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
