<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2018-10-22" />

<title>APA and Protien QTL overlap</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">APA and Protien QTL overlap</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>10/22/2018</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-02-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>threeprimeseq/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree606e562d86042f5e0b093d1b4cdf423502814038targetblank606e562a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/606e562d86042f5e0b093d1b4cdf423502814038" target="_blank">606e562</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree606e562d86042f5e0b093d1b4cdf423502814038targetblank606e562a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/perm_QTL_trans_noMP_5percov/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/4suDataIGV.Rmd
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/EvaleQTLs.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  analysis/verifybam_dubs.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/ApaQTLs/
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/DistTXN2Peak_genelocAnno/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeakCounts_noMP_5perc/
    Untracked:  data/PeakCounts_noMP_genelocanno/
    Untracked:  data/PeakUsage/
    Untracked:  data/PeakUsage_noMP/
    Untracked:  data/PeakUsage_noMP_GeneLocAnno/
    Untracked:  data/PeaksUsed/
    Untracked:  data/PeaksUsed_noMP_5percCov/
    Untracked:  data/RNAkalisto/
    Untracked:  data/RefSeq_annotations/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/WASP_STAT/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/apaQTL_examp_noMP/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_GeneLocAnno/
    Untracked:  data/diff_iso_proc/
    Untracked:  data/diff_iso_trans/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/molPheno_noMP/
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov_3UTR/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/protAndAPAAndExplmRes.Rda
    Untracked:  data/protAndAPAlmRes.Rda
    Untracked:  data/protAndExpressionlmRes.Rda
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  data/threePrimeSeqMetaData55Ind.txt
    Untracked:  data/threePrimeSeqMetaData55Ind.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/NewPeakPostMP.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/mispriming_approach.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlapMolQTL.opposite.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/peakQCPPlots.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/pipeline_55Ind.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/test.smash.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/apa_prot_overlap.html" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d8c6f3c5da410db72cab08afd52d4b05d4852812/analysis/apa_prot_overlap.Rmd" target="_blank">d8c6f3c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
<td>
add ven diagrams
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/70a8d68e3f30a742c9fe9e1dae6c6a37f229f981/analysis/apa_prot_overlap.Rmd" target="_blank">70a8d68</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-23
</td>
<td>
add anova pvalue to examples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/apa_prot_overlap.html" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/c8dbaf06efdfed0b7563c03c01ccc390b6cea6c0/analysis/apa_prot_overlap.Rmd" target="_blank">c8dbaf0</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
<td>
add apa prot rna overlap
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>I want to use this analysis to look at the genes with a APAQTL and a protein QTL. I am trying to understand how many of these are independent of RNA.</p>
<p>I will first look at genes with significant QTLs in both phenotypes. I can use the pipeline I created in <a href="https://brimittleman.github.io/threeprimeseq/swarmPlots_QTLs.html" class="uri">https://brimittleman.github.io/threeprimeseq/swarmPlots_QTLs.html</a> to vizualize these snps.</p>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.2.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ───────────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.0.0     ✔ purrr   0.2.5
✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.4.0
✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 3.5.2</code></pre>
<pre><code>── Conflicts ──────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre class="r"><code>library(ggpubr)</code></pre>
<pre><code>Loading required package: magrittr</code></pre>
<pre><code>
Attaching package: &#39;magrittr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    set_names</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    extract</code></pre>
<pre><code>
Attaching package: &#39;ggpubr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:cowplot&#39;:

    get_legend</code></pre>
<pre class="r"><code>library(VennDiagram)</code></pre>
<pre><code>Loading required package: grid</code></pre>
<pre><code>Loading required package: futile.logger</code></pre>
<pre><code>
Attaching package: &#39;VennDiagram&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggpubr&#39;:

    rotate</code></pre>
<pre class="r"><code>library(gridExtra)</code></pre>
<pre><code>
Attaching package: &#39;gridExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    combine</code></pre>
<p>Gene Names:</p>
<pre class="r"><code>geneNames=read.table(&quot;../data/ensemble_to_genename.txt&quot;, stringsAsFactors = F, header = T, sep=&quot;\t&quot;)</code></pre>
<p>Significant APA QTLS:</p>
<pre class="r"><code>nuclearAPA=read.table(&quot;../data/perm_QTL_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_transcript_permResBH.txt&quot;, stringsAsFactors = F, header = T) %&gt;% separate(pid, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;), sep=&quot;:&quot;) %&gt;% separate(id, into=c(&quot;Gene.name&quot;, &quot;strand&quot;, &quot;peaknum&quot;), sep=&quot;_&quot;) %&gt;% dplyr::filter(-log10(bh)&gt;1)
totalAPA=read.table(&quot;../data/perm_QTL_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_transcript_permResBH.txt&quot;, stringsAsFactors = F, header=T)  %&gt;% separate(pid, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;), sep=&quot;:&quot;) %&gt;% separate(id, into=c(&quot;Gene.name&quot;, &quot;strand&quot;, &quot;peaknum&quot;), sep=&quot;_&quot;) %&gt;% dplyr::filter(-log10(bh)&gt;1)</code></pre>
<p>Significant Protien QTLs</p>
<pre class="r"><code>protQTL=read.table(&quot;../data/other_qtls/fastqtl_qqnorm_prot.fixed.perm.out&quot;, col.names = c(&quot;Gene.stable.ID&quot;, &quot;nvar&quot;, &quot;shape1&quot;, &quot;shape2&quot;, &quot;dummy&quot;, &quot;sid&quot;, &quot;dist&quot;, &quot;npval&quot;, &quot;slope&quot;, &quot;ppval&quot;, &quot;bpval&quot;),stringsAsFactors=F) %&gt;% inner_join(geneNames, by=&quot;Gene.stable.ID&quot;) %&gt;% dplyr::select(&quot;Gene.name&quot;, &quot;nvar&quot;, &quot;shape1&quot;, &quot;shape2&quot;, &quot;dummy&quot;, &quot;sid&quot;, &quot;dist&quot;, &quot;npval&quot;, &quot;slope&quot;, &quot;ppval&quot;, &quot;bpval&quot;)

protQTL$bh=p.adjust(protQTL$bpval, method=&quot;fdr&quot;)
protQTL_sig= protQTL %&gt;% dplyr::filter(-log10(bh)&gt;1)</code></pre>
<div id="gene-level" class="section level2">
<h2>Gene Level</h2>
<p>Overlap the QTLs by gene name:</p>
<pre class="r"><code>genesBothTot=protQTL_sig %&gt;%  inner_join(totalAPA, by=c(&quot;Gene.name&quot;))
genesBotNuc=protQTL_sig %&gt;%  inner_join(nuclearAPA, by=c(&quot;Gene.name&quot;))</code></pre>
<p>These are the genes that have a significant QTL in both.They are not the same snp. This may be because I am using the permuted snps. I will use the APA snp to make the plot.</p>
<pre class="r"><code>plotQTL_func= function(SNP, peak, gene){
  apaN_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_PeakAPANuclear.&quot;, SNP, peak, &quot;.txt&quot;, sep = &quot;&quot; ), header=T)
  apaT_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_PeakAPATotal.&quot;, SNP, peak, &quot;.txt&quot;, sep = &quot;&quot; ), header=T)
  su30_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_4su_30_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header = T)
  su60_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_4su_60_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header=T)
  RNA_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_RNAseq_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;),header=T)
  RNAg_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_RNAseqGeuvadis_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header = T)
  ribo_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_ribo_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;),header=T)
  prot_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_prot.&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header=T)

  
  ggplot_func= function(file, molPhen,GENE,allOverlap_T){
    file = file %&gt;% mutate(genotype=Allele1 + Allele2)
    file$genotype= as.factor(as.character(file$genotype))
    plot=ggplot(file, aes(y=Pheno, x=genotype, by=genotype, fill=genotype)) + geom_boxplot(width=.25) + geom_jitter() + labs(y=&quot;Phenotpye&quot;,title=paste(molPhen, GENE, sep=&quot;: &quot;)) + scale_fill_brewer(palette=&quot;Paired&quot;) + stat_compare_means(method = &quot;anova&quot;,  label.y.npc = &quot;top&quot;)
    return(plot)
  }

  
  apaNplot=ggplot_func(apaN_file, &quot;Apa Nuclear&quot;, gene)
  apaTplot=ggplot_func(apaT_file, &quot;Apa Total&quot;, gene)
  su30plot=ggplot_func(su30_file, &quot;4su30&quot;,gene)
  su60plot=ggplot_func(su60_file, &quot;4su60&quot;,gene)
  RNAplot=ggplot_func(RNA_file, &quot;RNA&quot;,gene)
  RNAgPlot=ggplot_func(RNAg_file, &quot;RNAg&quot;,gene)
  riboPlot= ggplot_func(ribo_file, &quot;Ribo&quot;,gene)
  protplot=ggplot_func(prot_file, &quot;Protein&quot;,gene)
  
  full_plot= plot_grid(apaNplot,apaTplot, RNAplot, protplot,nrow=2)
  return (full_plot)
}</code></pre>
<p>Total:</p>
<ul>
<li>MRPL43 peak44585 10:102740271</li>
</ul>
<pre class="bash"><code>grep MRPL43 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000055950


python createQTLsnpAPAPhenTable.py 10 10:102740271  peak44585 Total
python createQTLsnpAPAPhenTable.py 10 10:102740271  peak44585  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;10&quot; &quot;10:102740271&quot; &quot;ENSG00000055950&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*10:102740271* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;10:102740271&quot;, peak=&quot;peak44585&quot;, gene=&quot;ENSG00000055950&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-8-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-8-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Nuclear:</p>
<ul>
<li>SWAP70 peak49384 11:9732917</li>
</ul>
<pre class="bash"><code>grep SWAP70 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000133789


python createQTLsnpAPAPhenTable.py 11 11:9732917  peak49384 Total
python createQTLsnpAPAPhenTable.py 11 11:9732917  peak49384  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;11&quot; &quot;11:9732917&quot; &quot;ENSG00000133789&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*11:9732917* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;11:9732917&quot;, peak=&quot;peak49384&quot;, gene=&quot;ENSG00000133789&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-10-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-10-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>DHRS7B peak132739 17:21102458</li>
</ul>
<pre class="bash"><code>grep DHRS7B /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000109016


python createQTLsnpAPAPhenTable.py 17 17:21102458  peak132739 Total
python createQTLsnpAPAPhenTable.py 17 17:21102458  peak132739  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;17&quot; &quot;17:21102458&quot; &quot;ENSG00000109016&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*17:21102458* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;17:21102458&quot;, peak=&quot;peak132739&quot;, gene=&quot;ENSG00000109016&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-12-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-12-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>UBA6 peak240167 4:68502794</li>
</ul>
<pre class="bash"><code>grep UBA6 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000033178


python createQTLsnpAPAPhenTable.py 4 4:68502794  peak240167 Total
python createQTLsnpAPAPhenTable.py 4 4:68502794  peak240167  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;4&quot; &quot;4:68502794&quot; &quot;ENSG00000033178&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*4:68502794* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;4:68502794&quot;, peak=&quot;peak240167&quot;, gene=&quot;ENSG00000033178&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-14-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-14-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This is not the most effective way to do this because I am overlapping by gene then looking at the effect of the apaQTL snp. I want a method that will look directly at the effect of one snp. I can use the overlap files I created based on the APA qtls in other phenotypes. I can overlap the phenotypes and look for snps that have low pvalues in APA and protien.</p>
</div>
<div id="snp-level" class="section level2">
<h2>SNP level</h2>
<div id="total" class="section level3">
<h3>Total</h3>
<p>I want the overlap where I started in APA qtls and found the snp in the mol file. I am starting with the total.</p>
<pre class="r"><code>totAPAinsu30=read.table(&quot;../data/mol_overlap/APA2molTotal/TotAPAqtlsPval4su30.txt&quot;, header = T, stringsAsFactors = F)
totAPAinsu60=read.table(&quot;../data/mol_overlap/APA2molTotal/TotAPAqtlsPval4su60.txt&quot;, header = T, stringsAsFactors = F)
totAPAinRNA=read.table(&quot;../data/mol_overlap/APA2molTotal/TotAPAqtlsPvalRNA.txt&quot;, header = T, stringsAsFactors = F)
totAPAinRNAg=read.table(&quot;../data/mol_overlap/APA2molTotal/TotAPAqtlsPvalRNAg.txt&quot;, header = T, stringsAsFactors = F)
totAPAinRibo=read.table(&quot;../data/mol_overlap/APA2molTotal/TotAPAqtlsPvalribo.txt&quot;, header = T, stringsAsFactors = F)
totAPAinProt=read.table(&quot;../data/mol_overlap/APA2molTotal/TotAPAqtlsPvalProtein.txt&quot;, header = T, stringsAsFactors = F)


allOverlap_T=totAPAinsu30 %&gt;%  full_join(totAPAinsu60, by=c(&quot;Gene.name&quot;, &quot;sid&quot;)) %&gt;%  full_join(totAPAinRNA, by=c(&quot;Gene.name&quot;, &quot;sid&quot;)) %&gt;%  full_join(totAPAinRNAg, by=c(&quot;Gene.name&quot;, &quot;sid&quot;))  %&gt;%  full_join(totAPAinRibo, by=c(&quot;Gene.name&quot;, &quot;sid&quot;))  %&gt;%  full_join(totAPAinProt, by=c(&quot;Gene.name&quot;, &quot;sid&quot;)) 

colnames(allOverlap_T)=c(&quot;Gene.name&quot;, &quot;sid&quot;, &quot;su30&quot;, &quot;su60&quot;, &quot;RNA&quot;, &quot;RNAg&quot;, &quot;ribo&quot;, &quot;prot&quot;)</code></pre>
<pre class="r"><code>plot(sort(allOverlap_T$prot))</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-16-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot(allOverlap_T$RNA ~ allOverlap_T$prot)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-2">
Past versions of unnamed-chunk-16-2.png
</button>
</p>
<div id="fig-unnamed-chunk-16-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-16-2.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I want to make a ggplot of these where I color them by RNA pvalue:</p>
<pre class="r"><code>allOverlap_T_lowP=allOverlap_T %&gt;% dplyr::filter(prot&lt;.05)
ggplot(allOverlap_T_lowP, aes(x=RNA, y=prot)) + geom_point()+ geom_text(aes(label=Gene.name),hjust=0, vjust=0) + geom_vline(xintercept = .05, col=&quot;red&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-17-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(allOverlap_T_lowP, aes(x=RNAg, y=prot)) + geom_point()+ geom_text(aes(label=Gene.name),hjust=0, vjust=0) + geom_vline(xintercept = .05, col=&quot;red&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-18-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I can use these to look for examples of SNPs that are significant in prot but not in RNA.</p>
<p>Look at some of these:</p>
<p>Total RNA:<br />
* SACM1L<br />
* EBI3<br />
* FBXL18<br />
* PSMF1<br />
* COX17</p>
<p>Total RNAg:<br />
* EBI3<br />
* FBXL18<br />
* APBB1IP * PSMF1</p>
<p>Look at some examples of genes that come up in both.</p>
<p>EBI3 peak152751 19:4236475</p>
<p>Expressed in B lymphocytes in response to EB virus.</p>
<pre class="bash"><code>grep EBI3 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000105246


python createQTLsnpAPAPhenTable.py 19 19:4236475  peak152751 Total
python createQTLsnpAPAPhenTable.py 19 19:4236475  peak152751  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;19&quot; &quot;19:4236475&quot; &quot;ENSG00000105246&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*19:4236475* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;19:4236475&quot;, peak=&quot;peak152751&quot;, gene=&quot;ENSG00000105246&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-20-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-20-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>FBXL18 peak291746 7:5524129</p>
<p>“The protein encoded by this gene is a member of a family of proteins that contain an approximately 40-amino acid F-box motif. This motif is important for interaction with SKP1 and for targeting some proteins for degradation.” genecards.org</p>
<pre class="bash"><code>grep FBXL18 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000155034


python createQTLsnpAPAPhenTable.py 7 7:5524129  peak291746 Total
python createQTLsnpAPAPhenTable.py 7 7:5524129  peak291746  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;7&quot; &quot;7:5524129&quot; &quot;ENSG00000155034&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*7:5524129* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;7:5524129&quot;, peak=&quot;peak291746&quot;, gene=&quot;ENSG00000155034&quot;)</code></pre>
<pre><code>Warning: Removed 4 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 4 rows containing non-finite values (stat_compare_means).</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-22-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-22-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>PSMF1 peak193648 20:1131308</li>
</ul>
<p>This gene codes the 26S proteasome.</p>
<pre class="bash"><code>grep PSMF1 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000125818


python createQTLsnpAPAPhenTable.py 20 20:1131308   peak193648 Total
python createQTLsnpAPAPhenTable.py 20 20:1131308   peak193648  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;20&quot; &quot;20:1131308&quot; &quot;ENSG00000125818&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*20:1131308* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;20:1131308&quot;, peak=&quot;peak193648&quot;, gene=&quot;ENSG00000125818&quot;)</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite values (stat_compare_means).</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-24-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-24-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I want to know the number of these that are &lt;.05 in protien and above .1 in RNA</p>
<pre class="r"><code>allOverlap_T_lowP_highRNA=allOverlap_T %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(RNA&gt;.05)
allOverlap_T_lowP_highRNAg=allOverlap_T %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(RNAg&gt;.05)</code></pre>
<p>8 snps with &lt; .05 for protein. Of those 6 have RNA pvalues greater than .05</p>
<pre class="r"><code>6/8</code></pre>
<pre><code>[1] 0.75</code></pre>
</div>
<div id="nuclear" class="section level3">
<h3>Nuclear</h3>
<pre class="r"><code>nucAPAinsu30=read.table(&quot;../data/mol_overlap/APA2molNuclear/NucAPAqtlsPval4su30.txt&quot;, header = T, stringsAsFactors = F)
nucAPAinsu60=read.table(&quot;../data/mol_overlap/APA2molNuclear/NucAPAqtlsPval4su60.txt&quot;, header = T, stringsAsFactors = F)
nucAPAinRNA=read.table(&quot;../data/mol_overlap/APA2molNuclear/NucAPAqtlsPvalRNA.txt&quot;, header = T, stringsAsFactors = F)
nucAPAinRNAg=read.table(&quot;../data/mol_overlap/APA2molNuclear/NucAPAqtlsPvalRNAg.txt&quot;, header = T, stringsAsFactors = F)
nucAPAinRibo=read.table(&quot;../data/mol_overlap/APA2molNuclear/NucAPAqtlsPvalribo.txt&quot;, header = T, stringsAsFactors = F)
nucAPAinProt=read.table(&quot;../data/mol_overlap/APA2molNuclear/NucAPAqtlsPvalProtein.txt&quot;, header = T, stringsAsFactors = F)


allOverlap_N=nucAPAinsu30 %&gt;%  full_join(nucAPAinsu60, by=c(&quot;Gene.name&quot;, &quot;sid&quot;)) %&gt;%  full_join(nucAPAinRNA, by=c(&quot;Gene.name&quot;, &quot;sid&quot;)) %&gt;%  full_join(nucAPAinRNAg, by=c(&quot;Gene.name&quot;, &quot;sid&quot;))  %&gt;%  full_join(nucAPAinRibo, by=c(&quot;Gene.name&quot;, &quot;sid&quot;))  %&gt;%  full_join(nucAPAinProt, by=c(&quot;Gene.name&quot;, &quot;sid&quot;)) 

colnames(allOverlap_N)=c(&quot;Gene.name&quot;, &quot;sid&quot;, &quot;su30&quot;, &quot;su60&quot;, &quot;RNA&quot;, &quot;RNAg&quot;, &quot;ribo&quot;, &quot;prot&quot;)</code></pre>
<pre class="r"><code>#subset by prot &lt; .05  

allOverlap_N_lowP=allOverlap_N %&gt;% dplyr::filter(prot&lt;.05)
ggplot(allOverlap_N_lowP, aes(x=RNA, y=prot)) + geom_point()+ geom_text(aes(label=Gene.name),hjust=0, vjust=0)+ geom_vline(xintercept = .05, col=&quot;red&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-28-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(allOverlap_N_lowP, aes(x=RNAg, y=prot)) + geom_point()+ geom_text(aes(label=Gene.name),hjust=0, vjust=0)+ geom_vline(xintercept = .05, col=&quot;red&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-29-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>allOverlap_N_lowP_highRNA=allOverlap_N %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(RNA&gt;.05)
allOverlap_N_lowP_highRNAg=allOverlap_N %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(RNAg&gt;.05)</code></pre>
<p>39 snps with &lt; .05 for protein. Of those 28 have RNA pvalues greater than .05</p>
<pre class="r"><code>28/39</code></pre>
<pre><code>[1] 0.7179487</code></pre>
<pre class="r"><code>inBothN= allOverlap_N_lowP_highRNAg %&gt;%  inner_join(allOverlap_N_lowP_highRNA, by=c(&quot;Gene.name&quot;, &quot;sid&quot;, &quot;su30&quot;, &quot;su60&quot;, &quot;RNA&quot;, &quot;RNAg&quot;, &quot;ribo&quot;, &quot;prot&quot;)) %&gt;% arrange(desc(RNA))</code></pre>
<pre class="r"><code>inBothN$Gene.name[1:5]</code></pre>
<pre><code>[1] &quot;MSMO1&quot;  &quot;LYAR&quot;   &quot;CD2BP2&quot; &quot;KDM2A&quot;  &quot;RINT1&quot; </code></pre>
<ul>
<li>MSMO1 4:166260601 peak249109</li>
</ul>
<p>contains metal binding motifs, known alternative splice isoforms</p>
<pre class="bash"><code>grep MSMO1 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000052802


python createQTLsnpAPAPhenTable.py 4 4:166260601  peak249109 Total
python createQTLsnpAPAPhenTable.py 4 4:166260601  peak249109  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;4&quot; &quot;4:166260601&quot; &quot;ENSG00000052802&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*4:166260601* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;4:166260601&quot;, peak=&quot;peak249109&quot;, gene=&quot;ENSG00000052802&quot;)</code></pre>
<pre><code>Warning: Removed 3 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 3 rows containing non-finite values (stat_compare_means).</code></pre>
<pre><code>Warning: Removed 3 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-35-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-35-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>LYAR peak235215 4:4196045</li>
</ul>
<p>involved in processing pre-rRNA</p>
<pre class="bash"><code>grep LYAR /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000145220


python createQTLsnpAPAPhenTable.py 4 4:4196045  peak235215 Total
python createQTLsnpAPAPhenTable.py 4 4:4196045  peak235215  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;4&quot; &quot;4:4196045&quot; &quot;ENSG00000145220&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*4:4196045* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;4:4196045&quot;, peak=&quot;peak235215&quot;, gene=&quot;ENSG00000145220&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-37-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-37-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>CD2BP2 peak122237 16:29898001</p>
<p>From genecards: “in the cytoplasm, the encoded protein binds the cytoplasmic tail of human surface antigen CD2 via its C-terminal GYF domain, and regulate CD2-triggered T lymphocyte activation. In the nucleus, this protein is a component of the U5 small nuclear ribonucleoprotein complex and is involved in RNA splicing.”</p>
<pre class="bash"><code>grep CD2BP2 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000169217


python createQTLsnpAPAPhenTable.py 16 16:29898001 peak122237 Total
python createQTLsnpAPAPhenTable.py 16 16:29898001 peak122237  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;16&quot; &quot;16:29898001&quot; &quot;ENSG00000169217&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*16:29898001* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;16:29898001&quot;, peak=&quot;peak122237&quot;, gene=&quot;ENSG00000169217&quot;)</code></pre>
<pre><code>Warning: Removed 5 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 5 rows containing non-finite values (stat_compare_means).</code></pre>
<pre><code>Warning: Removed 5 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-39-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-39-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>KDM2A peak55622 11:66851583</p>
<pre class="bash"><code>grep KDM2A /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000173120


python createQTLsnpAPAPhenTable.py 11 11:66851583 peak55622 Total
python createQTLsnpAPAPhenTable.py 11 11:66851583 peak55622  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;11&quot; &quot;11:66851583&quot; &quot;ENSG00000173120&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*11:66851583* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;11:66851583&quot;, peak=&quot;peak55622&quot;, gene=&quot;ENSG00000173120&quot;)</code></pre>
<pre><code>Warning: Removed 8 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 8 rows containing non-finite values (stat_compare_means).</code></pre>
<pre><code>Warning: Removed 8 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-41-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-41-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>RINT1 peak303436 7:105155320</p>
<p>Interacts with double strand break repair protiens, regulates cell cycle and telomere length</p>
<pre class="bash"><code>grep RINT1 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000135249


python createQTLsnpAPAPhenTable.py 7 7:105155320 peak303436 Total
python createQTLsnpAPAPhenTable.py 7 7:105155320 peak303436  Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;7&quot; &quot;7:105155320&quot; &quot;ENSG00000135249&quot;

#into apaExamp
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*7:105155320* .
</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;7:105155320&quot;, peak=&quot;peak303436&quot;, gene=&quot;ENSG00000135249&quot;)</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite values (stat_compare_means).</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-43-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-1">
Past versions of unnamed-chunk-43-1.png
</button>
</p>
<div id="fig-unnamed-chunk-43-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-43-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-43-1.png" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="significance" class="section level3">
<h3>Significance</h3>
<p>In the next step I need to add significance to the boxplots and think more about the significance cutoffs.</p>
<p>Maybe I can compare 2 other phenotypes for &lt;.05 and &gt;.05 to see if the percentage is less than what I see for RNA and protein.</p>
<pre class="r"><code>sigProt=allOverlap_N_lowP %&gt;% nrow()
paste(&quot;Sig Prot&quot;, sigProt, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot 39&quot;</code></pre>
<pre class="r"><code>allOverlap_N_lowP_higrna=allOverlap_N %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(RNA&gt;.05) %&gt;%nrow()
paste(&quot;Sig Prot not RNA&quot;, allOverlap_N_lowP_higrna, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot not RNA 28&quot;</code></pre>
<pre class="r"><code>allOverlap_N_lowP_higribo=allOverlap_N %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(ribo&gt;.05) %&gt;%nrow()
paste(&quot;Sig Prot not ribo&quot;, allOverlap_N_lowP_higribo, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot not ribo 23&quot;</code></pre>
<pre class="r"><code>allOverlap_N_lowP_higsu30=allOverlap_N %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(su30&gt;.05) %&gt;% nrow()
paste(&quot;Sig Prot not 4su30&quot;, allOverlap_N_lowP_higsu30, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot not 4su30 31&quot;</code></pre>
<pre class="r"><code>allOverlap_N_lowP_higsu60=allOverlap_N %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(su60&gt;.05) %&gt;% nrow()
paste(&quot;Sig Prot not 4su60&quot;, allOverlap_N_lowP_higsu60, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot not 4su60 27&quot;</code></pre>
<pre class="r"><code>sigProtT=allOverlap_T_lowP %&gt;% nrow()
paste(&quot;Sig Prot&quot;, sigProtT, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot 8&quot;</code></pre>
<pre class="r"><code>allOverlap_T_lowP_higrna=allOverlap_T %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(RNA&gt;.05) %&gt;%nrow()
paste(&quot;Sig Prot not RNA&quot;, allOverlap_T_lowP_higrna, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot not RNA 6&quot;</code></pre>
<pre class="r"><code>allOverlap_T_lowP_higribo=allOverlap_T %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(ribo&gt;.05) %&gt;%nrow()
paste(&quot;Sig Prot not ribo&quot;, allOverlap_T_lowP_higribo, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot not ribo 5&quot;</code></pre>
<pre class="r"><code>allOverlap_T_lowP_higsu30=allOverlap_T %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(su30&gt;.05) %&gt;% nrow()
paste(&quot;Sig Prot not 4su30&quot;, allOverlap_T_lowP_higsu30, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot not 4su30 6&quot;</code></pre>
<pre class="r"><code>allOverlap_T_lowP_higsu60=allOverlap_T %&gt;% dplyr::filter(prot&lt;.05) %&gt;% dplyr::filter(su60&gt;.05) %&gt;% nrow()
paste(&quot;Sig Prot not 4su60&quot;, allOverlap_T_lowP_higsu60, sep=&quot; &quot;)</code></pre>
<pre><code>[1] &quot;Sig Prot not 4su60 5&quot;</code></pre>
</div>
<div id="closer-to-a-global-scale" class="section level3">
<h3>CLoser to a global scale</h3>
<p>In order to think about this more broadly I am going to plot overlaps:</p>
<p>Given all of these are APA qtls, how many are significant in:</p>
<ul>
<li>protein<br />
</li>
<li>RNA and protein</li>
<li>RNA but not protein</li>
<li>Not in RNA or Protein</li>
</ul>
<p>Do this first for total:</p>
<pre class="r"><code>p_N=allOverlap_N %&gt;% filter(prot&lt;.05) %&gt;% nrow()
RP_N=allOverlap_N %&gt;% filter(prot&lt;.05 &amp; RNA &lt;.05) %&gt;% nrow()
RnotP_N=allOverlap_N %&gt;% filter(prot&gt;.05 &amp; RNA &lt;.05) %&gt;% nrow()
notRnotP_N=allOverlap_N %&gt;% filter(prot&gt;.05 &amp; RNA &gt;.05) %&gt;% nrow()
notRP_N= allOverlap_N %&gt;% filter(prot&lt;.05 &amp; RNA &gt;.05) %&gt;% nrow()


overlapNames=c(&quot;Protein&quot;, &quot;RNAandProtein&quot;, &quot;RNAnotProt&quot;, &quot;Neither&quot;, &quot;ProteinnotRNA&quot;)
NucRPoverlap=c(p_N/nrow(allOverlap_N),RP_N/nrow(allOverlap_N), RnotP_N/nrow(allOverlap_N), notRnotP_N/nrow(allOverlap_N),notRP_N/nrow(allOverlap_N) )


NucOverlapDF=data.frame(overlapNames,NucRPoverlap)
ggplot(NucOverlapDF, aes(x=overlapNames, y=NucRPoverlap)) + geom_bar(stat=&quot;identity&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-46-1">
Past versions of unnamed-chunk-46-1.png
</button>
</p>
<div id="fig-unnamed-chunk-46-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-46-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>DO the same for total:</p>
<pre class="r"><code>p_T=allOverlap_T %&gt;% filter(prot&lt;.05) %&gt;% nrow()
RP_T=allOverlap_T %&gt;% filter(prot&lt;.05 &amp; RNA &lt;.05) %&gt;% nrow()
RnotP_T=allOverlap_T %&gt;% filter(prot&gt;.05 &amp; RNA &lt;.05) %&gt;% nrow()
notRnotP_T=allOverlap_T %&gt;% filter(prot&gt;.05 &amp; RNA &gt;.05) %&gt;% nrow()
notRP_T= allOverlap_T %&gt;% filter(prot&lt;.05 &amp; RNA &gt;.05) %&gt;% nrow()

TotRPoverlap=c(p_T/nrow(allOverlap_T),RP_T/nrow(allOverlap_T), RnotP_T/nrow(allOverlap_T), notRnotP_T/nrow(allOverlap_T),notRP_T/nrow(allOverlap_T))
TotOverlapDF=data.frame(overlapNames,TotRPoverlap)
ggplot(TotOverlapDF, aes(x=overlapNames, y=TotRPoverlap)) + geom_bar(stat=&quot;identity&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-47-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>allOverlapDF=TotOverlapDF %&gt;% full_join(NucOverlapDF, by=&quot;overlapNames&quot;)

allOverlapDF_melt=melt(allOverlapDF, id.vars=&quot;overlapNames&quot;)

ggplot(allOverlapDF_melt, aes(x=overlapNames, y=value, by=variable, fill=variable)) + geom_bar(stat=&quot;identity&quot;, position=&quot;dodge&quot;) + scale_fill_manual(values=c(&quot;darkviolet&quot;, &quot;deepskyblue3&quot;)) + labs(y=&quot;Proportion of APA QTLs&quot;, x=&quot;Category&quot;)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-48-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>oT=overlapplotT=draw.pairwise.venn(area1=.08, area2=.07, cross.area = .02, category = c(&quot;Protein&quot;, &quot;RNA&quot;),lty = rep(&quot;solid&quot;, 2), fill = c(&quot;Blue&quot;, &quot;Orange&quot;), alpha = rep(0.5, 2), cat.pos = c(0, 0), cat.dist = rep(0.025, 2), euler.d = T, scaled=T)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-49-1">
Past versions of unnamed-chunk-49-1.png
</button>
</p>
<div id="fig-unnamed-chunk-49-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-49-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(&quot;../output/plots/overlapProtRNAvenT.png&quot;) 
grid.arrange(gTree(children=oT), top=&quot;Total: Protein and RNA QTL overlap&quot;, bottom=&quot;Neither=.37&quot;)

dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>oN=overlapplotN=draw.pairwise.venn(area1=.056, area2=.096, cross.area = .016, category = c(&quot;Protein&quot;, &quot;RNA&quot;),lty = rep(&quot;solid&quot;, 2), fill = c(&quot;Blue&quot;, &quot;Orange&quot;), alpha = rep(0.5, 2), cat.pos = c(0, 0), cat.dist = rep(0.025, 2), euler.d = T, scaled=T)</code></pre>
<p><img src="figure/apa_prot_overlap.Rmd/unnamed-chunk-50-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-50-1">
Past versions of unnamed-chunk-50-1.png
</button>
</p>
<div id="fig-unnamed-chunk-50-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d59b3c136382a196f2ce31f4ea61f8109bf97dbd/docs/figure/apa_prot_overlap.Rmd/unnamed-chunk-50-1.png" target="_blank">d59b3c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(&quot;../output/plots/overlapProtRNAvenN.png&quot;) 

grid.arrange(gTree(children=oN), top=&quot;Nuclear:Protein and RNA QTL overlap&quot;, bottom=&quot;Neither=.38&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
This doesnt look statistically significant but it does look like the total fraction has more pQTLs that are not eQTLs and in the nuclear you get the opposite effect.<br />
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] bindrcpp_0.2.2      gridExtra_2.3       VennDiagram_1.6.20 
 [4] futile.logger_1.4.3 ggpubr_0.1.8        magrittr_1.5       
 [7] cowplot_0.9.3       data.table_1.11.8   forcats_0.3.0      
[10] stringr_1.4.0       dplyr_0.7.6         purrr_0.2.5        
[13] readr_1.1.1         tidyr_0.8.1         tibble_1.4.2       
[16] ggplot2_3.0.0       tidyverse_1.2.1     workflowr_1.2.0    

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4     reshape2_1.4.3       haven_1.1.2         
 [4] lattice_0.20-35      colorspace_1.3-2     htmltools_0.3.6     
 [7] yaml_2.2.0           rlang_0.2.2          pillar_1.3.0        
[10] glue_1.3.0           withr_2.1.2          RColorBrewer_1.1-2  
[13] lambda.r_1.2.3       modelr_0.1.2         readxl_1.1.0        
[16] bindr_0.1.1          plyr_1.8.4           munsell_0.5.0       
[19] gtable_0.2.0         cellranger_1.1.0     rvest_0.3.2         
[22] evaluate_0.13        labeling_0.3         knitr_1.20          
[25] broom_0.5.0          Rcpp_0.12.19         formatR_1.5         
[28] scales_1.0.0         backports_1.1.2      jsonlite_1.6        
[31] fs_1.2.6             hms_0.4.2            digest_0.6.17       
[34] stringi_1.2.4        rprojroot_1.3-2      cli_1.0.1           
[37] tools_3.5.1          lazyeval_0.2.1       futile.options_1.0.1
[40] crayon_1.3.4         whisker_0.3-2        pkgconfig_2.0.2     
[43] xml2_1.2.0           lubridate_1.7.4      assertthat_0.2.0    
[46] rmarkdown_1.11       httr_1.3.1           rstudioapi_0.9.0    
[49] R6_2.3.0             nlme_3.1-137         git2r_0.24.0        
[52] compiler_3.5.1      </code></pre>
</div>
</div>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
