<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Approach to better assign genes to peaks</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Approach to better assign genes to peaks</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>1/28/2019</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-02-14</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/73ef57ee28916b2de34600c675d1494f67caf6a8" target="_blank">73ef57e</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/perm_QTL_trans_noMP_5percov/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/4suDataIGV.Rmd
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/EvaleQTLs.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  analysis/verifybam_dubs.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/ApaQTLs/
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/DistTXN2Peak_genelocAnno/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeakCounts_noMP_5perc/
    Untracked:  data/PeakCounts_noMP_genelocanno/
    Untracked:  data/PeakUsage/
    Untracked:  data/PeakUsage_noMP/
    Untracked:  data/PeakUsage_noMP_GeneLocAnno/
    Untracked:  data/PeaksUsed/
    Untracked:  data/PeaksUsed_noMP_5percCov/
    Untracked:  data/RNAkalisto/
    Untracked:  data/RefSeq_annotations/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/WASP_STAT/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/apaQTL_examp_noMP/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_GeneLocAnno/
    Untracked:  data/diff_iso_proc/
    Untracked:  data/diff_iso_trans/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/molPheno_noMP/
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov_3UTR/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/protAndAPAAndExplmRes.Rda
    Untracked:  data/protAndAPAlmRes.Rda
    Untracked:  data/protAndExpressionlmRes.Rda
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  data/threePrimeSeqMetaData55Ind.txt
    Untracked:  data/threePrimeSeqMetaData55Ind.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/mispriming_approach.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlapMolQTL.opposite.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/peakQCPPlots.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/pipeline_55Ind.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   code/Snakefile

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/73ef57ee28916b2de34600c675d1494f67caf6a8/analysis/ApproachForGeneAssignment.Rmd" target="_blank">73ef57e</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-02-14
</td>
<td style="text-align:left;">
add fdr cutoff for leafcutter results
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/5d371f7dd38859bb9ee2ee7f73d75d9b923dd8f0/docs/ApproachForGeneAssignment.html" target="_blank">5d371f7</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-02-13
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/4e0fbe4afbe04daade70f5f33e849acb0b254b0d/analysis/ApproachForGeneAssignment.Rmd" target="_blank">4e0fbe4</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-02-13
</td>
<td style="text-align:left;">
add gene location plot
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/f91ab35a90c5485f29aebe06f7f51ff8ed3c4863/docs/ApproachForGeneAssignment.html" target="_blank">f91ab35</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/4541f4c4a7026705ec8c0d7a6905b3175659dfc4/analysis/ApproachForGeneAssignment.Rmd" target="_blank">4541f4c</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
add correlation plot
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/a70015c1d8d14b738c8c8ac74fe9efe5495f5eb4/docs/ApproachForGeneAssignment.html" target="_blank">a70015c</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/f846c50e6ceb4dcc00a323b6e854c01b9aa040df/analysis/ApproachForGeneAssignment.Rmd" target="_blank">f846c50</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
add distance plots
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/5c81ca7391db50d591685dd342c1a689b8f789ed/docs/ApproachForGeneAssignment.html" target="_blank">5c81ca7</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/c6310a2cdca263082892a5a750cbf7bff06a4730/analysis/ApproachForGeneAssignment.Rmd" target="_blank">c6310a2</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
QTL res
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/35c4e885a682c3e5eddc721f5ba22e145a16ca6d/docs/ApproachForGeneAssignment.html" target="_blank">35c4e88</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/a8a9acb19700d3ea88446e267d432f7b71127151/analysis/ApproachForGeneAssignment.Rmd" target="_blank">a8a9acb</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-30
</td>
<td style="text-align:left;">
add explanation for midlevel
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/736a503136062df42145237599d44efe94f16f01/docs/ApproachForGeneAssignment.html" target="_blank">736a503</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-29
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/a7f02950e5518547f051485ec325b61c729d2d87/analysis/ApproachForGeneAssignment.Rmd" target="_blank">a7f0295</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-29
</td>
<td style="text-align:left;">
evaluate top delta PSI
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/26951df7ecf6cc14d32737f5bb1cac7718295849/docs/ApproachForGeneAssignment.html" target="_blank">26951df</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-29
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/1def3fb49474592a08e1bf59b50960a3ba595a1c/docs/ApproachForGeneAssignment.html" target="_blank">1def3fb</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-29
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/d194ec23d501c5bed0510512de0ea4de58ea2a81/analysis/ApproachForGeneAssignment.Rmd" target="_blank">d194ec2</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-29
</td>
<td style="text-align:left;">
lc results
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/2701515bea9e53173dbdb1b6cd036cb4965b76c8/docs/ApproachForGeneAssignment.html" target="_blank">2701515</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-29
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/956f4a823c0e00684406e31873a02e4ada0beaf1/analysis/ApproachForGeneAssignment.Rmd" target="_blank">956f4a8</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-29
</td>
<td style="text-align:left;">
peak usage 5%
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/4e74f810baeb55ba265be42de715ff7b8650b13d/docs/ApproachForGeneAssignment.html" target="_blank">4e74f81</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-28
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/e7ed8fc62089d72607a313cc641cd01976e857ce/analysis/ApproachForGeneAssignment.Rmd" target="_blank">e7ed8fc</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-28
</td>
<td style="text-align:left;">
fix map func
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/a97adbc8c88ed1a800aabcfa7cc7fad5ab2df63e/docs/ApproachForGeneAssignment.html" target="_blank">a97adbc</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-28
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/a2911ff6b8b52d9320e90518db9ee0dc6433ff75/analysis/ApproachForGeneAssignment.Rmd" target="_blank">a2911ff</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-28
</td>
<td style="text-align:left;">
start analysis for new gene peak assignemnt
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="new-approach-idea" class="section level2">
<h2>New Approach Idea</h2>
<p>Lin et al: An in-depth map of polyadenylation sites in cancer 2012: -Mapped locations to annotated locations in UCUS browser: “The mapped locations were annotated using the UCSC genome browser tables ( 26 ). When a locus could be attributed to multiple possible annotations, the locus was assigned with a single annotation in the following priority order: 3′ UTRs (sense), coding sequences (CDS, sense), 5′ UTRs (sense), intron (sense), non-coding RNAs (ncRNAs, sense), 5′ UTR antisense, CDS antisense, 3′ UTR antisense, intron antisense, promoter antisense, ncRNA antisense and intergenic”</p>
<p>I want to download this annotation and try this. I am using the ncbi_refseq annotations. I will download regions of the genome seperatly and then merge the files.</p>
<ul>
<li><p>5’ UTR</p></li>
<li><p>Coding Exon</p></li>
<li><p>Intron</p></li>
<li><p>3’ UTR</p></li>
<li><p>(downstream 5000)-downstream proximal region</p></li>
</ul>
<p>I also want a dictionary with the transcripts and the gene names for the annotation. This information will come from the Transcript2GeneName file. In this file the transcript ID is in column1 and the gene name column 13.</p>
<p>I have downloaded all of the these to data/RefSeq_annotations. I will concatinate all of these for a full annotation dataset, I will then sort this file. The file is ncbiRefSeq_allAnnotation.sort.bed</p>
<p>Using this I can create an annotation in a bed file I can use for the overlap with my peaks. This will include getting the transcript to gene annotations. I will transfer the files to midway in my genome annotation directory and work with them there.</p>
<p>Format full refseq annotation:</p>
<pre class="bash"><code>TXN2Gene_file=&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/Transcript2GeneName.dms&quot;

gene_dic={}

for ln in open(TXN2Gene_file,&quot;r&quot;):
   txn=ln.split()[1]
   gene=ln.split()[12]
   gene_dic[txn]=gene

outF=open(&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_FormatedallAnnotation.sort.bed&quot;,&quot;w&quot;)

inFile=&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_allAnnotation.sort.bed&quot;  


for ln in open(inFile, &quot;r&quot;):
   chrom, start, end, name, score, strand = ln.split()
   chrom_fix=chrom[3:]
   txn=name.split(&quot;_&quot;)[:2]
   txnF=&quot;_&quot;.join(txn)
   gene=gene_dic[txnF]
   type=name.split(&quot;_&quot;)[2]
   id=type + &quot;:&quot; + gene
   outF.write(&quot;%s\t%s\t%s\t%s\t%s\t%s\n&quot;%(chrom_fix, start, end, id, score, strand))

outF.close()</code></pre>
</div>
<div id="map-peaks-with-new-annotation" class="section level2">
<h2>Map peaks with new annotation</h2>
<p>I want to create a file with my peaks mapped to these regions. I will include a structure for when there is a tie and put intergenic if it is not found. I need to do an intersect that gives me all of the IDs. After this I can use python to parse the hiarchy.</p>
<p>I can use bedtools map for this. I want all of the data to come back.</p>
<p>-c 4 -o distinct<br />
-S opposite strand</p>
<p>I will do this on the peaks before I looked at usage.</p>
<p>mapnoMPPeaks2GenomeLoc.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=mapnoMPPeaks2GenomeLoc
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mapnoMPPeaks2GenomeLoc.out
#SBATCH --error=mapnoMPPeaks2GenomeLoc.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

#annotation: /project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_FormatedallAnnotation.sort.bed

#peaks:  /project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_filtered/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR.bed


bedtools map -a /project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_filtered/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR.bed -b /project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_FormatedallAnnotation.sort.bed -c 4 -S -o distinct &gt; /project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLoc.bed
</code></pre>
<p>Look at how many get no annotation. I can do this interactively in python.</p>
<pre class="bash"><code>num=0
for ln in open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLoc.bed&quot;, &quot;r&quot;):
    location=ln.split()[7]
    if location==&quot;.&quot;:
        num +=1

print(num)</code></pre>
<p>This shows me that 85% of the peaks fall into one of these annotations.</p>
<p>Now I need to sort out the peaks with multiple annotations.</p>
<ol start="4" style="list-style-type: decimal">
<li><p>5’ UTR</p></li>
<li><p>Coding Exon</p></li>
<li><p>Intron</p></li>
<li><p>3’ UTR</p></li>
<li><p>(downstream 5000)-downstream proximal region</p></li>
</ol>
<p>I can write this out as the SAF I need. GeneID (peak1:1:14404:14484:-:OR4F16) Chr Start End Strand. For this I cannot include the peaks with no gene association. I can go back to this if i need to in the future.</p>
<p>processGenLocPeakAnno2SAF.py</p>
<pre class="bash"><code>inFile=&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLoc.bed&quot;
outFile=open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.SAF&quot; , &quot;w&quot;)

outFile.write(&quot;GeneID\tChr\tStart\tEnd\tStrand\n&quot;)
for ln in open(inFile, &quot;r&quot;):
    chrom, start, end, peak, cov, strand, score, anno = ln.split()
    if anno==&quot;.&quot;: 
        continue  
    anno_lst=anno.split(&quot;,&quot;)
    if len(anno_lst)==1:
        gene=anno_lst[0].split(&quot;:&quot;)[1]
        print(&quot;1 gene&quot;)
        peak_i=int(peak)
        start_i=int(start)
        end_i=int(end)
        ID=&quot;peak%d:%s:%d:%d:%s:%s&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
        outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
    else:
        type_dic={}
        for each in anno_lst:
            type_dic[each.split(&quot;:&quot;)[0]]=each.split(&quot;:&quot;)[1]
        if &quot;utr3&quot; in type_dic.keys():
            gene=type_dic[&quot;utr3&quot;]
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
        elif &quot;end&quot; in type_dic.keys():
            gene=type_dic[&quot;end&quot;]
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
        elif &quot;cds&quot; in type_dic.keys():
            gene=type_dic[&quot;cds&quot;] 
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
        elif &quot;utr5&quot; in type_dic.keys():
            gene=type_dic[&quot;utr5&quot;]
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
        elif &quot;intron&quot; in type_dic.keys():
            gene=type_dic[&quot;intron&quot;]
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))

outFile.close()</code></pre>
</div>
<div id="map-reads-to-new-annotated-peaks" class="section level2">
<h2>Map reads to new annotated peaks</h2>
<p>This has all of the peaks with at least one gene annotation. (104555) I can run feature counts on this to start getting the usage.</p>
<p>GeneLocAnno_fc_TN_noMP.sh</p>
<p>Because I mapped the genes opposite. The reads are now on in the same direction as the peaks</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=GeneLocAnno_fc_TN_noMP
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=GeneLocAnno_fc_TN_noMP.out
#SBATCH --error=GeneLocAnno_fc_TN_noMP.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

featureCounts -O -a /project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.SAF -F SAF -o /project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fc /project2/gilad/briana/threeprimeseq/data/bam_NoMP_sort/*T-combined-sort.noMP.sort.bam -s 1

featureCounts -O -a /project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.SAF -F SAF -o /project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fc /project2/gilad/briana/threeprimeseq/data/bam_NoMP_sort/*N-combined-sort.noMP.sort.bam -s 1</code></pre>
<p>Around 4 mill mapping in nuclear and 7 mill in total.</p>
<p>fix_head_fc_geneLoc_tot_noMP.py</p>
<pre class="bash"><code>infile= open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fc&quot;, &quot;r&quot;)
fout = open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;,&#39;w&#39;)
for line, i in enumerate(infile):
    if line == 1:
        i_list=i.split()
        libraries=i_list[:6]
        for sample in i_list[6:]:
            full = sample.split(&quot;/&quot;)[7]
            samp= full.split(&quot;-&quot;)[2:4]
            lim=&quot;_&quot;
            samp_st=lim.join(samp)
            libraries.append(samp_st)
        first_line= &quot;\t&quot;.join(libraries)
        fout.write(first_line + &#39;\n&#39;)
    else :
        fout.write(i)
fout.close()</code></pre>
<p>fix_head_fc_geneLoc_nuc_noMP.py</p>
<pre class="bash"><code>infile= open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fc&quot;, &quot;r&quot;)
fout = open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;,&#39;w&#39;)
for line, i in enumerate(infile):
    if line == 1:
        i_list=i.split()
        libraries=i_list[:6]
        for sample in i_list[6:]:
            full = sample.split(&quot;/&quot;)[7]
            samp= full.split(&quot;-&quot;)[2:4]
            lim=&quot;_&quot;
            samp_st=lim.join(samp)
            libraries.append(samp_st)
        first_line= &quot;\t&quot;.join(libraries)
        fout.write(first_line + &#39;\n&#39;)
    else :
        fout.write(i)
fout.close()
</code></pre>
<p>I can use the fileIDs</p>
<p>/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP/file_id_mapping_nuclear_Transcript.txt and /project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP/file_id_mapping_total_Transcript.txt</p>
</div>
<div id="make-phenotypes" class="section level2">
<h2>Make phenotypes</h2>
<p>I need to make a file with the gene start and ends because I use these in the phenotypes for QTL mapping. I can do this with the project2/gilad/briana/genome_anotation_data/RefSeq_annotations/Transcript2GeneName.dms file. I want the file to look like chr,start,end,txn:gene,.,strand I will sue the start and end of the coding seq</p>
<p>getGeneEnds.py</p>
<pre class="bash"><code>TXN2Gene_file=open(&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/Transcript2GeneName.dms&quot;,&quot;r&quot;)

outFile=open(&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_endAllGenes.bed&quot;, &quot;w&quot;)

for i, ln in enumerate(TXN2Gene_file):
    if i &gt;0 :
        chrom=ln.split()[2]
        chromf=chrom[3:]
        start=int(ln.split()[6])
        end=int(ln.split()[7])
        txn=ln.split()[1]
        genename=ln.split()[12]
        id=txn + &quot;:&quot; + genename
        strand=ln.split()[3]
        score=&quot;.&quot;
        outFile.write(&quot;%s\t%s\t%s\t%s\t%s\t%s\n&quot;%(chromf, start, end, id, score, strand))

outFile.close()</code></pre>
<p>Sort the bed file:</p>
<pre class="bash"><code>sort -k1,1 -k2,2n /project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_endAllGenes.bed &gt; /project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_endAllGenes.sort.bed</code></pre>
<p>makePhenoRefSeqPeaks_GeneLoc_Total_noMP.py</p>
<pre class="bash"><code>#PYTHON 3

dic_IND = {}
dic_BAM = {}

for ln in open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP/file_id_mapping_total_Transcript.txt&quot;):
    bam, IND = ln.split(&quot;\t&quot;)
    IND = IND.strip()
    dic_IND[bam] = IND
    if IND not in dic_BAM:
        dic_BAM[IND] = []
    dic_BAM[IND].append(bam)


#now I have ind dic with keys as the bam and ind as the values
#I also have a bam dic with ind as the keys and bam as the values  
    
inds=list(dic_BAM.keys()) #list of ind libraries  

#gene start and end dictionaries: 
dic_geneS = {}
dic_geneE = {}
for ln in open(&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_endAllGenes.sort.bed&quot;):
    chrom, start, end, geneID, score, strand = ln.split(&#39;\t&#39;)
    gene= geneID.split(&quot;:&quot;)[1]
#    if &quot;-&quot; in gene:
 #       gene=gene.split(&quot;-&quot;)[0]
    if gene not in dic_geneS:
        dic_geneS[gene]=int(start)
        dic_geneE[gene]=int(end)
        


#list of genes   

count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;, &quot;r&quot;)
genes=[]
for line , i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        gene=id_list[5]
        if gene not in genes:
            genes.append(gene)
            
#make the ind and gene dic  
dic_dub={}
for g in genes:
    dic_dub[g]={}
    for i in inds:
        dic_dub[g][i]=0


#populate the dictionary  
count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;, &quot;r&quot;)
for line, i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        g= id_list[5]
        values=list(i_list[6:])
        list_list=[]
        for ind,val in zip(inds, values):
            list_list.append([ind, val])
        for num, name in enumerate(list_list):
            dic_dub[g][list_list[num][0]] += int(list_list[num][1])
        

#write the file by acessing the dictionary and putting values in the table ver the value in the dic 
        

fout=open(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.fc&quot;,&quot;w&quot;)
peak=[&quot;chrom&quot;]
inds_noL=[]
for each in inds:
    indsNA= &quot;NA&quot; + each[:-2]
    inds_noL.append(indsNA) 
fout.write(&quot; &quot;.join(peak + inds_noL) + &#39;\n&#39; )


count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;, &quot;r&quot;)
for line , i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        gene=id_list[5]
        start=dic_geneS[id_list[5]]
        end=dic_geneE[id_list[5]]
        buff=[]
        buff.append(&quot;chr%s:%d:%d:%s_%s_%s&quot;%(id_list[1], start, end, id_list[5], id_list[4], id_list[0]))
        for x,y in zip(i_list[6:], inds):
            b=int(dic_dub[gene][y])
            t=int(x)
            buff.append(&quot;%d/%d&quot;%(t,b))
        fout.write(&quot; &quot;.join(buff)+ &#39;\n&#39;)
        
fout.close()</code></pre>
<p>makePhenoRefSeqPeaks_GeneLoc_Nuclear_noMP.py</p>
<pre class="bash"><code>
#PYTHON 3

dic_IND = {}
dic_BAM = {}

for ln in open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP/file_id_mapping_nuclear_Transcript.txt&quot;):
    bam, IND = ln.split(&quot;\t&quot;)
    IND = IND.strip()
    dic_IND[bam] = IND
    if IND not in dic_BAM:
        dic_BAM[IND] = []
    dic_BAM[IND].append(bam)


#now I have ind dic with keys as the bam and ind as the values
#I also have a bam dic with ind as the keys and bam as the values  
    
inds=list(dic_BAM.keys()) #list of ind libraries  

#gene start and end dictionaries: 
dic_geneS = {}
dic_geneE = {}
for ln in open(&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_endAllGenes.sort.bed&quot;):
    chrom, start, end, geneID, score, strand = ln.split(&#39;\t&#39;)
    gene= geneID.split(&quot;:&quot;)[1]
    #if &quot;-&quot; in gene:
     #   gene=gene.split(&quot;-&quot;)[0]
    if gene not in dic_geneS:
        dic_geneS[gene]=int(start)
        dic_geneE[gene]=int(end)
        


#list of genes   

count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;, &quot;r&quot;)
genes=[]
for line , i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        gene=id_list[5]
        if gene not in genes:
            genes.append(gene)
            
#make the ind and gene dic  
dic_dub={}
for g in genes:
    dic_dub[g]={}
    for i in inds:
        dic_dub[g][i]=0


#populate the dictionary  
count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;, &quot;r&quot;)
for line, i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        g= id_list[5]
        values=list(i_list[6:])
        list_list=[]
        for ind,val in zip(inds, values):
            list_list.append([ind, val])
        for num, name in enumerate(list_list):
            dic_dub[g][list_list[num][0]] += int(list_list[num][1])
        

#write the file by acessing the dictionary and putting values in the table ver the value in the dic 
        

fout=open(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.fc&quot;,&quot;w&quot;)
peak=[&quot;chrom&quot;]
inds_noL=[]
for each in inds:
    indsNA= &quot;NA&quot; + each[:-2]
    inds_noL.append(indsNA) 
fout.write(&quot; &quot;.join(peak + inds_noL) + &#39;\n&#39; )


count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;, &quot;r&quot;)
for line , i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        gene=id_list[5]
        start=dic_geneS[id_list[5]]
        end=dic_geneE[id_list[5]]
        buff=[]
        buff.append(&quot;chr%s:%d:%d:%s_%s_%s&quot;%(id_list[1], start, end, id_list[5], id_list[4], id_list[0]))
        for x,y in zip(i_list[6:], inds):
            b=int(dic_dub[gene][y])
            t=int(x)
            buff.append(&quot;%d/%d&quot;%(t,b))
        fout.write(&quot; &quot;.join(buff)+ &#39;\n&#39;)
        
fout.close()</code></pre>
<p>Run make pheno files:</p>
<p>run_makePhen_sep_GeneLocAnno_noMP.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_makePhen_sep_GeneLocAnno_noMP
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=run_makePhen_sep_GeneLocAnno_noMP.out
#SBATCH --error=run_makePhen_sep_GeneLocAnno_noMP.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

python makePhenoRefSeqPeaks_GeneLoc_Total_noMP.py  

python makePhenoRefSeqPeaks_GeneLoc_Nuclear_noMP.py  
</code></pre>
</div>
<div id="convert-to-usage" class="section level2">
<h2>Convert to Usage</h2>
<p>pheno2CountOnly_genelocAnno.R</p>
<pre class="r"><code>library(reshape2)
library(tidyverse)


totalPeakUs=read.table(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.fc&quot;, header = T, stringsAsFactors = F) %&gt;% separate(chrom, sep = &quot;:&quot;, into = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep=&quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;))
nuclearPeakUs=read.table(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.fc&quot;, header = T, stringsAsFactors = F) %&gt;% separate(chrom, sep = &quot;:&quot;, into = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep=&quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;))



write.table(totalPeakUs[,7:dim(totalPeakUs)[2]], file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.CountsOnly&quot;,quote=FALSE, col.names = F, row.names = F)

write.table(nuclearPeakUs[,7:dim(nuclearPeakUs)[2]], file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.CountsOnly&quot;,quote=FALSE, col.names = F, row.names = F)</code></pre>
<p>convertCount2Numeric_noMP_GeneLocAnno.py</p>
<pre class="bash"><code>def convert(infile, outfile):
  final=open(outfile, &quot;w&quot;)
  for ln in open(infile, &quot;r&quot;):
    line_list=ln.split()
    new_list=[]
    for i in line_list:
      num, dem = i.split(&quot;/&quot;)
      if dem == &quot;0&quot;:
        perc = &quot;0.00&quot;
      else:
        perc = int(num)/int(dem)
        perc=round(perc,2)
        perc= str(perc)
      new_list.append(perc)
    final.write(&quot;\t&quot;.join(new_list)+ &#39;\n&#39;)
  final.close()
  
convert(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.CountsOnly&quot;,&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.CountsOnlyNumeric.txt&quot;)


convert(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.CountsOnly&quot;,&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.CountsOnlyNumeric.txt&quot;)
</code></pre>
<p>Pull these into R here so I can filter peaks with 5% and understand how many per gene.</p>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.1.1
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ───────────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.0.0     ✔ purrr   0.2.5
✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.3.1
✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>── Conflicts ──────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre class="r"><code>library(ggpubr)</code></pre>
<pre><code>Loading required package: magrittr</code></pre>
<pre><code>
Attaching package: &#39;magrittr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    set_names</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    extract</code></pre>
<pre><code>
Attaching package: &#39;ggpubr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:cowplot&#39;:

    get_legend</code></pre>
<pre class="r"><code>totalPeakUs=read.table(&quot;../data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.fc&quot;, header = T, stringsAsFactors = F) %&gt;% separate(chrom, sep = &quot;:&quot;, into = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep=&quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;))</code></pre>
<pre><code>Warning: Expected 3 pieces. Additional pieces discarded in 4 rows [13402,
13403, 13404, 101569].</code></pre>
<pre class="r"><code>nuclearPeakUs=read.table(&quot;../data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.fc&quot;, header = T, stringsAsFactors = F) %&gt;% separate(chrom, sep = &quot;:&quot;, into = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep=&quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;))</code></pre>
<pre><code>Warning: Expected 3 pieces. Additional pieces discarded in 4 rows [13402,
13403, 13404, 101569].</code></pre>
<pre class="r"><code>ind=colnames(totalPeakUs)[7:dim(totalPeakUs)[2]]
totalPeakUs_CountNum=read.table(&quot;../data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.CountsOnlyNumeric.txt&quot;, col.names = ind)

nuclearPeakUs_CountNum=read.table(&quot;../data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.CountsOnlyNumeric.txt&quot;, col.names = ind)</code></pre>
<pre class="r"><code>#numeric with anno
totalPeak=as.data.frame(cbind(totalPeakUs[,1:6], totalPeakUs_CountNum))
nuclearPeak=as.data.frame(cbind(nuclearPeakUs[,1:6], nuclearPeakUs_CountNum))

#mean
totalPeakUs_CountNum_mean=rowMeans(totalPeakUs_CountNum)
nuclearPeakUs_CountNum_mean=rowMeans(nuclearPeakUs_CountNum)


#append mean to anno
TotalPeakUSMean=as.data.frame(cbind(totalPeakUs[,1:6],totalPeakUs_CountNum_mean))
NuclearPeakUSMean=as.data.frame(cbind(nuclearPeakUs[,1:6],nuclearPeakUs_CountNum_mean))</code></pre>
<p>Filter on the mean</p>
<pre class="r"><code>TotalPeakUSMean_filt=TotalPeakUSMean %&gt;% filter(totalPeakUs_CountNum_mean&gt;=.05) %&gt;% group_by(gene) %&gt;% summarise(Npeaks=n())
totalPeaksPerGene=TotalPeakUSMean_filt %&gt;% group_by(Npeaks) %&gt;% summarise(GenesWithNPeaks=n())


NuclearPeakUSMean_filt=NuclearPeakUSMean %&gt;% filter(nuclearPeakUs_CountNum_mean&gt;=.05) %&gt;% group_by(gene) %&gt;% summarise(Npeaks=n())
nuclearPeaksPerGene=NuclearPeakUSMean_filt %&gt;% group_by(Npeaks) %&gt;% summarise(GenesWithNPeaks=n())
nuclearPeaksPerGene$GenesWithNPeaks=as.integer(nuclearPeaksPerGene$GenesWithNPeaks)</code></pre>
<p>Peak num level</p>
<pre class="r"><code>nPeaksBoth=totalPeaksPerGene %&gt;% full_join(nuclearPeaksPerGene, by=&quot;Npeaks&quot;)
colnames(nPeaksBoth)= c(&quot;Peaks&quot;, &quot;Total&quot;, &quot;Nuclear&quot;)
nPeaksBoth$Total= nPeaksBoth$Total %&gt;% replace_na(0)

#melt nPeaksBoth
nPeaksBoth_melt=melt(nPeaksBoth, id.var=&quot;Peaks&quot;)
colnames(nPeaksBoth_melt)= c(&quot;Peaks&quot;, &quot;Fraction&quot;, &quot;Genes&quot;)

#plot
peakUsage5perc=ggplot(nPeaksBoth_melt, aes(x=Peaks, y=Genes, fill=Fraction)) + geom_bar(stat=&quot;identity&quot;, position = &quot;dodge&quot;) + labs(title=&quot;Number of Genes with &gt;5% Peak Usage \n cleaned for mispriming&quot;) + theme(axis.text.y = element_text(size=12),axis.title.y=element_text(size=10,face=&quot;bold&quot;), axis.title.x=element_text(size=12,face=&quot;bold&quot;))+ scale_fill_manual(values=c(&quot;darkviolet&quot;,&quot;deepskyblue3&quot;))  + facet_grid(~Fraction)

peakUsage5perc</code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-20-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/2701515bea9e53173dbdb1b6cd036cb4965b76c8/docs/figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-20-1.png" target="_blank">2701515</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-29
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>This is a similar distribution to the other annotations.</p>
<p>Save this plot:</p>
<pre class="r"><code>ggsave(peakUsage5perc, file=&quot;../output/plots/QC_plots/peakUsage5perc_noMP_geneLocAnno.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<p>Genes covered with these annoations.</p>
<pre class="r"><code>#nuclear
nrow(NuclearPeakUSMean_filt) </code></pre>
<pre><code>[1] 14851</code></pre>
<pre class="r"><code>#total
nrow(TotalPeakUSMean_filt) </code></pre>
<pre><code>[1] 14852</code></pre>
<p>There are more genes in this. We expect this because i have included LINCs.</p>
<p>Look at number of peaks in each set.</p>
<pre class="r"><code>#nuclear  
NuclearPeakUSMean %&gt;% filter(nuclearPeakUs_CountNum_mean&gt;=.05) %&gt;% nrow()</code></pre>
<pre><code>[1] 37370</code></pre>
<pre class="r"><code>#total
TotalPeakUSMean %&gt;% filter(totalPeakUs_CountNum_mean&gt;=.05) %&gt;% nrow()</code></pre>
<pre><code>[1] 33002</code></pre>
<p>There are a few less peaks. This is expected give we cut 15% of the peaks because they are not within 5kb of an annotated gene.</p>
<p>Write out these peaks</p>
<pre class="r"><code>NuclearPeakUSMean_5perc=NuclearPeakUSMean %&gt;% filter(nuclearPeakUs_CountNum_mean&gt;=.05)
write.table(NuclearPeakUSMean_5perc,file=&quot;../data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt&quot;, row.names=F, col.names=F, quote = F)


TotalPeakUSMean_5per= TotalPeakUSMean %&gt;% filter(totalPeakUs_CountNum_mean&gt;=.05) 
write.table(TotalPeakUSMean_5per,file=&quot;../data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt&quot;, row.names=F, col.names=F, quote = F)</code></pre>
<p>I want to filter these peaks so I can rerun the leafcutter analysis and look at some of the results. I transfered these to /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/.</p>
<p>filterPheno_bothFraction_GeneLocAnno_5perc.py</p>
<pre class="bash"><code>#python  

totalokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt&quot;

totalokPeaks5perc={}
for ln in open(totalokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    totalokPeaks5perc[peakname]=&quot;&quot;


nuclearokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt&quot;
nuclearokPeaks5perc={}
for ln in open(nuclearokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    nuclearokPeaks5perc[peakname]=&quot;&quot;
    
    
totalPhenoBefore=open(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.fc&quot;,&quot;r&quot;)
totalPhenoAfter=open(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc&quot;, &quot;w&quot;)
for num, ln in enumerate(totalPhenoBefore):
    if num ==0:
        totalPhenoAfter.write(ln)
    else:  
        id=ln.split()[0].split(&quot;:&quot;)[3].split(&quot;_&quot;)[2]
        if id in totalokPeaks5perc.keys():
            totalPhenoAfter.write(ln)
totalPhenoAfter.close()  

nuclearPhenoBefore=open(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.fc&quot;,&quot;r&quot;)
nuclearPhenoAfter=open(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc&quot;, &quot;w&quot;)
for num, ln in enumerate(nuclearPhenoBefore):
    if num ==0:
        nuclearPhenoAfter.write(ln)
    else:  
        id=ln.split()[0].split(&quot;:&quot;)[3].split(&quot;_&quot;)[2]
        if id in nuclearokPeaks5perc.keys():
            nuclearPhenoAfter.write(ln)
nuclearPhenoAfter.close() </code></pre>
</div>
<div id="diff-iso-in-leafcutter" class="section level2">
<h2>Diff Iso in leafcutter</h2>
<p>Now I can use these peaks to get counts for leafcutter.</p>
<p>I want a file that will have the peaks from total or nuclear.</p>
<p>I am starting here with the SAF file</p>
<p>filternamePeaks5percCov_GeneLocAnno.py</p>
<pre class="bash"><code>assignedPeaks=open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.SAF&quot;,&quot;r&quot;)
outFile=open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.5percCov.SAF&quot;, &quot;w&quot;)

totalokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt&quot;
nuclearokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt&quot;

allPeakOk={}
for ln in open(nuclearokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    peaknum=peakname[4:]
    allPeakOk[peaknum]=&quot;&quot;
for ln in open(totalokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    peaknum=peakname[4:]
    if peaknum not in allPeakOk.keys():
        allPeakOk[peaknum]=&quot;&quot;
        
for i, ln in enumerate(assignedPeaks): 
    if i == 0:
        outFile.write(ln)
    else:
        ID=ln.split()[0]
        peak=ID.split(&quot;:&quot;)[0]
        peak_num=peak[4:]
        if peak_num in allPeakOk.keys():
             outFile.write(ln)
outFile.close()
    </code></pre>
<p>Now I will run feature counts, remembering that i want to look at the same strand as the peaks.</p>
<p>bothFrac_processed_GeneLocAnno_FC.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=bothFrac_processed_GeneLocAnno_FC
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=bothFrac_processed_GeneLocAnno_FC.out
#SBATCH --error=bothFrac_processed_GeneLocAnno_FC.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env


featureCounts -O -a /project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.5percCov.SAF -F SAF -o /project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_processed_GeneLocAnno_bothFrac/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed.fc /project2/gilad/briana/threeprimeseq/data/bam_NoMP_sort/*sort.bam -s 1</code></pre>
<p>Fix header for this:</p>
<p>fix_head_fc_procBothFrac_GeneLocAnno.py</p>
<pre class="bash"><code>#python 

infile= open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_processed_GeneLocAnno_bothFrac/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed.fc&quot;, &quot;r&quot;)
fout = open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_processed_GeneLocAnno_bothFrac/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed_fixed.fc&quot;,&#39;w&#39;)
for line, i in enumerate(infile):
    if line == 1:
        i_list=i.split()
        libraries = i_list[:6]
        print(libraries)
        for sample in i_list[6:]:
            full = sample.split(&quot;/&quot;)[7]
            samp= full.split(&quot;-&quot;)[2:4]
            lim=&quot;_&quot;
            samp_st=lim.join(samp)
            libraries.append(samp_st)
        first_line= &quot;\t&quot;.join(libraries)
        fout.write(first_line + &#39;\n&#39;)
    else :
        fout.write(i)
fout.close()</code></pre>
<p>fc2leafphen_processed_GeneLocAnno.py</p>
<pre class="bash"><code>inFile= open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_processed_GeneLocAnno_bothFrac/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed_fixed.fc&quot;, &quot;r&quot;)
outFile= open(&quot;/project2/gilad/briana/threeprimeseq/data/pheno_DiffIso_processed_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed_forLC.fc&quot;, &quot;w&quot;)

for num, ln in enumerate(inFile):
        if num == 1:
            lines=ln.split()[6:]
            outFile.write(&quot; &quot;.join(lines)+&#39;\n&#39;)
        if num &gt; 1:
            ID=ln.split()[0]
            peak=ID.split(&quot;:&quot;)[0]
            chrom=ID.split(&quot;:&quot;)[1]
            start=ID.split(&quot;:&quot;)[2]
            start=int(start)
            end=ID.split(&quot;:&quot;)[3]
            end=int(end)
            strand=ID.split(&quot;:&quot;)[4]
            gene=ID.split(&quot;:&quot;)[5]
            new_ID=&quot;chr%s:%d:%d:%s&quot;%(chrom, start, end, gene)
            pheno=ln.split()[6:]
            pheno.insert(0, new_ID)
            outFile.write(&quot; &quot;.join(pheno)+&#39;\n&#39;)
            
outFile.close()</code></pre>
<p>subset_diffisopheno_processed_GeneLocAnno.py</p>
<pre class="bash"><code>def main(inFile, outFile, target):
    ifile=open(inFile, &quot;r&quot;)
    ofile=open(outFile, &quot;w&quot;)
    target=int(target)
    for num, ln in enumerate(ifile):
        if num == 0:
            ofile.write(ln)
        else:
            ID=ln.split()[0]
            chrom=ID.split(&quot;:&quot;)[0][3:]
            print(chrom)
            chrom=int(chrom)
            if chrom == target:
                ofile.write(ln)
            
if __name__ == &quot;__main__&quot;:
    import sys

    target = sys.argv[1]
    inFile = &quot;/project2/gilad/briana/threeprimeseq/data/pheno_DiffIso_processed_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed_forLC.fc&quot;
    outFile = &quot;/project2/gilad/briana/threeprimeseq/data/pheno_DiffIso_processed_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed_forLC_%s.txt&quot;%(target)
    main(inFile, outFile, target)</code></pre>
<p>run_subset_diffisopheno_processed_GeneLocAnno.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_subset_diffisopheno_processed_GeneLocAnno
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=run_subset_diffisopheno_processed_GeneLocAnno.out
#SBATCH --error=run_subset_diffisopheno_processed_GeneLocAnno.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

for i in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 
do
python subset_diffisopheno_processed_GeneLocAnno.py  $i 
done</code></pre>
<p>makeLCSampleList_processed_GeneLocAnno.py</p>
<pre class="bash"><code>outfile=open(&quot;/project2/gilad/briana/threeprimeseq/data/pheno_DiffIso_processed_GeneLocAnno/sample_groups.txt&quot;, &quot;w&quot;)
infile=open(&quot;/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_processed_GeneLocAnno_bothFrac/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed.fc&quot;, &quot;r&quot;)

for line, i in enumerate(infile):
    if line == 1:
        i_list=i.split()
        libraries=[]
        for sample in i_list[6:]:
            full = sample.split(&quot;/&quot;)[7]
            samp= full.split(&quot;-&quot;)[2:4]
            lim=&quot;_&quot;
            samp_st=lim.join(samp)
            libraries.append(samp_st)
        for l in libraries:
            if l[-1] == &quot;T&quot;:
                outfile.write(&quot;%s\tTotal\n&quot;%(l))
            else:
                outfile.write(&quot;%s\tNuclear\n&quot;%(l))
    else:
          next
                
outfile.close()</code></pre>
<p>run_leafcutter_ds_bychrom_processed_GeneLocAnno.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_leafcutter_ds_bychrom_processed_GeneLocAnno
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=run_leafcutter_ds_bychrom_processed_GeneLocAnno.out
#SBATCH --error=run_leafcutter_ds_bychrom_processed_GeneLocAnno.err
#SBATCH --partition=bigmem2
#SBATCH --mem=50G
#SBATCH --mail-type=END

module load R

for i in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 
do
Rscript /project2/gilad/briana/davidaknowles-leafcutter-c3d9474/scripts/leafcutter_ds.R --num_threads 4  /project2/gilad/briana/threeprimeseq/data/pheno_DiffIso_processed_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed_forLC_${i}.txt /project2/gilad/briana/threeprimeseq/data/pheno_DiffIso_processed_GeneLocAnno/sample_groups.txt -o /project2/gilad/briana/threeprimeseq/data/diff_iso_processed_GeneLocAnno/TN_diff_isoform_GeneLocAnno_chr${i}.txt 
done</code></pre>
<pre class="bash"><code>awk &#39;{if(NR&gt;1)print}&#39; /project2/gilad/briana/threeprimeseq/data/diff_iso_processed_GeneLocAnno/TN_diff_isoform_GeneLocAnno_chr*.txt_effect_sizes.txt &gt; /project2/gilad/briana/threeprimeseq/data/diff_iso_processed_GeneLocAnno/TN_diff_isoform_GeneLocAnno_AllChrom.txt_effect_sizes.txt

awk &#39;{if(NR&gt;1)print}&#39; /project2/gilad/briana/threeprimeseq/data/diff_iso_processed_GeneLocAnno/TN_diff_isoform_GeneLocAnno_chr*cluster_significance.txt &gt; /project2/gilad/briana/threeprimeseq/data/diff_iso_processed_GeneLocAnno/TN_diff_isoform_GeneLocAnno_AllChrom.txt_cluster_significance.txt</code></pre>
<p>Look at the significant peaks</p>
<pre class="r"><code>diffIso=read.table(&quot;../data/diff_iso_GeneLocAnno/TN_diff_isoform_GeneLocAnno_AllChrom.txt_cluster_significance.txt&quot;, header = F,col.names = c(&quot;status&quot;,   &quot;loglr&quot;,    &quot;df&quot;,   &quot;p&quot;,    &quot;cluster&quot;,  &quot;p.adjust&quot;),stringsAsFactors = F,sep=&quot;\t&quot;) %&gt;% filter(status == &quot;Success&quot;)


diffIso$p.adjust=as.numeric(as.character(diffIso$p.adjust))


qqplot(-log10(runif(nrow(diffIso))), -log10(diffIso$p.adjust),ylab=&quot;-log10 Total Adjusted Leafcutter pvalue&quot;, xlab=&quot;-log 10 Uniform expectation&quot;, main=&quot;Leafcutter differencial isoform analysis between fractions&quot;)
abline(0,1)</code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-35-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/5d371f7dd38859bb9ee2ee7f73d75d9b923dd8f0/docs/figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-35-1.png" target="_blank">5d371f7</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-02-13
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/1def3fb49474592a08e1bf59b50960a3ba595a1c/docs/figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-35-1.png" target="_blank">1def3fb</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-29
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>diffIso_10FDR=diffIso %&gt;% filter(-log10(p.adjust)&gt;1)

diffIso_10FDR_genes=diffIso_10FDR %&gt;% separate(cluster, into = c(&quot;chr&quot;, &quot;gene&quot;), sep=&quot;:&quot;) %&gt;% group_by(gene) %&gt;% tally()</code></pre>
<p>Look at the effect sizes and delta PSI values</p>
<pre class="r"><code>effectsize=read.table(&quot;../data/diff_iso_GeneLocAnno/TN_diff_isoform_GeneLocAnno_AllChrom.txt_effect_sizes.txt&quot;, stringsAsFactors = F, col.names=c(&#39;intron&#39;,  &#39;logef&#39; ,&#39;Nuclear&#39;, &#39;Total&#39;,&#39;deltapsi&#39;))

effectsize$deltapsi=as.numeric(as.character(effectsize$deltapsi))</code></pre>
<pre><code>Warning: NAs introduced by coercion</code></pre>
<pre class="r"><code>effectsize$logef=as.numeric(as.character(effectsize$logef))</code></pre>
<pre><code>Warning: NAs introduced by coercion</code></pre>
<pre class="r"><code>plot(sort(effectsize$deltapsi),main=&quot;Leafcutter delta PSI&quot;, ylab=&quot;Delta PSI&quot;, xlab=&quot;Peak Index&quot;)</code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(sort(effectsize$logef),main=&quot;Leafcutter logef&quot;, ylab=&quot;log ef&quot;, xlab=&quot;Peak Index&quot;)</code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-37-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>effectsize_dpsi= effectsize %&gt;% filter(abs(deltapsi) &gt; .2) 

effectsize_dpsi_gene= effectsize %&gt;% filter(abs(deltapsi) &gt; .2) %&gt;% separate(intron, into=c(&quot;chr&quot;, &#39;start&#39;, &#39;end&#39;,&#39;gene&#39;), sep=&quot;:&quot;) %&gt;% group_by(gene) %&gt;% tally()</code></pre>
<p>Intersect effectsize_dpsi genes and the diffIso_10FDR genes.</p>
<pre class="r"><code>inboth=effectsize_dpsi_gene %&gt;% inner_join(diffIso_10FDR_genes, by=&quot;gene&quot;)</code></pre>
<p>I will look at the top PSI values to check these assignments:</p>
<pre class="r"><code>filterHighPSI=effectsize %&gt;% filter(abs(deltapsi)&gt;.4) %&gt;% arrange(deltapsi)
slice(filterHighPSI, 1:10)</code></pre>
<pre><code>                              intron      logef           Nuclear
1       chr16:81123538:81123627:GCSH -1.7908757 0.724883044705174
2     chr3:25705115:25705187:MIR4442 -1.3243205 0.660590394608791
3         chr6:84007319:84007404:ME1 -1.6868552 0.671530253986571
4  chr19:36806427:36806515:LINC00665 -1.1761709 0.809837765186814
5      chr14:47718556:47718593:MDGA2 -1.1825788 0.659822642518761
6      chr14:51976014:51976057:FRMD6 -1.5675947 0.636611882071359
7  chr20:58647920:58648008:C20orf197 -1.0998440 0.691727788494324
8      chr6:37812035:37812094:ZFAND3 -1.6108539 0.617570611488878
9     chr8:104953022:104953097:RIMS2 -1.3900918 0.611545558430189
10  chr7:147596894:147596932:CNTNAP2 -0.8272763  0.72731101101958
               Total   deltapsi
1  0.151199900846243 -0.5736831
2  0.121029754316441 -0.5395606
3  0.134625920458626 -0.5369043
4   0.28835529406143 -0.5214825
5  0.154119006269487 -0.5057036
6  0.139375449680577 -0.4972364
7  0.199171791192942 -0.4925560
8  0.127058493301525 -0.4905121
9   0.13467815341195 -0.4768674
10 0.259840841892183 -0.4674702</code></pre>
<p>Files to make: -bed file with all 5% peaks (named with gene)</p>
<p>Convert Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.5percCov.SAF to a bed file</p>
<p>Chrom, start, end, name, score, strand</p>
<p>peaksGeneLocAnno_5percSAF2Bed.py</p>
<pre class="bash"><code>inFile=&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.5percCov.SAF&quot;
outFile=open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.5percCov.bed&quot;, &quot;w&quot;)

for i, ln in enumerate(inFile):
  if i &gt;0:
     ID, chrom, start, end, strand = ln.split()
     gene=ID.split(&quot;:&quot;)[5]
     peaknum=ID.split(:)[0]
     newID=gene + &quot;:&quot;+ peaknum
     start_i=int(start)
     end_i=int(end)
     score=&quot;.&quot;
     outFile.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_i, end_i, newID, score, strand))
outFile.close()     </code></pre>
<div id="evaluate-top-differences" class="section level3">
<h3>Evaluate top differences</h3>
<ul>
<li>C20orf197</li>
</ul>
<p>peak72985 - in the three prime UTR<br />
total:0.08948718 nuclear: 0.3607692</p>
<p>the other peak is peak72984 - first exon of the gene<br />
total: 0.39769231 nuclear: 0.1264103</p>
<ul>
<li>GCSH:</li>
</ul>
<p>Total has 3 peaks and nuclear has 2 used at 5%.<br />
peak46139- internal intron peak<br />
Total:0.15743590 Nuclear: 0.6787179</p>
<p>The other common peak is peak46138 and is in the 3’ UTR Total: 0.73333333 Nuclear: 0.2943590</p>
<ul>
<li>ME1<br />
chr6:84007319:84007404-peak 102432 - intronic peak<br />
Total:0.09307692<br />
Nuclear: 0.54974359</li>
</ul>
<p>There are 2 more peaks for this gene. They are in the 3’ UTR</p>
<p>peak 102430- distal<br />
Total:0.31794872 Nuclear:0.17230769</p>
<p>peak 102431 - proximal<br />
Total: 0.43512821<br />
Nuclear: 0.20179487</p>
<ul>
<li>FHIT<br />
Total and nuclear use different peaks and only have 1 with over 5%.</li>
</ul>
<p>Total: peak79937 (0.2492308) - in the 3’ UTR<br />
Nucelar: peak79975- (0.0574359) intronic</p>
<p>There is a lot of coverage in this gene. Many low used peaks that are mostly noise.</p>
<ul>
<li>LINC00665</li>
</ul>
<p>peak57872 -last Intron Total: 0.2538462<br />
Nuclear: 0.7361538</p>
<p>peak57873 upstream intron<br />
total: 0.5923077 Nuclear: 0.2382051</p>
<p>This means the nuclear is the longer version.</p>
<ul>
<li>EBF1</li>
</ul>
<p>peak97306 - intronic<br />
Total: 0.07538462<br />
Nuclear: 0.1443590</p>
<p>This is the only peak used at 5% in nuclear. The total also has usage for 2 3’UTR peaks. peak97237-distal- 0.08512821 peak97240- prox- 0.13358974</p>
<p>this shows there is a lot of noise peaks in this gene</p>
<ul>
<li><p>FRMD6 This is confusing because the genes FRMD6 and FRMD6-AS1 and FRMD6-AS2 overlap.</p></li>
<li><p>CNTNAP2<br />
peak111687- Intronic Total: 0.2510256 Nuclear: 0.72102564</p></li>
</ul>
<p>Most used total peak is peak111684. This peak is upstream in another intron<br />
Total: 0.7371795<br />
Nuclear: 0.22769231</p>
<p>Looks like nuclear uses the longer version here</p>
<ul>
<li><p>MIR7-1 Not that much coverage in this location for the RNA, not sure I trust this</p></li>
<li><p>SLC27A4</p></li>
</ul>
<p>peak121617 3’ UTR of gene<br />
Total: 0.3171795 Nuclear: 0.72923077</p>
<p>Most used total peak is peak121616 total: 0.6269231<br />
nuclear: 0.14461538</p>
<p>This is in the first intron</p>
<p>Looks like the longer version is the nuclear again.</p>
</div>
<div id="evaluate-mid-level-differences" class="section level3">
<h3>Evaluate mid level differences</h3>
<pre class="r"><code>filtermidPSI=effectsize %&gt;% filter(abs(deltapsi)&gt;.1,  abs(deltapsi)&lt;.2 ) %&gt;% arrange(desc(deltapsi))
slice(filtermidPSI, 1:15)</code></pre>
<pre><code>                             intron     logef            Nuclear
1   chr7:116558622:116558788:CAPZA2 1.2845944  0.166180284419304
2    chr5:115167277:115167361:ATG12 0.9098819  0.259166063283374
3   chr2:136102694:136102780:ZRANB3 1.2476055 0.0955688783220675
4  chr19:19223611:19223703:SLC25A42 1.3724496  0.743116509482767
5      chr6:35541367:35541452:FKBP5 0.8631872  0.231509000435571
6       chr11:8413414:8413511:STK33 1.2505709   0.19314125556685
7   chr8:145747318:145747406:LRRC14 0.5786649  0.138310661259947
8    chr14:93306089:93306171:GOLGA5 0.9272530  0.255386508299208
9        chr6:49398073:49398145:MUT 0.5145073  0.293792753148935
10  chr14:74181826:74181916:ELMSAN1 0.7173858  0.327901407093237
11     chr7:72039499:72039585:TYW1B 1.1631075  0.203541290866254
12    chr6:83878048:83878191:DOPEY1 0.8605640  0.183715086476799
13  chr11:19197879:19197962:ZDHHC13 0.7324735  0.263254598861606
14    chr18:23971562:23971647:TAF4B 1.2354332 0.0851874732097812
15  chr13:43681462:43681563:DNAJC15 0.8978046  0.584785689718398
               Total  deltapsi
1  0.366129635692815 0.1999494
2  0.459039198885798 0.1998731
3  0.295437340049888 0.1998685
4  0.942932116943742 0.1998156
5  0.431306079077603 0.1997971
6  0.392894273645649 0.1997530
7  0.338035107525039 0.1997244
8  0.455093847920976 0.1997073
9  0.493491781248813 0.1996990
10 0.527419993916497 0.1995186
11 0.402974660777138 0.1994334
12 0.383135912460267 0.1994208
13  0.46253392311505 0.1992793
14 0.284422574325073 0.1992351
15 0.783952368302737 0.1991667</code></pre>
<ul>
<li>LOC101927905- not a great example because this is not a protein coding gene.</li>
</ul>
<p>peak23703<br />
Total: 0.2471795 Nuclear: 0.1784615</p>
<ul>
<li>LYPLAL1</li>
</ul>
<p>Peak 10544- Proximal 3’ UTR peak<br />
Total:0.60000000 Nuclear: 0.38487179</p>
<p>Peak 10545 - distal 3’ UTR<br />
Total:0.21615385 Nuclear: 0.21564103</p>
<p>Nuclear uses 2 peaks in the second to last intron of the gene.</p>
<ul>
<li>NRIP1</li>
</ul>
<p>peak77305 3’ UTR<br />
total:0.57333333 nuclear:0.32076923</p>
<p>Nuclear fraction has an internal peak used at 0.2800000 that is not used in total</p>
<ul>
<li><p>C15orf41- looked at this gene prior</p></li>
<li><p>DENND6A<br />
Peak79852 - 3’ UTR of the gene (most distal) Total:0.80897436 Nuclear:0.60512821</p></li>
</ul>
<p>Other 3’ UTR peak is 79853<br />
Total:0.07769231 Nucelar 0.06512821</p>
<p>Nuclear has 3 extra internal peaks used at at least 5%</p>
<ul>
<li><p>TPTE2P5 peak31262 one of the first introns of a long gene. This part of the gene overlaps with SUGT1P3<br />
total:0.27897436 nuclear: 0.10025641</p></li>
<li><p>COG4 peak45747- 3; UTR peak<br />
total:0.9135897 nuclear: 0.71179487</p></li>
</ul>
<p>Total has one more peak in an intron- 45748<br />
total:0.0700000 nuclear:0.22974359</p>
<p>Other nucelar peak is peak45749 at 0.05897436</p>
<p>Example of internal peak usage in nuclear</p>
<ul>
<li>GOT2 Peak 45272 3’ UTR<br />
Total:0.9528205 Nuclear: 0.7635897</li>
</ul>
<p>Other peak used in nuclear is in second to last intron</p>
<p>Peak 45274- nuclear 0.2307692</p>
<p>internal peak usage in nuclear</p>
<ul>
<li>TMEM14B<br />
Peak 98769 first 3’ UTR of the gene (this gene has 2 transcripts with 2 different 3’ UTRs. looks like this is the only one we get usage for) Total:0.6343590 Nuclear: 0.49512821</li>
</ul>
<p>Peal98768 is only used in the nuclear at 5% or above- it is used at 0.22743590</p>
<p>This means we get internal 3’ in this for the nuclear.</p>
<ul>
<li>MRE11</li>
</ul>
<p>peak 21611 3’ UTR peak<br />
Total: 0.47102564 nuclear: 0.27512821</p>
<p>There is a run on peak in both as well (beyond annotated UTR)</p>
<p>peak 21608 Total: 0.35230769 nucelar: 0.52256410</p>
<p>The other used peak for both fractions is 21616 and is &lt;10% in both</p>
<ul>
<li>IFT52</li>
</ul>
<p>peak 72169- 3’ UTR (only total peak) total:0.9548718 nucelar: 0.7610256</p>
<p>Other nuclear peak:<br />
peak 72168 in an intron<br />
nuclear: 0.2133333</p>
<p>Example of internal stopage in nuclear</p>
<ul>
<li>HIP1</li>
</ul>
<p>peak 108486 3’ UTR total: 0.83794872 nuclear 0.6253846</p>
<p>More internal peak usage in nuclear</p>
<ul>
<li>DZANK1<br />
Peak 71173 - 3’ UTR<br />
total:0.66358974 nuclear: 0.52717949</li>
</ul>
<p>5 other peaks in nuclear, 3 other in total.</p>
<ul>
<li>IRF2<br />
peak 90870<br />
total: 0.39051282 nuclear: 0.18538462</li>
</ul>
<p>the nuclear fraction uses peak90889 at 0.29025641. this is in the first intron on the gene</p>
<ul>
<li>KIF14<br />
peak9668 3’ UTR<br />
total:0.65743590 nuclear:0.5341026</li>
</ul>
<p>Total actually has one extra peak used here. This is only about 5%.</p>
<p>Other peaks:<br />
peak9674- second intron total: 0.07692308 nuclear: 0.1538462</p>
<p>peak9671-intronic<br />
total:0.11615385 nuclear: 0.1779487</p>
<p>Here is looks like more internal peaks in nuclear.</p>
</div>
</div>
<div id="run-qtls-with-these-annotations" class="section level2">
<h2>Run QTLs with these annotations</h2>
<p>/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs</p>
<pre class="bash"><code>#zip file 
gzip filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc
gzip filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc


module load python
#leafcutter script
python /project2/gilad/briana/threeprimeseq/code/prepare_phenotype_table.py filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz

python /project2/gilad/briana/threeprimeseq/code/prepare_phenotype_table.py filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz




#source activate three-prime-env
 sh filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz_prepare.sh
sh filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz_prepare.sh


#keep only 2 PCs
head -n 3 filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.PCs &gt; filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.2PCs
head -n 3 filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.PCs &gt; filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.2PCs</code></pre>
<p>Use same sample list:</p>
<p>“/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript/SAMPLE.txt”</p>
<div id="nominal" class="section level3">
<h3>Nominal</h3>
<p>APAqtl_nominal_GeneLocAnno_noMP_5percUsage.sh</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=APAqtl_nominal_GeneLocAnno_noMP_5percUsage
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=APAqtl_nominal_GeneLocAnno_noMP_5percUsage.out
#SBATCH --error=APAqtl_nominal_GeneLocAnno_noMP_5percUsage.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

for i in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 
do
/home/brimittleman/software/bin/FastQTL/bin/fastQTL.static --vcf /project2/gilad/briana/YRI_geno_hg19/chr$i.dose.filt.vcf.gz --cov /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.2PCs --bed /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_chr$i.gz --out /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_chr$i.nominal.out --chunk 1 1  --window 5e5 --include-samples /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript/SAMPLE.txt
done


for i in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 
do
/home/brimittleman/software/bin/FastQTL/bin/fastQTL.static --vcf /project2/gilad/briana/YRI_geno_hg19/chr$i.dose.filt.vcf.gz --cov /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.2PCs --bed /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_chr$i.gz --out /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_chr$i.nominal.out --chunk 1 1  --window 5e5 --include-samples /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript/SAMPLE.txt
done
</code></pre>
</div>
<div id="permuted" class="section level3">
<h3>Permuted</h3>
<p>APAqtl_perm_GeneLocAnno_noMP_5percUsage.sh</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=APAqtl_perm_GeneLocAnno_noMP_5percUsage
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=APAqtl_perm_GeneLocAnno_noMP_5percUsage.out
#SBATCH --error=APAqtl_perm_GeneLocAnno_noMP_5percUsage.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

for i in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 
do
/home/brimittleman/software/bin/FastQTL/bin/fastQTL.static  --permute 1000 --vcf /project2/gilad/briana/YRI_geno_hg19/chr$i.dose.filt.vcf.gz --cov /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.2PCs --bed /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_chr$i.gz --out /project2/gilad/briana/threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear_fixed.pheno_5perc.fc.gz.qqnorm_chr$i.perm.out --chunk 1 1  --window 5e5 --include-samples /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript/SAMPLE.txt
done


for i in 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 
do
/home/brimittleman/software/bin/FastQTL/bin/fastQTL.static   --permute 1000 --vcf /project2/gilad/briana/YRI_geno_hg19/chr$i.dose.filt.vcf.gz --cov /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.2PCs --bed /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_chr$i.gz --out /project2/gilad/briana/threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_chr$i.perm.out --chunk 1 1  --window 5e5 --include-samples /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript/SAMPLE.txt
done
</code></pre>
<p>APAqtlpermCorrectQQplot_GeneLocAnno_noMP_5perUs.R</p>
<pre class="r"><code>library(dplyr)


##total results
tot.perm= read.table(&quot;/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total.fixed.pheno_5perc_permRes.txt&quot;,head=F, stringsAsFactors=F, col.names = c(&quot;pid&quot;, &quot;nvar&quot;, &quot;shape1&quot;, &quot;shape2&quot;, &quot;dummy&quot;, &quot;sid&quot;, &quot;dist&quot;, &quot;npval&quot;, &quot;slope&quot;, &quot;ppval&quot;, &quot;bpval&quot;))

#BH correction
tot.perm$bh=p.adjust(tot.perm$bpval, method=&quot;fdr&quot;)

#plot qqplot
png(&quot;/project2/gilad/briana/threeprimeseq/output/plots/qqplot_total_APAperm_GeneLocAnno_noMP_5percCov.png&quot;) 
qqplot_total= qqplot(-log10(runif(nrow(tot.perm))), -log10(tot.perm$bpval),ylab=&quot;-log10 Total permuted pvalue&quot;, xlab=&quot;Uniform expectation&quot;, main=&quot;Total permuted pvalues for all snps\n Gene Loc Anno&quot;)
abline(0,1)
dev.off()

#write df with BH  

write.table(tot.perm, file = &quot;/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total.fixed.pheno_5perc_permResBH.txt&quot;, col.names = T, row.names = F, quote = F)

##nuclear results  


nuc.perm= read.table(&quot;/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear.fixed.pheno_5perc_permRes.txt&quot;,head=F, stringsAsFactors=F, col.names = c(&quot;pid&quot;, &quot;nvar&quot;, &quot;shape1&quot;, &quot;shape2&quot;, &quot;dummy&quot;, &quot;sid&quot;, &quot;dist&quot;, &quot;npval&quot;, &quot;slope&quot;, &quot;ppval&quot;, &quot;bpval&quot;))
nuc.perm$bh=p.adjust(nuc.perm$bpval, method=&quot;fdr&quot;)


#plot qqplot
png(&quot;/project2/gilad/briana/threeprimeseq/output/plots/qqplot_nuclear_APAperm_GeneLocAnno_noMP_5percCov.png&quot;) 
qqplot(-log10(runif(nrow(nuc.perm))), -log10(nuc.perm$bpval),ylab=&quot;-log10 Nuclear permuted pvalue&quot;, xlab=&quot;Uniform expectation&quot;, main=&quot;Nuclear permuted pvalues for all snps \n Gene Loc Anno&quot;)
abline(0,1)
dev.off()

# write df with BH
write.table(nuc.perm, file = &quot;/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear.fixed.pheno_5perc_permResBH.txt&quot;, col.names = T, row.names = F, quote = F)</code></pre>
<p>run_APAqtlpermCorrectQQplot_GeneLocAnno_noMP_5perUs.sh</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=run_APAqtlpermCorrectQQplot_GeneLocAnno_noMP_5perUs
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=run_APAqtlpermCorrectQQplot_GeneLocAnno_noMP_5perUs.out
#SBATCH --error=run_APAqtlpermCorrectQQplot_GeneLocAnno_noMP_5perUs.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env


Rscript APAqtlpermCorrectQQplot_GeneLocAnno_noMP_5perUs.R </code></pre>
<p>Pull these in:</p>
<pre class="r"><code>totQTLs=read.table(&quot;../data/perm_QTL_GeneLocAnno_noMP_5percov/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total.fixed.pheno_5perc_permResBH.txt&quot;, stringsAsFactors = F, header=T)

Sig_TotQTLs= totQTLs %&gt;% filter(-log10(bh)&gt;=1)
nrow(Sig_TotQTLs)</code></pre>
<pre><code>[1] 363</code></pre>
<pre class="r"><code>nucQTLs=read.table(&quot;../data/perm_QTL_GeneLocAnno_noMP_5percov/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear.fixed.pheno_5perc_permResBH.txt&quot;, stringsAsFactors = F, header=T)

Sig_NucQTLs= nucQTLs %&gt;% filter(-log10(bh)&gt;=1)
nrow(Sig_NucQTLs)</code></pre>
<pre><code>[1] 623</code></pre>
</div>
</div>
<div id="validatation-for-these-assignments" class="section level2">
<h2>Validatation for these assignments</h2>
<div id="distance-to-end-of-transcription" class="section level3">
<h3>Distance to end of transcription</h3>
<p>I want to compute some QC metrics for these peaks. First, I will compute the distance from the center of each peak to the end of TXN for the gene. I can get this from the /project2/gilad/briana/genome_anotation_data/RefSeq_annotations/Transcript2GeneName.dms file. With this file I can make a dictionary with the gene and end of txn. I will use the longest version by updating the dictionary if a key comes up again. For + strand its the furthest end, for the - strand its the first start.</p>
<p>GeneTXNEnd.py</p>
<pre class="bash"><code>TXN2Gene_file=&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/Transcript2GeneName.dms&quot;
fout=open(&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/GeneTXNEnd.txt&quot;, &quot;w&quot;)

        

for ln in open(TXN2Gene_file,&quot;r&quot;):
   gene=ln.split()[12]
   strand=ln.split()[3]
   txStart=ln.split()[4]
   txEnd=ln.split()[5]
   if strand == &quot;+&quot;:
      val=txEnd
   else:
      val=txStart
   fout.write(&quot;%s\t%s\t%s\n&quot;%(gene,strand, val))
fout.close()</code></pre>
<p>Run uniq on this file because many genes had multiple transcripts with the same ends.</p>
<pre class="bash"><code>uniq GeneTXNEnd.txt &gt; GeneTXNEndUniq.txt</code></pre>
<p>Now I can get just the longest trx with a dictionary and connect them to my peaks. I want the final file to have the dist to the end. First I need to get the average of peak/center of the peak. I will always do End - Peakcenter. I will put the strand of the gene, this is opposite of the peak.</p>
<p>GetDistTXNend2Peak.py</p>
<pre class="bash"><code>endF=open(&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/GeneTXNEndUniq.txt&quot;,&quot;r&quot;)

endDic={}

for i,ln in enumerate(endF):
    if i &gt; 0:
        gene, strand, end = ln.split()
        if gene not in endDic.keys():
            endDic[gene]=int(end)
        else:
            if strand == &quot;+&quot;:
                if endDic[gene] &gt; int(end):
                    endDic[gene]=int(end)
                else:
                    continue
            else: 
                if endDic[gene]&lt; int(end):
                    endDic[gene]=int(end)
                else:
                    continue

peakFile=&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.5percCov.bed&quot;

distPeak2EndTXN=open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/distPeak2EndTXN.txt&quot;,&quot;w&quot;)

n=0
for ln in open(peakFile, &quot;r&quot;):
    peakS=ln.split()[5]
    if peakS == &quot;+&quot;:
        strand = &quot;-&quot;
    else:
        strand = &quot;+&quot;
    start=ln.split()[1]
    start=int(start)
    end=ln.split()[2]
    end=int(end)
    mid=(start+end)/2
    gene=ln.split()[3].split(&quot;:&quot;)[0]
    peak=ln.split()[3].split(&quot;:&quot;)[1]
    if gene in endDic.keys():
        txnEnd=endDic[gene]
        distance=txnEnd-mid
        distPeak2EndTXN.write(&quot;%s\t%s\t%d\t%s\n&quot;%(peak, gene, distance, strand))
    else:
        print(&quot;not in file&quot;)
        n+=1
print(n)
distPeak2EndTXN.close()
        
</code></pre>
<p>bring to my computer:</p>
<pre class="r"><code>distTXN2Peak=read.table(&quot;../data/DistTXN2Peak_genelocAnno/distPeak2EndTXN.txt&quot;, col.names = c(&quot;Peak&quot;, &quot;name2&quot;, &quot;Distance&quot;, &quot;Gene_Strand&quot;),stringsAsFactors = F)</code></pre>
<p>Plot the ditribution of the ditances.</p>
<pre class="r"><code>summary(distTXN2Peak$Distance)</code></pre>
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
-43480518     -2617        22   -107542      2638   2049472 </code></pre>
<pre class="r"><code>ggplot(distTXN2Peak, aes(x=Distance)) + geom_density()</code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-52-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Look at furthest ones in igv.</p>
<pre class="r"><code>distTXN2Peak %&gt;% arrange(Distance) %&gt;% slice(1:15)</code></pre>
<pre><code>         Peak    name2  Distance Gene_Strand
1   peak50119   KANSL1 -43480518           -
2   peak50114   KANSL1 -43468229           -
3   peak50108   KANSL1 -43436524           -
4   peak50104   KANSL1 -43394661           -
5   peak50103   KANSL1 -43381414           -
6   peak50099   KANSL1 -43351333           -
7   peak50098   KANSL1 -43346626           -
8   peak50139      NSF -43257341           +
9   peak50138      NSF -43255873           +
10  peak50136      NSF -43231499           +
11  peak50135      NSF -43222606           +
12  peak50132      NSF -43212672           +
13 peak100477 HLA-DPB2 -28772478           +
14 peak100490 MIR219A1 -28770854           +
15 peak100352 HLA-DQA2 -28767889           +</code></pre>
<p>KANSL1</p>
<p>Gene comes us more than once.<br />
KANSL1 - 44107281 KANSL1 + 760700</p>
<p>NSF</p>
<p>Gene comes up more than once NSF + 44834830<br />
NSF + 1577443</p>
<p>HLA-DPB2- Many transcripts around 4383650 but one around 33096890 HLA-DPB2 + 33096890<br />
HLA-DPB2 + 4383650<br />
HLA-DPB2 + 4540678<br />
HLA-DPB2 + 4378196<br />
HLA-DPB2 + 4421814<br />
HLA-DPB2 + 4329060<br />
HLA-DPB2 + 4554153<br />
HLA-DPB2 + 4577055</p>
<p>MIR219A1: Many transcripts at one location and 1 at other<br />
MIR219A1 + 33175721 MIR219A1 + 4619551 MIR219A1 + 4457028 MIR219A1 + 4649482 MIR219A1 + 4407948 MIR219A1 + 4633041</p>
<p>HLA-DQA2: Many transcripts at one location and 1 at other<br />
HLA-DQA2 + 32714664 HLA-DQA2 + 4002942 HLA-DQA2 + 4160566 HLA-DQA2 + 3997229 HLA-DQA2 + 4051847 HLA-DQA2 + 3946758 HLA-DQA2 + 4171833 HLA-DQA2 + 4146332</p>
<p>Look at the other end of the Dist.</p>
<pre class="r"><code>distTXN2Peak %&gt;% arrange(desc(Distance)) %&gt;% slice(1:15)</code></pre>
<pre><code>         Peak     name2 Distance Gene_Strand
1   peak71042   MACROD2  2049472           +
2   peak71043   MACROD2  1661637           +
3  peak101832       EYS  1576814           -
4   peak32698      GPC5  1409165           +
5   peak32700      GPC5  1389871           +
6   peak32701      GPC5  1377331           +
7   peak32702      GPC5  1367899           +
8   peak32703      GPC5  1347401           +
9  peak101837       EYS  1332157           -
10  peak80287     ROBO2  1321995           +
11  peak32704      GPC5  1320325           +
12  peak80288     ROBO2  1305390           +
13  peak62336 LOC730100  1284160           +
14 peak111684   CNTNAP2  1253090           +
15  peak80293     ROBO2  1213859           +</code></pre>
<p>MACROD2 + 16033842</p>
<p>EYS - 64429875 EYS - 66039168 EYS - 66044805</p>
<p>Look at distribution of the absolute value:</p>
<pre class="r"><code>distTXN2Peak =distTXN2Peak %&gt;% mutate(AbsDist=abs(Distance))

summary(distTXN2Peak$AbsDist)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
       0       84     2626   134050    20651 43480518 </code></pre>
<pre class="r"><code>distTXN2PeakPlot=ggplot(distTXN2Peak, aes(x=AbsDist + 1)) + geom_density() + scale_x_log10() + labs(x=&quot;Absolute Distance between end of Transcription and center of Peak&quot;, title=&quot;Distribution of transcription to peak absolute distance&quot;) 
distTXN2PeakPlot</code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-55-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Make sure this is not different by strand</p>
<pre class="r"><code>ggplot(distTXN2Peak, aes(x=AbsDist, by=Gene_Strand, fill=Gene_Strand)) + geom_density(alpha=.5) + scale_x_log10()</code></pre>
<pre><code>Warning: Transformation introduced infinite values in continuous x-axis</code></pre>
<pre><code>Warning: Removed 4 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-56-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-56-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/a70015c1d8d14b738c8c8ac74fe9efe5495f5eb4/docs/figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-56-1.png" target="_blank">a70015c</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-31
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>I want to add to this the average gene length. I can get this from the txn start to txn end. Start is column 5 and end is column 6. I will filter for genes in the peaks.</p>
<pre class="r"><code>txnanno=read.table(&quot;../data/RefSeq_annotations/Transcript2GeneName.dms&quot;, header=T,stringsAsFactors = F) %&gt;% mutate(length=abs(txEnd-txStart)) %&gt;% semi_join(distTXN2Peak, by=&quot;name2&quot;)

mean(txnanno$length)</code></pre>
<pre><code>[1] 60808.79</code></pre>
<p>Now I want to add this value as a verticle line</p>
<pre class="r"><code>distTXN2PeakPlot=distTXN2PeakPlot+  geom_vline(xintercept=mean(txnanno$length), col=&quot;red&quot;) + annotate(&quot;text&quot;, x=1000000, y=.4, label=&quot;Average transcript length \n for genes in peaks&quot;, col=&#39;red&#39;)
distTXN2PeakPlot</code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-58-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation-with-rna-seq" class="section level3">
<h3>Correlation with RNA seq</h3>
<p>For this I need the counts table and the kalisto data</p>
<pre class="r"><code>library(tximport)

tx2gene=read.table(&quot;../data/RNAkalisto/ncbiRefSeq.txn2gene.txt&quot; ,header= F, sep=&quot;\t&quot;, stringsAsFactors = F)

txi.kallisto.tsv &lt;- tximport(&quot;../data/RNAkalisto/abundance.tsv&quot;, type = &quot;kallisto&quot;, tx2gene = tx2gene,countsFromAbundance=&quot;lengthScaledTPM&quot; )</code></pre>
<pre><code>Note: importing `abundance.h5` is typically faster than `abundance.tsv`</code></pre>
<pre><code>reading in files with read_tsv</code></pre>
<pre><code>1 
removing duplicated transcript rows from tx2gene
transcripts missing from tx2gene: 99
summarizing abundance
summarizing counts
summarizing length</code></pre>
<p>Count file: /project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_processed_GeneLocAnno_bothFrac/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed.fc (this has total and nuclear ) I can pull this in a filter on rows with T.</p>
<pre class="r"><code>peakCounts=read.table(&quot;../data/PeakCounts_noMP_genelocanno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_sm_quant_processed.fc&quot;, header=T, stringsAsFactors = F)[,1:8] %&gt;% separate(Geneid, into=c(&quot;peak&quot;, &quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;gene&quot;), sep=&quot;:&quot;) %&gt;% select(gene, X.project2.gilad.briana.threeprimeseq.data.bam_NoMP_sort.YL.SP.18486.T.combined.sort.noMP.sort.bam) %&gt;% filter(X.project2.gilad.briana.threeprimeseq.data.bam_NoMP_sort.YL.SP.18486.T.combined.sort.noMP.sort.bam&gt;10) %&gt;%  group_by(gene) %&gt;% summarize(GeneSum=sum(X.project2.gilad.briana.threeprimeseq.data.bam_NoMP_sort.YL.SP.18486.T.combined.sort.noMP.sort.bam)) %&gt;% mutate(GeneSumNorm=GeneSum/10.8) </code></pre>
<pre class="r"><code>TXN_abund=as.data.frame(txi.kallisto.tsv$abundance) %&gt;% rownames_to_column(var=&quot;gene&quot;)
colnames(TXN_abund)=c(&quot;gene&quot;, &quot;TPM&quot;)

TXN_Gene=TXN_abund %&gt;% inner_join(peakCounts,by=&quot;gene&quot;)</code></pre>
<pre class="r"><code>TXN_Gene=TXN_Gene %&gt;% filter(TPM&gt;0) %&gt;% filter(GeneSumNorm&gt;0)


summary(lm(log10(TPM)~log10(GeneSumNorm),TXN_Gene))</code></pre>
<pre><code>
Call:
lm(formula = log10(TPM) ~ log10(GeneSumNorm), data = TXN_Gene)

Residuals:
    Min      1Q  Median      3Q     Max 
-7.4932 -0.2517  0.0270  0.2906  3.6180 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)        0.342913   0.010367   33.08   &lt;2e-16 ***
log10(GeneSumNorm) 0.768062   0.007915   97.04   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.503 on 10596 degrees of freedom
Multiple R-squared:  0.4705,    Adjusted R-squared:  0.4705 
F-statistic:  9416 on 1 and 10596 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>corr_18486Tot=ggplot(TXN_Gene, aes(x=log10(TPM), y= log10(GeneSumNorm))) + geom_point() + labs(title=&quot;Total&quot;, x=&quot;log10 RNA seq TPM&quot;, y=&quot;log10 Peak count sum per gene&quot;)+ geom_smooth(aes(x=log10(TPM),y=log10(GeneSumNorm)),method = &quot;lm&quot;) + annotate(&quot;text&quot;,x=0, y=5,label=&quot;R2=.47&quot;) +geom_density2d(na.rm = TRUE, size = 1, colour = &#39;red&#39;) 

#+ geom_text(aes(label=gene),hjust=0, vjust=0)
       
corr_18486Tot </code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-62-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This is in line with the estimates from when I filtered out close genes and all of that. (<a href="https://brimittleman.github.io/threeprimeseq/InvestigatePeak2GeneAssignment.html" class="uri">https://brimittleman.github.io/threeprimeseq/InvestigatePeak2GeneAssignment.html</a>)</p>
</div>
<div id="where-are-the-peaks" class="section level3">
<h3>Where are the peaks</h3>
<p>I want to look at how many peaks fall into each of the gene assignment conditions. 5’ UTR<br />
Coding Exon<br />
Intron<br />
3’ UTR<br />
(downstream 5000)-downstream proximal region</p>
<p>processGenLocPeakAnno2SAF_withAnno.py</p>
<pre class="bash"><code>inFile=&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLoc.bed&quot;
outFile=open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed_withAnno.SAF&quot; , &quot;w&quot;)

outFile.write(&quot;GeneID\tChr\tStart\tEnd\tStrand\n&quot;)
for ln in open(inFile, &quot;r&quot;):
    chrom, start, end, peak, cov, strand, score, anno = ln.split()
    if anno==&quot;.&quot;: 
        continue  
    anno_lst=anno.split(&quot;,&quot;)
    if len(anno_lst)==1:
        gene=anno_lst[0].split(&quot;:&quot;)[1]
        print(&quot;1 gene&quot;)
        anno=anno_lst[0].split(&quot;:&quot;)[0]
        peak_i=int(peak)
        start_i=int(start)
        end_i=int(end)
        ID=&quot;peak%d:%s:%d:%d:%s:%s:%s&quot;%(peak_i, chrom, start_i, end_i, strand, gene,anno)
        outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
    else:
        type_dic={}
        for each in anno_lst:
            type_dic[each.split(&quot;:&quot;)[0]]=each.split(&quot;:&quot;)[1]
        if &quot;utr3&quot; in type_dic.keys():
            gene=type_dic[&quot;utr3&quot;]
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s:utr3&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
        elif &quot;end&quot; in type_dic.keys():
            gene=type_dic[&quot;end&quot;]
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s:end&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
        elif &quot;cds&quot; in type_dic.keys():
            gene=type_dic[&quot;cds&quot;] 
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s:cds&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
        elif &quot;utr5&quot; in type_dic.keys():
            gene=type_dic[&quot;utr5&quot;]
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s:utr5&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))
        elif &quot;intron&quot; in type_dic.keys():
            gene=type_dic[&quot;intron&quot;]
            peak_i=int(peak)
            start_i=int(start)
            end_i=int(end)
            ID=&quot;peak%d:%s:%d:%d:%s:%s:intron&quot;%(peak_i, chrom, start_i, end_i, strand, gene)
            outFile.write(&quot;%s\t%s\t%d\t%d\t%s\n&quot;%(ID, chrom, start_i, end_i, strand))

outFile.close()</code></pre>
<p>I now need to filter to 5%</p>
<p>filternamePeaks5percCov_GeneLocAnno_withAnno.py</p>
<pre class="bash"><code>assignedPeaks=open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed_withAnno.SAF&quot;,&quot;r&quot;)
outFile=open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.5percCov_withAnno.SAF&quot;, &quot;w&quot;)

totalokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt&quot;
nuclearokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt&quot;

allPeakOk={}
for ln in open(nuclearokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    peaknum=peakname[4:]
    allPeakOk[peaknum]=&quot;&quot;
for ln in open(totalokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    peaknum=peakname[4:]
    if peaknum not in allPeakOk.keys():
        allPeakOk[peaknum]=&quot;&quot;
        
for i, ln in enumerate(assignedPeaks): 
    if i == 0:
        outFile.write(ln)
    else:
        ID=ln.split()[0]
        peak=ID.split(&quot;:&quot;)[0]
        peak_num=peak[4:]
        if peak_num in allPeakOk.keys():
             outFile.write(ln)
outFile.close()
    </code></pre>
<pre class="r"><code>peakswAnno=read.table(&quot;../data/PeaksUsed_noMP_5percCov/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.5percCov_withAnno.SAF&quot;, header=T) %&gt;% separate(GeneID, into=c(&quot;Peak&quot;, &quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;gene&quot;, &quot;loc&quot;),sep=&quot;:&quot;) %&gt;% select(Peak, loc) %&gt;% group_by(loc) %&gt;% summarise(Num=n())</code></pre>
<pre class="r"><code>locationOfPeaks=ggplot(peakswAnno, aes(x=loc, y=Num)) + geom_bar(stat=&quot;identity&quot;, fill=&quot;blue&quot;) + labs(x=&quot;Gene Location&quot;, y=&quot;Number of Peaks&quot;, title=&quot;Location distribution for all PAS with 5% Usage&quot;)
locationOfPeaks</code></pre>
<p><img src="figure/ApproachForGeneAssignment.Rmd/unnamed-chunk-66-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(locationOfPeaks, file=&quot;../output/plots/PeakLocationByAnnotation.png&quot;)</code></pre>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tximport_1.8.0  bindrcpp_0.2.2  ggpubr_0.1.8    magrittr_1.5   
 [5] cowplot_0.9.3   reshape2_1.4.3  forcats_0.3.0   stringr_1.3.1  
 [9] dplyr_0.7.6     purrr_0.2.5     readr_1.1.1     tidyr_0.8.1    
[13] tibble_1.4.2    ggplot2_3.0.0   tidyverse_1.2.1 workflowr_1.1.1

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4  haven_1.1.2       lattice_0.20-35  
 [4] colorspace_1.3-2  htmltools_0.3.6   yaml_2.2.0       
 [7] rlang_0.2.2       R.oo_1.22.0       pillar_1.3.0     
[10] glue_1.3.0        withr_2.1.2       R.utils_2.7.0    
[13] modelr_0.1.2      readxl_1.1.0      bindr_0.1.1      
[16] plyr_1.8.4        munsell_0.5.0     gtable_0.2.0     
[19] cellranger_1.1.0  rvest_0.3.2       R.methodsS3_1.7.1
[22] evaluate_0.11     labeling_0.3      knitr_1.20       
[25] broom_0.5.0       Rcpp_0.12.19      scales_1.0.0     
[28] backports_1.1.2   jsonlite_1.5      hms_0.4.2        
[31] digest_0.6.17     stringi_1.2.4     grid_3.5.1       
[34] rprojroot_1.3-2   cli_1.0.1         tools_3.5.1      
[37] lazyeval_0.2.1    crayon_1.3.4      whisker_0.3-2    
[40] pkgconfig_2.0.2   MASS_7.3-50       xml2_1.2.0       
[43] lubridate_1.7.4   assertthat_0.2.0  rmarkdown_1.10   
[46] httr_1.3.1        rstudioapi_0.8    R6_2.3.0         
[49] nlme_3.1-137      git2r_0.23.0      compiler_3.5.1   </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
