<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Characterize apaQTLs</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Characterize apaQTLs</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2/16/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-02-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>threeprimeseq/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree9bc1a3800743858e5820db31d69c35fad97c6af6targetblank9bc1a38a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/9bc1a3800743858e5820db31d69c35fad97c6af6" target="_blank">9bc1a38</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree9bc1a3800743858e5820db31d69c35fad97c6af6targetblank9bc1a38a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/perm_QTL_trans_noMP_5percov/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/4suDataIGV.Rmd
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/EvaleQTLs.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  analysis/verifybam_dubs.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/AllPeak_counts/
    Untracked:  data/ApaQTLs/
    Untracked:  data/ApaQTLs_otherPhen/
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/DistTXN2Peak_genelocAnno/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/LocusZoom_proc/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeakCounts_noMP_5perc/
    Untracked:  data/PeakCounts_noMP_genelocanno/
    Untracked:  data/PeakUsage/
    Untracked:  data/PeakUsage_noMP/
    Untracked:  data/PeakUsage_noMP_GeneLocAnno/
    Untracked:  data/PeaksUsed/
    Untracked:  data/PeaksUsed_noMP_5percCov/
    Untracked:  data/QTL_overlap/
    Untracked:  data/RNAkalisto/
    Untracked:  data/RefSeq_annotations/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/WASP_STAT/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/apaExamp_proc/
    Untracked:  data/apaQTL_examp_noMP/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_GeneLocAnno/
    Untracked:  data/diff_iso_proc/
    Untracked:  data/diff_iso_trans/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/molPheno_noMP/
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov_3UTR/
    Untracked:  data/perm_QTL_diffWindow/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/protAndAPAAndExplmRes.Rda
    Untracked:  data/protAndAPAlmRes.Rda
    Untracked:  data/protAndExpressionlmRes.Rda
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  data/threePrimeSeqMetaData55Ind.txt
    Untracked:  data/threePrimeSeqMetaData55Ind.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  output/deeptools_plots/
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/NewPeakPostMP.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diffIsoAnalysisNewMapping.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/mispriming_approach.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlapMolQTL.opposite.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/peakQCPPlots.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/pipeline_55Ind.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/test.smash.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/9bc1a3800743858e5820db31d69c35fad97c6af6/analysis/characterize_apaQTLs.Rmd" target="_blank">9bc1a38</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
add eif2a lz
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/a20b63cf93370a96d805e6c1e3c5c2bb27a04eec/docs/characterize_apaQTLs.html" target="_blank">a20b63c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/bcce5dfd14928191a48c526894bb2782de40f424/analysis/characterize_apaQTLs.Rmd" target="_blank">bcce5df</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
add examples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/ab45fe833d06dbc7a0393e64700b8bc2c7041ef5/docs/characterize_apaQTLs.html" target="_blank">ab45fe8</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/75085b4069c6dca0c25927d3ce7fbee350ce5c93/analysis/characterize_apaQTLs.Rmd" target="_blank">75085b4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
add example plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/1dde5f9fe2ecf9b852387869a73040073dff209a/docs/characterize_apaQTLs.html" target="_blank">1dde5f9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/65b150e846888d82a396bdd64dc893f548d4f9a4/docs/characterize_apaQTLs.html" target="_blank">65b150e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a77fa79d58ef04e8f8451d17eabaf4edd2685448/analysis/characterize_apaQTLs.Rmd" target="_blank">a77fa79</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
add eQTL pval for overlap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/eea777a604e759c8dac2e4910255cc59d0393e2a/docs/characterize_apaQTLs.html" target="_blank">eea777a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/1b5032e5f42f8b4c91c9ff7e367f5e0dc1434d5a/analysis/characterize_apaQTLs.Rmd" target="_blank">1b5032e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
add chromHMM analysus
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/fe8c55bf8ebacd5c558d2fdcae4d8ffab52e47ce/docs/characterize_apaQTLs.html" target="_blank">fe8c55b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/2d86d89955517a186c92172abc87f00bc397787f/analysis/characterize_apaQTLs.Rmd" target="_blank">2d86d89</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
add pi1 anaysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/beb0d714e5bdb4af3c7bfaa998b39b49cbff51e8/docs/characterize_apaQTLs.html" target="_blank">beb0d71</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/3c1f0490368cd5508664674f57fee5ac5025eb95/analysis/characterize_apaQTLs.Rmd" target="_blank">3c1f049</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
add gwas overlap
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.2.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ──────────────────────────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.0.0     ✔ purrr   0.2.5
✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.4.0
✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 3.5.2</code></pre>
<pre><code>── Conflicts ─────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(qvalue)
library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<p>In this analysis I will look at the apaQTLs to draw biological insight. To do this I will run the following analysis:</p>
<ul>
<li><p>Look at chromatin regions for QTLs (chromHMM)</p></li>
<li><p>Overlap apaQTLs between fractions</p></li>
<li><p>Overlap apaQTLs with GWAS</p></li>
<li><p>QTL examples</p></li>
</ul>
<div id="chromatin-regions-for-qtls" class="section level2">
<h2>Chromatin regions for QTLs</h2>
<p>This will start with the sig QTL bed files:</p>
<ul>
<li><p>/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Nuclear.apaQTLs.sort.bed</p></li>
<li><p>/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Total.apaQTLs.sort.bed</p></li>
</ul>
<p>Creare a python script using pybedtools:</p>
<p>ChromHMM.QTLs.py</p>
<pre class="bash"><code>
import pybedtools 

sigNuc=pybedtools.BedTool(&#39;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Nuclear.apaQTLs.sort.bed&#39;) 

sigTot=pybedtools.BedTool(&#39;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Total.apaQTLs.sort.bed&#39;)

hmm=pybedtools.BedTool(&quot;/project2/gilad/briana/genome_anotation_data/GM12878.chromHMM.sort.bed&quot;)

#map hmm to snps  
Tot_overlapHMM=sigTot.map(hmm, c=4)

Nuc_overlapHMM=sigNuc.map(hmm,c=4)

#save results  

Tot_overlapHMM.saveas(&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Tot_overlapHMM.bed&quot;)

Nuc_overlapHMM.saveas(&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Nuc_overlapHMM.bed&quot;)</code></pre>
<p>Run in three-prime-env</p>
<pre class="r"><code>chromHmm=read.table(&quot;../data/ChromHmmOverlap/chromHMM_regions.txt&quot;, col.names = c(&quot;number&quot;, &quot;name&quot;), stringsAsFactors = F)
TotalOverlapHMM=read.table(&quot;../data/ApaQTLs/Tot_overlapHMM.bed&quot;, col.names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;pid&quot;, &quot;significance&quot;, &quot;strand&quot;, &quot;number&quot;))
TotalOverlapHMM_names=TotalOverlapHMM %&gt;% left_join(chromHmm, by=&quot;number&quot;)

NuclearOverlapHMM=read.table(&quot;../data/ApaQTLs/Nuc_overlapHMM.bed&quot;, col.names=c(&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;pid&quot;, &quot;significance&quot;, &quot;strand&quot;, &quot;number&quot;))
NuclearOverlapHMM_names=NuclearOverlapHMM %&gt;% left_join(chromHmm, by=&quot;number&quot;)


ggplot(TotalOverlapHMM_names, aes(x=name)) + geom_bar() + labs(title=&quot;ChromHMM labels for Total APAQtls&quot; , y=&quot;Number of SNPs&quot;, x=&quot;Region&quot;)+theme(axis.text.x = element_text(angle = 90, hjust = 1))</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ab45fe833d06dbc7a0393e64700b8bc2c7041ef5/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-3-1.png" target="_blank">ab45fe8</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/eea777a604e759c8dac2e4910255cc59d0393e2a/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-3-1.png" target="_blank">eea777a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(NuclearOverlapHMM_names, aes(x=name)) + geom_bar() + labs(title=&quot;ChromHMM labels for Nuclear APAQtls&quot; , y=&quot;Number of SNPs&quot;, x=&quot;Region&quot;)+theme(axis.text.x = element_text(angle = 90, hjust = 1))</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ab45fe833d06dbc7a0393e64700b8bc2c7041ef5/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-3-2.png" target="_blank">ab45fe8</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/eea777a604e759c8dac2e4910255cc59d0393e2a/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-3-2.png" target="_blank">eea777a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Group them to put on plot together</p>
<pre class="r"><code>NuclearOverlapHMM_names_byname= NuclearOverlapHMM_names %&gt;% group_by(name) %&gt;% summarise(Nuclear=n())
TotalOverlapHMM_names_byname= TotalOverlapHMM_names %&gt;% group_by(name) %&gt;% summarise(Total=n())


bothFracHMM= TotalOverlapHMM_names_byname %&gt;% full_join(NuclearOverlapHMM_names_byname, by=&quot;name&quot;)
bothFracHMM$Nuclear= bothFracHMM$Nuclear %&gt;% replace_na(0)</code></pre>
<p>melt and plot</p>
<pre class="r"><code>bothFracHMM_melt= melt(bothFracHMM,id.vars=&quot;name&quot;)
colnames(bothFracHMM_melt)=c(&quot;Region&quot;, &quot;Fraction&quot;, &quot;NQTLs&quot;)

apaQTLsChromregion=ggplot(bothFracHMM_melt, aes(x=Region, y=NQTLs, by=Fraction, fill=Fraction)) + geom_bar(stat=&quot;identity&quot;, position = &quot;dodge&quot;)+ theme(axis.text.x = element_text(angle = 90, hjust = 1)) +scale_fill_manual(values=c(&quot;darkviolet&quot;,&quot;deepskyblue3&quot;))  + labs(title=&quot;apaQTLs by chromatin region&quot;)
apaQTLsChromregion</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ab45fe833d06dbc7a0393e64700b8bc2c7041ef5/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-5-1.png" target="_blank">ab45fe8</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/eea777a604e759c8dac2e4910255cc59d0393e2a/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-5-1.png" target="_blank">eea777a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(apaQTLsChromregion, file=&quot;../output/plots/apaQTLsbyChromHMM.png&quot;, width = 7, height = 5)</code></pre>
<pre class="r"><code>bothFracHMM_prop= bothFracHMM %&gt;% mutate(Total_prop=Total/291) %&gt;% mutate(Nuclear_prop=Nuclear/615) %&gt;% select(-Total,-Nuclear) 

bothFracHMM_prop_melt= melt(bothFracHMM_prop,id.vars=&quot;name&quot;)
colnames(bothFracHMM_prop_melt)=c(&quot;Region&quot;, &quot;Fraction&quot;, &quot;PropQTLs&quot;)


propapaQTLsChromregion=ggplot(bothFracHMM_prop_melt, aes(x=Region, y=PropQTLs, by=Fraction, fill=Fraction)) + geom_bar(stat=&quot;identity&quot;, position = &quot;dodge&quot;)+ theme(axis.text.x = element_text(angle = 90, hjust = 1)) +scale_fill_manual(values=c(&quot;darkviolet&quot;,&quot;deepskyblue3&quot;))  + labs(y=&quot;Proportion of apaQTLs&quot;, title=&quot;Proportion of apaQTLs by chromatin region&quot;)
propapaQTLsChromregion</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ab45fe833d06dbc7a0393e64700b8bc2c7041ef5/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-6-1.png" target="_blank">ab45fe8</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/eea777a604e759c8dac2e4910255cc59d0393e2a/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-6-1.png" target="_blank">eea777a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(propapaQTLsChromregion, file=&quot;../output/plots/proportionapaQTLsbyChromHMM.png&quot;, width = 7, height = 5)</code></pre>
<p>I will need to get matched snps to look for enrichment in these values.</p>
</div>
<div id="overlap-apaqtls-between-fractions." class="section level2">
<h2>Overlap apaQTLs between fractions.</h2>
<p>Goal: Find the nominal pvalue for the significant snp peak pair in oposite fraction. I can make a dictionary with the total and nuclear QTLs then run through the nominal files to get the pvalues:</p>
<p>Start with apa QTLs:</p>
<ul>
<li><p>/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Nuclear.apaQTLs.sort.bed</p></li>
<li><p>/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Total.apaQTLs.sort.bed</p></li>
</ul>
<p>NomResFromOppFrac.py</p>
<pre class="bash"><code>

nucQTLs=&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Nuclear.apaQTLs.sort.bed&quot;  
totQTLs=&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Total.apaQTLs.sort.bed&quot;  

nucNom=&quot;/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt&quot;
totNom=&quot;/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt&quot;

outnuc=&quot;/project2/gilad/briana/threeprimeseq/data/QTL_Overlap/TotQTLs_inNucFractionRes.txt&quot;
outtot=&quot;/project2/gilad/briana/threeprimeseq/data/QTL_Overlap/NucQTLs_inTotFractionRes.txt&quot;


def oppFract(inRes, inQTL, out):
    fout=open(out, &quot;w&quot;)
    qtl_dic={}
    #SNP is key, peak is value
    for ln in open(inQTL,&quot;r&quot;):
        snp=ln.split()[2]
        chrom=ln.split()[0]
        peak=ln.split()[3].split(&quot;:&quot;)[0]
        qtl=str(chrom) + &quot;:&quot; + str(snp)
        if qtl not in qtl_dic.keys():
            qtl_dic[qtl]=[peak]
        else: 
            qtl_dic[qtl].append(peak)
    #print(qtl_dic)
    for ln in open(inRes, &quot;r&quot;):
        pval=ln.split()[3]
        snp=ln.split()[1]
        peak=ln.split()[0].split(&quot;:&quot;)[3].split(&quot;_&quot;)[-1]
        if snp in qtl_dic.keys():
            if peak in qtl_dic[snp]:
                fout.write(&quot;%s\t%s\t%s\n&quot;%(snp, peak, pval))   
    fout.close()
    

oppFract(nucNom, totQTLs,outnuc)  
oppFract(totNom, nucQTLs, outtot)  
                </code></pre>
<p>Run in bash: run_NomResFromOppFrac.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=NomResFromOppFrac
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=NomResFromOppFrac.out
#SBATCH --error=NomResFromOppFrac.err
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env


python NomResFromOppFrac.py</code></pre>
<pre class="r"><code>names=c(&quot;SNP&quot;, &quot;peak&quot;, &quot;pval&quot;)
NucQTLinTot=read.table(&quot;../data/QTL_overlap/NucQTLs_inTotFractionRes.txt&quot;, stringsAsFactors = F, col.names = names)
TotQTLinNuc=read.table(&quot;../data/QTL_overlap/TotQTLs_inNucFractionRes.txt&quot;, stringsAsFactors = F, col.names = names)</code></pre>
<p>Get pi values:</p>
<pre class="r"><code>qvalTot= pi0est(NucQTLinTot$pval, pi0.method = &quot;bootstrap&quot;)
1-qvalTot$pi0</code></pre>
<pre><code>[1] 0.8424242</code></pre>
<pre class="r"><code>qvalNuc= pi0est(TotQTLinNuc$pval, pi0.method = &quot;bootstrap&quot;)
1-qvalNuc$pi0</code></pre>
<pre><code>[1] 0.9197861</code></pre>
<p>plots:</p>
<pre class="r"><code>par(mfrow=c(1,2))
hist(NucQTLinTot$pval,xlab=&quot;Total apaQTL pvalue&quot;, main=&quot;Nuclear apaQTLs \nin Total Fraction&quot;)
text(.6,200, paste(&quot;pi_1=&quot;, round((1-qvalTot$pi0), digit=3), sep=&quot; &quot;))
hist(TotQTLinNuc$pval,xlab=&quot;Nuclear apaQTL pvalue&quot;, main=&quot;Total apaQTLs \nin Nuclear Fraction&quot;)
text(.6,125, paste(&quot;pi_1=&quot;, round((1-qvalNuc$pi0), digit=3), sep=&quot; &quot;))</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/eea777a604e759c8dac2e4910255cc59d0393e2a/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-12-1.png" target="_blank">eea777a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>png(&quot;../output/plots/apaFractionOverlapPi1.png&quot;, width=1000, height = 500)
par(mfrow=c(1,2))
hist(NucQTLinTot$pval,xlab=&quot;Total apaQTL pvalue&quot;, main=&quot;Nuclear apaQTLs \nin Total Fraction&quot;)
text(.6,200, paste(&quot;pi_1=&quot;, round((1-qvalTot$pi0), digit=3), sep=&quot; &quot;))
hist(TotQTLinNuc$pval,xlab=&quot;Nuclear apaQTL pvalue&quot;, main=&quot;Total apaQTLs \nin Nuclear Fraction&quot;)
text(.6,125, paste(&quot;pi_1=&quot;, round((1-qvalNuc$pi0), digit=3), sep=&quot; &quot;))
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<p>This provides evidence for high degree of QTL sharing with increased sharing total to nuclear. This demonstrates to me that there are nuclear QTLs that do not persist in the total fraction. I will want to learn more about these.</p>
</div>
<div id="overlap-with-gwas-catelog" class="section level2">
<h2>Overlap with GWAS catelog</h2>
<p>I did this analysis with the QTLs in the preprocessed 39 individual analysis. I will follow a similar pipeline here. I will find all of the snps in LD with the QTLs then test for these in the GWAS catelog. The pipeline I used to get the LD for all of the snp is shown <a href="apaQTLoverlapGWAS.Rmd">here</a>. The plink files are in /project2/gilad/briana/threeprimeseq/data/GWAS_overlap/. There are both map and ped files.</p>
<p>I can now adapt the subset_plink4QTLs.py file to take the current QTLs list. The file just has the QTLs with the chromosome and position. I can make this and put it in:</p>
<p>/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed</p>
<p>The 50mb QTLs are in /project2/gilad/briana/threeprimeseq/data/ApaQTLs.</p>
<ul>
<li>NuclearapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt<br />
</li>
<li>TotalapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt</li>
</ul>
<p>The QTL snps are in the 6th column.</p>
<pre class="bash"><code>cut -f6 -d&quot; &quot; /project2/gilad/briana/threeprimeseq/data/ApaQTLs/NuclearapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt | uniq &gt; /project2/gilad/briana/threeprimeseq/data/ApaQTLs/NuclearQTLs_uniq_50mb.txt
cut -f6 -d&quot; &quot;  /project2/gilad/briana/threeprimeseq/data/ApaQTLs/TotalapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt | uniq &gt; /project2/gilad/briana/threeprimeseq/data/ApaQTLs/TotalQTLs_uniq_50mb.txt</code></pre>
<p>I can convert these the the way they are in GEU snp files tony made (snp_num_pos)</p>
<p>QTLs2GeuSnps_proc.py</p>
<pre class="bash"><code>tot_in=open(&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/TotalQTLs_uniq_50mb.txt&quot;, &quot;r&quot;)  
nuc_in=open(&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/NuclearQTLs_uniq_50mb.txt&quot;, &quot;r&quot;)

tot_out=open(&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/TotalQTLs_uniq_50mb_GEU.txt&quot;, &quot;w&quot;) 
nuc_out=open(&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/NuclearQTLs_uniq_50mb_GEU.txt&quot;, &quot;w&quot;) 


def fix_file(fin, fout):
  for ln in fin:
    chrom, pos = ln.split(&quot;:&quot;)
    fout.write(&quot;snp_%s_%s&quot;%(chrom,pos))
  fout.close()
  

fix_file(tot_in, tot_out)
fix_file(nuc_in, nuc_out)</code></pre>
<p>subset_plink4QTLs_proc.py</p>
<pre class="bash"><code>def main(genFile, qtlFile, outFile):
  #convert snp file to a list: 
  def file_to_list(file):
    snp_list=[]
    for ln in file:
      snp=ln.strip()
      snp_list.append(snp)
    return(snp_list)

  gen=open(genFile,&quot;r&quot;)
  fout=open(outFile, &quot;w&quot;)
  qtls=open(qtlFile, &quot;r&quot;)
  qtl_list=file_to_list(qtls)
  for ln in gen:
      snp=ln.split()[2]
      if snp in qtl_list:
          fout.write(ln)
  fout.close()
    

if __name__ == &quot;__main__&quot;:
    import sys
    chrom=sys.argv[1]
    fraction=sys.argv[2]
    genFile = &quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap/geu_plinkYRI_LDchr%s.ld&quot;%(chrom)
    outFile= &quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/%sApaQTL_LD/chr%s.%sQTL.LD.geno.ld&quot;%(fraction,chrom,fraction)
    qtlFile= &quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/%sQTLs_uniq_50mb_GEU.txt&quot;%(fraction)
    main(genFile, qtlFile, outFile) </code></pre>
<p>run_subset_plink4QTLs_proc.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name= run_subset_plink4QTLs_proc
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=run_subset_plink4QTLs_proc.out
#SBATCH --error=run_subset_plink4QTLs_proc.err
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env


for i  in {1..22};
do
python subset_plink4QTLs_proc.py ${i} &quot;Total&quot;
done

for i  in {1..22};
do
python subset_plink4QTLs_proc.py ${i} &quot;Nuclear&quot;
done</code></pre>
<p>This added 2446 total snps and 6258 nuclear snps.</p>
<p>Cat and remove indels:</p>
<pre class="bash"><code>cat chr* &gt; allChr.TotalQTL.LD.gene.ld
grep -v indel allChr.TotalQTL.LD.gene.ld &gt; allChr.TotalQTL.LD.gene.ld_noIndel

cat chr* &gt; allChr.NuclearQTL.LD.gene.ld
grep -v indel allChr.NuclearQTL.LD.gene.ld &gt; allChr.NuclearQTL.LD.gene.ld_noIndel</code></pre>
<p>Make these bed files:</p>
<p>makeAlloverlapbed_proc.py</p>
<pre class="bash"><code>
#load files:  

QTL_total=open(&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/TotalQTLs_uniq_50mb_GEU.txt&quot;, &quot;r&quot;)
QTL_nuclear=open(&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/NuclearQTLs_uniq_50mb_GEU.txt&quot;, &quot;r&quot;)
LD_total=open(&quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/TotalApaQTL_LD/allChr.TotalQTL.LD.gene.ld_noIndel&quot;, &quot;r&quot;)
LD_nuclear=open(&quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/NuclearApaQTL_LD/allChr.NuclearQTL.LD.gene.ld_noIndel&quot;, &quot;r&quot;)
outFile= open(&quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/AllOverlapSnps.bed&quot;, &quot;w&quot;)

#function for qtl to bed format
def qtl2bed(fqtl, fraction, fout=outFile):
    for ln in fqtl:
        snp, chrom, pos = ln.split(&quot;_&quot;)
        start=int(pos)-1
        end= int(pos)
        fout.write(&quot;%s\t%d\t%d\tQTL_%s\n&quot;%(chrom, start, end,fraction))

#function for ld to bed format 
def ld2bed(fLD, fraction, fout=outFile):
    for ln in fLD:
        snpID=ln.split()[5]
        snp, chrom, pos= snpID.split(&quot;_&quot;)
        start=int(pos)-1
        end=int(pos)
        fout.write(&quot;%s\t%d\t%d\tLD_%s\n&quot;%(chrom, start, end,fraction))


#I will run each of these for both fractions to get all of the snps in the out file. 


qtl2bed(QTL_nuclear, &quot;Nuclear&quot;)
qtl2bed(QTL_total, &quot;Total&quot;)
ld2bed(LD_nuclear, &quot;Nuclear&quot;)
ld2bed(LD_total, &quot;Total&quot;)


outFile.close()</code></pre>
<p>Sort this:</p>
<pre class="bash"><code>sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/AllOverlapSnps.bed &gt; /project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/AllOverlapSnps.sort.bed</code></pre>
<p>Overlap with GWAS</p>
<p>I can use the overlapSNPsGWAS.py file I created in the previous rendition of this analysis but run it with these files.</p>
<p>run_overlapSNPsGWAS_proc.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_overlapSNPsGWAS_proc
#SBATCH --account=pi-yangili1
#SBATCH --time=5:00:00
#SBATCH --output=run_overlapSNPsGWAS_proc.out
#SBATCH --error=run_overlapSNPsGWAS_proc.err
#SBATCH --partition=broadwl
#SBATCH --mem=10G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env

python overlapSNPsGWAS.py  &quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/AllOverlapSnps.sort.bed&quot; &quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/AllSnps_GWASoverlapped.txt&quot;</code></pre>
<p>Total QTLs overlap: rs3117582 6:31620520</p>
<p>Total LD overlap:</p>
<ul>
<li>rs2282301 1:155868625<br />
</li>
<li>rs3596 12:95696420<br />
</li>
<li>rs2277862 20:34152782<br />
</li>
<li>rs2517713 6:29918099</li>
</ul>
<p>Nuclear QTL overlap:</p>
<p>rs7206971 17:45425115</p>
<p>Nucelar LD overlapL</p>
<ul>
<li>rs10889353 1:63118196<br />
</li>
<li>rs2282301 1:155868625</li>
<li>rs10859871 12:95711876<br />
</li>
<li>rs10133111 14:103377321</li>
<li>rs17382723 2:242053546<br />
</li>
<li>rs2277862 20:34152782<br />
</li>
<li>rs2298428 22:21982892<br />
</li>
<li>rs13160562 5:96111371<br />
</li>
<li>rs29784 5:172595308<br />
</li>
<li>rs2517713 6:29918099<br />
</li>
<li>rs3077 6:33033022</li>
</ul>
<p>Are these eQTLs?</p>
<p>Pull these snps from the eQTLs nominal pvalues:</p>
<p>Process the AllSnps_GWASoverlapped.txt file to have the RS id, the snp, and where it was found.</p>
<p>formatGWASOverlap.py</p>
<pre class="bash"><code>inFile=&quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/AllSnps_GWASoverlapped.txt&quot;

outFile=open(&quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/AllSnps_GWASoverlapped_format.txt&quot;,&quot;w&quot;)

for i in open(inFile, &quot;r&quot;):
   chrom=i.split()[0]
   loc=i.split()[2]
   snp=chrom + &quot;:&quot; + loc
   found=i.split()[3]
   rs=i.split()[7].split(&quot;:&quot;)[0]
   outFile.write(&quot;%s\t%s\t%s\n&quot;%(snp, rs, found))
outFile.close()
</code></pre>
<p>This is 14 uniq snps.</p>
<p>/project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out</p>
<p>I need to go throguh this file and check for the snps. I can do this by making a dictionary of the uniq snps in the overlap then testing the lines ib the nominal file. I want to keep the gene, snp, and pvalue for these associations<br />
getexpPvalOverlap.py</p>
<pre class="bash"><code>overlapSnps=&quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/AllSnps_GWASoverlapped_format.txt&quot;
outF=open(&quot;/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/eQTL_pval_GWASOverlap.txt&quot;,&quot;w&quot;)  
nomRes=&quot;/project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out&quot;  

snps_dic={}
for ln in open(overlapSnps, &quot;r&quot;):
    snp=ln.split()[0]
    if snp not in snps_dic.keys():
        snps_dic[snp]=&quot;&quot;  

for ln in open(nomRes,&quot;r&quot;):
    snp=ln.split()[1]
    if snp in snps_dic.keys():
        gene=ln.split()[0].split(&quot;.&quot;)[0]
        pval=ln.split()[3]
        outF.write(&quot;%s\t%s\t%s\n&quot;%(snp,gene,pval))
outF.close()</code></pre>
<p>I can pull this in and get the lowest pval for each one</p>
<pre class="r"><code>eQTLpvalOverlapGWAS=read.table(&quot;../data/ApaQTLs/eQTL_pval_GWASOverlap.txt&quot;, stringsAsFactors = F, col.names = c(&quot;snp&quot;, &quot;gene&quot;, &quot;pval&quot;))</code></pre>
<pre class="r"><code>eQTLpvalOverlapGWAS_min= eQTLpvalOverlapGWAS %&gt;% group_by(snp) %&gt;% summarise(pvalM=min(pval)) %&gt;% mutate(noSig=ifelse(pvalM&lt;.05, &quot;yes&quot;, &quot;no&quot;))</code></pre>
<p>FIlter by the non sig ones to look at as exmaples:</p>
<pre class="r"><code>eQTLpvalOverlapGWAS_NotSig= eQTLpvalOverlapGWAS_min %&gt;% filter(noSig==&quot;no&quot;)

eQTLpvalOverlapGWAS_NotSig</code></pre>
<pre><code># A tibble: 5 x 3
  snp          pvalM noSig
  &lt;chr&gt;        &lt;dbl&gt; &lt;chr&gt;
1 1:63118196  0.400  no   
2 12:95696420 0.0813 no   
3 12:95711876 0.0813 no   
4 2:242053546 0.0878 no   
5 5:172595308 0.0530 no   </code></pre>
<p>1:63118196 in LD with nuclear QTL (find which it is LD with )</p>
<pre class="bash"><code>grep snp_1_63118196 chr1.NuclearQTL.LD.geno.ld
#snp_1_63018852  
#r2 .91
</code></pre>
<p>12:95696420 LD with total QTL</p>
<pre class="bash"><code>grep snp_12_95696420 TotalApaQTL_LD/chr12.TotalQTL.LD.geno.ld
#snp_12_95672642
#r2 =1  
</code></pre>
<p>12:95711876 LD_Nuclear</p>
<pre class="bash"><code>grep snp_12_95711876 NuclearApaQTL_LD/chr12.NuclearQTL.LD.geno.ld
#snp_12_95688540
#r2 =1  </code></pre>
<p>2:242053546 LD_Nuclear</p>
<pre class="bash"><code>grep snp_2_242053546 NuclearApaQTL_LD/chr2.NuclearQTL.LD.geno.ld
#snp_2_242018994
#r2 = .94  </code></pre>
<p>5:172595308 LD Nuclear</p>
<pre class="bash"><code>grep snp_5_172595308 NuclearApaQTL_LD/chr5.NuclearQTL.LD.geno.ld
#snp_5_172580291
#r2 = .938    </code></pre>
</div>
<div id="exampls-plots" class="section level2">
<h2>Exampls plots:</h2>
<p>In <a href="swarmPlots_QTLs.html">this analysis file</a> I made boxplots for example QTLs. Here I can look at these.</p>
<div id="section" class="section level3">
<h3>1:63018852</h3>
<p>First example 1:63018852 (nuclear QTL) DOCK7 peak5329 (2 GWAS hits for this snp and gene)</p>
<pre class="bash"><code>grep DOCK7 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000116641  </code></pre>
<p>createQTLsnpAPAPhenTable_proc.py</p>
<pre class="bash"><code>def main(PhenFile, GenFile, outFile, snp, peak):
    fout=open(outFile, &quot;w&quot;)
    #Phen=open(PhenFile, &quot;r&quot;)
    Gen=open(GenFile, &quot;r&quot;)
    #get ind and pheno info
    def get_pheno():
      Phen=open(PhenFile, &quot;r&quot;)
      for num, ln in enumerate(Phen):
          if num == 0:
              indiv= ln.split()[4:]
          else:
              id=ln.split()[3].split(&quot;:&quot;)[3]
              peakID=id.split(&quot;_&quot;)[2]
              if peakID == peak:
                  pheno_list=ln.split()[4:]
                  pheno_data=list(zip(indiv,pheno_list))
                  #print(pheno_data)
                  return(pheno_data)
    def get_geno():
      for num, lnG in enumerate(Gen):
          if num == 13:
              Ind_geno=lnG.split()[9:]
          if num &gt;= 14: 
              sid= lnG.split()[2]
              if sid == snp: 
                  gen_list=lnG.split()[9:]
                  allele1=[]
                  allele2=[]
                  for i in gen_list:
                      genotype=i.split(&quot;:&quot;)[0]
                      allele1.append(genotype.split(&quot;|&quot;)[0])
                      allele2.append(genotype.split(&quot;|&quot;)[1])
            #now i have my indiv., phen, allele 1, alle 2     
                  geno_data=list(zip(Ind_geno, allele1, allele2))
                  #print(geno_data)
                  return(geno_data)

    phenotype=get_pheno()
    pheno_df=pd.DataFrame(data=phenotype,columns=[&quot;Ind&quot;, &quot;Pheno&quot;])
    #print(pheno_df)
    genotype=get_geno()
    geno_df=pd.DataFrame(data=genotype, columns=[&quot;Ind&quot;, &quot;Allele1&quot;, &quot;Allele2&quot;])
    #print(geno_df)
    full_df=pd.merge(geno_df, pheno_df, how=&quot;inner&quot;, on=&quot;Ind&quot;)
    full_df.to_csv(fout, sep=&quot;\t&quot;, encoding=&#39;utf-8&#39;, index=False)
    fout.close()
    

if __name__ == &quot;__main__&quot;:
    import sys
    import pandas as pd
    chrom=sys.argv[1]
    snp = sys.argv[2]
    peak = sys.argv[3]
    fraction=sys.argv[4]
    
    PhenFile = &quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.%s.fixed.pheno_5perc.fc.gz.phen_chr%s&quot;%(fraction, chrom)
    GenFile= &quot;/project2/gilad/briana/YRI_geno_hg19/chr%s.dose.filt.vcf&quot;%(chrom)
    outFile = &quot;/project2/gilad/briana/threeprimeseq/data/ApaQTL_proc_examples/qtlSNP_PeakAPA%s.%s%s.txt&quot;%(fraction, snp, peak)
    main(PhenFile, GenFile, outFile, snp, peak)</code></pre>
<p><strong>createQTLsnpMolPhenTable.py changed the output dir</strong></p>
<pre class="bash"><code>python createQTLsnpAPAPhenTable_proc.py 1 1:63018852  peak5329 Total
python createQTLsnpAPAPhenTable_proc.py 1 1:63018852  peak5329 Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;1&quot; &quot;1:63018852&quot; &quot;ENSG00000116641&quot;
</code></pre>
<p>Function to make plots:</p>
<pre class="r"><code>plotQTL_func= function(SNP, peak, gene){
  apaN_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_PeakAPANuclear.&quot;, SNP, peak, &quot;.txt&quot;, sep = &quot;&quot; ), header=T)
  apaT_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_PeakAPATotal.&quot;, SNP, peak, &quot;.txt&quot;, sep = &quot;&quot; ), header=T)
  su30_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_Peak_4su_30_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header = T)
  su60_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_Peak_4su_60_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header=T)
  RNA_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_Peak_RNAseq_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;),header=T)
  RNAg_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_Peak_RNAseqGeuvadis_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header = T)
  ribo_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_Peak_ribo_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;),header=T)
  prot_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_Peak_prot.&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header=T)
  
  ggplot_func= function(file, molPhen,GENE){
    file = file %&gt;% mutate(genotype=Allele1 + Allele2)
    file$genotype= as.factor(as.character(file$genotype))
    plot=ggplot(file, aes(y=Pheno, x=genotype, by=genotype, fill=genotype)) + geom_boxplot(width=.25) + geom_jitter() + labs(y=&quot;Phenotpye&quot;,title=paste(molPhen, GENE, sep=&quot;: &quot;)) + scale_fill_brewer(palette=&quot;Paired&quot;)
    return(plot)
  }
  
  apaNplot=ggplot_func(apaN_file, &quot;Apa Nuclear&quot;, gene)
  apaTplot=ggplot_func(apaT_file, &quot;Apa Total&quot;, gene)
  su30plot=ggplot_func(su30_file, &quot;4su30&quot;,gene)
  su60plot=ggplot_func(su60_file, &quot;4su60&quot;,gene)
  RNAplot=ggplot_func(RNA_file, &quot;RNA Seq&quot;,gene)
  RNAgPlot=ggplot_func(RNAg_file, &quot;RNA Seq Geuvadis&quot;,gene)
  riboPlot= ggplot_func(ribo_file, &quot;Ribo Seq&quot;,gene)
  protplot=ggplot_func(prot_file, &quot;Protein&quot;,gene)
  
  full_plot= plot_grid(apaNplot,apaTplot, RNAplot, protplot,nrow=2)
  return (full_plot)
}</code></pre>
<pre class="r"><code>DOC7_boxplots=plotQTL_func(SNP=&quot;1:63018852&quot;, peak=&quot;peak5329&quot;, gene=&quot;ENSG00000116641&quot;)
DOC7_boxplots</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a20b63cf93370a96d805e6c1e3c5c2bb27a04eec/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-36-1.png" target="_blank">a20b63c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ab45fe833d06dbc7a0393e64700b8bc2c7041ef5/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-36-1.png" target="_blank">ab45fe8</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(DOC7_boxplots, file=&quot;../output/plots/DOC7_boxplots.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<p>Only ApaPlot:</p>
<pre class="r"><code>plotAPAQTL_func= function(SNP, peak, gene){
  apaN_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_PeakAPANuclear.&quot;, SNP, peak, &quot;.txt&quot;, sep = &quot;&quot; ), header=T)
  apaT_file=read.table(paste(&quot;../data/apaExamp_proc/qtlSNP_PeakAPATotal.&quot;, SNP, peak, &quot;.txt&quot;, sep = &quot;&quot; ), header=T)
  
  ggplot_func= function(file, molPhen,GENE){
    file = file %&gt;% mutate(genotype=Allele1 + Allele2)
    file$genotype= as.factor(as.character(file$genotype))
    plot=ggplot(file, aes(y=Pheno, x=genotype, by=genotype, fill=genotype)) + geom_boxplot(width=.25) + geom_jitter() + labs(y=&quot;Phenotpye&quot;,title=paste(molPhen, GENE, sep=&quot;: &quot;)) + scale_fill_brewer(palette=&quot;Paired&quot;)
    return(plot)
  }
  
  apaNplot=ggplot_func(apaN_file, &quot;Apa Nuclear&quot;, gene)
  apaTplot=ggplot_func(apaT_file, &quot;Apa Total&quot;, gene)

  
  full_plot= plot_grid(apaNplot,apaTplot,nrow=1)
  return (full_plot)
}</code></pre>
<pre class="r"><code>plotAPAQTL_func(SNP=&quot;1:63018852&quot;, peak=&quot;peak5329&quot;, gene=&quot;DOCK7&quot;)</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-1">
Past versions of unnamed-chunk-38-1.png
</button>
</p>
<div id="fig-unnamed-chunk-38-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ab45fe833d06dbc7a0393e64700b8bc2c7041ef5/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-38-1.png" target="_blank">ab45fe8</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>FInd this in the catelog:</p>
<p>/project2/gilad/briana/genome_anotation_data/hg19GwasCatalog.sort.bed</p>
<p>1 63118195 63118196 rs10889353:19060911:DOCK7 0.32</p>
<p>In GWAS associated with LDL cholesteral and similar phenotypes</p>
<div id="locus-zoom-for-this-plot" class="section level4">
<h4>Locus zoom for this plot</h4>
<pre class="bash"><code>grep DOCK7 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000116641


grep peak5329  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom_proc/NuclearAPA.peak5329.DOCK7.nomNuc.txt

grep peak5329  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt  &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom_proc/TotalAPA.peak5329.DOCK7.nomTotal.txt

grep ENSG00000116641 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom_proc/RNA.DOCK7.nomTotal.txt

grep ENSG00000116641 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_prot.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom_proc/Prot.DOCK7.nomTotal.txt

 grep ENSG00000116641 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_ribo_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom_proc/Ribo.DOCK7.nomTotal.txt</code></pre>
<pre class="r"><code>APATotal_DOCK7_LZ=read.table(&quot;../data/LocusZoom_proc/TotalAPA.peak5329.DOCK7.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(APATotal_DOCK7_LZ,&quot;../data/LocusZoom_proc/apaTotalDOCK7_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


APANuclear_DOCK7_LZ=read.table(&quot;../data/LocusZoom_proc/NuclearAPA.peak5329.DOCK7.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(APANuclear_DOCK7_LZ,&quot;../data/LocusZoom_proc/apaNuclearDOCK7_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


prot_DOCK7_LZ=read.table(&quot;../data/LocusZoom_proc/Prot.DOCK7.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(prot_DOCK7_LZ,&quot;../data/LocusZoom_proc/ProtDOCK7_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


RNA_DOCK7_LZ=read.table(&quot;../data/LocusZoom_proc/RNA.DOCK7.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(RNA_DOCK7_LZ,&quot;../data/LocusZoom_proc/RNADOCK7_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


ribo_DOCK7_LZ=read.table(&quot;../data/LocusZoom_proc/Ribo.DOCK7.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(ribo_DOCK7_LZ,&quot;../data/LocusZoom_proc/RiboDOCK7_LZ.txt&quot;, col.names = T, row.names = F, quote = F)
#sed -e &#39;s/^/Chr/&#39;</code></pre>
<p>Put these on the locus zoom site.</p>
</div>
</div>
<div id="section-1" class="section level3">
<h3>12:95672642</h3>
<p>peak39092 VEZT 12:95672642</p>
<pre class="bash"><code>grep VEZT /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000028203 
gunzip /project2/gilad/briana/YRI_geno_hg19/chr12.dose.filt.vcf.gz

python createQTLsnpAPAPhenTable_proc.py 12 12:95672642 peak39092 Total
python createQTLsnpAPAPhenTable_proc.py 12 12:95672642   peak39092 Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;12&quot; &quot;12:95672642&quot; &quot;ENSG00000028203&quot;</code></pre>
<p>No protein for this gene</p>
<pre class="r"><code>plotQTL_func(SNP=&quot;12:95672642&quot;, peak=&quot;peak39092&quot;, gene=&quot;ENSG00000028203&quot;)</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a20b63cf93370a96d805e6c1e3c5c2bb27a04eec/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-42-1.png" target="_blank">a20b63c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ab45fe833d06dbc7a0393e64700b8bc2c7041ef5/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-42-1.png" target="_blank">ab45fe8</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="section-2" class="section level3">
<h3>12:95688540</h3>
<p>Nuclear QTL</p>
<p>VEZT peak39092</p>
<p>Same gene peak</p>
</div>
<div id="section-3" class="section level3">
<h3>2:242018994</h3>
<p>Nuclear QTL<br />
MTERF4 peak96962 2:242018994</p>
<pre class="bash"><code>grep MTERF4 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000122085 
gunzip /project2/gilad/briana/YRI_geno_hg19/chr2.dose.filt.vcf.gz

python createQTLsnpAPAPhenTable_proc.py 2 2:242018994 peak96962 Total
python createQTLsnpAPAPhenTable_proc.py 2 2:242018994   peak96962 Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;2&quot; &quot;2:242018994&quot; &quot;ENSG00000122085&quot;</code></pre>
<p>No protein for this gene</p>
<pre class="r"><code>plotQTL_func(SNP=&quot;2:242018994&quot;, peak=&quot;peak96962&quot;, gene=&quot;ENSG00000122085&quot;)</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a20b63cf93370a96d805e6c1e3c5c2bb27a04eec/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-44-1.png" target="_blank">a20b63c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ab45fe833d06dbc7a0393e64700b8bc2c7041ef5/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-44-1.png" target="_blank">ab45fe8</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>No protein data</p>
</div>
<div id="section-4" class="section level3">
<h3>5:172580291</h3>
<p>BNIP1 _peak135178 5:172580291 Nuclear QTL</p>
<pre class="bash"><code>grep BNIP1 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000113734 
gunzip /project2/gilad/briana/YRI_geno_hg19/chr5.dose.filt.vcf.gz

python createQTLsnpAPAPhenTable_proc.py 5 5:172580291 peak135178 Total
python createQTLsnpAPAPhenTable_proc.py 5 5:172580291   peak135178 Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;5&quot; &quot;5:172580291&quot; &quot;ENSG00000113734&quot;</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;5:172580291&quot;, peak=&quot;peak135178&quot;, gene=&quot;ENSG00000113734&quot;)</code></pre>
<pre><code>Warning: Removed 1 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-46-1">
Past versions of unnamed-chunk-46-1.png
</button>
</p>
<div id="fig-unnamed-chunk-46-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a20b63cf93370a96d805e6c1e3c5c2bb27a04eec/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-46-1.png" target="_blank">a20b63c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Not sure whats up with total</p>
</div>
<div id="eif2a" class="section level3">
<h3>EIF2A</h3>
<p>3:150302010 rs14434 peak114357</p>
<pre class="bash"><code>grep EIF2A /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000144895 
gunzip /project2/gilad/briana/YRI_geno_hg19/chr3.dose.filt.vcf.gz

python createQTLsnpAPAPhenTable_proc.py 3 3:150302010 peak114357 Total
python createQTLsnpAPAPhenTable_proc.py 3 3:150302010 peak114357 Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;3&quot; &quot;3:150302010&quot; &quot;ENSG00000144895&quot;</code></pre>
<pre class="r"><code>EIF2a_plot=plotQTL_func(SNP=&quot;3:150302010&quot;, peak=&quot;peak114357&quot;, gene=&quot;ENSG00000144895&quot;)
EIF2a_plot</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a20b63cf93370a96d805e6c1e3c5c2bb27a04eec/docs/figure/characterize_apaQTLs.Rmd/unnamed-chunk-48-1.png" target="_blank">a20b63c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(EIF2a_plot, file=&quot;../output/plots/EIF2a_boxplots.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<div id="locus-zoom-for-this-one" class="section level4">
<h4>Locus zoom for this one</h4>
<p>(only have to redo the APA plots)</p>
<pre class="bash"><code>grep EIF2A /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000144895


grep peak114357  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom_proc/NuclearAPA.peak114357.EIF2A.nomNuc.txt

grep peak114357 /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt  &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom_proc/TotalAPA.peak114357.EIF2A.nomTotal.txt
</code></pre>
<pre class="r"><code>APATotal_EIF2A_LZ=read.table(&quot;../data/LocusZoom_proc/TotalAPA.peak114357.EIF2A.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(APATotal_EIF2A_LZ,&quot;../data/LocusZoom_proc/apaTotalEIF2A_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


APANuclear_EIF2A_LZ=read.table(&quot;../data/LocusZoom_proc/NuclearAPA.peak114357.EIF2A.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(APANuclear_EIF2A_LZ,&quot;../data/LocusZoom_proc/apaNuclearEIF2A_LZ.txt&quot;, col.names = T, row.names = F, quote = F)</code></pre>
</div>
</div>
<div id="churc1" class="section level3">
<h3>Churc1</h3>
<p>14:65389250 peak48989 TotalQTL rs10131002</p>
<pre class="bash"><code>grep CHURC1 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ENSG00000258289 
gunzip /project2/gilad/briana/YRI_geno_hg19/chr14.dose.filt.vcf.gz

python createQTLsnpAPAPhenTable_proc.py 14 14:65389250 peak48989 Total
python createQTLsnpAPAPhenTable_proc.py 14 14:65389250   peak48989 Nuclear

sbatch run_createQTLsnpMolPhenTable.sh &quot;14&quot; &quot;14:65389250&quot; &quot;ENSG00000258289&quot;</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;14:65389250&quot;, peak=&quot;peak48989&quot;, gene=&quot;ENSG00000258289&quot;)</code></pre>
<p><img src="figure/characterize_apaQTLs.Rmd/unnamed-chunk-52-1.png" width="672" style="display: block; margin: auto;" /> No protein but this is a good example of one that is an eQTL</p>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] bindrcpp_0.2.2  cowplot_0.9.3   reshape2_1.4.3  qvalue_2.12.0  
 [5] forcats_0.3.0   stringr_1.4.0   dplyr_0.7.6     purrr_0.2.5    
 [9] readr_1.1.1     tidyr_0.8.1     tibble_1.4.2    ggplot2_3.0.0  
[13] tidyverse_1.2.1 workflowr_1.2.0

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4   splines_3.5.1      haven_1.1.2       
 [4] lattice_0.20-35    colorspace_1.3-2   htmltools_0.3.6   
 [7] yaml_2.2.0         utf8_1.1.4         rlang_0.2.2       
[10] pillar_1.3.0       glue_1.3.0         withr_2.1.2       
[13] RColorBrewer_1.1-2 modelr_0.1.2       readxl_1.1.0      
[16] bindr_0.1.1        plyr_1.8.4         munsell_0.5.0     
[19] gtable_0.2.0       cellranger_1.1.0   rvest_0.3.2       
[22] evaluate_0.13      labeling_0.3       knitr_1.20        
[25] fansi_0.4.0        broom_0.5.0        Rcpp_0.12.19      
[28] scales_1.0.0       backports_1.1.2    jsonlite_1.6      
[31] fs_1.2.6           hms_0.4.2          digest_0.6.17     
[34] stringi_1.2.4      grid_3.5.1         rprojroot_1.3-2   
[37] cli_1.0.1          tools_3.5.1        magrittr_1.5      
[40] lazyeval_0.2.1     crayon_1.3.4       whisker_0.3-2     
[43] pkgconfig_2.0.2    xml2_1.2.0         lubridate_1.7.4   
[46] assertthat_0.2.0   rmarkdown_1.11     httr_1.3.1        
[49] rstudioapi_0.9.0   R6_2.3.0           nlme_3.1-137      
[52] git2r_0.24.0       compiler_3.5.1    </code></pre>
</div>
</div>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
