<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2018-11-29" />

<title>Explaining pQTLS</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Explaining pQTLS</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>11/29/2018</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-12-03</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/17b336bff06fb3529bf11d09b97ee0de29434a22" target="_blank">17b336b</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/LocusZoom/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeaksUsed/
    Untracked:  data/RNAkalisto/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/apaExamp/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_trans/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/39indQC.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   code/Snakefile

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/17b336bff06fb3529bf11d09b97ee0de29434a22/analysis/explainpQTLs.Rmd" target="_blank">17b336b</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-12-03
</td>
<td style="text-align:left;">
add random
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/8bb97f621f41d8d8f0ff04f66ca34d1578902468/docs/explainpQTLs.html" target="_blank">8bb97f6</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-12-03
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/8b45f35390f99b64593a58dcc245e9040de54fb4/analysis/explainpQTLs.Rmd" target="_blank">8b45f35</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-12-03
</td>
<td style="text-align:left;">
add anova
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/dd06ccc5bcfebb073212f2c6895435258b6a9667/docs/explainpQTLs.html" target="_blank">dd06ccc</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-12-03
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/37ef750b1076803a9c5f4908d8f4aaeb87f42760/analysis/explainpQTLs.Rmd" target="_blank">37ef750</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-12-03
</td>
<td style="text-align:left;">
add graph for top 5 qtls
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/210b58b2a0f472cefb26b16fad513ea3ca49d9d7/docs/explainpQTLs.html" target="_blank">210b58b</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-29
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/413c8fddd492447ac0d51a7b928c56a3c941f41f/analysis/explainpQTLs.Rmd" target="_blank">413c8fd</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-29
</td>
<td style="text-align:left;">
add filter QTL analysis and start explain pqtl
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<p>In this analysis I want to look at how well the eQTLs and apaQTLs an explain pQTLs. I will use linear models on the effect sizes.</p>
<p><em>Libraries</em></p>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.1.1
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ───────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.0.0     ✔ purrr   0.2.5
✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.3.1
✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>── Conflicts ──────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre class="r"><code>library(broom)
library(cowplot)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<p>Input the pQTLs (10% FDR) and gene names</p>
<pre class="r"><code>geneNames=read.table(&quot;../data/ensemble_to_genename.txt&quot;, sep=&quot;\t&quot;, header=T,stringsAsFactors = F)
pQTL=read.table(&quot;../data/other_qtls/fastqtl_qqnorm_prot.fixed.perm.out&quot;, col.names = c(&quot;Gene.stable.ID&quot;, &quot;nvar&quot;, &quot;shape1&quot;, &quot;shape2&quot;, &quot;dummy&quot;, &quot;sid&quot;, &quot;dist&quot;, &quot;npval&quot;, &quot;slope&quot;, &quot;ppval&quot;, &quot;bpval&quot;), header = F, stringsAsFactors = F) %&gt;% inner_join(geneNames, by=&quot;Gene.stable.ID&quot;) %&gt;% dplyr::select(&quot;Gene.name&quot;,&quot;Gene.stable.ID&quot;, &quot;nvar&quot;, &quot;shape1&quot;, &quot;shape2&quot;, &quot;dummy&quot;, &quot;sid&quot;, &quot;dist&quot;, &quot;npval&quot;, &quot;slope&quot;, &quot;ppval&quot;, &quot;bpval&quot;)

pQTL$bh=p.adjust(pQTL$bpval, method=&quot;fdr&quot;)

pQTL_sig=pQTL %&gt;% filter(-log10(bh)&gt; 1) </code></pre>
<div id="example-ccdc51" class="section level2">
<h2>Example CCDC51</h2>
<p>Start with the exanple with the highest effect size:</p>
<p>CCDC51- ENSG00000164051 3:48476431</p>
<p>I need all of the results for this snp gene pair from the total, nuclear, and RNA nominal files. I can make a python script that will take the gene name and snp and create the relevent dataframe.</p>
<p>files: * /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_NomRes.txt<br />
* /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_NomRes.txt<br />
* /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out (need the ENSG ID)</p>
<p>APAandRNAfromProtQTLs.py</p>
<pre class="bash"><code>#use this by inserting a gene, gene_ensg (from prot), and snp for the protien QTLs
def main(gene, gene_ensg,snp):
    out_prot=open(&quot;/project2/gilad/briana/threeprimeseq/data/protQTL_otherphen/%s_%s_Protres.txt&quot;%(gene, snp), &quot;w&quot;)
    out_RNA=open(&quot;/project2/gilad/briana/threeprimeseq/data/protQTL_otherphen/%s_%s_RNAres.txt&quot;%(gene, snp), &quot;w&quot;)
    out_Total=open(&quot;/project2/gilad/briana/threeprimeseq/data/protQTL_otherphen/%s_%s_Totalres.txt&quot;%(gene, snp), &quot;w&quot;)
    out_Nuclear=open(&quot;/project2/gilad/briana/threeprimeseq/data/protQTL_otherphen/%s_%s_Nuclearres.txt&quot;%(gene, snp), &quot;w&quot;)
    for ln in open(&quot;/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_NomRes.txt&quot;, &quot;r&quot;):
       s=ln.split()[1]
       g=ln.split()[0].split(&quot;:&quot;)[3].split(&quot;_&quot;)[0]
       if g==gene:
           out_Nuclear.write(ln)
    for ln in open(&quot;/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_NomRes.txt&quot;, &quot;r&quot;):
       s=ln.split()[1]
       g=ln.split()[0].split(&quot;:&quot;)[3].split(&quot;_&quot;)[0]
       if g==gene:
           out_Total.write(ln)
    for ln in open(&quot;/project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out&quot;, &quot;r&quot;):
        s=ln.split()[1]
        g=ln.split()[0]
        if gene_ensg in g:
          out_RNA.write(ln) 
    for ln in open(&quot;/project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_prot.fixed.nominal.out&quot;, &quot;r&quot;):
        s=ln.split()[1]
        g=ln.split()[0]
        if gene_ensg in g:
          out_prot.write(ln) 

    out_Total.close()
    out_Nuclear.close()
    out_RNA.close()
    out_prot.close()

if __name__ == &quot;__main__&quot;:
    import sys
    gene=sys.argv[1]
    gene_ensg=sys.argv[2]
    snp=sys.argv[3]
    main(gene, gene_ensg, snp)
    
</code></pre>
<p>CCDC51_APAandRNAfromProtQTLs.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=CCDC51_APAandRNAfromProtQTLs
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=CCDC51_APAandRNAfromProtQTLs.out
#SBATCH --error=CCDC51_APAandRNAfromProtQTLs.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load python

python APAandRNAfromProtQTLs.py CCDC51 ENSG00000164051 3:48476431 </code></pre>
<p>I first want to look at the effect sizes. I am interested in understanding the directions of the effect sizes.</p>
<p>3:48728204. This genotype is in the genotype file twice. I am gonig to remove this snp from the analysis.</p>
<pre class="r"><code>nom_names=c(&quot;gene&quot;, &quot;snp&quot;, &quot;dist&quot;, &quot;pval&quot;, &quot;slope&quot;)
CCDC51_apaTot=read.table(&quot;../data/pQTL_otherphen/CCDC51_3:48476431_Totalres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope)
CCDC51_apaTot=filter(CCDC51_apaTot, !grepl(&quot;3:48728204&quot;,snp))
CCDC51_apaNuc=read.table(&quot;../data/pQTL_otherphen/CCDC51_3:48476431_Nuclearres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope)
CCDC51_apaNuc=filter(CCDC51_apaNuc, !grepl(&quot;3:48728204&quot;,snp))
CCDC51_RNA=read.table(&quot;../data/pQTL_otherphen/CCDC51_3:48476431_RNAres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% select(snp, slope)
CCDC51_RNA=filter(CCDC51_RNA, !grepl(&quot;3:48728204&quot;,snp))
CCDC51_Prot=read.table(&quot;../data/pQTL_otherphen/CCDC51_3:48476431_Protres.txt&quot;, col.names = nom_names, stringsAsFactors = F)%&gt;%  select(snp, slope)
CCDC51_Prot=filter(CCDC51_Prot, !grepl(&quot;3:48728204&quot;,snp))</code></pre>
<p>Start with RNA and protein because they are the most simple</p>
<pre class="r"><code>CCDC51_ProtRNA=CCDC51_RNA %&gt;% left_join(CCDC51_Prot, by=&quot;snp&quot;)
colnames(CCDC51_ProtRNA)=c(&quot;snp&quot;, &quot;RNA&quot;, &quot;Protein&quot;)</code></pre>
<p>Plot this:</p>
<pre class="r"><code>plot(CCDC51_ProtRNA$Protein ~ CCDC51_ProtRNA$RNA)
abline(lm(CCDC51_ProtRNA$Protein ~ CCDC51_ProtRNA$RNA))</code></pre>
<p><img src="figure/explainpQTLs.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-7-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/8bb97f621f41d8d8f0ff04f66ca34d1578902468/docs/figure/explainpQTLs.Rmd/unnamed-chunk-7-1.png" target="_blank">8bb97f6</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>For APA I need to get a data frame that has the snps by the effect sizes for each peak.I basically want to spread this df.</p>
<pre class="r"><code>CCDC51_apaTot_s=spread(CCDC51_apaTot, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<pre class="r"><code>CCDC51_apaNuc_s=spread(CCDC51_apaNuc, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<p>I can look at the protein ~ apaTotal.</p>
<pre class="r"><code>CCDC51_apaTotProt=CCDC51_apaTot_s %&gt;% left_join(CCDC51_Prot, by=&quot;snp&quot;)
colnames(CCDC51_apaTotProt)=c(&quot;snp&quot;, &quot;peak216857&quot;, &quot;peak216858&quot;, &quot;peak216859&quot;, &quot;peak216860&quot;, &quot;peak216867&quot;,&quot;Protein&quot;)</code></pre>
<p>Try with all of it:</p>
<pre class="r"><code>CCDC51_apaTotProtRNA=CCDC51_apaTot_s %&gt;% left_join(CCDC51_ProtRNA, by=&quot;snp&quot;)
colnames(CCDC51_apaTotProtRNA)=c(&quot;snp&quot;, &quot;peak216857&quot;, &quot;peak216858&quot;, &quot;peak216859&quot;, &quot;peak216860&quot;, &quot;peak216867&quot;,&quot;RNA&quot;,&quot;Protein&quot;)</code></pre>
<p>I want to use the same data for each model:</p>
<pre class="r"><code>CCDC51_lmProtRNA=lm(Protein ~ RNA, data=CCDC51_apaTotProtRNA)
CCDC51_lmProtRNA_sum=glance(CCDC51_lmProtRNA)

CCDC51_lmProtAPA=lm(Protein ~peak216857 + peak216858 + peak216859 + peak216860 + peak216867, data=CCDC51_apaTotProtRNA)
CCDC51_lmProtAPA_sum=glance(CCDC51_lmProtAPA)

CCDC51_lmProtAPARNA=lm(Protein ~RNA +peak216857 + peak216858 + peak216859 + peak216860 + peak216867, data=CCDC51_apaTotProtRNA)
CCDC51_lmProtAPARNA_sum=glance(CCDC51_lmProtAPARNA)
adjR2_CCDC51=cbind(&quot;CCDC51&quot;, CCDC51_lmProtRNA_sum[1,2], CCDC51_lmProtAPA_sum[1,2], CCDC51_lmProtAPARNA_sum[1,2])
colnames(adjR2_CCDC51)=c(&quot;Gene&quot;, &quot;RNA&quot;, &quot;APA&quot;, &quot;RNAandAPA&quot;)</code></pre>
</div>
<div id="example-dtd1" class="section level2">
<h2>Example DTD1</h2>
<p>DTD1- ENSG00000125821 20:18607243</p>
<p>DTD1_APAandRNAfromProtQTLs.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=DTD1_APAandRNAfromProtQTLs
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=DTD1_APAandRNAfromProtQTLs.out
#SBATCH --error=DTD1_APAandRNAfromProtQTLs.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load python

python APAandRNAfromProtQTLs.py DTD1 ENSG00000125821 20:18607243</code></pre>
<pre class="r"><code>nom_names=c(&quot;gene&quot;, &quot;snp&quot;, &quot;dist&quot;, &quot;pval&quot;, &quot;slope&quot;)

DTD1_RNA=read.table(&quot;../data/pQTL_otherphen/DTD1_20:18607243_RNAres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% select(snp, slope) %&gt;%  filter(!duplicated(snp))

DTD1_Prot=read.table(&quot;../data/pQTL_otherphen/DTD1_20:18607243_Protres.txt&quot;, col.names = nom_names, stringsAsFactors = F)%&gt;%  select(snp, slope) %&gt;%  filter(!duplicated(snp))



#FILTER snps in the list from Rna and prot
DTD1_apaTot=read.table(&quot;../data/pQTL_otherphen/DTD1_20:18607243_Totalres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope) %&gt;% group_by(peak) %&gt;%  filter(!duplicated(snp)) 
DTD1_apaNuc=read.table(&quot;../data/pQTL_otherphen/DTD1_20:18607243_Nuclearres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope) %&gt;% group_by(peak) %&gt;%  filter(!duplicated(snp)) </code></pre>
<p>Start with RNA and protein because they are the most simple</p>
<pre class="r"><code>DTD1_ProtRNA=DTD1_RNA %&gt;% left_join(DTD1_Prot, by=&quot;snp&quot;)
colnames(DTD1_ProtRNA)=c(&quot;snp&quot;, &quot;RNA&quot;, &quot;Protein&quot;)</code></pre>
<pre class="r"><code>DTD1_apaTot_s=spread(DTD1_apaTot, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<pre class="r"><code>DTD1_apaNuc_s=spread(DTD1_apaNuc, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<pre class="r"><code>DTD1_apaTotProt=DTD1_apaTot_s %&gt;% inner_join(DTD1_Prot, by=&quot;snp&quot;)

colnames(DTD1_apaTotProt)=c(&quot;snp&quot;, &quot;peak195416&quot;, &quot;peak195418&quot; ,&quot;peak195419&quot;, &quot;peak195420&quot;, &quot;peak195423&quot;, &quot;peak195424&quot;,&quot;peak195425&quot; ,&quot;peak195426&quot;, &quot;peak195427&quot;, &quot;peak195428&quot;, &quot;peak195429&quot;, &quot;peak195430&quot;, &quot;peak195431&quot;,&quot;peak195432&quot;, &quot;peak195433&quot; ,&quot;peak195434&quot; ,&quot;peak195436&quot;, &quot;peak195438&quot;, &quot;peak195443&quot;, &quot;peak195444&quot; ,&quot;peak195445&quot;, &quot;peak195446&quot;, &quot;peak195447&quot;, &quot;peak195449&quot; ,&quot;peak195450&quot;, &quot;peak195451&quot;, &quot;peak195452&quot;,&quot;peak195453&quot;, &quot;peak195454&quot;, &quot;peak195455&quot; ,&quot;protein&quot;)</code></pre>
<p>`</p>
<p>Try with all:</p>
<pre class="r"><code>DTD1_apaTotProtRNA=DTD1_apaTot_s %&gt;% inner_join(DTD1_ProtRNA, by=&quot;snp&quot;)

colnames(DTD1_apaTotProtRNA)=c(&quot;snp&quot;, &quot;peak195416&quot;, &quot;peak195418&quot; ,&quot;peak195419&quot;, &quot;peak195420&quot;, &quot;peak195423&quot;, &quot;peak195424&quot;,&quot;peak195425&quot; ,&quot;peak195426&quot;, &quot;peak195427&quot;, &quot;peak195428&quot;, &quot;peak195429&quot;, &quot;peak195430&quot;, &quot;peak195431&quot;,&quot;peak195432&quot;, &quot;peak195433&quot; ,&quot;peak195434&quot; ,&quot;peak195436&quot;, &quot;peak195438&quot;, &quot;peak195443&quot;, &quot;peak195444&quot; ,&quot;peak195445&quot;, &quot;peak195446&quot;, &quot;peak195447&quot;, &quot;peak195449&quot; ,&quot;peak195450&quot;, &quot;peak195451&quot;, &quot;peak195452&quot;,&quot;peak195453&quot;, &quot;peak195454&quot;, &quot;peak195455&quot; ,&quot;RNA&quot;, &quot;protein&quot;)</code></pre>
<p>Model:</p>
<pre class="r"><code>DTD1_lmProtRNA=lm(protein ~ RNA, data=DTD1_apaTotProtRNA)
DTD1_lmProtRNA_sum=glance(DTD1_lmProtRNA)

DTD1_lmProtAPA=lm(protein ~ peak195416+ peak195418 +peak195419+peak195420+ peak195423+ peak195424+peak195425 +peak195426+ peak195427+ peak195428+ peak195429+ peak195430 + peak195431+peak195432 + peak195433+peak195434+peak195436+peak195438+peak195443+peak195444+peak195445+peak195446+peak195447+peak195449+peak195450+peak195451+peak195452+peak195453+ peak195454 +peak195455, data=DTD1_apaTotProtRNA)
DTD1_lmProtAPA_sum=glance(DTD1_lmProtAPA)

DTD1_lmProtAPARNA=lm(protein ~RNA+ peak195416+ peak195418 +peak195419+peak195420+ peak195423+ peak195424+peak195425 +peak195426+ peak195427+ peak195428+ peak195429+ peak195430 + peak195431+peak195432 + peak195433+peak195434+peak195436+peak195438+peak195443+peak195444+peak195445+peak195446+peak195447+peak195449+peak195450+peak195451+peak195452+peak195453+ peak195454 +peak195455, data=DTD1_apaTotProtRNA)
DTD1_lmProtAPARNA_sum=glance(DTD1_lmProtAPARNA)

adjR2_DTD1=cbind(&quot;DTD1&quot;,DTD1_lmProtRNA_sum[1,2], DTD1_lmProtAPA_sum[1,2], DTD1_lmProtAPARNA_sum[1,2])
colnames(adjR2_DTD1)=c(&quot;Gene&quot;, &quot;RNA&quot;, &quot;APA&quot;, &quot;RNAandAPA&quot;)</code></pre>
</div>
<div id="example-dhrs7b" class="section level2">
<h2>Example DHRS7B</h2>
<p>DHRS7B ENSG00000109016 17:21036822 DHRS7B_APAandRNAfromProtQTLs.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=DHRS7B_APAandRNAfromProtQTLs
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=DHRS7B_APAandRNAfromProtQTLs.out
#SBATCH --error=DHRS7B_APAandRNAfromProtQTLs.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load python

python APAandRNAfromProtQTLs.py DHRS7B ENSG00000109016 17:21036822</code></pre>
<pre class="r"><code>nom_names=c(&quot;gene&quot;, &quot;snp&quot;, &quot;dist&quot;, &quot;pval&quot;, &quot;slope&quot;)

DHRS7B_RNA=read.table(&quot;../data/pQTL_otherphen/DHRS7B_17:21036822_RNAres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% select(snp, slope) %&gt;%  filter(!duplicated(snp))

DHRS7B_Prot=read.table(&quot;../data/pQTL_otherphen/DHRS7B_17:21036822_Protres.txt&quot;, col.names = nom_names, stringsAsFactors = F)%&gt;%  select(snp, slope) %&gt;%  filter(!duplicated(snp))



#FILTER snps in the list from Rna and prot
DHRS7B_apaTot=read.table(&quot;../data/pQTL_otherphen/DHRS7B_17:21036822_Totalres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope) %&gt;% group_by(peak) %&gt;%  filter(!duplicated(snp)) 
DHRS7B_apaNuc=read.table(&quot;../data/pQTL_otherphen/DHRS7B_17:21036822_Nuclearres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope) %&gt;% group_by(peak) %&gt;%  filter(!duplicated(snp)) </code></pre>
<p>Start with RNA and protein because they are the most simple</p>
<pre class="r"><code>DHRS7B_ProtRNA=DHRS7B_RNA %&gt;% left_join(DHRS7B_Prot, by=&quot;snp&quot;)
colnames(DHRS7B_ProtRNA)=c(&quot;snp&quot;, &quot;RNA&quot;, &quot;Protein&quot;)</code></pre>
<p>For APA I need to get a data frame that has the snps by the effect sizes for each peak.I basically want to spread this df.</p>
<pre class="r"><code>DHRS7B_apaTot_s=spread(DHRS7B_apaTot, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<pre class="r"><code>DHRS7B_apaNuc_s=spread(DHRS7B_apaNuc, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<p>I can look at the protein ~ apaTotal.</p>
<pre class="r"><code>DHRS7B_apaTotProt=DHRS7B_apaTot_s %&gt;% left_join(DHRS7B_Prot, by=&quot;snp&quot;)
colnames(DHRS7B_apaTotProt)=c(names(DHRS7B_apaTot_s),&quot;Protein&quot;)</code></pre>
<p>Try with all:</p>
<pre class="r"><code>DHRS7B_apaTotProtRNA=DHRS7B_apaTot_s %&gt;% inner_join(DHRS7B_ProtRNA, by=&quot;snp&quot;)

colnames(DHRS7B_apaTotProtRNA)=c(names(DHRS7B_apaTot_s),&quot;RNA&quot;, &quot;protein&quot;)</code></pre>
<pre class="r"><code>DHRS7B_lmProtRNA=lm(protein ~ RNA, data=DHRS7B_apaTotProtRNA)

DHRS7B_lmProtRNA_sum=glance(DHRS7B_lmProtRNA)

DHRS7B_lmProtAPA=lm(protein ~ peak132690+peak132692+peak132693+peak132694+peak132720+peak132721+peak132722+peak132723+peak132724+peak132725+peak132728+peak132729+peak132730+peak132732+peak132733+peak132734+peak132735+peak132738+peak132739+peak132740+peak132741+peak132742+peak132744+peak132745 +peak132746+peak132747+peak132748+peak132749+peak132750+peak132751+peak132753+peak132754+ peak132755+peak132756+peak132757+peak132760+peak132761+peak132762+peak132763+peak132764+peak132766+peak132774+peak132775+peak132777+peak132778+peak132780, data=DHRS7B_apaTotProtRNA)
DHRS7B_lmProtAPA_sum=glance(DHRS7B_lmProtAPA)


DHRS7B_lmProtAPARNA=lm(protein ~ peak132690+peak132692+peak132693+peak132694+peak132720+peak132721+peak132722+peak132723+peak132724+peak132725+peak132728+peak132729+peak132730+peak132732+peak132733+peak132734+peak132735+peak132738+peak132739+peak132740+peak132741+peak132742+peak132744+peak132745 +peak132746+peak132747+peak132748+peak132749+peak132750+peak132751+peak132753+peak132754+ peak132755+peak132756+peak132757+peak132760+peak132761+peak132762+peak132763+peak132764+peak132766+peak132774+peak132775+peak132777+peak132778+peak132780, data=DHRS7B_apaTotProtRNA)

DHRS7B_lmProtAPARNA_sum=glance(DHRS7B_lmProtAPARNA)

adjR2_DHRS7B=cbind(&quot;DHRS7B&quot;,DHRS7B_lmProtRNA_sum[1,2], DHRS7B_lmProtAPA_sum[1,2], DHRS7B_lmProtAPARNA_sum[1,2])
colnames(adjR2_DHRS7B)=c(&quot;Gene&quot;,  &quot;RNA&quot;, &quot;APA&quot;, &quot;RNAandAPA&quot;)</code></pre>
</div>
<div id="example-mmab" class="section level2">
<h2>Example MMAB</h2>
<p>ENSG00000139428 MMAB 12:109997847</p>
<p>MMAB_APAandRNAfromProtQTLs.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=MMAB_APAandRNAfromProtQTLs
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MMAB_APAandRNAfromProtQTLs.out
#SBATCH --error=MMAB_APAandRNAfromProtQTLs.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load python

python APAandRNAfromProtQTLs.py MMAB ENSG00000139428 12:109997847 </code></pre>
<pre class="r"><code>nom_names=c(&quot;gene&quot;, &quot;snp&quot;, &quot;dist&quot;, &quot;pval&quot;, &quot;slope&quot;)

MMAB_RNA=read.table(&quot;../data/pQTL_otherphen/MMAB_12:109997847_RNAres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% select(snp, slope) %&gt;%  filter(!duplicated(snp))

MMAB_Prot=read.table(&quot;../data/pQTL_otherphen/MMAB_12:109997847_Protres.txt&quot;, col.names = nom_names, stringsAsFactors = F)%&gt;%  select(snp, slope) %&gt;%  filter(!duplicated(snp))



#FILTER snps in the list from Rna and prot
MMAB_apaTot=read.table(&quot;../data/pQTL_otherphen/MMAB_12:109997847_Totalres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope) %&gt;% group_by(peak) %&gt;%  filter(!duplicated(snp)) 
MMAB_apaNuc=read.table(&quot;../data/pQTL_otherphen/MMAB_12:109997847_Nuclearres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope) %&gt;% group_by(peak) %&gt;%  filter(!duplicated(snp)) </code></pre>
<p>Start with RNA and protein because they are the most simple</p>
<pre class="r"><code>MMAB_ProtRNA=MMAB_RNA %&gt;% left_join(MMAB_Prot, by=&quot;snp&quot;)
colnames(MMAB_ProtRNA)=c(&quot;snp&quot;, &quot;RNA&quot;, &quot;Protein&quot;)</code></pre>
<p>For APA I need to get a data frame that has the snps by the effect sizes for each peak.I basically want to spread this df.</p>
<pre class="r"><code>MMAB_apaTot_s=spread(MMAB_apaTot, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<pre class="r"><code>MMAB_apaNuc_s=spread(MMAB_apaNuc, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<p>I can look at the protein ~ apaTotal.</p>
<pre class="r"><code>MMAB_apaTotProt=MMAB_apaTot_s %&gt;% left_join(MMAB_Prot, by=&quot;snp&quot;)
colnames(MMAB_apaTotProt)=c(names(MMAB_apaTot_s),&quot;Protein&quot;)</code></pre>
<p>Try with all:</p>
<pre class="r"><code>MMAB_apaTotProtRNA=MMAB_apaTot_s %&gt;% inner_join(MMAB_ProtRNA, by=&quot;snp&quot;)

colnames(MMAB_apaTotProtRNA)=c(names(MMAB_apaTot_s),&quot;RNA&quot;, &quot;protein&quot;)</code></pre>
<pre class="r"><code>MMAB_lmProtRNA=lm(protein ~ RNA, data=MMAB_apaTotProtRNA)
MMAB_lmProtRNA_sum=glance(MMAB_lmProtRNA)


MMAB_lmProtAPA=lm(protein ~ peak78754+peak78755+peak78756+peak78757+peak78758+peak78759+peak78760+peak78761+peak78762+peak78763+peak78764+peak78765+peak78766, data=MMAB_apaTotProtRNA)
MMAB_lmProtAPA_sum=glance(MMAB_lmProtAPA)


MMAB_lmProtAPARNA=lm(protein ~RNA+ peak78754+peak78755+peak78756+peak78757+peak78758+peak78759+peak78760+peak78761+peak78762+peak78763+peak78764+peak78765+peak78766, data=MMAB_apaTotProtRNA)
MMAB_lmProtAPARNA_sum=glance(MMAB_lmProtAPARNA)

adjR2_MMAB=cbind(&quot;MMAB&quot;,MMAB_lmProtRNA_sum[1,2], MMAB_lmProtAPA_sum[1,2], MMAB_lmProtAPARNA_sum[1,2])
colnames(adjR2_MMAB)=c(&quot;Gene&quot;,  &quot;RNA&quot;, &quot;APA&quot;, &quot;RNAandAPA&quot;)</code></pre>
</div>
<div id="example-fyttd1" class="section level2">
<h2>Example FYTTD1</h2>
<p>FYTTD1 ENSG00000122068 3:197035232</p>
<p>FYTTD1_APAandRNAfromProtQTLs.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=FYTTD1_APAandRNAfromProtQTLs
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=FYTTD1_APAandRNAfromProtQTLs.out
#SBATCH --error=FYTTD1_APAandRNAfromProtQTLs.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load python

python APAandRNAfromProtQTLs.py FYTTD1 ENSG00000122068 3:197035232  </code></pre>
<pre class="r"><code>nom_names=c(&quot;gene&quot;, &quot;snp&quot;, &quot;dist&quot;, &quot;pval&quot;, &quot;slope&quot;)

FYTTD1_RNA=read.table(&quot;../data/pQTL_otherphen/FYTTD1_3:197035232_RNAres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% select(snp, slope) %&gt;%  filter(!duplicated(snp))

FYTTD1_Prot=read.table(&quot;../data/pQTL_otherphen/FYTTD1_3:197035232_Protres.txt&quot;, col.names = nom_names, stringsAsFactors = F)%&gt;%  select(snp, slope) %&gt;%  filter(!duplicated(snp))



#FILTER snps in the list from Rna and prot
FYTTD1_apaTot=read.table(&quot;../data/pQTL_otherphen/FYTTD1_3:197035232_Totalres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope) %&gt;% group_by(peak) %&gt;%  filter(!duplicated(snp)) 
FYTTD1_apaNuc=read.table(&quot;../data/pQTL_otherphen/FYTTD1_3:197035232_Nuclearres.txt&quot;, col.names = nom_names, stringsAsFactors = F) %&gt;% separate(gene, sep = &quot;:&quot;, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep = &quot;_&quot;, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;)) %&gt;% select(snp,peak, slope) %&gt;% group_by(peak) %&gt;%  filter(!duplicated(snp)) </code></pre>
<p>Start with RNA and protein because they are the most simple</p>
<pre class="r"><code>FYTTD1_ProtRNA=FYTTD1_RNA %&gt;% left_join(FYTTD1_Prot, by=&quot;snp&quot;)
colnames(FYTTD1_ProtRNA)=c(&quot;snp&quot;, &quot;RNA&quot;, &quot;Protein&quot;)</code></pre>
<p>For APA I need to get a data frame that has the snps by the effect sizes for each peak.I basically want to spread this df.</p>
<pre class="r"><code>FYTTD1_apaTot_s=spread(FYTTD1_apaTot, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<pre class="r"><code>FYTTD1_apaNuc_s=spread(FYTTD1_apaNuc, &quot;peak&quot;, &quot;slope&quot;,drop = F)</code></pre>
<p>I can look at the protein ~ apaTotal.</p>
<pre class="r"><code>FYTTD1_apaTotProt=FYTTD1_apaTot_s %&gt;% left_join(FYTTD1_Prot, by=&quot;snp&quot;)
colnames(FYTTD1_apaTotProt)=c(names(FYTTD1_apaTot_s),&quot;Protein&quot;)</code></pre>
<pre class="r"><code>FYTTD1_apaTotProtRNA=FYTTD1_apaTot_s %&gt;% inner_join(FYTTD1_ProtRNA, by=&quot;snp&quot;)

colnames(FYTTD1_apaTotProtRNA)=c(names(FYTTD1_apaTot_s),&quot;RNA&quot;, &quot;protein&quot;)</code></pre>
<p>Model:</p>
<pre class="r"><code>FYTTD1_lmProtRNA=lm(protein ~ RNA, data=FYTTD1_apaTotProtRNA)
FYTTD1_lmProtRNA_sum=glance(FYTTD1_lmProtRNA)

FYTTD1_lmProtAPA=lm(protein ~ peak234016 +peak234082 +peak234088+peak234089+peak234090+peak234092+peak234093+peak234094+peak234095+peak234099+peak234101+peak234103+peak234104+peak234105+peak234106+peak234107+peak234108+peak234111+peak234113+peak234114+peak234117+peak234119+peak234120+peak234122+peak234123+peak234124+peak234125+peak234131+peak234135, data=FYTTD1_apaTotProtRNA)
FYTTD1_lmProtAPA_sum=glance(FYTTD1_lmProtAPA)


FYTTD1_lmProtAPARNA=lm(protein ~RNA+ peak234016 +peak234082 +peak234088+peak234089+peak234090+peak234092+peak234093+peak234094+peak234095+peak234099+peak234101+peak234103+peak234104+peak234105+peak234106+peak234107+peak234108+peak234111+peak234113+peak234114+peak234117+peak234119+peak234120+peak234122+peak234123+peak234124+peak234125+peak234131+peak234135, data=FYTTD1_apaTotProtRNA)
FYTTD1_lmProtAPARNA_sum=glance(FYTTD1_lmProtAPARNA)

adjR2_FYTTD1=cbind(&quot;FYTTD1&quot;,FYTTD1_lmProtRNA_sum[1,2], FYTTD1_lmProtAPA_sum[1,2], FYTTD1_lmProtAPARNA_sum[1,2])
colnames(adjR2_FYTTD1)=c(&quot;Gene&quot;,  &quot;RNA&quot;, &quot;APA&quot;, &quot;RNAandAPA&quot;)</code></pre>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>In each of these examples the total APA explains more of the protein variability. I am going to plot the adjR2 for each gene and model</p>
<pre class="r"><code>adjR2=rbind(adjR2_CCDC51,adjR2_DTD1,adjR2_DHRS7B, adjR2_MMAB, adjR2_FYTTD1)

adjR2_melt=melt(adjR2, id.vars=&quot;Gene&quot;)
colnames(adjR2_melt)=c(&quot;Gene&quot;, &quot;Model&quot;, &quot;AdjR2&quot;)</code></pre>
<p>Plot this</p>
<pre class="r"><code>ggplot(adjR2_melt, aes(x=Gene, y=AdjR2, by=Model, fill=Model)) + geom_bar(stat=&quot;identity&quot;, position = &quot;dodge&quot;) + labs(title=&quot;Adjusted R2 for variation\n in protein effect sizes explained in each model&quot;,x=&quot;pQTL Gene&quot;) + scale_fill_brewer(palette=&quot;Paired&quot;)</code></pre>
<p><img src="figure/explainpQTLs.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-46-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/8bb97f621f41d8d8f0ff04f66ca34d1578902468/docs/figure/explainpQTLs.Rmd/unnamed-chunk-46-1.png" target="_blank">8bb97f6</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="residuals-method" class="section level2">
<h2>Residuals method:</h2>
<p>I want to use the resid(lm) method to get the residuals for the RNA model and see if I can explain some of these with the APA information:</p>
<div id="get-rna-model-residuals" class="section level3">
<h3>Get RNA model residuals</h3>
<pre class="r"><code>CCDC51_RNA_Resid=resid(CCDC51_lmProtRNA)
DTD1_RNA_Resid=resid(DTD1_lmProtRNA)
DHRS7B_RNA_Resid=resid(DHRS7B_lmProtRNA)
MMAB_RNA_Resid=resid(MMAB_lmProtRNA)
FYTTD1_RNA_Resid=resid(FYTTD1_lmProtRNA)</code></pre>
</div>
<div id="model-apa-against-residuals" class="section level3">
<h3>Model APA against residuals</h3>
<div id="ccdc51" class="section level4">
<h4>CCDC51</h4>
<pre class="r"><code>CCDC51_lmResidAPA=lm(CCDC51_RNA_Resid ~ CCDC51_apaTotProt$peak216857 +CCDC51_apaTotProt$peak216858 + CCDC51_apaTotProt$peak216859 + CCDC51_apaTotProt$peak216860 + CCDC51_apaTotProt$peak216867)
CCDC51_lmResidRNA_sum=glance(CCDC51_lmResidAPA)</code></pre>
</div>
<div id="dtd1" class="section level4">
<h4>DTD1</h4>
<pre class="r"><code>DTD1_lmResidAPA=lm(DTD1_RNA_Resid ~ DTD1_apaTotProtRNA$peak195416+ DTD1_apaTotProtRNA$peak195418 +DTD1_apaTotProtRNA$peak195419+DTD1_apaTotProtRNA$peak195420+ DTD1_apaTotProtRNA$peak195423+ DTD1_apaTotProtRNA$peak195424+DTD1_apaTotProtRNA$peak195425 +DTD1_apaTotProtRNA$peak195426+ DTD1_apaTotProtRNA$peak195427+ DTD1_apaTotProtRNA$peak195428+ DTD1_apaTotProtRNA$peak195429+ DTD1_apaTotProtRNA$peak195430 + DTD1_apaTotProtRNA$peak195431+DTD1_apaTotProtRNA$peak195432 + DTD1_apaTotProtRNA$peak195433+DTD1_apaTotProtRNA$peak195434+DTD1_apaTotProtRNA$peak195436+DTD1_apaTotProtRNA$peak195438+DTD1_apaTotProtRNA$peak195443+DTD1_apaTotProtRNA$peak195444+DTD1_apaTotProtRNA$peak195445+DTD1_apaTotProtRNA$peak195446+DTD1_apaTotProtRNA$peak195447+DTD1_apaTotProtRNA$peak195449+DTD1_apaTotProtRNA$peak195450+DTD1_apaTotProtRNA$peak195451+DTD1_apaTotProtRNA$peak195452+DTD1_apaTotProtRNA$peak195453+ DTD1_apaTotProtRNA$peak195454 +DTD1_apaTotProtRNA$peak195455)
DTD1_lmResidRNA_sum=glance(DTD1_lmResidAPA)</code></pre>
</div>
<div id="dhrs7b" class="section level4">
<h4>DHRS7B</h4>
<pre class="r"><code>DHRS7B_lmResidAPA=lm(DHRS7B_RNA_Resid ~ DHRS7B_apaTotProtRNA$peak132690+DHRS7B_apaTotProtRNA$peak132692+DHRS7B_apaTotProtRNA$peak132693+DHRS7B_apaTotProtRNA$peak132694+DHRS7B_apaTotProtRNA$peak132720+DHRS7B_apaTotProtRNA$peak132721+DHRS7B_apaTotProtRNA$peak132722+DHRS7B_apaTotProtRNA$peak132723+DHRS7B_apaTotProtRNA$peak132724+DHRS7B_apaTotProtRNA$peak132725+DHRS7B_apaTotProtRNA$peak132728+DHRS7B_apaTotProtRNA$peak132729+DHRS7B_apaTotProtRNA$peak132730+DHRS7B_apaTotProtRNA$peak132732+DHRS7B_apaTotProtRNA$peak132733+DHRS7B_apaTotProtRNA$peak132734+DHRS7B_apaTotProtRNA$peak132735+DHRS7B_apaTotProtRNA$peak132738+DHRS7B_apaTotProtRNA$peak132739+DHRS7B_apaTotProtRNA$peak132740+DHRS7B_apaTotProtRNA$peak132741+DHRS7B_apaTotProtRNA$peak132742+DHRS7B_apaTotProtRNA$peak132744+DHRS7B_apaTotProtRNA$peak132745 +DHRS7B_apaTotProtRNA$peak132746+DHRS7B_apaTotProtRNA$peak132747+DHRS7B_apaTotProtRNA$peak132748+DHRS7B_apaTotProtRNA$peak132749+DHRS7B_apaTotProtRNA$peak132750+DHRS7B_apaTotProtRNA$peak132751+DHRS7B_apaTotProtRNA$peak132753+DHRS7B_apaTotProtRNA$peak132754+ DHRS7B_apaTotProtRNA$peak132755+DHRS7B_apaTotProtRNA$peak132756+DHRS7B_apaTotProtRNA$peak132757+DHRS7B_apaTotProtRNA$peak132760+DHRS7B_apaTotProtRNA$peak132761+DHRS7B_apaTotProtRNA$peak132762+DHRS7B_apaTotProtRNA$peak132763+DHRS7B_apaTotProtRNA$peak132764+DHRS7B_apaTotProtRNA$peak132766+DHRS7B_apaTotProtRNA$peak132774+DHRS7B_apaTotProtRNA$peak132775+DHRS7B_apaTotProtRNA$peak132777+DHRS7B_apaTotProtRNA$peak132778+DHRS7B_apaTotProtRNA$peak132780)
DHRS7B_lmResidRNA_sum=glance(DHRS7B_lmResidAPA)</code></pre>
</div>
<div id="mmab" class="section level4">
<h4>MMAB</h4>
<pre class="r"><code>MMAB_lmResidAPA=lm(MMAB_RNA_Resid ~ MMAB_apaTotProtRNA$peak78754+MMAB_apaTotProtRNA$peak78755+MMAB_apaTotProtRNA$peak78756+MMAB_apaTotProtRNA$peak78757+MMAB_apaTotProtRNA$peak78758+MMAB_apaTotProtRNA$peak78759+MMAB_apaTotProtRNA$peak78760+MMAB_apaTotProtRNA$peak78761+MMAB_apaTotProtRNA$peak78762+MMAB_apaTotProtRNA$peak78763+MMAB_apaTotProtRNA$peak78764+MMAB_apaTotProtRNA$peak78765+MMAB_apaTotProtRNA$peak78766)
MMAB_lmResidRNA_sum=glance(MMAB_lmResidAPA)</code></pre>
</div>
<div id="fyttd1" class="section level4">
<h4>FYTTD1</h4>
<pre class="r"><code>FYTTD1_lmResidAPA=lm(FYTTD1_RNA_Resid ~ FYTTD1_apaTotProtRNA$peak234016 +FYTTD1_apaTotProtRNA$peak234082 +FYTTD1_apaTotProtRNA$peak234088+FYTTD1_apaTotProtRNA$peak234089+FYTTD1_apaTotProtRNA$peak234090+FYTTD1_apaTotProtRNA$peak234092+FYTTD1_apaTotProtRNA$peak234093+FYTTD1_apaTotProtRNA$peak234094+FYTTD1_apaTotProtRNA$peak234095+FYTTD1_apaTotProtRNA$peak234099+FYTTD1_apaTotProtRNA$peak234101+FYTTD1_apaTotProtRNA$peak234103+FYTTD1_apaTotProtRNA$peak234104+FYTTD1_apaTotProtRNA$peak234105+FYTTD1_apaTotProtRNA$peak234106+FYTTD1_apaTotProtRNA$peak234107+FYTTD1_apaTotProtRNA$peak234108+FYTTD1_apaTotProtRNA$peak234111+FYTTD1_apaTotProtRNA$peak234113+FYTTD1_apaTotProtRNA$peak234114+FYTTD1_apaTotProtRNA$peak234117+FYTTD1_apaTotProtRNA$peak234119+FYTTD1_apaTotProtRNA$peak234120+FYTTD1_apaTotProtRNA$peak234122+FYTTD1_apaTotProtRNA$peak234123+FYTTD1_apaTotProtRNA$peak234124+FYTTD1_apaTotProtRNA$peak234125+FYTTD1_apaTotProtRNA$peak234131+FYTTD1_apaTotProtRNA$peak234135)
FYTTD1_lmResidRNA_sum=glance(FYTTD1_lmResidAPA)</code></pre>
</div>
</div>
<div id="plot-with-residuals" class="section level3">
<h3>Plot with Residuals</h3>
<pre class="r"><code>RNA_Residuals=rbind(CCDC51_lmResidRNA_sum[1,2],DTD1_lmResidRNA_sum[1,2],DHRS7B_lmResidRNA_sum[1,2],MMAB_lmResidRNA_sum[1,2],FYTTD1_lmResidRNA_sum[1,2])
adjR2_resid=cbind(adjR2,RNA_Residuals)
colnames(adjR2_resid)=c(&quot;Gene&quot;, &quot;RNA&quot;, &quot;APA&quot;, &quot;RNAandAPA&quot;, &quot;APAinRNAResiduals&quot;)

adjR2_resid_melt=melt(adjR2_resid, id.vars=&quot;Gene&quot;)
colnames(adjR2_resid_melt)=c(&quot;Gene&quot;, &quot;Model&quot;, &quot;AdjR2&quot;)
adjR2_resid_melt_sub=adjR2_resid_melt %&gt;% filter(Model==&quot;APAinRNAResiduals&quot; | Model== &quot;RNA&quot;)</code></pre>
<pre class="r"><code>ggplot(adjR2_resid_melt_sub, aes(x=Gene, y=AdjR2, by=Model, fill=Model)) + geom_bar(stat=&quot;identity&quot;, position = &quot;dodge&quot;) + labs(title=&quot;Adjusted R2 for variation\n in protein effect sizes explained in each model&quot;,x=&quot;pQTL Gene&quot;) + scale_fill_brewer(palette=&quot;Paired&quot;) </code></pre>
<p><img src="figure/explainpQTLs.Rmd/unnamed-chunk-54-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-54-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/8bb97f621f41d8d8f0ff04f66ca34d1578902468/docs/figure/explainpQTLs.Rmd/unnamed-chunk-54-1.png" target="_blank">8bb97f6</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-12-03
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>There is most likely an overfitting problem.</p>
</div>
</div>
<div id="inference" class="section level2">
<h2>Inference</h2>
<p>RSS= residuial sum of squares</p>
<pre class="r"><code>#subset model 
RSS_sub=anova(CCDC51_lmProtRNA)[&quot;Residuals&quot;, &quot;Sum Sq&quot;]
df_sub=1466

#full model
RSS_full=anova(CCDC51_lmProtAPARNA)[&quot;Residuals&quot;, &quot;Sum Sq&quot;]
df_full=14661
anova(CCDC51_lmProtAPARNA)</code></pre>
<pre><code>Analysis of Variance Table

Response: Protein
             Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
RNA           1  19.427  19.427 175.950 &lt; 2.2e-16 ***
peak216857    1 100.409 100.409 909.411 &lt; 2.2e-16 ***
peak216858    1  26.726  26.726 242.057 &lt; 2.2e-16 ***
peak216859    1   3.528   3.528  31.951 1.898e-08 ***
peak216860    1  13.273  13.273 120.211 &lt; 2.2e-16 ***
peak216867    1   6.939   6.939  62.846 4.402e-15 ***
Residuals  1461 161.310   0.110                      
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>F_CCDC51=((RSS_sub-RSS_full)/(df_sub-df_full))/(RSS_full/df_full)
F_CCDC51</code></pre>
<pre><code>[1] -1.039217</code></pre>
<pre class="r"><code>#i can do this with the annova function 
anova(CCDC51_lmProtRNA,CCDC51_lmProtAPARNA)</code></pre>
<pre><code>Analysis of Variance Table

Model 1: Protein ~ RNA
Model 2: Protein ~ RNA + peak216857 + peak216858 + peak216859 + peak216860 + 
    peak216867
  Res.Df    RSS Df Sum of Sq     F    Pr(&gt;F)    
1   1466 312.18                                 
2   1461 161.31  5    150.87 273.3 &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>anova(MMAB_lmProtRNA,MMAB_lmProtAPARNA)</code></pre>
<pre><code>Analysis of Variance Table

Model 1: protein ~ RNA
Model 2: protein ~ RNA + peak78754 + peak78755 + peak78756 + peak78757 + 
    peak78758 + peak78759 + peak78760 + peak78761 + peak78762 + 
    peak78763 + peak78764 + peak78765 + peak78766
  Res.Df    RSS Df Sum of Sq      F    Pr(&gt;F)    
1   2942 529.17                                  
2   2929 245.38 13    283.79 260.57 &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>anova(DHRS7B_lmProtRNA,DHRS7B_lmProtAPARNA)</code></pre>
<pre><code>Analysis of Variance Table

Model 1: protein ~ RNA
Model 2: protein ~ peak132690 + peak132692 + peak132693 + peak132694 + 
    peak132720 + peak132721 + peak132722 + peak132723 + peak132724 + 
    peak132725 + peak132728 + peak132729 + peak132730 + peak132732 + 
    peak132733 + peak132734 + peak132735 + peak132738 + peak132739 + 
    peak132740 + peak132741 + peak132742 + peak132744 + peak132745 + 
    peak132746 + peak132747 + peak132748 + peak132749 + peak132750 + 
    peak132751 + peak132753 + peak132754 + peak132755 + peak132756 + 
    peak132757 + peak132760 + peak132761 + peak132762 + peak132763 + 
    peak132764 + peak132766 + peak132774 + peak132775 + peak132777 + 
    peak132778 + peak132780
  Res.Df    RSS Df Sum of Sq     F    Pr(&gt;F)    
1   3071 479602                                 
2   3026   2144 45    477458 14978 &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>anova(DHRS7B_lmProtRNA,DHRS7B_lmProtAPARNA)</code></pre>
<pre><code>Analysis of Variance Table

Model 1: protein ~ RNA
Model 2: protein ~ peak132690 + peak132692 + peak132693 + peak132694 + 
    peak132720 + peak132721 + peak132722 + peak132723 + peak132724 + 
    peak132725 + peak132728 + peak132729 + peak132730 + peak132732 + 
    peak132733 + peak132734 + peak132735 + peak132738 + peak132739 + 
    peak132740 + peak132741 + peak132742 + peak132744 + peak132745 + 
    peak132746 + peak132747 + peak132748 + peak132749 + peak132750 + 
    peak132751 + peak132753 + peak132754 + peak132755 + peak132756 + 
    peak132757 + peak132760 + peak132761 + peak132762 + peak132763 + 
    peak132764 + peak132766 + peak132774 + peak132775 + peak132777 + 
    peak132778 + peak132780
  Res.Df    RSS Df Sum of Sq     F    Pr(&gt;F)    
1   3071 479602                                 
2   3026   2144 45    477458 14978 &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<pre class="r"><code>anova(FYTTD1_lmProtRNA,FYTTD1_lmProtAPARNA)</code></pre>
<pre><code>Analysis of Variance Table

Model 1: protein ~ RNA
Model 2: protein ~ RNA + peak234016 + peak234082 + peak234088 + peak234089 + 
    peak234090 + peak234092 + peak234093 + peak234094 + peak234095 + 
    peak234099 + peak234101 + peak234103 + peak234104 + peak234105 + 
    peak234106 + peak234107 + peak234108 + peak234111 + peak234113 + 
    peak234114 + peak234117 + peak234119 + peak234120 + peak234122 + 
    peak234123 + peak234124 + peak234125 + peak234131 + peak234135
  Res.Df     RSS Df Sum of Sq      F    Pr(&gt;F)    
1   3409 143.379                                  
2   3380  77.295 29    66.083 99.645 &lt; 2.2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>The models are significantly different</p>
<div id="random-values" class="section level3">
<h3>Random values</h3>
<pre class="r"><code>CCDC51_apaTotProtRNA_rand=transform(CCDC51_apaTotProtRNA,peak216857=sample(peak216857), peak216858=sample(peak216858),peak216859=sample(peak216859), peak216860=sample(peak216860) ,peak216867=sample(peak216867))


CCDC51_lmProtAPA_rand=lm(CCDC51_RNA_Resid ~ CCDC51_apaTotProtRNA_rand$peak216857 + CCDC51_apaTotProtRNA_rand$peak216858 + CCDC51_apaTotProtRNA_rand$peak216859 + CCDC51_apaTotProtRNA_rand$peak216860 + CCDC51_apaTotProtRNA_rand$peak216867)


summary(CCDC51_lmProtAPA_rand)</code></pre>
<pre><code>
Call:
lm(formula = CCDC51_RNA_Resid ~ CCDC51_apaTotProtRNA_rand$peak216857 + 
    CCDC51_apaTotProtRNA_rand$peak216858 + CCDC51_apaTotProtRNA_rand$peak216859 + 
    CCDC51_apaTotProtRNA_rand$peak216860 + CCDC51_apaTotProtRNA_rand$peak216867)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.28924 -0.25246 -0.06547  0.25200  2.14234 

Coefficients:
                                      Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)                           0.008342   0.012878   0.648   0.5172
CCDC51_apaTotProtRNA_rand$peak216857  0.080512   0.042575   1.891   0.0588
CCDC51_apaTotProtRNA_rand$peak216858  0.020707   0.030660   0.675   0.4995
CCDC51_apaTotProtRNA_rand$peak216859 -0.046768   0.046426  -1.007   0.3139
CCDC51_apaTotProtRNA_rand$peak216860 -0.027177   0.042205  -0.644   0.5197
CCDC51_apaTotProtRNA_rand$peak216867 -0.031990   0.040653  -0.787   0.4315
                                      
(Intercept)                           
CCDC51_apaTotProtRNA_rand$peak216857 .
CCDC51_apaTotProtRNA_rand$peak216858  
CCDC51_apaTotProtRNA_rand$peak216859  
CCDC51_apaTotProtRNA_rand$peak216860  
CCDC51_apaTotProtRNA_rand$peak216867  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.4611 on 1462 degrees of freedom
Multiple R-squared:  0.004125,  Adjusted R-squared:  0.0007196 
F-statistic: 1.211 on 5 and 1462 DF,  p-value: 0.3014</code></pre>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] bindrcpp_0.2.2  cowplot_0.9.3   broom_0.5.0     reshape2_1.4.3 
 [5] forcats_0.3.0   stringr_1.3.1   dplyr_0.7.6     purrr_0.2.5    
 [9] readr_1.1.1     tidyr_0.8.1     tibble_1.4.2    ggplot2_3.0.0  
[13] tidyverse_1.2.1 workflowr_1.1.1

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4   haven_1.1.2        lattice_0.20-35   
 [4] colorspace_1.3-2   htmltools_0.3.6    yaml_2.2.0        
 [7] rlang_0.2.2        R.oo_1.22.0        pillar_1.3.0      
[10] glue_1.3.0         withr_2.1.2        R.utils_2.7.0     
[13] RColorBrewer_1.1-2 modelr_0.1.2       readxl_1.1.0      
[16] bindr_0.1.1        plyr_1.8.4         munsell_0.5.0     
[19] gtable_0.2.0       cellranger_1.1.0   rvest_0.3.2       
[22] R.methodsS3_1.7.1  evaluate_0.11      labeling_0.3      
[25] knitr_1.20         Rcpp_0.12.19       scales_1.0.0      
[28] backports_1.1.2    jsonlite_1.5       hms_0.4.2         
[31] digest_0.6.17      stringi_1.2.4      grid_3.5.1        
[34] rprojroot_1.3-2    cli_1.0.1          tools_3.5.1       
[37] magrittr_1.5       lazyeval_0.2.1     crayon_1.3.4      
[40] whisker_0.3-2      pkgconfig_2.0.2    xml2_1.2.0        
[43] lubridate_1.7.4    assertthat_0.2.0   rmarkdown_1.10    
[46] httr_1.3.1         rstudioapi_0.8     R6_2.3.0          
[49] nlme_3.1-137       git2r_0.23.0       compiler_3.5.1    </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
