<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Chromatin marks and PAS</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Chromatin marks and PAS</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>3/20/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-03-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>threeprimeseq/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtreef3740ea146d6ce93131363aa0579cbd2f08ae054targetblankf3740eaa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/f3740ea146d6ce93131363aa0579cbd2f08ae054" target="_blank">f3740ea</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtreef3740ea146d6ce93131363aa0579cbd2f08ae054targetblankf3740eaa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/perm_QTL_trans_noMP_5percov/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/4suDataIGV.Rmd
    Untracked:  analysis/AdaptTonyExampleplots.Rmd
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/EvaleQTLs.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/groSeqAnalysis.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/peaksWithMisprimming.Rmd
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  analysis/verifybam_dubs.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/AllPeak_counts/
    Untracked:  data/ApaQTLs/
    Untracked:  data/ApaQTLs_otherPhen/
    Untracked:  data/CTCF/
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/DistTXN2Peak_genelocAnno/
    Untracked:  data/EmpiricalDists/
    Untracked:  data/FeatureoverlapPeaks/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/GWAS_overlap/
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/LocusZoom_Unexp/
    Untracked:  data/LocusZoom_proc/
    Untracked:  data/MatchedSnps/
    Untracked:  data/NucSpecQTL/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeakCounts_noMP_5perc/
    Untracked:  data/PeakCounts_noMP_genelocanno/
    Untracked:  data/PeakUsage/
    Untracked:  data/PeakUsage_noMP/
    Untracked:  data/PeakUsage_noMP_GeneLocAnno/
    Untracked:  data/PeaksUsed/
    Untracked:  data/PeaksUsed_noMP_5percCov/
    Untracked:  data/PolyA_DB/
    Untracked:  data/QTL_overlap/
    Untracked:  data/RNAdecay/
    Untracked:  data/RNAkalisto/
    Untracked:  data/RefSeq_annotations/
    Untracked:  data/Replicates_usage/
    Untracked:  data/Signal_Loc/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/WASP_STAT/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/apaExamp_proc/
    Untracked:  data/apaQTL_examp_noMP/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_GeneLocAnno/
    Untracked:  data/diff_iso_proc/
    Untracked:  data/diff_iso_trans/
    Untracked:  data/eQTL_inAPA/
    Untracked:  data/eQTLs_Lietal/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/molPheno_noMP/
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_inAPA/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/pacbio_cov/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/peaks4DT/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov_3UTR/
    Untracked:  data/perm_QTL_diffWindow/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/protAndAPAAndExplmRes.Rda
    Untracked:  data/protAndAPAlmRes.Rda
    Untracked:  data/protAndExpressionlmRes.Rda
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  data/threePrimeSeqMetaData55Ind.txt
    Untracked:  data/threePrimeSeqMetaData55Ind.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_redobatch4.xlsx
    Untracked:  data/~$threePrimeSeqMetaData55Ind_redobatch4.xlsx
    Untracked:  manuscript/
    Untracked:  output/LZ/
    Untracked:  output/deeptools_plots/
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/NewPeakPostMP.Rmd
    Modified:   analysis/NuclearSpecQTL.Rmd
    Modified:   analysis/PeakToXper.Rmd
    Modified:   analysis/RNAdecayAndAPA.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/characterize_apaQTLs.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diffIsoAnalysisNewMapping.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/fixBWChromNames.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/initialPacBioQuant.Rmd
    Modified:   analysis/mispriming_approach.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlapMolQTL.opposite.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/peakQCPPlots.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/pipeline_55Ind.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/test.smash.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   analysis/unexplainedeQTL_analysis.Rmd
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/f3740ea146d6ce93131363aa0579cbd2f08ae054/analysis/HistoneModandPAS.Rmd" target="_blank">f3740ea</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-25
</td>
<td>
add top and bottom dnase
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/3eac46483466f9e52a552878598dabc4f9f107c2/docs/HistoneModandPAS.html" target="_blank">3eac464</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/4939dfe6fc6695823f2828c8b07d127241cd90d5/analysis/HistoneModandPAS.Rmd" target="_blank">4939dfe</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-23
</td>
<td>
look at mnase at all categories
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/1b7f0885f9b36cd072ea20de25d613487aa07c9e/docs/HistoneModandPAS.html" target="_blank">1b7f088</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/c02e9279dd89b96d3a8f9df7385ec632fd7fc1e3/analysis/HistoneModandPAS.Rmd" target="_blank">c02e927</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-21
</td>
<td>
add mnase merge chipseq
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/a6b0fe4ff82b681fdb4b2a981efcdaa0b440bf91/docs/HistoneModandPAS.html" target="_blank">a6b0fe4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/54168fdd5f406aad21c63d2e5050e99b9cc80f27/analysis/HistoneModandPAS.Rmd" target="_blank">54168fd</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-20
</td>
<td>
add histone mod analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ──────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.1.0       ✔ purrr   0.3.1  
✔ tibble  2.0.1       ✔ dplyr   0.8.0.1
✔ tidyr   0.8.3       ✔ stringr 1.4.0  
✔ readr   1.3.1       ✔ forcats 0.4.0  </code></pre>
<pre><code>Warning: package &#39;tibble&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;purrr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;forcats&#39; was built under R version 3.5.2</code></pre>
<pre><code>── Conflicts ─────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.2.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>Warning: package &#39;data.table&#39; was built under R version 3.5.2</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>Warning: package &#39;cowplot&#39; was built under R version 3.5.2</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<p><a href="http://science.sciencemag.org/content/352/6291/aad9926.full?ijkey=fkp/DIzVNS9RY&amp;keytype=ref&amp;siteid=sci" class="uri">http://science.sciencemag.org/content/352/6291/aad9926.full?ijkey=fkp/DIzVNS9RY&amp;keytype=ref&amp;siteid=sci</a></p>
<p>This article talks about chromatin modifications for heterochromatin downstream of PAS. I will look at enrichment for repressive histone marks downstream of my called PAS.</p>
<p>Repressive marks H3K27me3, H3K9me3</p>
<p><a href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeUwHistone/" class="uri">http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeUwHistone/</a></p>
<p>H3k27me3 H3k36me3 H3k4me3</p>
<p>Deeptools plot</p>
<p>h3k27me3DTmypeaks.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=h3k27me3DTmypeaks
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=h3k27me3DTmypeaks.out
#SBATCH --error=h3k27me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep1.bigWig -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3K27me3_myPeaksNompfilt.gz


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3K27me3_myPeaksNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;H3K27me3 at All Called PAS&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3K27me3_myPeaksNompfilt.png</code></pre>
<p>Put all of the marks on one plot:</p>
<p>I also want to just use the last base of the peak APAPAS_5percCov_fixedStrand.bed histonemarksDTmypeaks.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=histonemarksDTmypeaks
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=histonemarksDTmypeaks.out
#SBATCH --error=histonemarksDTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k36me3StdRawRep1.bigWig  /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k4me3StdRawRep1.bigWig -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/HistMarks_myPeaksNompfilt.gz


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/HistMarks_myPeaksNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;Histone marks at All Called PAS&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/HistMarks_myPeaksNompfilt.png</code></pre>
<p>Scales are too different to put these on the same spot:</p>
<p>H3k27me3DTmyPAS.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=H3k27me3DTmypeaks.sh
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=H3k27me3DTmypeaks.out
#SBATCH --error=H3k27me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bw  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 5000 -a 5000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt.gz --outFileNameMatrix /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt_matrix.txt --outFileSortedRegions /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt_SortedRegions.txt 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;H3k27me3 at All Called PAS&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt.png</code></pre>
<p>H3k36me3DTmyPAS.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=H3k27me3DTmypeaks.sh
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=H3k36me3DTmypeaks.out
#SBATCH --error=H3k36me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bw   -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 5000 -a 5000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt.gz --outFileNameMatrix /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt_matrix.txt --outFileSortedRegions /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt_SortedRegions.txt 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;H3k36me3 at All Called PAS&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt.png</code></pre>
<p>H3k4me3DTmyPAS.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=H3k4me3DTmypeaks.sh
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=H3k36me3DTmypeaks.out
#SBATCH --error=H3k36me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bw  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 5000 -a 5000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt.gz --outFileNameMatrix project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt_matrix.txt --outFileSortedRegions /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt_SortedRegions.txt 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;H3k4me3 at All Called PAS&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt.png</code></pre>
<div id="download-both-replicates-for-these-and-merge" class="section level2">
<h2>Download both replicates for these and merge:</h2>
<p>mergeH3k27me3.sh</p>
<pre class="bash"><code>
#!/bin/bash

#SBATCH --job-name=mergeH3k27me3
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeH3k27me3.out
#SBATCH --error=mergeH3k27me3.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env


bigWigMerge /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep2.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bedGraph


sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bedGraph /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.sort.bedGraph

bedGraphToBigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.sort.bedGraph /project2/gilad/briana/genome_anotation_data/chrom.length.chr.txt  /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bw


</code></pre>
<p>mergeH3k36me3.sh</p>
<pre class="bash"><code>
#!/bin/bash

#SBATCH --job-name=mergeH3k36me3
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeH3k36me3.out
#SBATCH --error=mergeH3k36me3.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env


bigWigMerge /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k36me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k36me3StdRawRep2.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bedGraph


sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bedGraph &gt; /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.sort.bedGraph

bedGraphToBigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.sort.bedGraph /project2/gilad/briana/genome_anotation_data/chrom.length.chr.txt  /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bw


</code></pre>
<p>mergeH3k4me3.sh</p>
<pre class="bash"><code>
#!/bin/bash

#SBATCH --job-name=mergeH3k4me3
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeH3k4me3.out
#SBATCH --error=mergeH3k4me3.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env


bigWigMerge /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k4me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k4me3StdRawRep2.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bedGraph


sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bedGraph &gt;/project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.sort.bedGraph

bedGraphToBigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.sort.bedGraph /project2/gilad/briana/genome_anotation_data/chrom.length.chr.txt  /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bw


</code></pre>
</div>
<div id="mnase" class="section level2">
<h2>MNASE:</h2>
<p>MNASEmyPAS.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=MNASEmyPAS
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAS.out
#SBATCH --error=MNASEmyPAS.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;MNASE at All Called PAS&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNompfilt.png</code></pre>
<p>Try second mnase track.</p>
<p>MNASEmyPAS_secondfile.sh</p>
<pre class="bash"><code>
#!/bin/bash

#SBATCH --job-name=MNASEmyPAS_secondfile
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAS_secondfile.out
#SBATCH --error=MNASEmyPAS_secondfile.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeSydhNsomeGm12878Sig.bigWig  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase2_myPeaksNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase2_myPeaksNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;MNASE at All Called PAS&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase2_myPeaksNompfilt.png
</code></pre>
<p>Convert to PAS ratehr than peak: APAPeaks_5percCov_fixedStrand_INTRON.bed</p>
<pre class="bash"><code>python Peak2PAS.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_INTRON.bed /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_INTRON.bed</code></pre>
<p>Run this with intronic vs utr<br />
MNASEmyPASIntron.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=MNASEmyPASIntron
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASIntron.out
#SBATCH --error=MNASEmyPASIntron.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_INTRON.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksIntronNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksIntronNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;MNASE at Intronic Called PAS&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksIntronNompfilt.png</code></pre>
<p>Nuclear specific:</p>
<pre class="bash"><code>python Peak2PAS.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc.bed /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc.bed
</code></pre>
<p>MNASEmyPASNuclear.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=MNASEmyPASNuclear
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASNuclear.out
#SBATCH --error=MNASEmyPASNuclear.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;MNASE at Nuclear specific PAS&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearNompfilt.png</code></pre>
<p>Nuclear Intronic:</p>
<p>APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron.bed</p>
<pre class="bash"><code>python Peak2PAS.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron.bed /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc_Intron.bed
</code></pre>
<p>MNASEmyPASNuclearIntronic.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=MNASEmyPASNuclearIntronic
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASNuclearIntronic.out
#SBATCH --error=MNASEmyPASNuclearIntronic.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc_Intron.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearIntronNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearIntronNompfilt.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;MNASE at Nuclear specific PAS in Intron&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearIntronNompfilt.png</code></pre>
<div id="sepatate-by-mean-usage-strongweak" class="section level3">
<h3>Sepatate by mean usage (strong/weak)</h3>
<div id="total" class="section level4">
<h4>Total</h4>
<p>Mean usage:</p>
<p>top 20% and bottom 20%. by mean usage</p>
<pre class="r"><code>meanUsageTot=read.table(&quot;../data/PeaksUsed_noMP_5percCov/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt&quot;, col.names = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;,&quot;gene&quot;, &quot;strand&quot;, &quot;name&quot;, &quot;meanUsage&quot;)) %&gt;% mutate(perc=ntile(meanUsage,n=100)) 


meanUsageTot_bot20=meanUsageTot %&gt;% filter(perc &lt;20) %&gt;% dplyr::select(name)
meanUsageTot_top20=meanUsageTot %&gt;% filter(perc &gt;80)%&gt;% dplyr::select(name)</code></pre>
<p>Write out the peaks:</p>
<pre class="r"><code>write.table(meanUsageTot_bot20, file=&quot;../data/PeaksUsed_noMP_5percCov/TotalPeaksBottom20Usage.txt&quot;, quote=F, row.names = F, col.names = F, sep=&quot;\t&quot;)

write.table(meanUsageTot_top20, file=&quot;../data/PeaksUsed_noMP_5percCov/TotalPeaksTop20Usage.txt&quot;, quote=F, row.names = F, col.names = F, sep=&quot;\t&quot;)</code></pre>
<p>Copy to /project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/</p>
<p>Subset PAS file:</p>
<p>subsetPAStottop20perc.py</p>
<pre class="bash"><code>top20=open(&quot;/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/TotalPeaksTop20Usage.txt&quot;, &quot;r&quot;)


AllPas=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed&quot;, &quot;r&quot;)

Top20PAS=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_top20inTotal.bed&quot;, &quot;w&quot;)





def subsetPAS(use, outpas, PAS):
    okPAS={}
    for ln in use:
        peak=ln.strip()
        okPAS[peak]=&quot;&quot;
    for ln in PAS:
        peaknum=ln.split()[3].split(&quot;:&quot;)[-1]
        print
        if peaknum in okPAS.keys():
            outpas.write(ln)
    outpas.close()


subsetPAS(top20, Top20PAS, AllPas)</code></pre>
<p>subsetPAStotbottom20perc.py</p>
<pre class="bash"><code>
bottom20=open(&quot;/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/TotalPeaksBottom20Usage.txt&quot;, &quot;r&quot;)

AllPas=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed&quot;, &quot;r&quot;)


Bottom20PAS=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_bottom20inTotal.bed&quot;, &quot;w&quot;)




def subsetPAS(use, outpas, PAS):
    okPAS={}
    for ln in use:
        peak=ln.strip()
        okPAS[peak]=&quot;&quot;
    for ln in PAS:
        peaknum=ln.split()[3].split(&quot;:&quot;)[-1]
        print
        if peaknum in okPAS.keys():
            outpas.write(ln)
    outpas.close()


subsetPAS(bottom20, Bottom20PAS, AllPas)
</code></pre>
<p>Deeptools plots for these:</p>
<p>MNASEmyPAStop20tot.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=MNASEmyPAStop20tot
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAStop20tot.out
#SBATCH --error=MNASEmyPAStop20tot.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_top20inTotal.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20tot.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20tot.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;MNASE at Top 20% Total Usage&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20tot.png</code></pre>
<p>MNASEmyPASbottom20tot.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=MNASEmyPASbottom20tot
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASbottom20tot.out
#SBATCH --error=MNASEmyPASbottom20tot.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_bottom20inTotal.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20tot.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20tot.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;MNASE at Bottom 20% Total Usage&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20tot.png</code></pre>
</div>
<div id="nuclear" class="section level4">
<h4>Nuclear</h4>
<p>Mean usage:</p>
<p>top 20% and bottom 20%. by mean usage</p>
<pre class="r"><code>meanUsageNuc=read.table(&quot;../data/PeaksUsed_noMP_5percCov/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt&quot;, col.names = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;,&quot;gene&quot;, &quot;strand&quot;, &quot;name&quot;, &quot;meanUsage&quot;)) %&gt;% mutate(perc=ntile(meanUsage,n=100)) 


meanUsageNuc_bot20=meanUsageNuc %&gt;% filter(perc &lt;20) %&gt;% dplyr::select(name)
meanUsageNuc_top20=meanUsageNuc %&gt;% filter(perc &gt;80)%&gt;% dplyr::select(name)</code></pre>
<p>Write out the peaks:</p>
<pre class="r"><code>write.table(meanUsageNuc_bot20, file=&quot;../data/PeaksUsed_noMP_5percCov/NuclearPeaksBottom20Usage.txt&quot;, quote=F, row.names = F, col.names = F, sep=&quot;\t&quot;)

write.table(meanUsageNuc_top20, file=&quot;../data/PeaksUsed_noMP_5percCov/NuclearPeaksTop20Usage.txt&quot;, quote=F, row.names = F, col.names = F, sep=&quot;\t&quot;)</code></pre>
<p>Copy to /project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/</p>
<p>Subset PAS file:</p>
<p>subsetPASnuctop20perc.py</p>
<pre class="bash"><code>top20=open(&quot;/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/NuclearPeaksTop20Usage.txt&quot;, &quot;r&quot;)


AllPas=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed&quot;, &quot;r&quot;)

Top20PAS=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_top20inNuclear.bed&quot;, &quot;w&quot;)





def subsetPAS(use, outpas, PAS):
    okPAS={}
    for ln in use:
        peak=ln.strip()
        okPAS[peak]=&quot;&quot;
    for ln in PAS:
        peaknum=ln.split()[3].split(&quot;:&quot;)[-1]
        print
        if peaknum in okPAS.keys():
            outpas.write(ln)
    outpas.close()


subsetPAS(top20, Top20PAS, AllPas)</code></pre>
<p>subsetPASnucbottom20perc.py</p>
<pre class="bash"><code>
bottom20=open(&quot;/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/NuclearPeaksBottom20Usage.txt&quot;, &quot;r&quot;)

AllPas=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed&quot;, &quot;r&quot;)


Bottom20PAS=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_bottom20inNuclear.bed&quot;, &quot;w&quot;)




def subsetPAS(use, outpas, PAS):
    okPAS={}
    for ln in use:
        peak=ln.strip()
        okPAS[peak]=&quot;&quot;
    for ln in PAS:
        peaknum=ln.split()[3].split(&quot;:&quot;)[-1]
        print
        if peaknum in okPAS.keys():
            outpas.write(ln)
    outpas.close()


subsetPAS(bottom20, Bottom20PAS, AllPas)
</code></pre>
<p>Deeptools plots for these:</p>
<p>MNASEmyPAStop20nuc.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=MNASEmyPAStop20nuc
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAStop20nuc.out
#SBATCH --error=MNASEmyPAStop20nuc.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_top20inNuclear.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20nuc.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20nuc.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;MNASE at Top 20% Nuclear Usage&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20nuc.png</code></pre>
<p>MNASEmyPASbottom20nuc.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=MNASEmyPASbottom20nuc
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASbottom20nuc.out
#SBATCH --error=MNASEmyPASbottom20nuc.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_bottom20inNuclear.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20nuc.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20nuc.gz --refPointLabel &quot;Called PAS&quot; --plotTitle &quot;MNASE at Bottom 20% Nuclear Usage&quot; --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20nuc.png</code></pre>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] cowplot_0.9.4     data.table_1.12.0 workflowr_1.2.0  
 [4] forcats_0.4.0     stringr_1.4.0     dplyr_0.8.0.1    
 [7] purrr_0.3.1       readr_1.3.1       tidyr_0.8.3      
[10] tibble_2.0.1      ggplot2_3.1.0     tidyverse_1.2.1  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0       cellranger_1.1.0 plyr_1.8.4       pillar_1.3.1    
 [5] compiler_3.5.1   git2r_0.24.0     tools_3.5.1      digest_0.6.18   
 [9] lubridate_1.7.4  jsonlite_1.6     evaluate_0.13    nlme_3.1-137    
[13] gtable_0.2.0     lattice_0.20-38  pkgconfig_2.0.2  rlang_0.3.1     
[17] cli_1.0.1        rstudioapi_0.9.0 yaml_2.2.0       haven_2.1.0     
[21] xfun_0.5         withr_2.1.2      xml2_1.2.0       httr_1.4.0      
[25] knitr_1.21       hms_0.4.2        generics_0.0.2   fs_1.2.6        
[29] rprojroot_1.3-2  grid_3.5.1       tidyselect_0.2.5 glue_1.3.0      
[33] R6_2.4.0         readxl_1.3.0     rmarkdown_1.11   modelr_0.1.4    
[37] magrittr_1.5     whisker_0.3-2    backports_1.1.3  scales_1.0.0    
[41] htmltools_0.3.6  rvest_0.3.2      assertthat_0.2.0 colorspace_1.4-0
[45] stringi_1.3.1    lazyeval_0.2.1   munsell_0.5.0    broom_0.5.1     
[49] crayon_1.3.4    </code></pre>
</div>
</div>
</div>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
