<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->





<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-04-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>threeprimeseq/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree4f1ef747e56ec8f5f404b83076896bc340a26453targetblank4f1ef74a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/4f1ef747e56ec8f5f404b83076896bc340a26453" target="_blank">4f1ef74</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree4f1ef747e56ec8f5f404b83076896bc340a26453targetblank4f1ef74a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/perm_QTL_trans_noMP_5percov/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/4suDataIGV.Rmd
    Untracked:  analysis/AdaptTonyExampleplots.Rmd
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/EvaleQTLs.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/groSeqAnalysis.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/peaksWithMisprimming.Rmd
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  analysis/verifybam_dubs.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/Snakefile_proc
    Untracked:  code/cluster_proc.json
    Untracked:  code/config_proc.yaml
    Untracked:  code/snakemake.batch_proc
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  code/submit-snakemake_proc.sh
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/AllPeak_counts/
    Untracked:  data/ApaQTLs/
    Untracked:  data/ApaQTLs_otherPhen/
    Untracked:  data/CTCF/
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/DistTXN2Peak_genelocAnno/
    Untracked:  data/EmpiricalDists/
    Untracked:  data/ExampleQTLplot2/
    Untracked:  data/FeatureoverlapPeaks/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/GWAS_overlap/
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/LocusZoom_Unexp/
    Untracked:  data/LocusZoom_proc/
    Untracked:  data/MatchedSnps/
    Untracked:  data/NucSpecQTL/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeakCounts_noMP_5perc/
    Untracked:  data/PeakCounts_noMP_genelocanno/
    Untracked:  data/PeakUsage/
    Untracked:  data/PeakUsage_noMP/
    Untracked:  data/PeakUsage_noMP_GeneLocAnno/
    Untracked:  data/PeaksUsed/
    Untracked:  data/PeaksUsed_noMP_5percCov/
    Untracked:  data/PolyA_DB/
    Untracked:  data/QTL_overlap/
    Untracked:  data/RNAdecay/
    Untracked:  data/RNAkalisto/
    Untracked:  data/RefSeq_annotations/
    Untracked:  data/Replicates_usage/
    Untracked:  data/Signal_Loc/
    Untracked:  data/TNcompExamp/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/WASP_STAT/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/apaExamp_proc/
    Untracked:  data/apaQTL_examp_noMP/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_GeneLocAnno/
    Untracked:  data/diff_iso_proc/
    Untracked:  data/diff_iso_trans/
    Untracked:  data/eQTL_inAPA/
    Untracked:  data/eQTLs_Lietal/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/molPheno_noMP/
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_inAPA/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/pacbio_cov/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/peaks4DT/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov_3UTR/
    Untracked:  data/perm_QTL_diffWindow/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/protAndAPAAndExplmRes.Rda
    Untracked:  data/protAndAPAlmRes.Rda
    Untracked:  data/protAndExpressionlmRes.Rda
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/reseqCounts/
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  data/threePrimeSeqMetaData55Ind.txt
    Untracked:  data/threePrimeSeqMetaData55Ind.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_redobatch4.xlsx
    Untracked:  data/~$threePrimeSeqMetaData55Ind_redobatch4.xlsx
    Untracked:  manuscript/
    Untracked:  newbatch4files.csv
    Untracked:  newbatch4files.xlsx
    Untracked:  newbatch4fileswscript.txt
    Untracked:  output/APAqtlExamp/
    Untracked:  output/LZ/
    Untracked:  output/deeptools_plots/
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/ExampleQTLPlot2.Rmd
    Modified:   analysis/HistoneModandPAS.Rmd
    Modified:   analysis/IndPeakUsageDiff.Rmd
    Modified:   analysis/NewPeakPostMP.Rmd
    Modified:   analysis/NuclearSpecQTL.Rmd
    Modified:   analysis/PeakToXper.Rmd
    Modified:   analysis/RNAdecayAndAPA.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/characterize_apaQTLs.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diffIsoAnalysisNewMapping.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/exampleQTLfacetboxplot.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/fixBWChromNames.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/initialPacBioQuant.Rmd
    Modified:   analysis/mispriming_approach.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlapMolQTL.opposite.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/peakQCPPlots.Rmd
    Modified:   analysis/peakQCplotsSTARprocessing.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/pipeline_55Ind.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/test.smash.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   analysis/unexplainedeQTL_analysis.Rmd
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/4f1ef747e56ec8f5f404b83076896bc340a26453/analysis/index.Rmd" target="_blank">4f1ef74</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-25
</td>
<td>
add corrrelation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/d41ba5551ae27bad5c3c954d775a6192f80be66e/docs/index.html" target="_blank">d41ba55</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/44b054416947c2bad379addc34fecdac83920f9c/analysis/index.Rmd" target="_blank">44b0544</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-15
</td>
<td>
add facet bp examples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/1f4ea03e116e10a0012f9a716a6851d7789a4c63/docs/index.html" target="_blank">1f4ea03</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/70732bbcc6bd78d0388b605903acefa4e13ed4b4/analysis/index.Rmd" target="_blank">70732bb</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-04-10
</td>
<td>
add scripts for new snake
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/0ff28d4957bd5a0d80d0028a4bcd44637de65da9/docs/index.html" target="_blank">0ff28d4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-04-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a79791b96d06e58cd6c843eea29e8521fe5231ce/analysis/index.Rmd" target="_blank">a79791b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-04-03
</td>
<td>
start tot nuc example plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/608791d67a8113172c8473e2743b886600b19e86/docs/index.html" target="_blank">608791d</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/535642fe8d3b526767d20c4898806a544ff84291/analysis/index.Rmd" target="_blank">535642f</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-26
</td>
<td>
fix example plot html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/5244c0fda391b0e74a73e9cd89c3434dd4761e5e/docs/index.html" target="_blank">5244c0f</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/a6b0fe4ff82b681fdb4b2a981efcdaa0b440bf91/docs/index.html" target="_blank">a6b0fe4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/54168fdd5f406aad21c63d2e5050e99b9cc80f27/analysis/index.Rmd" target="_blank">54168fd</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-20
</td>
<td>
add histone mod analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/9aa10032efaf24eaa48170105df1b94ecd45e80e/docs/index.html" target="_blank">9aa1003</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d924da6d7a394ab66063c7872534805d7a4f8202/analysis/index.Rmd" target="_blank">d924da6</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-19
</td>
<td>
add ctcf analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/128366906da17e5b80b7abe6e206c98c8c124cef/docs/index.html" target="_blank">1283669</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/5d6ac937df2222499a1a0993bfc85b083557ccc0/analysis/index.Rmd" target="_blank">5d6ac93</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-15
</td>
<td>
add decay analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/f69638470a5d03a3c836d8356d741d8cf930aabf/docs/index.html" target="_blank">f696384</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/01c22a6b794b2f5471ad28dd180485717e6b8c46/analysis/index.Rmd" target="_blank">01c22a6</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-13
</td>
<td>
eQTL and unexp eQTL res
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/f88b41fc7e35c093a5866488bd949fa5f60e541f/docs/index.html" target="_blank">f88b41f</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/f9e94cfa5421b15c3786b50cd787034b7150203a/analysis/index.Rmd" target="_blank">f9e94cf</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-13
</td>
<td>
add net seq analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/4b2da32f2a5a7c725d211e4e6434550568cd84ab/docs/index.html" target="_blank">4b2da32</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/8a6b54e054a49b181763f8fd740bac65a30919ed/docs/index.html" target="_blank">8a6b54e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/8871122719aceb2b378022b813b069f09fd6225d/analysis/index.Rmd" target="_blank">8871122</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-11
</td>
<td>
subset to x%
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/9d234b6f770ae88cad8dd3627a9029e84d610d41/docs/index.html" target="_blank">9d234b6</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/55a488c4129dd63a9130fbe87c2082fc68a46de9/docs/index.html" target="_blank">55a488c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ea54d47841fcff17fafb16f7b485ddc2d50861ad/analysis/index.Rmd" target="_blank">ea54d47</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-09
</td>
<td>
add new GWAS overlap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/ba63ea22968208956eeb9dff3f6fb3573863fed1/docs/index.html" target="_blank">ba63ea2</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/c200503cbf5505dd25e7f30321019a5758f6e9b4/analysis/index.Rmd" target="_blank">c200503</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
<td>
add signal site loc analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/08ca500a0c5718df214c6d722375b53bbc200210/docs/index.html" target="_blank">08ca500</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d5d08090e3553d1e536665adcdb6846f551af58c/analysis/index.Rmd" target="_blank">d5d0809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-04
</td>
<td>
add analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/901e191f5be2e2a853f754ae6274042cf49111c3/docs/index.html" target="_blank">901e191</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a96297c66918b1ba972743ba7124344cf76da185/analysis/index.Rmd" target="_blank">a96297c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-02
</td>
<td>
add signal site enrich analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/19ad929e25eada8de76cf98b9834a7e6bd4c38dc/docs/index.html" target="_blank">19ad929</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/f812272be3782d6f215df58f3e1326ad704a313b/analysis/index.Rmd" target="_blank">f812272</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-01
</td>
<td>
add erna and rep element analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/a259bd93bd6de74279295ee615ac6442108514c4/docs/index.html" target="_blank">a259bd9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/eb1a05c66815af97bc057d4c8c67624e66194348/analysis/index.Rmd" target="_blank">eb1a05c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
<td>
add pacbio analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/b74a969e73a748845e343b055aa1c987c334dc6a/docs/index.html" target="_blank">b74a969</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/2336f876998f368497b100cdd933b2c8f4f2ab6e/analysis/index.Rmd" target="_blank">2336f87</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-25
</td>
<td>
add unexplained QTL analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/194c94d7f4a7100e0297e57d6e41ae36a37bff31/docs/index.html" target="_blank">194c94d</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/41500e46b28e993851aa3bf6c0eab12a8c122d12/analysis/index.Rmd" target="_blank">41500e4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
add prot and rna
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/cefce8bdf6ed0313464d163ac695b11970d6899a/docs/index.html" target="_blank">cefce8b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/beb0d714e5bdb4af3c7bfaa998b39b49cbff51e8/docs/index.html" target="_blank">beb0d71</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/3c1f0490368cd5508664674f57fee5ac5025eb95/analysis/index.Rmd" target="_blank">3c1f049</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
add gwas overlap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/4ea438ecbc784c72fd7334c43e790d79f564aabb/docs/index.html" target="_blank">4ea438e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/bcb2f86006d7004d0560ed04a6ccd2a32f1c8f21/analysis/index.Rmd" target="_blank">bcb2f86</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-18
</td>
<td>
add qtl by per and diff iso
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/879be332f32f95550851f1ad6c41dcf69cb18d01/docs/index.html" target="_blank">879be33</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/f8c76ea9596e24e3cf8ae8994c66ebb89ae77c01/analysis/index.Rmd" target="_blank">f8c76ea</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-16
</td>
<td>
move peak QC plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/c2c77b631c7a234efb7abff3cbfa2a187c5586dd/docs/index.html" target="_blank">c2c77b6</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/b7a0e322230bf7234c98dc7518d4232b49cfda22/analysis/index.Rmd" target="_blank">b7a0e32</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-15
</td>
<td>
add window size qtl analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/4f17cca4f0d8625a915a916d0ab5e8c885a556e0/docs/index.html" target="_blank">4f17cca</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/b8cfd6ffc84bc9e2752e94ce76a747fe266bcc10/docs/index.html" target="_blank">b8cfd6f</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/3fea6443e9bd446a5da083b8b9ef810c249dca1a/analysis/index.Rmd" target="_blank">3fea644</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-07
</td>
<td>
add accountmapbias
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/450b3897ccf23b42719f2134a819affd08d61c0c/docs/index.html" target="_blank">450b389</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/6bac9f93c4b0b305fb44aaedbc730cdfe6e91567/analysis/index.Rmd" target="_blank">6bac9f9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-06
</td>
<td>
add distance plots for QC on APAqtls
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/cc6b1ee6f99f9021ccbee65119f77db8d2590725/docs/index.html" target="_blank">cc6b1ee</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/3445a3b2e1afc8fcbb21ef640a92b116324ab272/docs/index.html" target="_blank">3445a3b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/957a0ee240598e92f053139047f94037724100ee/analysis/index.Rmd" target="_blank">957a0ee</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-01
</td>
<td>
initiate 55 ind pipeline
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/a97adbc8c88ed1a800aabcfa7cc7fad5ab2df63e/docs/index.html" target="_blank">a97adbc</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a2911ff6b8b52d9320e90518db9ee0dc6433ff75/analysis/index.Rmd" target="_blank">a2911ff</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-28
</td>
<td>
start analysis for new gene peak assignemnt
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/712016289cf8bcd1ed567697099d7eaccc430e52/docs/index.html" target="_blank">7120162</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/9df8d7e46f7a0c276cc794dd1e69de0b64be90f1/analysis/index.Rmd" target="_blank">9df8d7e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-24
</td>
<td>
add linear model code with res for stephens lab
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/aefc33040326e05c44fd4305c5d26249f41ef089/docs/index.html" target="_blank">aefc330</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/fd184be6b12e4f11a774d09588aa5d3fdc5a26fd/analysis/index.Rmd" target="_blank">fd184be</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-22
</td>
<td>
add code for leafcutter on processed
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/6e2265324731fff1d31c0476dac4bfc5e9bf5761/docs/index.html" target="_blank">6e22653</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/7eaae548c220b51aa01de98d9fcb1dd54a0eeb21/analysis/index.Rmd" target="_blank">7eaae54</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-17
</td>
<td>
select peaks to use in deeptools plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/a5d48fd8f5408c3f49eadb94058a4b57b576e3a6/docs/index.html" target="_blank">a5d48fd</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/6aa94e46191d361cdb422647f48195789fdcf789/analysis/index.Rmd" target="_blank">6aa94e4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-16
</td>
<td>
plots for 5% usage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/e088c558ed7e0ea70b9287e27926bf2e08171251/docs/index.html" target="_blank">e088c55</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/6bc9243ded950fc7e317fd6ef901657801c66b46/analysis/index.Rmd" target="_blank">6bc9243</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-14
</td>
<td>
evaluate clean reads, make new file for misprime filter
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/580e24472c52b98efb062d7a15c0b436b2079a78/docs/index.html" target="_blank">580e244</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/42fcbdd1ff21ebf0e08b3df26501f37430f2290c/analysis/index.Rmd" target="_blank">42fcbdd</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-11
</td>
<td>
initialize mispriming approach file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/4d935ce0b3a14790b1c1af714ee809938c63fe79/docs/index.html" target="_blank">4d935ce</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/c3b3bbb4729ecbebed9f1b530d086b5a8f6024d9/analysis/index.Rmd" target="_blank">c3b3bbb</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-01-03
</td>
<td>
start covariate correlation with pc analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/785b8216e916fcc9ecfa66ee7f35622fd210b602/docs/index.html" target="_blank">785b821</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d99a6f3a3875fd4b2482084467c9875b1579f9ca/analysis/index.Rmd" target="_blank">d99a6f3</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-12-20
</td>
<td>
initializa comparison analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/6da90e9c6405976ca4cbeb59c6c07a30365e195f/docs/index.html" target="_blank">6da90e9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-12-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/1b3647af4a89d265d4326caa129d77cb256746f7/docs/index.html" target="_blank">1b3647a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-12-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/28e2214af39f9fa00f16b8464715af602c9f6d06/analysis/index.Rmd" target="_blank">28e2214</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-12-10
</td>
<td>
start qtls by included PC analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/cdfa5b2f6c15a6820451283a9dd704e13f5d4e6f/docs/index.html" target="_blank">cdfa5b2</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-12-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/655b58288654c0cff03f96faf4b235bb80d91391/analysis/index.Rmd" target="_blank">655b582</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-12-05
</td>
<td>
PCA with batch and read count
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/210b58b2a0f472cefb26b16fad513ea3ca49d9d7/docs/index.html" target="_blank">210b58b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/413c8fddd492447ac0d51a7b928c56a3c941f41f/analysis/index.Rmd" target="_blank">413c8fd</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-29
</td>
<td>
add filter QTL analysis and start explain pqtl
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/813a5000acdf4ddf85a19f3ae61c80bcfe41927a/docs/index.html" target="_blank">813a500</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/23c62c988586bb3cd15114d6c7bc28fba051c0fa/analysis/index.Rmd" target="_blank">23c62c9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-15
</td>
<td>
add locus zoom initial analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/e5b4d6a1db201c68f309fd701007d0b01610b8b2/docs/index.html" target="_blank">e5b4d6a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a9d3372ad4c9fc6ac9b95045a3abe7535cf8bc01/analysis/index.Rmd" target="_blank">a9d3372</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-15
</td>
<td>
fix gwas bullet
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/24821f2e800b26456114ff1400996f16ebbaff3d/docs/index.html" target="_blank">24821f2</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/2ec5ffd37187beac4a5124fbb700b2ea52a1e75a/docs/index.html" target="_blank">2ec5ffd</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/962e39bcfc173365546499232b60d586f6470f38/analysis/index.Rmd" target="_blank">962e39b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-07
</td>
<td>
move chromhmm analysis to its own analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/3de08482d4164845cac29ac8b3fccc3d0a8e2eea/docs/index.html" target="_blank">3de0848</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/b5f744f599ad5ec725eb3e31b86e804979f568b2/analysis/index.Rmd" target="_blank">b5f744f</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-11-06
</td>
<td>
initiate flash2mash
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/d01545f42315390a1a4fdac3f228b11a78b5e403/docs/index.html" target="_blank">d01545f</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/9c9486dfcd74721fe66d7e3a0eee0f51fef2566a/analysis/index.Rmd" target="_blank">9c9486d</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-30
</td>
<td>
add genoem tracks code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/0428c8cc7dbe48e523ec661d5a0215e784135d7b/docs/index.html" target="_blank">0428c8c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/42d0e3d7c7875581d9b5bd1d63dc658b03e21f11/analysis/index.Rmd" target="_blank">42d0e3d</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-29
</td>
<td>
add gwas overlap to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/de860f0e5f84639460e9157c55f0024274bfb7d4/docs/index.html" target="_blank">de860f0</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/96a97f4b407baa2e87100144f096a2a9c483e874/analysis/index.Rmd" target="_blank">96a97f4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-24
</td>
<td>
add nuclear characterization
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/5649809b5974649e2d31ecf7dc6441beb7e28081/docs/index.html" target="_blank">5649809</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/c8dbaf06efdfed0b7563c03c01ccc390b6cea6c0/analysis/index.Rmd" target="_blank">c8dbaf0</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-22
</td>
<td>
add apa prot rna overlap
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/eb02fbc3fdca9d239eeba2bac84bd1a4d20dc434/docs/index.html" target="_blank">eb02fbc</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/f819c8e3a4ef75111646f1ca1e8343711c4133f3/analysis/index.Rmd" target="_blank">f819c8e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-11
</td>
<td>
add distance metric analsis total apaQTL
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/a7e5fa22643969bcb59ea9f7ab3696f6b82430d9/docs/index.html" target="_blank">a7e5fa2</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/1e7fbbb20e4aadbe860965f014b2aa62b19fe821/analysis/index.Rmd" target="_blank">1e7fbbb</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-11
</td>
<td>
update format
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/729c084f3118cae3ba8e7adde5c23797025f43ff/docs/index.html" target="_blank">729c084</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/89f119a4d395b475d4f3e663f1509fb0b372b973/analysis/index.Rmd" target="_blank">89f119a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-11
</td>
<td>
update format
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/22aa0872ec0621164f376f8225f985be6112e20f/docs/index.html" target="_blank">22aa087</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/11f9dfa2e7e97a7279afc91e4f1a8f2cdb1f856e/analysis/index.Rmd" target="_blank">11f9dfa</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-08
</td>
<td>
overlap molQTL opp dir
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/077ed603f855c9fd95c9e77bb1008695e8ba3891/docs/index.html" target="_blank">077ed60</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/7369f8ebf5a521bb3cd265f35ae26915f39b5bee/docs/index.html" target="_blank">7369f8e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/f66679bc1d71db8c924eba94a35f4643d179ae7e/analysis/index.Rmd" target="_blank">f66679b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-02
</td>
<td>
overlap plots at peak level
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/2e0d959a51368bcfa9ebc00b54b6b86195e8835a/docs/index.html" target="_blank">2e0d959</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/9d6ee033c3a681f38437f2a0cee61cef97ac51da/analysis/index.Rmd" target="_blank">9d6ee03</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-01
</td>
<td>
add 4su plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/da0f70ef6a8662ab8ac9ee900915ed61e2e31fd6/docs/index.html" target="_blank">da0f70e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/fecb046a72c0ed7df318d9642fa62e4c7af56cd8/analysis/index.Rmd" target="_blank">fecb046</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-01
</td>
<td>
fix overlapfile
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/ac983db135c4ca1872cc58b4efea050275858722/docs/index.html" target="_blank">ac983db</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/35142fb308209cce516f74504f2f1a3abcfdff51/analysis/index.Rmd" target="_blank">35142fb</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-10-01
</td>
<td>
overlap QTL plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/149d033c96899a702b4c0a7320484cb980e3fb56/docs/index.html" target="_blank">149d033</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/aaed5fd02287453b28ab53c96adb5d51b5c385dd/docs/index.html" target="_blank">aaed5fd</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/eda266ec447da55ce6a5e62e907e634cbc547a20/analysis/index.Rmd" target="_blank">eda266e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-09-26
</td>
<td>
test peak to gene transcript dist
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/47cd4bf7e1f277a76f2c44c6869c62f2211868e9/docs/index.html" target="_blank">47cd4bf</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-09-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/b1bcf9916b400de7a1ae40bb2335c8ef064c9048/docs/index.html" target="_blank">b1bcf99</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-09-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/d3bb2874e82a9160c3a68644ff092be0c8d799d1/docs/index.html" target="_blank">d3bb287</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-09-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/f7934ce5afa448c50043b68c5920186b0c66f460/analysis/index.Rmd" target="_blank">f7934ce</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-09-24
</td>
<td>
wflow_publish(c(“index.Rmd”, “39indQC.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/f92a58fb8dd5bf15e528a9d1762e34ae3c25e005/docs/index.html" target="_blank">f92a58f</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-09-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/46b7343493337a9c20b5ce490b66cbb942467169/analysis/index.Rmd" target="_blank">46b7343</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-09-06
</td>
<td>
add overlap analysis with code to subset
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/b4bda1b4c6cb9cd811f4133bca3184c00583c706/docs/index.html" target="_blank">b4bda1b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-08-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/e000583e8d094de80cd8cb12bc6cb3933f15a645/docs/index.html" target="_blank">e000583</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-08-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a5f527674983b1420a3c82e09ac39e69218d690b/analysis/index.Rmd" target="_blank">a5f5276</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-08-30
</td>
<td>
initialize diff iso pipeline
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/c6dc97bf8c0353ab9179b44a1f9ed8210f4c0adb/docs/index.html" target="_blank">c6dc97b</a>
</td>
<td>
brimittleman
</td>
<td>
2018-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/5564e259b7b24257b0d4e275873dafca7b024166/docs/index.html" target="_blank">5564e25</a>
</td>
<td>
brimittleman
</td>
<td>
2018-08-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/6b1b51c30cedc1bbdf21009104f126a8b4add401/analysis/index.Rmd" target="_blank">6b1b51c</a>
</td>
<td>
brimittleman
</td>
<td>
2018-08-20
</td>
<td>
start qtl analsis, add to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/c67380e570e74625b061e504b345fae77f0e3a6e/docs/index.html" target="_blank">c67380e</a>
</td>
<td>
brimittleman
</td>
<td>
2018-08-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/38bfbaf6b3e1b6f8e8fd397af1b86d34dd711a9e/docs/index.html" target="_blank">38bfbaf</a>
</td>
<td>
brimittleman
</td>
<td>
2018-08-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/03761cbbf00f95021204eefca436dbb1f2146cbc/analysis/index.Rmd" target="_blank">03761cb</a>
</td>
<td>
brimittleman
</td>
<td>
2018-08-09
</td>
<td>
start peak explore analysis for 54 libraries
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/e5a8da629657f8df6a23af081905e7dcab9fe98d/docs/index.html" target="_blank">e5a8da6</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/422a428cf40c5283c7dc0ba0d3cc21d8739cddb3/analysis/index.Rmd" target="_blank">422a428</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-30
</td>
<td>
add peak cove pipeline and combined lane qc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/0398fdb061b51ffc376cca2a37bee74727674489/docs/index.html" target="_blank">0398fdb</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/2d41f11ceda8b51e78e2a040bc3ab99084c228a9/analysis/index.Rmd" target="_blank">2d41f11</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-25
</td>
<td>
add expand smash to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/3a5a8fe9a053a464ef64ed3c411bdedf82003927/docs/index.html" target="_blank">3a5a8fe</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/a34f9c17e0111475b79b8ba65678d7cd03ccf7e8/docs/index.html" target="_blank">a34f9c1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ca9cee943f269ffa18819efc8aa0ffa44f39f971/analysis/index.Rmd" target="_blank">ca9cee9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-17
</td>
<td>
add smash test to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/c3ca148d27b4e58f712221b619052926fd20a50e/docs/index.html" target="_blank">c3ca148</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/dc4a51af1b101200e1685b760ac3f5c78db359b3/docs/index.html" target="_blank">dc4a51a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/4fede8412f8277c95afc167709e9e370c386cf8b/analysis/index.Rmd" target="_blank">4fede84</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-16
</td>
<td>
add eval brain analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/cfc8f437a3e2701980064f4423d8042c6144dd1d/docs/index.html" target="_blank">cfc8f43</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/4a847692c90977347c2a192e8e68e1d2d02b6fa5/analysis/index.Rmd" target="_blank">4a84769</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-13
</td>
<td>
add analysis for coparing coverage of RNAseq and 3’ seq
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/210cd2b1dff2619ce69c54368d8d332412aa1677/docs/index.html" target="_blank">210cd2b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/8b622c3a9568a7f7f8c639b02a1da43563cebbd3/analysis/index.Rmd" target="_blank">8b622c3</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-07-05
</td>
<td>
add denovo to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/ee777df63592d6f252dd40a4cdc2c94baf53ffa9/docs/index.html" target="_blank">ee777df</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/789d8ef3fbeb727e66635f978480d6b768b290dd/analysis/index.Rmd" target="_blank">789d8ef</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-26
</td>
<td>
start test macs analysis. download package
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/30b415e3b95d62eafebb6a4824a29fba6c727be2/docs/index.html" target="_blank">30b415e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/08b593457c104b333a203741789e2d632e85bfa1/docs/index.html" target="_blank">08b5934</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/642faf0176cbdb3b251de2e62b9f5dc9d385bbdf/analysis/index.Rmd" target="_blank">642faf0</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-13
</td>
<td>
start PAS enrichment analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/4ef7d85554ed0db97548af29c6d994675af4837e/docs/index.html" target="_blank">4ef7d85</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/03c7b1448ea0c93a1d7a0b2d589e8ef964bf7bba/analysis/index.Rmd" target="_blank">03c7b14</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-11
</td>
<td>
start A analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/062808d7f8f0132374041057f946a2c28c32797b/docs/index.html" target="_blank">062808d</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/86cfd9ff2fdd20c9bcc0f2872ca016b383d81700/docs/index.html" target="_blank">86cfd9f</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a19683c3a42318377499f44fdd98f7af8ce4d2fd/analysis/index.Rmd" target="_blank">a19683c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-06-06
</td>
<td>
start dif isoform analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/710cf6a06e3904e99384341936f5c8b4a2a67428/docs/index.html" target="_blank">710cf6a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d58bc138fd71227816b58cffda2ccf2df7199808/analysis/index.Rmd" target="_blank">d58bc13</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-29
</td>
<td>
start 200 bp analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/96ecbb125b7504e34275073c4369760db2ef8c3a/docs/index.html" target="_blank">96ecbb1</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/2076ce905888ca32e1fc086f15e954c7dde05450/analysis/index.Rmd" target="_blank">2076ce9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-26
</td>
<td>
initial commit, gene level analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/b5cdd59432005e3c407edeab96984f4d90c65b45/docs/index.html" target="_blank">b5cdd59</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/9a49459368772f030622712864c469ecee83a53a/analysis/index.Rmd" target="_blank">9a49459</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-26
</td>
<td>
start map qc analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/924952b3c14f5a58131b1b99ae8be276d3b2d37f/docs/index.html" target="_blank">924952b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/f8ce0fcfc549967f30ad08e19962c8135bb5dc27/analysis/index.Rmd" target="_blank">f8ce0fc</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-09
</td>
<td>
Update to 1.0
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/f8ce0fcfc549967f30ad08e19962c8135bb5dc27/docs/index.html" target="_blank">f8ce0fc</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-09
</td>
<td>
Update to 1.0
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/3dd80ef12a6e2e8ba83dc7987ec22d667d4b7492/docs/index.html" target="_blank">3dd80ef</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/519447d91969a0dc7db57d52cc61ecd71a9c53ab/analysis/index.Rmd" target="_blank">519447d</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-08
</td>
<td>
Add total.nuc.rnaseq analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/889c9c6fa9085578dbb6d3e233c1ee787daef080/docs/index.html" target="_blank">889c9c6</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/9d196cf4b24ebee34e9624c46536a9c5aeca4f44/docs/index.html" target="_blank">9d196cf</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/972ef3effabfa330e5e75404d603f5ed2fb2f5df/docs/index.html" target="_blank">972ef3e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/6cc5321541eca63223ad1bd790071d38ebb5bd7d/analysis/index.Rmd" target="_blank">6cc5321</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-02
</td>
<td>
add home page information
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/e12dd6cde00d646f994e07a4b5bd67138ad08494/docs/index.html" target="_blank">e12dd6c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/0e95115337380d333adf3ee43adceea878e04ff8/analysis/index.Rmd" target="_blank">0e95115</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-01
</td>
<td>
add home page information
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/4c26d4c1641e60f953fe953c04c7e6151ec4511e/docs/index.html" target="_blank">4c26d4c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/539892a301ff49ee88909de2718bf6abac3f4155/analysis/index.Rmd" target="_blank">539892a</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2018-05-01
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Briana Mittleman</p>
<p>This is the site I will use for my human 3’ seq project. I will have my workflow and analysis here. The three prime seq protocol I am using is Lexogen, QuantSeq 3’ mRNA-Seq library kit. The first analysis will be on the total and nucleic fractions of the LCL used in the net-seq analysis found at <a href="https://brimittleman.github.io/Net-seq/" class="uri">https://brimittleman.github.io/Net-seq/</a></p>
<div id="initial-analysis-on-rna-seq-of-nuclear-and-total-fractions" class="section level2">
<h2>Initial analysis on RNA seq of nuclear and total fractions:</h2>
<ul>
<li><a href="total.nuc.rnaseq.html">Initial data QC</a></li>
</ul>
<div id="initial-analysis-on-first-32-3-seq-libraries" class="section level3">
<h3>Initial analysis on first 32 3’ seq libraries:</h3>
<ul>
<li><a href="initial_map_qc.html">Initial mapping QC</a></li>
<li><a href="initial_gene_qc.html">Initial gene leve QC</a><br />
</li>
<li><a href="cov.analysis.threeprime1.html">Compare coverage in Three prime seq to RNAseq</a><br />
</li>
<li><a href="cov200bpwind.html">Coverage in 200bp bins</a><br />
</li>
<li><a href="dif.iso.usage.leafcutter.html">Differential isoform usage between fractions</a><br />
</li>
<li><a href="explore.filters.html">Explore filter cuttoffs</a><br />
</li>
<li><a href="enrichmentPAS.html">Enrichment at PAS</a><br />
</li>
<li><a href="comb.lanes.qc.html">Second Lane data</a></li>
</ul>
</div>
</div>
<div id="explore-and-filter-mis-primming" class="section level2">
<h2>Explore and filter mis-primming:</h2>
<ul>
<li><a href="a.content.html">Look at number of As in bins</a><br />
</li>
<li><a href="filter_As.html">Create script to filter mapped A’s</a><br />
</li>
<li><a href="cleanupdtseq.internalpriming.html">Test cleanUpdTSeq for calling internal priming</a><br />
</li>
<li><a href="peak.cov.pipeline.html">Pipeline to create peaks and clean them</a></li>
</ul>
</div>
<div id="peak-calling" class="section level2">
<h2>Peak calling:</h2>
<ul>
<li><a href="test.max2.html">Testing macs2 on 3’seq data</a><br />
</li>
<li><a href="deNovopeakcalling.html">Call peak regions de-novo</a></li>
<li><a href="eval.brain.html">Compare peaks in Derti Brain set</a><br />
</li>
<li><a href="test.smash.html">Test SMASH smoothing</a><br />
</li>
<li><a href="expand.smash.html">Expand SMASH analysis for use on midway</a><br />
</li>
<li><a href="dataprocfigures.html">Data Processing Figure design</a></li>
</ul>
<div id="peak-coveragenominal-qtlsdiff-iso" class="section level3">
<h3>Peak coverage/Nominal QTLs/Diff iso:</h3>
<ul>
<li><a href="28ind.peak.explore.html">Start evaluating peak quantificaiton</a><br />
</li>
<li><a href="pheno.leaf.comb.html">Prepare the data for leafcutter QTL calling</a></li>
<li><a href="apaQTLwLeafcutter.html">Call nominal QTLs</a><br />
</li>
<li><a href="diff_iso_pipeline.html">Differential Isoform pipeline</a></li>
</ul>
</div>
<div id="rerun-analysis-opp-strand" class="section level3">
<h3>Rerun analysis opp strand:</h3>
<ul>
<li><a href="peakOverlap_oppstrand.html">Rerun QTL analysis with off strand</a></li>
</ul>
</div>
<div id="evaluate-and-learn-from-qtls" class="section level3">
<h3>Evaluate and learn from QTLs:</h3>
<ul>
<li><a href="overlap_qtls.html">Overlap QTLs with other molecular QTLs</a></li>
</ul>
</div>
</div>
<div id="analysis-for-all-39-indivduals" class="section level2">
<h2>Analysis for all 39 indivduals:</h2>
<ul>
<li><p><a href="39indQC.html">Read and peak QC plots</a></p></li>
<li><p><a href="genometrack_figs.html">Genome tracks to visualize reads</a></p></li>
<li><p><a href="apaQTLsAllInd.html">APAqtl for full ind set</a></p></li>
<li><p><a href="PeakToGeneAssignment.html">QTL and Diff iso wih correct peak to gene assignment</a></p></li>
<li><p><a href="conditionAPAqtlFraction.html">Condition Total and Nuclear QTLs on each other</a></p></li>
<li><p><a href="characterizeTotalApaQtls.html">Characterize Total Apa QTLS</a></p></li>
<li><p><a href="characterizeNuclearApaQtls.html">Characterize Nuclear Apa QTLS</a></p></li>
<li><p><a href="chromHmm_enrichment.html">ChromHMM enrichment analysis</a></p></li>
<li><p><a href="apaQTLoverlapGWAS.html">Overlap with GWAS</a></p></li>
<li><p><a href="peakGeneCovEx.html">Visualize peaks in significant genes</a></p></li>
<li><p><a href="modifyLeafcutter.html">Look at filtering low peaks and modify leafcutter</a></p></li>
</ul>
<div id="molecular-qtl-analysis-and-overlap" class="section level3">
<h3>Molecular QTL analysis and Overlap:</h3>
<ul>
<li><p><a href="callMolQTLS.html">Rerun molecular QTLs</a></p></li>
<li><p><a href="overlapMolQTL.html">Make overlap QTL plots</a></p></li>
<li><p><a href="swarmPlots_QTLs.html">Plot top apaQTLs in other phenotypes</a></p></li>
<li><p><a href="locusZoom.html">LocusZoom plot for example QTLs</a></p></li>
<li><p><a href="overlapMolQTL.opposite.html">Overlap QTLs starting with APAqtls</a></p></li>
<li><p><a href="apa_prot_overlap.html">Investigate APA protein RNA relationship</a></p></li>
<li><p><a href="flash2mash.html">Use flash and mash to look at sharing</a></p></li>
<li><p><a href="explainpQTLs.html">Linear models to explain pQTLS</a></p></li>
</ul>
<p>Until now I have been focusing on way to develop the story for the paper. The next steps will be to systematically understand the data in order to filter to high quality data to use in the final analysis.</p>
<ul>
<li><p><a href="understandPeaks.html">Investigate Peak quality</a></p></li>
<li><p><a href="QTLsbyPCnum.html">Investigate Effect of included PCs on QTL number</a></p></li>
<li><p><a href="InvestigatePeak2GeneAssignment.html">Peak to gene assignment</a></p></li>
<li><p><a href="CompareLianoglouData.html">Look at our peaks compared to peaks from Lianogluo et al</a></p></li>
<li><p><a href="CovariateOnPCs.html">Explore cofactors agaisnt PCs for 39ind</a></p></li>
<li><p><a href="mispriming_approach.html">Deal with Mispriming problem</a></p></li>
<li><p><a href="NewPeakPostMP.html">Create new peaks after mispriming filter</a></p></li>
<li><p><a href="peakQCPPlots.html">QC plots peaks &gt;5% usage</a></p></li>
<li><p><a href="proxDeepToolsPlot.html">Deep Tools plots for internal most</a></p></li>
</ul>
</div>
</div>
<div id="data-analysis-post-mp-cleaning-and-filtering" class="section level2">
<h2>Data analysis post MP cleaning and filtering</h2>
<p>In the previous sections (specifically <a href="NewPeakPostMP.html">here</a>) I reran some QC and the QTL analysis with the filtered and clean data. Here I want to look at some of the other analysis with the processed data.</p>
<ul>
<li><p><a href="FractionDiffProcessed.html">Processed data - Different between fractions</a></p></li>
<li><p><a href="PreAshExplore.html">Pre ash processing for Stephens lab</a></p></li>
<li><p><a href="ApproachForGeneAssignment.html">New approach for assisgning peaks to a gene</a></p></li>
</ul>
</div>
<div id="individuals" class="section level2">
<h2>55 Individuals</h2>
<p>I now have 55 individuals. I am going to call peaks and QTLs in this data.</p>
<ul>
<li><p><a href="pipeline_55Ind.html">Full QTL pipeline in 55 ind (cleaning, peaks, assign genes, filter usage, QTL, differential isoform analysis)</a></p></li>
<li><p><a href="IndPeakUsageDiff.html">Individual differences in peak usage</a></p></li>
<li><p><a href="QConApaQTLs.html">QC plots for understanding apaQTLs</a></p></li>
</ul>
</div>
<div id="new-processing-pipeline---star-with-wasp" class="section level2">
<h2>New processing pipeline - STAR with WASP</h2>
<ul>
<li><p><a href="accountMapBias.html">Test if mapping bias leads to false positive QTLs</a></p></li>
<li><p><a href="TestQTLbySizeRegion.html">Number of QTLs by window size</a></p></li>
<li><p><a href="peakQCplotsSTARprocessing.html">Peak QC plots</a></p></li>
<li><p><a href="apaQTLsbyCountPercentile.html">apaQTLs by Read Count percentile</a></p></li>
<li><p><a href="diffIsoAnalysisNewMapping.html">Leafcutter differential PAU analysis</a></p></li>
<li><p><a href="characterize_apaQTLs.html">Characterize apaQTLs</a></p></li>
<li><p><a href="ProtandRNApvals.html">Look at processed apaQTLs in other fractions</a></p></li>
<li><p><a href="unexplainedeQTL_analysis.html">Can we explain non-chromatin eQTLs</a></p></li>
<li><p><a href="initialPacBioQuant.html">Explore coverage in PAS by pacbio</a></p></li>
<li><p><a href="eRNAandRepElements.html">Ask if nuclear peaks are due to eRNA or rep elements</a></p></li>
<li><p><a href="SignalSiteEnrich.html">Signal site enrichment</a></p></li>
<li><p><a href="SignalSiteLoc.html">Signal Site Location</a></p></li>
<li><p><a href="OnePeakGeneAnalysis.html">Genes with 1 peak analysis</a></p></li>
<li><p><a href="OverlapFullGWASCatelog.html">Redo GWAS overlap with full catelog</a></p></li>
<li><p><a href="PeakToXper.html">Look at enrichment in 3’ and RNA seq up to x% usage peaks</a></p></li>
<li><p><a href="NuclearSpecQTL.html">Specificity between fraction QTLs</a></p></li>
<li><p><a href="EmpDistforOverlaps.html">Overlap molQTLs with apa associations</a></p></li>
<li><p><a href="ExampleQTLPlot2.html">Heatmaps to look at qtls</a></p></li>
<li><p><a href="TotvNucExamp.html">Heatmap to look at fraction differences</a></p></li>
<li><p><a href="exampleQTLfacetboxplot.html">New example idea with facet boxplots</a></p></li>
</ul>
<div id="explore-other-molecular-phenotypes-and-the-relationship-with-apa" class="section level3">
<h3>Explore other molecular phenotypes and the relationship with APA</h3>
<ul>
<li><p><a href="RNAdecayAndAPA.html">Relationship between APA and decay</a></p></li>
<li><p><a href="CTCFandAPA.html">Overlap with CTCF data and QTLs</a></p></li>
<li><p><a href="NetSeqatPeaks.html">Explore relationships with peaks and net seq</a></p></li>
<li><p><a href="HistoneModandPAS.html">Explore models for chromatin and APA</a></p></li>
</ul>
</div>
<div id="reprocess-after-resequencing" class="section level3">
<h3>Reprocess after resequencing</h3>
<p>After resequencing I am going to start using a new data folder to differentiate the new processing. This will be data_reseq. I will also create a new snakefile for the new pipeline.</p>
<ul>
<li><p><a href="helpScripts_newsnake.html">Helper scripts for new snakefile</a></p></li>
<li><p><a href="corrbetweensamples.html">Test correlation between ind counts</a></p></li>
</ul>
</div>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
