<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>PacPio Reads in my peaks</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">PacPio Reads in my peaks</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2/26/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-03-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>threeprimeseq/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree15d73efe7a1e98ef3988a03f334e9defa70148b3targetblank15d73efa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/15d73efe7a1e98ef3988a03f334e9defa70148b3" target="_blank">15d73ef</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree15d73efe7a1e98ef3988a03f334e9defa70148b3targetblank15d73efa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/perm_QTL_trans_noMP_5percov/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/4suDataIGV.Rmd
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/EvaleQTLs.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/groSeqAnalysis.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  analysis/verifybam_dubs.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/AllPeak_counts/
    Untracked:  data/ApaQTLs/
    Untracked:  data/ApaQTLs_otherPhen/
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/DistTXN2Peak_genelocAnno/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/LocusZoom_Unexp/
    Untracked:  data/LocusZoom_proc/
    Untracked:  data/MatchedSnps/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeakCounts_noMP_5perc/
    Untracked:  data/PeakCounts_noMP_genelocanno/
    Untracked:  data/PeakUsage/
    Untracked:  data/PeakUsage_noMP/
    Untracked:  data/PeakUsage_noMP_GeneLocAnno/
    Untracked:  data/PeaksUsed/
    Untracked:  data/PeaksUsed_noMP_5percCov/
    Untracked:  data/QTL_overlap/
    Untracked:  data/RNAkalisto/
    Untracked:  data/RefSeq_annotations/
    Untracked:  data/Replicates_usage/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/WASP_STAT/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/apaExamp_proc/
    Untracked:  data/apaQTL_examp_noMP/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_GeneLocAnno/
    Untracked:  data/diff_iso_proc/
    Untracked:  data/diff_iso_trans/
    Untracked:  data/eQTLs_Lietal/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/molPheno_noMP/
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/pacbio_cov/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/peaks4DT/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov_3UTR/
    Untracked:  data/perm_QTL_diffWindow/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/protAndAPAAndExplmRes.Rda
    Untracked:  data/protAndAPAlmRes.Rda
    Untracked:  data/protAndExpressionlmRes.Rda
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  data/threePrimeSeqMetaData55Ind.txt
    Untracked:  data/threePrimeSeqMetaData55Ind.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  output/LZ/
    Untracked:  output/deeptools_plots/
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/NewPeakPostMP.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/fixBWChromNames.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/mispriming_approach.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlapMolQTL.opposite.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/peakQCPPlots.Rmd
    Modified:   analysis/peakQCplotsSTARprocessing.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/pipeline_55Ind.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/test.smash.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   analysis/unexplainedeQTL_analysis.Rmd
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/15d73efe7a1e98ef3988a03f334e9defa70148b3/analysis/initialPacBioQuant.Rmd" target="_blank">15d73ef</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-01
</td>
<td>
pick lines
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/a259bd93bd6de74279295ee615ac6442108514c4/docs/initialPacBioQuant.html" target="_blank">a259bd9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/eb1a05c66815af97bc057d4c8c67624e66194348/analysis/initialPacBioQuant.Rmd" target="_blank">eb1a05c</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
<td>
add pacbio analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/f832bb0306b7d49147779cced6fc169978aa23ca/docs/initialPacBioQuant.html" target="_blank">f832bb0</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/6637b213a6d3ec1d5a99a7d2adad2e6a17da87ef/analysis/initialPacBioQuant.Rmd" target="_blank">6637b21</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-26
</td>
<td>
add avg total usage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/b27ba8659b45d2453d47ae75413834a7b7330548/docs/initialPacBioQuant.html" target="_blank">b27ba86</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/c5dfa4b1e09eeaac050b5d96cb41ff66eef6f255/analysis/initialPacBioQuant.Rmd" target="_blank">c5dfa4b</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-26
</td>
<td>
fix file for ankeeta
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="look-at-published-pacbio" class="section level2">
<h2>Look at published pacbio</h2>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.2.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ──────────────────────────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.0.0     ✔ purrr   0.2.5
✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.4.0
✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 3.5.2</code></pre>
<pre><code>── Conflicts ─────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<p>Ankeeta has been working with 3 pac bio libraries for whole LCLs. The meged bam file has 4,164,259 reads. I want to look at how many of these reads cover my peaks. It would be best to know how many reads ends</p>
<p>I need to fix the strand for my peaks and give them to her.</p>
<p>fixPeaks4Ankeeta.py</p>
<pre class="bash"><code>In=open(&quot;/project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed_withAnno.SAF&quot;,&quot;r&quot;)
Out=&quot;/project2/yangili1/PAPeaks_STARMap_GeneLocAnno.bed&quot;

def fix_strand(Fin,Fout):
    fout=open(Fout,&quot;w&quot;)
    for n, ln in enumerate(Fin):
        if n == 0: 
            continue
        else: 
            id, chrom, start, end, strand = ln.split()
            if strand==&quot;+&quot;:
                chromF=&quot;chr&quot; + chrom
                peak=id.split(&quot;:&quot;)[0]
                geneLoc=id.split(&quot;:&quot;)[5:]
                geneLocF=&quot;:&quot;.join(geneLoc)
                newID=peak + &quot;:&quot; + geneLocF
                score=&quot;.&quot;
                fout.write(&quot;%s\t%s\t%s\t%s\t%s\t-\n&quot;%(chromF,start,end,newID,score))
            else:
                chromF=&quot;chr&quot; + chrom
                peak=id.split(&quot;:&quot;)[0]
                geneLoc=id.split(&quot;:&quot;)[5:]
                geneLocF=&quot;:&quot;.join(geneLoc)
                newID=peak + &quot;:&quot; + geneLocF
                score=&quot;.&quot;
                fout.write(&quot;%s\t%s\t%s\t%s\t%s\t+\n&quot;%(chromF,start,end,newID,score))
    fout.close()
    
    
fix_strand(In, Out)</code></pre>
<p>Add average usage to this:</p>
<p>use similar code to filter_5percUsagePeaks.R</p>
<p>counts only numeric are in /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.CountsOnlyNumeric.txt I will take the mean for each row of this and use it as the score in the bed file.</p>
<p>Run this interactively</p>
<pre class="r"><code>library(dplyr)
totUsage=read.table(&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.CountsOnlyNumeric.txt&quot;, header=F)
peakBed=read.table(&quot;/project2/yangili1/PAPeaks_STARMap_GeneLocAnno.bed&quot;, header=F, col.names = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;ID&quot;, &quot;score&quot;, &quot;strand&quot;), stringsAsFactors = F)


MeanUsage=rowMeans(totUsage)

outBed=as.data.frame(cbind(peakBed, MeanUsage)) %&gt;% select(chr, start, end, ID, MeanUsage, strand)

write.table(outBed,file=&quot;/project2/yangili1/PAPeaks_STARMap_GeneLocAnno_withMeanUsage.bed&quot;, row.names=F, col.names=F, quote = F, sep=&quot;\t&quot;)  </code></pre>
<p>Result from pac bio overlap:</p>
<ul>
<li>/project2/yangili1/ankeetashah/APA_tools/peaks/threeprime_noMP_pacbio_UTR.coverage</li>
<li>project2/yangili1/ankeetashah/APA_tools/peaks/threeprime_noMP_pacbio_intron.coverage</li>
</ul>
<p>Make some plots for this: Distribution of reads ending at each peak</p>
<pre class="r"><code>covNames=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;ID&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;cov&quot;)
utrCov=read.table(&quot;../data/pacbio_cov/threeprime_noMP_pacbio_UTR.coverage&quot;, stringsAsFactors = F, col.names = covNames)
intronCov=read.table(&quot;../data/pacbio_cov/threeprime_noMP_pacbio_intron.coverage&quot;, stringsAsFactors = F,col.names = covNames) %&gt;% separate(ID, into=c(&quot;peak&quot;, &quot;gene&quot;,&quot;loc&quot;), sep=&quot;:&quot;) %&gt;% filter(loc==&quot;intron&quot;)</code></pre>
<p>Plot the distributions:</p>
<pre class="r"><code>ggplot(utrCov, aes(x=log10(cov + 1))) + geom_density() + labs(title=&quot;PacBio reads ending at each UTR peak&quot;, x=&quot;log10(nReads+1)&quot;)</code></pre>
<p><img src="figure/initialPacBioQuant.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a259bd93bd6de74279295ee615ac6442108514c4/docs/figure/initialPacBioQuant.Rmd/unnamed-chunk-5-1.png" target="_blank">a259bd9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(intronCov, aes(x=log10(cov + 1))) + geom_density() + labs(title=&quot;PacBio reads ending at each intronic peak&quot;, x=&quot;log10(nReads+1)&quot;)</code></pre>
<p><img src="figure/initialPacBioQuant.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a259bd93bd6de74279295ee615ac6442108514c4/docs/figure/initialPacBioQuant.Rmd/unnamed-chunk-6-1.png" target="_blank">a259bd9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>summary(intronCov$cov)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
   0.000    0.000    0.000    0.611    0.000 4145.000 </code></pre>
<pre class="r"><code>summary(utrCov$cov)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
    0.00     1.00     5.00    24.93    18.00 15301.00 </code></pre>
<p>Proportion of peaks with coverage</p>
<pre class="r"><code>intronCov_0=intronCov %&gt;% filter(cov &gt; 0) %&gt;% nrow()/ nrow(intronCov)
intronCov_10=intronCov %&gt;% filter(cov &gt;= 10) %&gt;% nrow()/nrow(intronCov)
intronCov_100=intronCov %&gt;% filter(cov &gt;= 100) %&gt;% nrow()/nrow(intronCov)
intronCov_1000=intronCov %&gt;% filter(cov &gt;= 1000) %&gt;% nrow()/nrow(intronCov)



utrCov_0=utrCov %&gt;% filter(cov &gt; 0) %&gt;% nrow() / nrow(utrCov)
utrCov_10=utrCov %&gt;% filter(cov &gt;= 10) %&gt;% nrow()/ nrow(utrCov)
utrCov_100=utrCov %&gt;% filter(cov &gt;= 100) %&gt;% nrow()/ nrow(utrCov)
utrCov_1000=utrCov %&gt;% filter(cov &gt;= 1000) %&gt;% nrow()/ nrow(utrCov)


Reads=c(&quot;1 read&quot;, &quot;10 reads&quot;, &quot;100 reads&quot;, &quot;1000 reads&quot;)
UTR=c(utrCov_0, utrCov_10, utrCov_100, utrCov_1000)
Intron=c(intronCov_0,intronCov_10,intronCov_100,intronCov_1000)

covDF=as.data.frame(cbind(Reads,UTR,Intron))
covDF$UTR=as.numeric(as.character(covDF$UTR))
covDF$Intron=as.numeric(as.character(covDF$Intron))


covDF_melt=melt(covDF, id.vars = &quot;Reads&quot;)
colnames(covDF_melt)=c(&quot;ReadCutoff&quot;, &quot;Location&quot;, &quot;Proportion&quot; )</code></pre>
<pre class="r"><code>propcovbycutff=ggplot(covDF_melt, aes(x=ReadCutoff, fill=Location, y=Proportion, by=Location)) + geom_bar(position=&quot;dodge&quot;,stat=&quot;identity&quot; ) + labs(y=&quot;Proportion of PAS covered&quot;, x=&quot;At least X reads ending at PAS&quot;, title=&quot;Proportion of PAS with PacBio read ending in it&quot;) + scale_fill_manual(values=c(&quot;blue&quot;,&quot;red&quot;)) + annotate(&quot;text&quot;, label=&quot;UTR PAS = 29,687&quot;, x=&quot;1000 reads&quot;, y=.7) + annotate(&quot;text&quot;, label=&quot;Intronic PAS = 87,733&quot;, x=&quot;1000 reads&quot;, y=.6)

propcovbycutff</code></pre>
<p><img src="figure/initialPacBioQuant.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a259bd93bd6de74279295ee615ac6442108514c4/docs/figure/initialPacBioQuant.Rmd/unnamed-chunk-9-1.png" target="_blank">a259bd9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(propcovbycutff, filename = &quot;../output/plots/PacBioReadsEndingAtPAS.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<p>Subset for peaks that passed the 5% coverage cutoff.</p>
<pre class="r"><code>tot5Perc=read.table(&quot;../data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt&quot;, stringsAsFactors = F, col.names = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;, &quot;avgUsage&quot;))
nuc5Perc=read.table(&quot;../data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt&quot;,col.names = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;, &quot;avgUsage&quot;),stringsAsFactors = F)</code></pre>
<pre class="r"><code>intronCov_tot5perc= intronCov %&gt;% semi_join(tot5Perc, by=&quot;peak&quot;)

intronCov_nuc5perc= intronCov %&gt;% semi_join(nuc5Perc, by=&quot;peak&quot;)</code></pre>
<pre class="r"><code>intronCov_tot5perc_0= intronCov_tot5perc %&gt;% filter(cov&gt; 0) %&gt;% nrow()/ nrow(intronCov_tot5perc)
intronCov_tot5perc_10= intronCov_tot5perc %&gt;% filter(cov&gt;= 10) %&gt;% nrow()/ nrow(intronCov_tot5perc)
intronCov_tot5perc_100= intronCov_tot5perc %&gt;% filter(cov&gt;= 100) %&gt;% nrow()/ nrow(intronCov_tot5perc)
intronCov_tot5perc_1000= intronCov_tot5perc %&gt;% filter(cov&gt;= 1000) %&gt;% nrow()/ nrow(intronCov_tot5perc)


intronCov_nuc5perc_0= intronCov_nuc5perc %&gt;% filter(cov&gt; 0) %&gt;% nrow()/ nrow(intronCov_nuc5perc)
intronCov_nuc5perc_10= intronCov_nuc5perc %&gt;% filter(cov&gt;= 10) %&gt;% nrow()/ nrow(intronCov_nuc5perc)
intronCov_nuc5perc_100= intronCov_nuc5perc %&gt;% filter(cov&gt;= 100) %&gt;% nrow()/ nrow(intronCov_nuc5perc)
intronCov_nuc5perc_1000= intronCov_nuc5perc %&gt;% filter(cov&gt;= 1000) %&gt;% nrow()/ nrow(intronCov_nuc5perc)

Reads=c(&quot;1 read&quot;, &quot;10 reads&quot;, &quot;100 reads&quot;, &quot;1000 reads&quot;)
Total=c(intronCov_tot5perc_0, intronCov_tot5perc_10, intronCov_tot5perc_100, intronCov_tot5perc_1000)
Nuclear=c(intronCov_nuc5perc_0,intronCov_nuc5perc_10,intronCov_nuc5perc_100,intronCov_nuc5perc_1000)

cov_5perDF=as.data.frame(cbind(Reads,Total,Nuclear))
cov_5perDF$Total=as.numeric(as.character(cov_5perDF$Total))
cov_5perDF$Nuclear=as.numeric(as.character(cov_5perDF$Nuclear))

cov_5perDF_melt=melt(cov_5perDF, id.vars = &quot;Reads&quot;)
colnames(cov_5perDF_melt)=c(&quot;ReadCutoff&quot;, &quot;Fraction&quot;, &quot;Proportion&quot; )</code></pre>
<pre class="r"><code>ggplot(cov_5perDF_melt, aes(x=ReadCutoff, fill=Fraction, y=Proportion, by=Fraction)) + geom_bar(position=&quot;dodge&quot;,stat=&quot;identity&quot; ) + labs(y=&quot;Proportion of PAS covered&quot;, x=&quot;At least X reads ending at PAS&quot;, title=&quot;Proportion of PAS with PacBio read ending in it \n Intron PAS with 5% mean Usage&quot;) + scale_fill_brewer(palette = &quot;Set1&quot;) + annotate(&quot;text&quot;, label=&quot;Total PAS = 16,662&quot;, x=&quot;1000 reads&quot;, y=.28) + annotate(&quot;text&quot;, label=&quot;Nuclear PAS = 18,829&quot;, x=&quot;1000 reads&quot;, y=.25)</code></pre>
<p><img src="figure/initialPacBioQuant.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a259bd93bd6de74279295ee615ac6442108514c4/docs/figure/initialPacBioQuant.Rmd/unnamed-chunk-13-1.png" target="_blank">a259bd9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Do this for the UTR</p>
<pre class="r"><code>UTRCov_tot5perc= utrCov %&gt;% separate(ID, into=c(&quot;peak&quot;, &quot;gene&quot;,&quot;loc&quot;), sep=&quot;:&quot;) %&gt;% semi_join(tot5Perc, by=&quot;peak&quot;)

UTRCov_nuc5perc= utrCov %&gt;% separate(ID, into=c(&quot;peak&quot;, &quot;gene&quot;,&quot;loc&quot;), sep=&quot;:&quot;) %&gt;% semi_join(nuc5Perc, by=&quot;peak&quot;)</code></pre>
<pre class="r"><code>utrCov_tot5perc_0= UTRCov_tot5perc %&gt;% filter(cov&gt; 0) %&gt;% nrow()/ nrow(UTRCov_tot5perc)
utrCov_tot5perc_10= UTRCov_tot5perc %&gt;% filter(cov&gt;= 10) %&gt;% nrow()/ nrow(UTRCov_tot5perc)
utrCov_tot5perc_100= UTRCov_tot5perc %&gt;% filter(cov&gt;= 100) %&gt;% nrow()/ nrow(UTRCov_tot5perc)
utrCov_tot5perc_1000= UTRCov_tot5perc %&gt;% filter(cov&gt;= 1000) %&gt;% nrow()/ nrow(UTRCov_tot5perc)


utrCov_nuc5perc_0= UTRCov_nuc5perc %&gt;% filter(cov&gt; 0) %&gt;% nrow()/ nrow(UTRCov_nuc5perc)
utrCov_nuc5perc_10= UTRCov_nuc5perc %&gt;% filter(cov&gt;= 10) %&gt;% nrow()/ nrow(UTRCov_nuc5perc)
utrCov_nuc5perc_100= UTRCov_nuc5perc %&gt;% filter(cov&gt;= 100) %&gt;% nrow()/ nrow(UTRCov_nuc5perc)
utrCov_nuc5perc_1000= UTRCov_nuc5perc %&gt;% filter(cov&gt;= 1000) %&gt;% nrow()/ nrow(UTRCov_nuc5perc)


Total_UTR=c(utrCov_tot5perc_0, utrCov_tot5perc_10, utrCov_tot5perc_100, utrCov_tot5perc_1000)
Nuclear_UTR=c(utrCov_nuc5perc_0,utrCov_nuc5perc_10,utrCov_nuc5perc_100,utrCov_nuc5perc_1000)

covUTR_5perDF=as.data.frame(cbind(Reads,Total=Total_UTR,Nuclear=Nuclear_UTR))
covUTR_5perDF$Total=as.numeric(as.character(covUTR_5perDF$Total))
covUTR_5perDF$Nuclear=as.numeric(as.character(covUTR_5perDF$Nuclear))

covUTR_5perDF_melt=melt(covUTR_5perDF, id.vars = &quot;Reads&quot;)
colnames(covUTR_5perDF_melt)=c(&quot;ReadCutoff&quot;, &quot;Fraction&quot;, &quot;Proportion&quot; )</code></pre>
<pre class="r"><code>ggplot(covUTR_5perDF_melt, aes(x=ReadCutoff, fill=Fraction, y=Proportion, by=Fraction)) + geom_bar(position=&quot;dodge&quot;,stat=&quot;identity&quot; ) + labs(y=&quot;Proportion of PAS covered&quot;, x=&quot;At least X reads ending at PAS&quot;, title=&quot;Proportion of PAS with PacBio read ending in it \n UTR PAS with 5% mean Usage&quot;) + scale_fill_brewer(palette = &quot;Set1&quot;) + annotate(&quot;text&quot;, label=&quot;Total PAS = 5,984&quot;, x=&quot;1000 reads&quot;, y=.7) + annotate(&quot;text&quot;, label=&quot;Nuclear PAS = 6,793&quot;, x=&quot;1000 reads&quot;, y=.6)</code></pre>
<p><img src="figure/initialPacBioQuant.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a259bd93bd6de74279295ee615ac6442108514c4/docs/figure/initialPacBioQuant.Rmd/unnamed-chunk-16-1.png" target="_blank">a259bd9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Make the first plot (utr v intron for the 5%) Process in excel</p>
<pre class="r"><code>allProp=read.table(&quot;../data/pacbio_cov/PacBioPropCov.txt&quot;, head=T, stringsAsFactors = F)
allProp$Read=as.factor(allProp$Read)
allPropPlot=ggplot(allProp, aes(x=Read, y=Proportion, by=Location, fill=Location)) + geom_bar(position=&quot;dodge&quot;,stat=&quot;identity&quot; ) +facet_grid(~Fraction) + labs(y=&quot;Proportion of PAS covered&quot;, x=&quot;At least X reads ending at PAS&quot;, title=&quot;Proportion of PAS with PacBio read ending in it \n PAS with 5% mean Usage&quot;)+scale_fill_manual(values=c(&quot;red&quot;,&quot;blue&quot;))

allPropPlot</code></pre>
<p><img src="figure/initialPacBioQuant.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/a259bd93bd6de74279295ee615ac6442108514c4/docs/figure/initialPacBioQuant.Rmd/unnamed-chunk-17-1.png" target="_blank">a259bd9</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(allPropPlot, filename = &quot;../output/plots/PacBioReadsEndingAtPAS_5percCov.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<pre class="r"><code>UTRCov_tot5perc %&gt;% semi_join(UTRCov_nuc5perc, by=&quot;peak&quot;) %&gt;% nrow()</code></pre>
<pre><code>[1] 5538</code></pre>
<pre class="r"><code>intronCov_tot5perc %&gt;% semi_join(intronCov_nuc5perc, by=&quot;peak&quot;) %&gt;% nrow()</code></pre>
<pre><code>[1] 15270</code></pre>
<pre class="r"><code>tot5Perc_peaks =tot5Perc %&gt;% select(peak)
nuc5Perc_peaks=nuc5Perc %&gt;% select(peak)
nrow(tot5Perc)</code></pre>
<pre><code>[1] 33002</code></pre>
<pre class="r"><code>nrow(nuc5Perc)</code></pre>
<pre><code>[1] 37370</code></pre>
<pre class="r"><code>ineither=tot5Perc_peaks %&gt;% full_join(nuc5Perc_peaks, by=&quot;peak&quot;)
nrow(ineither)</code></pre>
<pre><code>[1] 40066</code></pre>
</div>
<div id="pick-lines-for-ours" class="section level2">
<h2>Pick lines for ours</h2>
<p>I need to look at the top nuclear specific QTLs and chose individuals with both homozygous alleles.</p>
<pre class="r"><code>totQTL=read.table(&quot;../data/ApaQTLs/TotalapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt&quot;, head=F,stringsAsFactors = F, col.names = c(&#39;pid&#39;, &#39;nvar&#39;, &#39;shape1&#39;, &#39;shape2&#39; ,&#39;dummy&#39; ,&#39;sid&#39; ,&#39;dist&#39;, &#39;npval&#39;, &#39;slope&#39;, &#39;ppval&#39;, &#39;bpval&#39; ,&#39;bh&#39;)) %&gt;% separate(pid, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;geneID&quot;), sep=&quot;:&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;),sep=&quot;_&quot;) 

NucQTL=read.table(&quot;../data/ApaQTLs/NuclearapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt&quot;, head=F,stringsAsFactors = F, col.names = c(&#39;pid&#39;, &#39;nvar&#39;, &#39;shape1&#39;, &#39;shape2&#39; ,&#39;dummy&#39; ,&#39;sid&#39; ,&#39;dist&#39;, &#39;npval&#39;, &#39;slope&#39;, &#39;ppval&#39;, &#39;bpval&#39; ,&#39;bh&#39;)) %&gt;% separate(pid, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;geneID&quot;), sep=&quot;:&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;),sep=&quot;_&quot;) </code></pre>
<p>Filter out the genes with a total qtl from the nuclear qlts genes:</p>
<pre class="r"><code>NucOnlyQTL= NucQTL %&gt;% anti_join(totQTL, by=&quot;gene&quot;) %&gt;% arrange(slope) 

head(NucOnlyQTL)</code></pre>
<pre><code>  chr     start       end      gene strand       peak nvar   shape1
1   1 149585233 149585320 LINC00869      -   peak9493   40 1.066010
2   1 249211696 249211944     PGBD2      -  peak16578  142 0.991565
3   6  32519163  32519261  HLA-DRB6      + peak138621 2701 1.003040
4   1 113667733 113667823     LRIG2      -   peak8001  123 0.947699
5   6 109765267 109765350    MICAL1      + peak142627  122 0.886992
6  19   6535487   6535939    TNFSF9      -  peak77427  219 0.986695
    shape2   dummy         sid   dist       npval     slope       ppval
1  6.36770 48.4641 1:149601043  15809 3.31180e-04 -13.63700 0.000999001
2 27.35320 40.7121 1:249232758  21061 4.94006e-08  -7.46381 0.000999001
3  6.50355 35.2570  6:32505477 -13687 1.34996e-06  -2.94929 0.000999001
4 15.94130 42.7219 1:113678884  11150 3.66278e-09  -2.40161 0.000999001
5 15.36680 46.6074 6:109754433 -10835 1.03391e-05  -2.10134 0.001998000
6 34.27590 39.0277  19:6551945  16457 2.70575e-07  -2.05682 0.000999001
        bpval           bh
1 1.48525e-03 0.0949692158
2 2.05616e-05 0.0036973232
3 2.71955e-04 0.0297604242
4 1.27763e-06 0.0003351679
5 6.94153e-04 0.0569309509
6 1.76541e-04 0.0210644761</code></pre>
<p>See if any of these are also nuclear enriched from diff iso:</p>
<pre class="r"><code>nucMoreUsed=read.table(&quot;../data/diff_iso_GeneLocAnno/SigPeaksHigherInNuc.txt&quot;, header = T) %&gt;% separate(intron, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;),sep=&quot;:&quot;)</code></pre>
<p>Join the nuc only by these peaks:</p>
<pre class="r"><code>NucOnlyQTL_diff=NucOnlyQTL %&gt;% inner_join(nucMoreUsed, by=c(&quot;start&quot;, &quot;end&quot;, &quot;gene&quot;))%&gt;% arrange(abs(slope))   %&gt;% select(gene, peak, sid, slope,deltapsi )

NucOnlyQTL_diff</code></pre>
<pre><code>           gene       peak         sid     slope   deltapsi
1        IFNGR2 peak101664 21:34785672  0.460097 -0.2277754
2         RCSD1  peak11626 1:167610743  0.524358 -0.2000519
3          STX8  peak66383  17:9481747 -0.542269 -0.2115276
4         ZNRD1 peak138165  6:30017632 -0.774940 -0.2165452
5       FAM207A peak102641 21:46396659  0.923893 -0.2533298
6         DOCK7   peak5329  1:63018852  0.935357 -0.2357179
7        SMURF2  peak71606 17:62663377  1.102180 -0.3116123
8     LINC00476 peak167478  9:98613783 -1.105940 -0.2108329
9         ACTR2  peak86635  2:65478371 -1.112100 -0.2392005
10        FOXN2  peak85407  2:48547443 -1.192880 -0.2458012
11         DPF2  peak28278 11:65109233 -1.198550 -0.2090628
12        CDC42   peak1782  1:22435723  1.279150 -0.2717837
13       ZNF738  peak79187 19:21543803  1.489360 -0.2156677
14 LOC101927151  peak79354 19:28291331 -1.685500 -0.2078720
15       RPUSD2  peak53419 15:40861040 -1.716780 -0.2243680
16         NFIC  peak77091  19:3383733 -1.885840 -0.2067161</code></pre>
<p>Look at these and the snps:</p>
<p>RPUSD2: 15:40861040<br />
ACTR2 2:65478371 SMURF2 17:62663377 ZNF738 19:21543803</p>
<p>Pull in the genotypes for these:</p>
<pre class="r"><code>headerGeno=read.table(&quot;../data/pacbio_cov/indiv.geno.header.txt&quot;, stringsAsFactors = F) %&gt;% t() 

headerGeno= as.vector(headerGeno)</code></pre>
<div id="section" class="section level3">
<h3>15:40861040</h3>
<pre class="r"><code>RPUSD2_gene=read.table(&quot;../data/pacbio_cov/geno_15:408610.txt&quot;, stringsAsFactors = F, col.names  =headerGeno)</code></pre>
<p>No homo</p>
</div>
<div id="section-1" class="section level3">
<h3>2:65478371</h3>
<pre class="r"><code>ACTR2_gene=read.table(&quot;../data/pacbio_cov/geno_2:65478371.txt&quot;, stringsAsFactors = F, col.names  =headerGeno)</code></pre>
<p>1|1: 19144,18486</p>
<p>0|0 : 18498, 18501, 18504, 18507, 18511, 18519, 18520, 18522, 18523, 18859, 18861, 18868, 18870, 18873, 18874, 18910, 18912, 18913, 18923, 18924, 19093, 19099, 19101, 19108, 19121, 19127, 19128, 19129, 19131, 19140, 19149, 19160, 19172,19189, 19190, 19214, 19223, 19225, 19235, 19238, 19239, 19248, 19256</p>
</div>
<div id="section-2" class="section level3">
<h3>19:21543803</h3>
<pre class="r"><code>ZNF738_geno=read.table(&quot;../data/pacbio_cov/geno_19:21543803.txt&quot;, stringsAsFactors = F, col.names  =headerGeno )</code></pre>
<p>1|1: 18486, 18487, 18488, 19498, 18502, 18507,18508, 18517, 18520, 18522, 18523, 18852, 18856, 18858, 18859, 18862, 18867, 18868, 18870, 18871, 18873, 18874, 18907, 18909, 18912, 18913, 18916, 18917, 18933,18934, 19093, 19096, 10900, 19107, 19108, 19113, 19114, 19116, 19177, 19118, 19119, 19121, 19122, 19128, 19130, 19131, 19137, 19138, 19140, 19141, 19146, 19153, 19159, 19160, 19175, 19176, 19185, 19189, 19190, 19197, 19200, 19203,19204, 19206, 19207, 19222, 19225, 19236, 19248, 19256, 19257</p>
<p>0|0: 18498, 18504, 18511, 18519, 19127, 19143, 19147, 19210, 19226, 19239</p>
</div>
<div id="section-3" class="section level3">
<h3>17:62663377</h3>
<pre class="r"><code>SMURF2_geno=read.table(&quot;../data/pacbio_cov/geno_17:62663377.txt&quot;, stringsAsFactors = F, col.names  =headerGeno )</code></pre>
<p>1|1- 18501, 18868, 18871, 19171, 19226, 19235</p>
<p>0|0- 18498, 18499, 18504, 18505, 18508, 18516, 18517, 18519, 18522, 18523, 18852, 18853, 18855, 18856, 18859, 18861,18862, 18870, 18874, 18907, 18909, 18933, 18934, 19096, 19102, 19113, 19116, 19117, 19119, 19122, 19129, 19130, 19131, 19137, 19140, 19141, 19143, 19147, 19149, 19152, 19153, 19160, 19175, 19176, 19184, 19190, 19198, 19200, 19206, 19207, 19210, 19213, 19124, 19222, 19223, 19225, 19236, 19248, 19256</p>
</div>
<div id="to-use" class="section level3">
<h3>TO use:</h3>
19144,18486,18504,19239,18501<br />
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] bindrcpp_0.2.2  cowplot_0.9.3   reshape2_1.4.3  forcats_0.3.0  
 [5] stringr_1.4.0   dplyr_0.7.6     purrr_0.2.5     readr_1.1.1    
 [9] tidyr_0.8.1     tibble_1.4.2    ggplot2_3.0.0   tidyverse_1.2.1
[13] workflowr_1.2.0

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4   haven_1.1.2        lattice_0.20-35   
 [4] colorspace_1.3-2   htmltools_0.3.6    yaml_2.2.0        
 [7] rlang_0.2.2        pillar_1.3.0       glue_1.3.0        
[10] withr_2.1.2        RColorBrewer_1.1-2 modelr_0.1.2      
[13] readxl_1.1.0       bindr_0.1.1        plyr_1.8.4        
[16] munsell_0.5.0      gtable_0.2.0       cellranger_1.1.0  
[19] rvest_0.3.2        evaluate_0.13      labeling_0.3      
[22] knitr_1.20         broom_0.5.0        Rcpp_0.12.19      
[25] scales_1.0.0       backports_1.1.2    jsonlite_1.6      
[28] fs_1.2.6           hms_0.4.2          digest_0.6.17     
[31] stringi_1.2.4      grid_3.5.1         rprojroot_1.3-2   
[34] cli_1.0.1          tools_3.5.1        magrittr_1.5      
[37] lazyeval_0.2.1     crayon_1.3.4       whisker_0.3-2     
[40] pkgconfig_2.0.2    xml2_1.2.0         lubridate_1.7.4   
[43] assertthat_0.2.0   rmarkdown_1.11     httr_1.3.1        
[46] rstudioapi_0.9.0   R6_2.3.0           nlme_3.1-137      
[49] git2r_0.24.0       compiler_3.5.1    </code></pre>
</div>
</div>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
