<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />

<meta name="date" content="2018-11-15" />

<title>LocusZoom Plot</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">LocusZoom Plot</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>11/15/2018</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-11-19</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/bac8c20c16ad2960381cba88df6ac99beda943a6" target="_blank">bac8c20</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/LocusZoom/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeaksUsed/
    Untracked:  data/RNAkalisto/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/apaExamp/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_trans/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/other_qtls/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/39indQC.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   code/Snakefile

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/bac8c20c16ad2960381cba88df6ac99beda943a6/analysis/locusZoom.Rmd" target="_blank">bac8c20</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-19
</td>
<td style="text-align:left;">
export files for locus zoom site
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/0940e7c5edfd1c4f20b02b47a87084e71b310e9e/docs/locusZoom.html" target="_blank">0940e7c</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/a856d098d90cc5e431fda8c783aeb270a8e6664a/analysis/locusZoom.Rmd" target="_blank">a856d09</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
<td style="text-align:left;">
fix format
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/410f25c72a15eb6d32f39498fb046641b4eecb5a/docs/locusZoom.html" target="_blank">410f25c</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/eca9b137f44d7d29d1b1f94db2ad473b5ca77cb2/analysis/locusZoom.Rmd" target="_blank">eca9b13</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
<td style="text-align:left;">
add samcl1
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/locusZoom.html" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/4fb0d81892afb37eb09ddc5540cc9deefd597533/analysis/locusZoom.Rmd" target="_blank">4fb0d81</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
<td style="text-align:left;">
add more examples
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/617a3b7fb2f5c6037694b01b409c55f4a6dca406/docs/locusZoom.html" target="_blank">617a3b7</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-15
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/e79d21a515aeedb21a40e2d7533a02609c759d05/analysis/locusZoom.Rmd" target="_blank">e79d21a</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-15
</td>
<td style="text-align:left;">
add LD color to plot
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/813a5000acdf4ddf85a19f3ae61c80bcfe41927a/docs/locusZoom.html" target="_blank">813a500</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-15
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/23c62c988586bb3cd15114d6c7bc28fba051c0fa/analysis/locusZoom.Rmd" target="_blank">23c62c9</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-15
</td>
<td style="text-align:left;">
add locus zoom initial analysis
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<p>In this analysis I will create locus zoom plots for the example QTLs that look to be associated in APA and protein but not in RNA.</p>
<div id="eif2a" class="section level2">
<h2>EIF2A</h2>
<p>I will first do this for the EIF2A totalAPA example. peak228606, 3:150302010.</p>
<p>To run this analysis, I will need the nominal pvalues for this peak/gene. I can then plot the snp location against the pvalue. After I have this working, I can add the r2 values.</p>
<p>EIF2A==ENSG00000144895</p>
<p>grep EIF2A /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt</p>
<pre class="bash"><code>grep peak228606 /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/TotalAPA.peak228606.EIF2A.nomTotal.txt


grep ENSG00000144895 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/RNA.EIF2A.nomTotal.txt

grep ENSG00000144895 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_prot.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Prot.EIF2A.nomTotal.txt

 grep ENSG00000144895 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_ribo_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Ribo.EIF2A.nomTotal.txt
</code></pre>
<p>FastQTL results for nominal: * phenoID</p>
<ul>
<li><p>SID</p></li>
<li><p>Distance</p></li>
<li><p>Nominal Pval</p></li>
<li><p>Slope</p></li>
</ul>
<p>Librarys</p>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.1.1
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(reshape2)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ───────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.0.0     ✔ purrr   0.2.5
✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.3.1
✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>── Conflicts ──────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(VennDiagram)</code></pre>
<pre><code>Loading required package: grid</code></pre>
<pre><code>Loading required package: futile.logger</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre><code>The following objects are masked from &#39;package:reshape2&#39;:

    dcast, melt</code></pre>
<pre class="r"><code>library(ggpubr)</code></pre>
<pre><code>Loading required package: magrittr</code></pre>
<pre><code>
Attaching package: &#39;magrittr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    set_names</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    extract</code></pre>
<pre><code>
Attaching package: &#39;ggpubr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:VennDiagram&#39;:

    rotate</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggpubr&#39;:

    get_legend</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre class="r"><code>APA=read.table(&quot;../data/LocusZoom/TotalAPA.peak228606.EIF2A.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APAPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APAPval)
APA$Location=as.integer(APA$Location)
Prot=read.table(&quot;../data/LocusZoom/Prot.EIF2A.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;ProtPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, ProtPval)
Prot$Location=as.integer(Prot$Location)
RNA=read.table(&quot;../data/LocusZoom/RNA.EIF2A.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RnaPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RnaPval)
RNA$Location=as.integer(RNA$Location)
Ribo=read.table(&quot;../data/LocusZoom/Ribo.EIF2A.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RiboPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RiboPval)
Ribo$Location=as.integer(Ribo$Location)</code></pre>
<p>I can join these by the snps that are tested for all three.</p>
<pre class="r"><code>allPheno=APA %&gt;% inner_join(Prot, by=&quot;Location&quot;) %&gt;% inner_join(Ribo, by=&quot;Location&quot;) %&gt;% inner_join(RNA, by=&quot;Location&quot;)</code></pre>
<p>First I can just plot these as is and see what happens:</p>
<pre class="r"><code>allPhen_melt= melt(allPheno, id.vars=&quot;Location&quot;)</code></pre>
<pre class="r"><code>ggplot(allPhen_melt,aes(x=Location, y=value)) + geom_point() + facet_grid( rows=vars(variable))</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-6-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/figure/locusZoom.Rmd/unnamed-chunk-6-1.png" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/813a5000acdf4ddf85a19f3ae61c80bcfe41927a/docs/figure/locusZoom.Rmd/unnamed-chunk-6-1.png" target="_blank">813a500</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-15
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>I need to zoom in around my locus 150302010</p>
<pre class="r"><code>allPheno_filt=allPheno %&gt;% filter(Location&gt; 150297010 &amp; Location &lt; 150307010)

allPhen_filt_melt= melt(allPheno_filt, id.vars=&quot;Location&quot;)

ggplot(allPhen_filt_melt,aes(x=Location, y=-log10(value))) + geom_point() + facet_grid( rows=vars(variable)) + geom_vline(xintercept=150302010, linetype=&quot;dashed&quot;, color = &quot;red&quot;) + theme(axis.line=element_line()) + theme(panel.grid.major = element_line(&quot;lightgray&quot;,0.25), panel.grid.minor = element_line(&quot;lightgray&quot;,0.25)) + labs(x=&quot;Chromosome 3 Location&quot;, y=&quot;-Log 10 Pvalue&quot;, title=&quot;Locus Zoom for EIF2A:peak228606&quot;)</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-7-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/figure/locusZoom.Rmd/unnamed-chunk-7-1.png" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/813a5000acdf4ddf85a19f3ae61c80bcfe41927a/docs/figure/locusZoom.Rmd/unnamed-chunk-7-1.png" target="_blank">813a500</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-15
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Plot each seperatly because power is different.</p>
<pre class="r"><code>ggplot(allPhen_filt_melt,aes(x=Location, y=-log10(value))) + geom_point() + facet_grid( rows=vars(variable),scales=&quot;free&quot;) + geom_vline(xintercept=150302010, linetype=&quot;dashed&quot;, color = &quot;red&quot;) + theme(axis.line=element_line()) + theme(panel.grid.major = element_line(&quot;lightgray&quot;,0.25), panel.grid.minor = element_line(&quot;lightgray&quot;,0.25)) + labs(x=&quot;Chromosome 3 Location&quot;, y=&quot;-Log 10 Pvalue&quot;, title=&quot;Locus Zoom for EIF2A:peak228606&quot;)</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-8-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/figure/locusZoom.Rmd/unnamed-chunk-8-1.png" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>The next step is to add the LD structure. I can do this with PLINK and the files I made for the GWAS overlap.</p>
<p>RunPlink_EIF2A.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=RunPlink_EIF2A
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=RunPlink_EIF2A.out
#SBATCH --error=RunPlink_EIF2A.err
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END

module load plink


plink --ped /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr3.ped  --map /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr3.map --r2  --ld-snp 3:150302010 --ld-window-kb 1000 --ld-window 99999  --out /project2/gilad/briana/threeprimeseq/data/LocusZoom/EIF2A_leadsnp.txt</code></pre>
<pre class="r"><code>LD_structure=read.table(&quot;../data/LocusZoom/EIF2A_leadsnp.txt.ld&quot;, header=T) %&gt;% select(BP_B, R2) 
colnames(LD_structure)=c(&quot;Location&quot;, &quot;R2&quot;)

allPheno_filt2=allPheno %&gt;% filter(Location&gt; 150292010 &amp; Location &lt; 150312010)
allPheno_filt_LD=allPheno_filt2 %&gt;% inner_join(LD_structure, by=&quot;Location&quot;)


allPheno_filt_LD_melt=melt(allPheno_filt_LD, id.vars=c(&quot;Location&quot;, &quot;R2&quot;))</code></pre>
<pre class="r"><code>lockedscale=ggplot(allPheno_filt_LD_melt, aes(x=Location, y=-log10(value), col=R2)) +  geom_point() + facet_grid( rows=vars(variable)) + geom_vline(xintercept=150302010, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


freescale=ggplot(allPheno_filt_LD_melt, aes(x=Location, y=-log10(value), col=R2)) +  geom_point() + facet_grid( rows=vars(variable), scales = &quot;free&quot;) + geom_vline(xintercept=150302010, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()</code></pre>
<pre class="r"><code>plot_grid(lockedscale,freescale, align = &quot;v&quot;, ncol=1)</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-12-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/figure/locusZoom.Rmd/unnamed-chunk-12-1.png" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Try on the same plot:</p>
<pre class="r"><code>ggplot(allPheno_filt_LD_melt, aes(x=Location, y=-log10(value), col=variable, by =variable)) +  geom_point() + geom_vline(xintercept=150302010, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-13-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/figure/locusZoom.Rmd/unnamed-chunk-13-1.png" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>rs14434 <a href="https://www.ncbi.nlm.nih.gov/variation/view/?q=rs14434&amp;assm=GCF_000001405.33" class="uri">https://www.ncbi.nlm.nih.gov/variation/view/?q=rs14434&amp;assm=GCF_000001405.33</a></p>
</div>
<div id="rint1" class="section level2">
<h2>RINT1</h2>
<p>RINT1 is a nuclear QTL. peak303436 7:105155320 ENSG00000135249</p>
<pre class="bash"><code>grep peak303436  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/TotalAPA.peak303436.RINT1.nomNuc.txt

grep peak303436  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/TotalAPA.peak303436.RINT1.nomTotal.txt

grep ENSG00000135249 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/RNA.RINT1.nomTotal.txt

grep ENSG00000135249 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_prot.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Prot.RINT1.nomTotal.txt

 grep ENSG00000135249 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_ribo_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Ribo.RINT1.nomTotal.txt
</code></pre>
<p>RunPlink_RINT1.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=RunPlink_RINT1
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=RunPlink_RINT1.out
#SBATCH --error=RunPlink_RINT1.err
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END

module load plink


plink --ped /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr7.ped  --map /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr7.map --r2  --ld-snp 7:105155320 --ld-window-kb 1000 --ld-window 99999  --out /project2/gilad/briana/threeprimeseq/data/LocusZoom/RINT1_leadsnp</code></pre>
<pre class="r"><code>APA_Total_RINT1=read.table(&quot;../data/LocusZoom/TotalAPA.peak303436.RINT1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_TotalPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_TotalPval)
APA_Total_RINT1$Location=as.integer(APA_Total_RINT1$Location)

APA_Nuclear_RINT1=read.table(&quot;../data/LocusZoom/TotalAPA.peak303436.RINT1.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_NuclearPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_NuclearPval)
APA_Nuclear_RINT1$Location=as.integer(APA_Nuclear_RINT1$Location)

Prot_RINT1=read.table(&quot;../data/LocusZoom/Prot.RINT1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;ProtPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, ProtPval)
Prot_RINT1$Location=as.integer(Prot_RINT1$Location)
RNA_RINT1=read.table(&quot;../data/LocusZoom/RNA.RINT1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RnaPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RnaPval)
RNA_RINT1$Location=as.integer(RNA_RINT1$Location)
Ribo_RINT1=read.table(&quot;../data/LocusZoom/Ribo.RINT1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RiboPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RiboPval)
Ribo_RINT1$Location=as.integer(Ribo_RINT1$Location)

LD_structure_RINT1=read.table(&quot;../data/LocusZoom/RINT1_leadsnp.ld&quot;, header=T) %&gt;% select(BP_B, R2) 
colnames(LD_structure_RINT1)=c(&quot;Location&quot;, &quot;R2&quot;)</code></pre>
<p>I can join these by the snps that are tested for all three. Filter 1kb up and downstream</p>
<pre class="r"><code>allPheno_RINT1=APA_Total_RINT1 %&gt;% inner_join(APA_Nuclear_RINT1, by=&quot;Location&quot;) %&gt;% inner_join(Prot_RINT1, by=&quot;Location&quot;) %&gt;% inner_join(Ribo_RINT1, by=&quot;Location&quot;) %&gt;% inner_join(RNA_RINT1, by=&quot;Location&quot;) %&gt;% inner_join(LD_structure_RINT1, by=&quot;Location&quot;) %&gt;% filter(Location&gt; 105154320 &amp; Location &lt; 105156320)

allPheno_RINT1_melt=melt(allPheno_RINT1, id.vars=c(&quot;Location&quot;, &quot;R2&quot;))


lockedscale_RINT1=ggplot(allPheno_RINT1_melt, aes(x=Location, y=-log10(value), col=R2)) +  geom_point() + facet_grid( rows=vars(variable)) + geom_vline(xintercept=105155320, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


freescale_RINT1=ggplot(allPheno_RINT1_melt, aes(x=Location, y=-log10(value), col=R2)) +  geom_point() + facet_grid( rows=vars(variable), scales = &quot;free&quot;) + geom_vline(xintercept=105155320, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


plot_grid(lockedscale_RINT1,freescale_RINT1, align = &quot;v&quot;, ncol=1)</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-17-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/figure/locusZoom.Rmd/unnamed-chunk-17-1.png" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>rs2463632 (7:105155320): it is an intronic variant in PUS7</p>
<p>PUS7 chr7:105,080,108-105,162,714 RINT1 chr7:105,172,532-105,208,124</p>
<p>This snp is in the intron on the gene directly upstream of RINT1.</p>
</div>
<div id="lyar" class="section level2">
<h2>LYAR</h2>
<p>This is a nuclear QTL as well. peak235215 4:4196045 ENSG00000145220</p>
<p>RunLocusZoom_LYAR.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=RunLocusZoom_LYAR
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=RunLocusZoom_LYAR.out
#SBATCH --error=RunLocusZoom_LYAR.err
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END

module load plink


grep peak235215  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/NuclearAPA.peak303436.LYAR.nomNuc.txt

grep peak235215  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/TotalAPA.peak303436.LYAR.nomTotal.txt

grep ENSG00000145220 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/RNA.LYAR.nomTotal.txt

grep ENSG00000145220 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_prot.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Prot.LYAR.nomTotal.txt

 grep ENSG00000145220 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_ribo_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Ribo.LYAR.nomTotal.txt


plink --ped /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr4.ped  --map /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr4.map --r2  --ld-snp 4:4196045 --ld-window-kb 1000 --ld-window 99999  --out /project2/gilad/briana/threeprimeseq/data/LocusZoom/LYAR_leadsnp.txt</code></pre>
<p>Move to my computer:</p>
<pre class="r"><code>APA_Total_LYAR=read.table(&quot;../data/LocusZoom/TotalAPA.peak303436.LYAR.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_TotalPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_TotalPval)
APA_Total_LYAR$Location=as.integer(APA_Total_LYAR$Location)

APA_Nuclear_LYAR=read.table(&quot;../data/LocusZoom/NuclearAPA.peak303436.LYAR.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_NuclearPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_NuclearPval)
APA_Nuclear_LYAR$Location=as.integer(APA_Nuclear_LYAR$Location)

Prot_LYAR=read.table(&quot;../data/LocusZoom/Prot.LYAR.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;ProtPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, ProtPval)
Prot_LYAR$Location=as.integer(Prot_LYAR$Location)
RNA_LYAR=read.table(&quot;../data/LocusZoom/RNA.LYAR.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RnaPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RnaPval)
RNA_LYAR$Location=as.integer(RNA_LYAR$Location)
Ribo_LYAR=read.table(&quot;../data/LocusZoom/Ribo.LYAR.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RiboPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RiboPval)
Ribo_LYAR$Location=as.integer(Ribo_LYAR$Location)

LD_structure_LYAR=read.table(&quot;../data/LocusZoom/LYAR_leadsnp.txt.ld&quot;, header=T) %&gt;% select(BP_B, R2) 
colnames(LD_structure_LYAR)=c(&quot;Location&quot;, &quot;R2&quot;)


allPheno_LYAR=APA_Total_LYAR %&gt;% inner_join(APA_Nuclear_LYAR, by=&quot;Location&quot;) %&gt;% inner_join(Prot_LYAR, by=&quot;Location&quot;) %&gt;% inner_join(Ribo_LYAR, by=&quot;Location&quot;) %&gt;% inner_join(RNA_LYAR, by=&quot;Location&quot;) %&gt;% inner_join(LD_structure_LYAR, by=&quot;Location&quot;) %&gt;% filter(Location&gt; 4191045 &amp; Location &lt; 4201045)

allPheno_LYAR_melt=melt(allPheno_LYAR, id.vars=c(&quot;Location&quot;, &quot;R2&quot;))


lockedscale_LYAR=ggplot(allPheno_LYAR_melt, aes(x=Location, y=-log10(value), col=R2)) +  geom_point() + facet_grid( rows=vars(variable)) + geom_vline(xintercept=4196045, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


freescale_LYAR=ggplot(allPheno_LYAR_melt, aes(x=Location, y=-log10(value), col=R2)) +  geom_point() + facet_grid( rows=vars(variable), scales = &quot;free&quot;) + geom_vline(xintercept=4196045, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


plot_grid(lockedscale_LYAR,freescale_LYAR, align = &quot;v&quot;, ncol=1)</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-19-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/figure/locusZoom.Rmd/unnamed-chunk-19-1.png" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Snp is in an intron OTOP1 gene 2 genes upstream. rs7682186</p>
</div>
<div id="psmf1" class="section level2">
<h2>PSMF1</h2>
<p>Total QTL peak193648 20:1131308 ENSG00000125818</p>
<p>RunLocusZoom_PSMF1.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=RunLocusZoom_PSMF1
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=RunLocusZoom_PSMF1.out
#SBATCH --error=RunLocusZoom_PSMF1.err
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END

module load plink


grep peak193648  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/NuclearAPA.peak193648.PSMF1.nomNuc.txt

grep peak193648  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/TotalAPA.peak193648.PSMF1.nomTotal.txt

grep ENSG00000125818 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/RNA.PSMF1.nomTotal.txt

grep ENSG00000125818 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_prot.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Prot.PSMF1.nomTotal.txt

 grep ENSG00000125818 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_ribo_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Ribo.PSMF1.nomTotal.txt


plink --ped /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr20.ped  --map /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr20.map --r2  --ld-snp 20:1131308 --ld-window-kb 1000 --ld-window 99999  --out /project2/gilad/briana/threeprimeseq/data/LocusZoom/PSMF1_leadsnp.txt</code></pre>
<p>Move to computer</p>
<pre class="r"><code>APA_Total_PSMF1=read.table(&quot;../data/LocusZoom/TotalAPA.peak193648.PSMF1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_TotalPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_TotalPval)
APA_Total_PSMF1$Location=as.integer(APA_Total_PSMF1$Location)

APA_Nuclear_PSMF1=read.table(&quot;../data/LocusZoom/NuclearAPA.peak193648.PSMF1.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_NuclearPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_NuclearPval)
APA_Nuclear_PSMF1$Location=as.integer(APA_Nuclear_PSMF1$Location)

Prot_PSMF1=read.table(&quot;../data/LocusZoom/Prot.PSMF1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;ProtPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, ProtPval)
Prot_PSMF1$Location=as.integer(Prot_PSMF1$Location)
RNA_PSMF1=read.table(&quot;../data/LocusZoom/RNA.PSMF1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RnaPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RnaPval)
RNA_PSMF1$Location=as.integer(RNA_PSMF1$Location)
Ribo_PSMF1=read.table(&quot;../data/LocusZoom/Ribo.PSMF1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RiboPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RiboPval)
Ribo_PSMF1$Location=as.integer(Ribo_PSMF1$Location)

LD_structure_PSMF1=read.table(&quot;../data/LocusZoom/PSMF1_leadsnp.txt.ld&quot;, header=T) %&gt;% select(BP_B, R2) 
colnames(LD_structure_PSMF1)=c(&quot;Location&quot;, &quot;R2&quot;)


allPheno_PSMF1=APA_Total_PSMF1 %&gt;% inner_join(APA_Nuclear_PSMF1, by=&quot;Location&quot;) %&gt;% inner_join(Prot_PSMF1, by=&quot;Location&quot;) %&gt;% inner_join(Ribo_PSMF1, by=&quot;Location&quot;) %&gt;% inner_join(RNA_PSMF1, by=&quot;Location&quot;) %&gt;%  inner_join(LD_structure_PSMF1, by=&quot;Location&quot;) %&gt;% filter(Location&gt; 1121308 &amp; Location &lt; 1181308)
allPheno_PSMF1_melt=melt(allPheno_PSMF1, id.vars=c(&quot;Location&quot;, &quot;R2&quot;))


lockedscale_PSMF1=ggplot(allPheno_PSMF1_melt, aes(x=Location, y=-log10(value),col=R2)) +  geom_point() + facet_grid( rows=vars(variable)) + geom_vline(xintercept=1131308, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


freescale_PSMF1=ggplot(allPheno_PSMF1_melt, aes(x=Location, y=-log10(value), col=R2)) +  geom_point() + facet_grid( rows=vars(variable), scales = &quot;free&quot;) + geom_vline(xintercept=1131308, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


plot_grid(lockedscale_PSMF1,freescale_PSMF1, align = &quot;v&quot;, ncol=1)</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-21-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/figure/locusZoom.Rmd/unnamed-chunk-21-1.png" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>This varriant is in an intron of the PSMF1 gene. rs56398212</p>
</div>
<div id="ebi3" class="section level2">
<h2>EBI3</h2>
<p>This is a total and a nuclear QTL peak152751, ENSG00000105246 19:4236475</p>
<p>RunLocusZoom_EBI3.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=RunLocusZoom_EBI3
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=RunLocusZoom_EBI3.out
#SBATCH --error=RunLocusZoom_EBI3.err
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END

module load plink


grep peak152751  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/NuclearAPA.peak152751.EBI3.nomNuc.txt

grep peak152751  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/TotalAPA.peak152751.EBI3.nomTotal.txt

grep ENSG00000105246 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/RNA.EBI3.nomTotal.txt

grep ENSG00000105246 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_prot.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Prot.EBI3.nomTotal.txt

 grep ENSG00000105246 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_ribo_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Ribo.EBI3.nomTotal.txt


plink --ped /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr19.ped  --map /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr19.map --r2  --ld-snp 19:4236475 --ld-window-kb 1000 --ld-window 99999  --out /project2/gilad/briana/threeprimeseq/data/LocusZoom/EBI3_leadsnp.txt</code></pre>
<p>Move to comp</p>
<pre class="r"><code>APA_Total_EBI3=read.table(&quot;../data/LocusZoom/TotalAPA.peak152751.EBI3.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_TotalPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_TotalPval)
APA_Total_EBI3$Location=as.integer(APA_Total_EBI3$Location)

APA_Nuclear_EBI3=read.table(&quot;../data/LocusZoom/NuclearAPA.peak152751.EBI3.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_NuclearPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_NuclearPval)
APA_Nuclear_EBI3$Location=as.integer(APA_Nuclear_EBI3$Location)

Prot_EBI3=read.table(&quot;../data/LocusZoom/Prot.EBI3.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;ProtPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, ProtPval)
Prot_EBI3$Location=as.integer(Prot_EBI3$Location)
RNA_EBI3=read.table(&quot;../data/LocusZoom/RNA.EBI3.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RnaPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RnaPval)
RNA_EBI3$Location=as.integer(RNA_EBI3$Location)
Ribo_EBI3=read.table(&quot;../data/LocusZoom/Ribo.EBI3.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RiboPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RiboPval)
Ribo_EBI3$Location=as.integer(Ribo_EBI3$Location)

LD_structure_EBI3=read.table(&quot;../data/LocusZoom/EBI3_leadsnp.txt.ld&quot;, header=T) %&gt;% select(BP_B, R2) 
colnames(LD_structure_EBI3)=c(&quot;Location&quot;, &quot;R2&quot;)


allPheno_EBI3=APA_Total_EBI3 %&gt;% inner_join(APA_Nuclear_EBI3, by=&quot;Location&quot;) %&gt;% inner_join(Prot_EBI3, by=&quot;Location&quot;) %&gt;% inner_join(Ribo_EBI3, by=&quot;Location&quot;) %&gt;% inner_join(RNA_EBI3, by=&quot;Location&quot;) %&gt;%  inner_join(LD_structure_EBI3, by=&quot;Location&quot;) %&gt;% filter(Location&gt; 4231475 &amp; Location &lt; 4241475)
allPheno_EBI3_melt=melt(allPheno_EBI3, id.vars=c(&quot;Location&quot;, &quot;R2&quot;))


lockedscale_EBI3=ggplot(allPheno_EBI3_melt, aes(x=Location, y=-log10(value),col=R2)) +  geom_point() + facet_grid( rows=vars(variable)) + geom_vline(xintercept=4236475, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


freescale_EBI3=ggplot(allPheno_EBI3_melt, aes(x=Location, y=-log10(value), col=R2)) +  geom_point() + facet_grid( rows=vars(variable), scales = &quot;free&quot;) + geom_vline(xintercept=4236475, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


plot_grid(lockedscale_EBI3,freescale_EBI3, align = &quot;v&quot;, ncol=1)</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-23-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/b2b7368a0c997d9a785f6fedf5c37b93d6ef672e/docs/figure/locusZoom.Rmd/unnamed-chunk-23-1.png" target="_blank">b2b7368</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Snp is in the last intron of EBI3. It looks like the lead protien snp is the one directly upstream. rs353704. The region is CCCCAC. The preceeding SNP is rs353705. The relevent peak is 19:4236433:4236517. The snp is in the peak. This is interesting because the alternative allele decreases usage of this peak and the protein.</p>
</div>
<div id="sacm1l" class="section level2">
<h2>SACM1L</h2>
<p>There are 3 QTLs in the total and nuclear for this. I am gonig to focus on the hit that has the same snp peak assocaition.</p>
<p>peak216086 3:45780980 ENSG00000211456</p>
<p>RunLocusZoom_SACM1L.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=RunLocusZoom_SACM1L
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=RunLocusZoom_SACM1L.out
#SBATCH --error=RunLocusZoom_SACM1L.err
#SBATCH --partition=broadwl
#SBATCH --mem=30G
#SBATCH --mail-type=END

module load plink


grep peak216086  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/NuclearAPA.peak216086.SACM1L.nomNuc.txt

grep peak216086  /project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_NomRes.txt &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/TotalAPA.peak216086.SACM1L.nomTotal.txt

grep ENSG00000211456 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_RNAseq_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/RNA.SACM1L.nomTotal.txt

grep ENSG00000211456 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_prot.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Prot.SACM1L.nomTotal.txt

 grep ENSG00000211456 /project2/gilad/briana/threeprimeseq/data/molecular_QTLs/nom/fastqtl_qqnorm_ribo_phase2.fixed.nominal.out &gt; /project2/gilad/briana/threeprimeseq/data/LocusZoom/Ribo.SACM1L.nomTotal.txt


plink --ped /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr3.ped  --map /project2/gilad/briana/YRI_geno_hg19/plinkYRIgeno_chr3.map --r2  --ld-snp 3:45780980 --ld-window-kb 1000 --ld-window 99999  --out /project2/gilad/briana/threeprimeseq/data/LocusZoom/SACM1L_leadsnp.txt</code></pre>
<p>.</p>
<p>Move to comp</p>
<pre class="r"><code>APA_Total_SACM1L=read.table(&quot;../data/LocusZoom/TotalAPA.peak216086.SACM1L.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_TotalPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_TotalPval)
APA_Total_SACM1L$Location=as.integer(APA_Total_SACM1L$Location)

APA_Nuclear_SACM1L=read.table(&quot;../data/LocusZoom/NuclearAPA.peak216086.SACM1L.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;APA_NuclearPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;) %&gt;% select( Location, APA_NuclearPval)
APA_Nuclear_SACM1L$Location=as.integer(APA_Nuclear_SACM1L$Location)

Prot_SACM1L=read.table(&quot;../data/LocusZoom/Prot.SACM1L.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;ProtPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, ProtPval)
Prot_SACM1L$Location=as.integer(Prot_SACM1L$Location)
RNA_SACM1L=read.table(&quot;../data/LocusZoom/RNA.SACM1L.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RnaPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RnaPval)
RNA_SACM1L$Location=as.integer(RNA_SACM1L$Location)
Ribo_SACM1L=read.table(&quot;../data/LocusZoom/Ribo.SACM1L.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SID&quot;, &quot;Dist&quot;, &quot;RiboPval&quot;,&quot;slope&quot;)) %&gt;% separate(SID, into=c(&quot;Chrom&quot;, &quot;Location&quot;), sep=&quot;:&quot;)%&gt;% select( Location, RiboPval)
Ribo_SACM1L$Location=as.integer(Ribo_SACM1L$Location)

LD_structure_SACM1L=read.table(&quot;../data/LocusZoom/SACM1L_leadsnp.txt.ld&quot;, header=T) %&gt;% select(BP_B, R2) 
colnames(LD_structure_SACM1L)=c(&quot;Location&quot;, &quot;R2&quot;)


allPheno_SACM1L=APA_Total_SACM1L %&gt;% inner_join(APA_Nuclear_SACM1L, by=&quot;Location&quot;) %&gt;% inner_join(Prot_SACM1L, by=&quot;Location&quot;) %&gt;% inner_join(Ribo_SACM1L, by=&quot;Location&quot;) %&gt;% inner_join(RNA_SACM1L, by=&quot;Location&quot;) %&gt;%  inner_join(LD_structure_SACM1L, by=&quot;Location&quot;) %&gt;% filter(Location&gt; 45770980 &amp; Location &lt; 45790980)
allPheno_SACM1L_melt=melt(allPheno_SACM1L, id.vars=c(&quot;Location&quot;, &quot;R2&quot;))


lockedscale_SACM1L=ggplot(allPheno_SACM1L_melt, aes(x=Location, y=-log10(value),col=R2)) +  geom_point() + facet_grid( rows=vars(variable)) + geom_vline(xintercept=45780980, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


freescale_SACM1L=ggplot(allPheno_SACM1L_melt, aes(x=Location, y=-log10(value), col=R2)) +  geom_point() + facet_grid( rows=vars(variable), scales = &quot;free&quot;) + geom_vline(xintercept=45780980, linetype=&quot;dashed&quot;, color = &quot;red&quot;) +  theme_linedraw()


plot_grid(lockedscale_SACM1L,freescale_SACM1L, align = &quot;v&quot;, ncol=1)</code></pre>
<p><img src="figure/locusZoom.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-25-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/410f25c72a15eb6d32f39498fb046641b4eecb5a/docs/figure/locusZoom.Rmd/unnamed-chunk-25-1.png" target="_blank">410f25c</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-11-16
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>The snp is in an intron of the SAMCL1 gene rs80065472. The peak is in the UTR of the gene.</p>
</div>
<div id="locuszoom-online" class="section level2">
<h2>LocusZoom online</h2>
<pre class="r"><code>APA_LZ=read.table(&quot;../data/LocusZoom/TotalAPA.peak228606.EIF2A.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(APA_LZ,&quot;../data/LocusZoom/apaEIF21LZ.txt&quot;, col.names = T, row.names = F, quote = F)
#sed -e &#39;s/^/Chr/&#39; apaEIF21LZ.txt &gt; apaEIF21LZ_chr.txt 

prot_LZ=read.table(&quot;../data/LocusZoom/Prot.EIF2A.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(prot_LZ,&quot;../data/LocusZoom/ProtEIF21LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;ProtEIF21LZ.txt &gt; ProtEIF21LZ_chr.txt 


RNA_LZ=read.table(&quot;../data/LocusZoom/RNA.EIF2A.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(RNA_LZ,&quot;../data/LocusZoom/RNAEIF21LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39; RNAEIF21LZ.txt &gt; RNAEIF21LZ_chr.txt 


ribo_LZ=read.table(&quot;../data/LocusZoom/Ribo.EIF2A.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(ribo_LZ,&quot;../data/LocusZoom/RiboEIF21LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39; RiboEIF21LZ.txt &gt; RiboEIF21LZ_chr.txt</code></pre>
<p>Do this for another qtl:</p>
<pre class="r"><code>APATotal_SACM1L_LZ=read.table(&quot;../data/LocusZoom/TotalAPA.peak216086.SACM1L.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(APATotal_SACM1L_LZ,&quot;../data/LocusZoom/apaTotalSACM1L_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


APANuclear_SACM1L_LZ=read.table(&quot;../data/LocusZoom/NuclearAPA.peak216086.SACM1L.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(APANuclear_SACM1L_LZ,&quot;../data/LocusZoom/apaNuclearSACM1L_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


prot_SACM1L_LZ=read.table(&quot;../data/LocusZoom/Prot.SACM1L.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(prot_SACM1L_LZ,&quot;../data/LocusZoom/ProtSACM1L_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


RNA_SACM1L_LZ=read.table(&quot;../data/LocusZoom/RNA.SACM1L.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(RNA_SACM1L_LZ,&quot;../data/LocusZoom/RNASACM1L_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


ribo_SACM1L_LZ=read.table(&quot;../data/LocusZoom/Ribo.SACM1L.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(ribo_SACM1L_LZ,&quot;../data/LocusZoom/RiboSACM1L_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39; </code></pre>
<p>One more example: LYAR</p>
<pre class="r"><code>APATotal_LYAR_LZ=read.table(&quot;../data/LocusZoom/TotalAPA.peak303436.LYAR.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(APATotal_LYAR_LZ,&quot;../data/LocusZoom/apaTotalLYAR_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


APANuclear_LYAR_LZ=read.table(&quot;../data/LocusZoom/NuclearAPA.peak303436.LYAR.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(APANuclear_LYAR_LZ,&quot;../data/LocusZoom/apaNuclearLYAR_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


prot_LYAR_LZ=read.table(&quot;../data/LocusZoom/Prot.LYAR.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(prot_LYAR_LZ,&quot;../data/LocusZoom/ProtLYAR_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


RNA_LYAR_LZ=read.table(&quot;../data/LocusZoom/RNA.LYAR.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(RNA_LYAR_LZ,&quot;../data/LocusZoom/RNALYAR_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


ribo_LYAR_LZ=read.table(&quot;../data/LocusZoom/Ribo.LYAR.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(ribo_LYAR_LZ,&quot;../data/LocusZoom/RiboLYAR_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39; </code></pre>
<p>RINT1</p>
<pre class="r"><code>APATotal_RINT1_LZ=read.table(&quot;../data/LocusZoom/TotalAPA.peak303436.RINT1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(APATotal_RINT1_LZ,&quot;../data/LocusZoom/apaTotalRINT1_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


APANuclear_RINT1_LZ=read.table(&quot;../data/LocusZoom/TotalAPA.peak303436.RINT1.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(APANuclear_RINT1_LZ,&quot;../data/LocusZoom/apaNuclearRINT1_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


prot_RINT1_LZ=read.table(&quot;../data/LocusZoom/Prot.RINT1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(prot_RINT1_LZ,&quot;../data/LocusZoom/ProtRINT1_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


RNA_RINT1_LZ=read.table(&quot;../data/LocusZoom/RNA.RINT1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(RNA_RINT1_LZ,&quot;../data/LocusZoom/RNARINT1_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


ribo_RINT1_LZ=read.table(&quot;../data/LocusZoom/Ribo.RINT1.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(ribo_RINT1_LZ,&quot;../data/LocusZoom/RiboRINT1_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39; </code></pre>
<p>EBI3 rs353704.</p>
<pre class="r"><code>APATotal_EBI3_LZ=read.table(&quot;../data/LocusZoom/TotalAPA.peak152751.EBI3.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(APATotal_EBI3_LZ,&quot;../data/LocusZoom/apaTotalEBI3_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


APANuclear_EBI3_LZ=read.table(&quot;../data/LocusZoom/NuclearAPA.peak152751.EBI3.nomNuc.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(APANuclear_EBI3_LZ,&quot;../data/LocusZoom/apaNuclearEBI3_LZ.txt&quot;, col.names = T, row.names = F, quote = F)


prot_EBI3_LZ=read.table(&quot;../data/LocusZoom/Prot.EBI3.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)
write.table(prot_EBI3_LZ,&quot;../data/LocusZoom/ProtEBI3_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


RNA_EBI3_LZ=read.table(&quot;../data/LocusZoom/RNA.EBI3.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(RNA_EBI3_LZ,&quot;../data/LocusZoom/RNAEBI3_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39;


ribo_EBI3_LZ=read.table(&quot;../data/LocusZoom/Ribo.EBI3.nomTotal.txt&quot;, stringsAsFactors = F, col.names = c(&quot;PeakID&quot;, &quot;SNP&quot;, &quot;Dist&quot;, &quot;P&quot;,&quot;slope&quot;))  %&gt;% select( SNP, P)

write.table(ribo_EBI3_LZ,&quot;../data/LocusZoom/RiboEBI3_LZ.txt&quot;, col.names = T, row.names = F, quote = F)

#sed -e &#39;s/^/Chr/&#39; </code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] bindrcpp_0.2.2      cowplot_0.9.3       ggpubr_0.1.8       
 [4] magrittr_1.5        data.table_1.11.8   VennDiagram_1.6.20 
 [7] futile.logger_1.4.3 forcats_0.3.0       stringr_1.3.1      
[10] dplyr_0.7.6         purrr_0.2.5         readr_1.1.1        
[13] tidyr_0.8.1         tibble_1.4.2        ggplot2_3.0.0      
[16] tidyverse_1.2.1     reshape2_1.4.3      workflowr_1.1.1    

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4     haven_1.1.2          lattice_0.20-35     
 [4] colorspace_1.3-2     htmltools_0.3.6      yaml_2.2.0          
 [7] rlang_0.2.2          R.oo_1.22.0          pillar_1.3.0        
[10] glue_1.3.0           withr_2.1.2          R.utils_2.7.0       
[13] lambda.r_1.2.3       modelr_0.1.2         readxl_1.1.0        
[16] bindr_0.1.1          plyr_1.8.4           munsell_0.5.0       
[19] gtable_0.2.0         cellranger_1.1.0     rvest_0.3.2         
[22] R.methodsS3_1.7.1    evaluate_0.11        labeling_0.3        
[25] knitr_1.20           broom_0.5.0          Rcpp_0.12.19        
[28] formatR_1.5          backports_1.1.2      scales_1.0.0        
[31] jsonlite_1.5         hms_0.4.2            digest_0.6.17       
[34] stringi_1.2.4        rprojroot_1.3-2      cli_1.0.1           
[37] tools_3.5.1          lazyeval_0.2.1       futile.options_1.0.1
[40] crayon_1.3.4         whisker_0.3-2        pkgconfig_2.0.2     
[43] xml2_1.2.0           lubridate_1.7.4      assertthat_0.2.0    
[46] rmarkdown_1.10       httr_1.3.1           rstudioapi_0.8      
[49] R6_2.3.0             nlme_3.1-137         git2r_0.23.0        
[52] compiler_3.5.1      </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
