<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Merge Replicates</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Merge Replicates</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>2/21/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-02-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>threeprimeseq/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtreebd4b3dca639f45243a9559f2516499e43434a0edtargetblankbd4b3dca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/bd4b3dca639f45243a9559f2516499e43434a0ed" target="_blank">bd4b3dc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtreebd4b3dca639f45243a9559f2516499e43434a0edtargetblankbd4b3dca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/perm_QTL_trans_noMP_5percov/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/4suDataIGV.Rmd
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/EvaleQTLs.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/fixBWChromNames.Rmd
    Untracked:  analysis/groSeqAnalysis.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  analysis/verifybam_dubs.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/AllPeak_counts/
    Untracked:  data/ApaQTLs/
    Untracked:  data/ApaQTLs_otherPhen/
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/DistTXN2Peak_genelocAnno/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/LocusZoom_proc/
    Untracked:  data/MatchedSnps/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeakCounts_noMP_5perc/
    Untracked:  data/PeakCounts_noMP_genelocanno/
    Untracked:  data/PeakUsage/
    Untracked:  data/PeakUsage_noMP/
    Untracked:  data/PeakUsage_noMP_GeneLocAnno/
    Untracked:  data/PeaksUsed/
    Untracked:  data/PeaksUsed_noMP_5percCov/
    Untracked:  data/QTL_overlap/
    Untracked:  data/RNAkalisto/
    Untracked:  data/RefSeq_annotations/
    Untracked:  data/Replicates_usage/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/WASP_STAT/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/apaExamp_proc/
    Untracked:  data/apaQTL_examp_noMP/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_GeneLocAnno/
    Untracked:  data/diff_iso_proc/
    Untracked:  data/diff_iso_trans/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/molPheno_noMP/
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov_3UTR/
    Untracked:  data/perm_QTL_diffWindow/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/protAndAPAAndExplmRes.Rda
    Untracked:  data/protAndAPAlmRes.Rda
    Untracked:  data/protAndExpressionlmRes.Rda
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  data/threePrimeSeqMetaData55Ind.txt
    Untracked:  data/threePrimeSeqMetaData55Ind.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  data/~$threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  output/deeptools_plots/
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/NewPeakPostMP.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diffIsoAnalysisNewMapping.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/mispriming_approach.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlapMolQTL.opposite.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/peakQCPPlots.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/pipeline_55Ind.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/test.smash.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/bd4b3dca639f45243a9559f2516499e43434a0ed/analysis/mergeReplicates.Rmd" target="_blank">bd4b3dc</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-02-25
</td>
<td>
add usage diff heatplot
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.2.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ──────────────────────────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.0.0     ✔ purrr   0.2.5
✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.4.0
✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 3.5.2</code></pre>
<pre><code>── Conflicts ─────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<p>In this analysis I will merge replicates to see if more seq reads helps with coverage per gene and variance between peak usage score.<br />
##Process replicate samples</p>
<p>/project2/gilad/briana/threeprimeseq/data/sort/<br />
/project2/gilad/briana/threeprimeseq/data/Replicates/<br />
/project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam/</p>
<p>18499-N-batch4.combined.STARwWASP.bamAligned.filtered.sort.bam YL-SP-18499-N-combined-sort.bam 18499-T-batch4.combined.STARwWASP.bamAligned.filtered.sort.bam YL-SP-18499-T-combined-sort.bam</p>
<p>18912-N-batch4.combined.STARwWASP.bamAligned.filtered.sort.bam YL-SP-18912-N-combined-sort.bam 18912-T-batch4.combined.STARwWASP.bamAligned.filtered.sort.bam YL-SP-18912-N-combined-sort.bam</p>
<p>19093-N-batch4.combined.STARwWASP.bamAligned.filtered.sort.bam YL-SP-19093-N-combined-sort.bam 19093-T-batch4.combined.STARwWASP.bamAligned.filtered.sort.bam YL-SP-19093-T-combined-sort.bam</p>
<p>19140-N-batch4.combined.STARwWASP.bamAligned.filtered.sort.bam YL-SP-19140-N-combined-sort.bam 19140-T-batch4.combined.STARwWASP.bamAligned.filtered.sort.bam YL-SP-19140-T-combined-sort.bam</p>
<p>mergeReplicateBams.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=mergeReplicateBams
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeReplicateBams.out
#SBATCH --error=mergeReplicateBams.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env


for i in &quot;18499-N&quot; &quot;18499-T&quot; &quot;18912-N&quot; &quot;18912-T&quot;  &quot;19093-N&quot; &quot;19093-T&quot; &quot;19140-N&quot; &quot;19140-T&quot; 
do 
samtools merge /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam/YL-SP-$i-mergedReps-combined.bam /project2/gilad/briana/threeprimeseq/data/sort/YL-SP-$i-combined-sort.bam /project2/gilad/briana/threeprimeseq/data/Replicates/$i-batch4.combined.STARwWASP.bamAligned.filtered.sort.bam 
samtools sort /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam/YL-SP-$i-mergedReps-combined.bam &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam/YL-SP-$i-mergedReps-combined-sort.bam
samtools index /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam/YL-SP-$i-mergedReps-combined-sort.bam
done

</code></pre>
<p>Bam to bed:</p>
<p>bam2BedandSort.replicates.sh</p>
<pre class="bash"><code>#!/bin/bash


#SBATCH --job-name=bam2BedandSort.replicates
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=bam2BedandSort.replicates.out
#SBATCH --error=bam2BedandSort.replicates.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env 


for i in $(ls /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam/*-sort.bam )
do
describer=$(echo ${i} |sed -e &#39;s/.*YL-SP-//&#39;  | sed -e &quot;s/-mergedReps-combined-sort.bam//&quot;)
bedtools bamtobed -i $i &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBed/YL-SP-$describer-mergedReps-combined.bed
sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBed/YL-SP-$describer-mergedReps-combined.bed &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBed/YL-SP-$describer-mergedReps-combined-sort.bed
done 
</code></pre>
<p>Next step is to get the 10 bp upstream:</p>
<p>Upstream10Bases_replicates.py</p>
<pre class="bash"><code>#python  
def main(Fin, Fout):
  outBed=open(Fout, &quot;w&quot;)
  chrom_lengths=open(&quot;/project2/gilad/briana/genome_anotation_data/chrom_lengths2.sort.bed&quot;,&quot;r&quot;)
  #make a dictionary with chrom lengths
  length_dic={}
  for i in chrom_lengths:
    chrom, start, end = i.split()
    length_dic[str(chrom)]=int(end)  

#write file 
  for ln in open(Fin):
    chrom, start, end, name, score, strand = ln.split()
    chrom=str(chrom)
    if strand==&quot;+&quot;:
      start_new=int(start)-10
      if start_new &lt;= 1:
        start_new = 0 
      end_new= int(start)
      if end_new == 0:
        end_new=1
      outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new, end_new, name, score, strand))
    if strand == &quot;-&quot;:
      start_new=int(end)
      end_new=int(end) + 10
      if end_new &gt;= length_dic[chrom]:
        end_new = length_dic[chrom] 
        start_new=end_new-1 
      outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new, end_new, name, score, strand))
  outBed.close()  

if __name__ == &quot;__main__&quot;:
    import sys
    inFile = sys.argv[1]
    fileNoPath=inFile.split(&quot;/&quot;)[-1]
    fileshort=fileNoPath[:-4]
    outFile=&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_bed_10up/&quot; + fileshort + &quot;10up.bed&quot;
    main(inFile, outFile)</code></pre>
<p>wrap_Upstream10Bases_rep.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=w_Upstream10Bases_rep
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=w_Upstream10Bases_rep.out
#SBATCH --error=w_Upstream10Bases_rep.err
#SBATCH --partition=broadwl
#SBATCH --mem=8G
#SBATCH --mail-type=END


module load Anaconda3  

source activate three-prime-env


for i in $(ls /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBed/*sort.bed); do
            python  Upstream10Bases_replicates.py  $i 
        done</code></pre>
<p>Nuc10BasesUp_replicate.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=Nuc10BasesUp_replicate
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=Nuc10BasesUp_replicate.out
#SBATCH --error=Nuc10BasesUp_replicate.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

for i in $(ls /project2/gilad/briana/threeprimeseq/data/Replicates_bed_10up/*);do
   describer=$(echo ${i} | sed -e &#39;s/.*YL-SP-//&#39; | sed -e &quot;s/-mergedReps-combined-sort10up.bed$//&quot;)
   bedtools nuc -s -seq -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed $i &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_nuc_10up/TenBaseUP.${describer}.txt   
done</code></pre>
<p>replicates_filterMissprimingInNuc10.py</p>
<pre class="bash"><code>def main(Fin, Fout):
  outBed=open(Fout, &quot;w&quot;)
  inBed=open(Fin, &quot;r&quot;)
  for ind, ln in enumerate(inBed):
    if ind &gt;=1:
      chrom,start, end, name, score, strand, pctAT, pctGC, A, C, G, T, N, Other, Length, Sequence = ln.split()
      Tperc= float(T) / float(Length)
      if Tperc &lt; .7:
          if &quot;TTTTTT&quot; not in Sequence:
              start_new=int(start)
              end_new=int(end)
              outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new, end_new , name, score, strand))
  outBed.close()

if __name__ == &quot;__main__&quot;:
    import sys
    inFile = sys.argv[1]
    fileNoPath=inFile.split(&quot;/&quot;)[-1]
    sampleName=fileNoPath.split(&quot;.&quot;)[1]
    outFile=&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_nuc_10up_CleanReads/TenBaseUP.&quot; + sampleName + &quot;.CleanReads.bed&quot;
    main(inFile, outFile)</code></pre>
<p>run_filterMissprimingInNuc10_rep.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_filterMissprimingInNuc10_rep
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=run_filterMissprimingInNuc10_rep.out
#SBATCH --error=c
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

for i in $(ls /project2/gilad/briana/threeprimeseq/data/Replicates_nuc_10up/*);do
    python filterMissprimingInNuc10.py $i  
done </code></pre>
<p>filterSortBedbyCleanedBed_rep.R</p>
<pre class="r"><code>#!/bin/rscripts

# usage: Rscirpt --vanilla  filterSortBedbyCleanedBed.R identifier

#this script takes in the sorted bed file and the clean reads, it will clean the bed file   


library(dplyr)
library(tidyr)
library(data.table)


args = commandArgs(trailingOnly=TRUE)
identifier=args[1]


sortBedName= paste(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_mergedBed/YL-SP-&quot;, identifier, &quot;-mergedReps-combined-sort.bed&quot;, sep=&quot;&quot;)

CleanName= paste(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_nuc_10up_CleanReads/TenBaseUP.&quot;, identifier, &quot;.CleanReads.bed&quot;, sep=&quot;&quot;)

outFile= paste(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_bed_sort_CleanedMP/YL-SP-&quot;, identifier, &quot;-combined-sort.clean.bed&quot;, sep=&quot;&quot;)  

bedFile=fread(sortBedName, col.names = c(&quot;Chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;))

cleanFile=fread(CleanName, col.names = c(&quot;Chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;))

intersection=bedFile %&gt;% semi_join(cleanFile, by=&quot;name&quot;)

fwrite(intersection, file=outFile,quote = F, col.names = F, row.names = F, sep=&quot;\t&quot;)</code></pre>
<p>run_filterSortBedbyCleanedBed_replicates.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_filterSortBedbyCleanedBed
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=run_filterSortBedbyCleanedBed.out
#SBATCH --error=run_filterSortBedbyCleanedBed.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env  

for i in $(ls /project2/gilad/briana/threeprimeseq/data/Replicates_nuc_10up_CleanReads/*);do
   describer=$(echo ${i} | sed -e &#39;s/.*TenBaseUP.//&#39; | sed -e &quot;s/.CleanReads.bed//&quot;)
   Rscript --vanilla  filterSortBedbyCleanedBed_rep.R  ${describer}
done 
</code></pre>
<p>sort_filterSortBedbyCleanedBed_rep.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=sort_filterSortBedbyCleanedBed_rep
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=sort_filterSortBedbyCleanedBed_rep.out
#SBATCH --error=sort_filterSortBedbyCleanedBed_rep.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env  


for i in $(ls /project2/gilad/briana/threeprimeseq/data/Replicates_bed_sort_CleanedMP/*);do
  describer=$(echo ${i} | sed -e &#39;s/.*YL-SP-//&#39; | sed -e &quot;s/-combined-sort.clean.bed//&quot;)
  bedtools sort -faidx /project2/gilad/briana/threeprimeseq/code/chromOrder.num.txt -i  /project2/gilad/briana/threeprimeseq/data/Replicates_bed_sort_CleanedMP/YL-SP-${describer}-combined-sort.clean.bed &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_bed_sort_CleanedMP/YL-SP-${describer}-combined-sort.clean.sorted.bed
done</code></pre>
<p>filterBamforMP.pysam2_rep.py</p>
<pre class="bash"><code>#!/usr/bin/env python


&quot;&quot;&quot;
Usage: python filterBamforMP.pysam2_rep.py &lt;describer&gt;
&quot;&quot;&quot;


def main(Bin, Bamin, out):
    okRead={}
    for ln in open(Bin, &quot;r&quot;):
        chrom, start_new , end_new , name, score, strand = ln.split()
        okRead[name] = &quot;&quot;
    #pysam to read in bam allignments
    bamfile = pysam.AlignmentFile(Bamin, &quot;rb&quot;)
    finalBam =  pysam.AlignmentFile(out, &quot;wb&quot;, template=bamfile)
    #read name is the first col in each bam file
    n=0
    for read in bamfile.fetch():
        read_name=read.query_name
        #if statement about name  
        if read_name in okRead.keys():
            finalBam.write(read)
        if n % 1000==0 : print(n)
        n+=1 
    bamfile.close()
    finalBam.close()

    
if __name__ == &quot;__main__&quot;:
    import sys, pysam
    describer = sys.argv[1]
    inBed= &quot;/project2/gilad/briana/threeprimeseq/data/Replicates_bed_sort_CleanedMP/YL-SP-&quot; + describer + &quot;-combined-sort.clean.sorted.bed&quot;
    inBam=&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam/YL-SP-&quot; + describer + &quot;-mergedReps-combined-sort.bam&quot;
    outBam=&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam_NoMP/YL-SP-&quot; + describer + &quot;-combined-MergedReps_sort.noMP.bam&quot;
    main(inBed, inBam, outBam)</code></pre>
<p>wrap_filterBamforMP.pysam2_rep.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=wrap_filterBamforMP.pysam2_rep
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=wrap_filterBamforMP.pysam2_rep.out
#SBATCH --error=wrap_filterBamforMP.pysam2_rep.err
#SBATCH --partition=broadwl
#SBATCH --mem=32G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env  

for i in $(ls /project2/gilad/briana/threeprimeseq/data/Replicates_bed_sort_CleanedMP/*sort.clean.sorted.bed);do
   describer=$(echo ${i} | sed -e &#39;s/.*YL-SP-//&#39; | sed -e &quot;s/-combined-sort.clean.sorted.bed//&quot;)
   python filterBamforMP.pysam2_rep.py ${describer}
done</code></pre>
<p>Read counts after cleaning:<br />
18499N-12417636 18499T-17359225 18912N-11020463 18912T-21554082 19093N-13089487 19093T-24167003<br />
19140N- 14668172 19140T- 23520072</p>
<p>SortIndexBam_noMP_rep.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=SortIndexBam_noMP_rep
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=SortIndexBam_noMP_rep.out
#SBATCH --error=SortIndexBam_noMP_rep.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

#module load Anaconda3
#source activate three-prime-env 

module load samtools 

for i in $(ls /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam_NoMP/*);do
 describer=$(echo ${i} | sed -e &#39;s/.*YL-SP-//&#39; | sed -e &quot;s/-combined-MergedReps_sort.noMP.bam//&quot;)
  samtools sort /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam_NoMP/YL-SP-${describer}-combined-MergedReps_sort.noMP.bam &gt;  /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam_NoMP/YL-SP-${describer}-combined-MergedReps_sort.noMP.sort.bam 
  samtools index /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam_NoMP/YL-SP-${describer}-combined-MergedReps_sort.noMP.sort.bam 
done  </code></pre>
<p>Run feature counts on the peaks<br />
GeneLocAnno_fc_TN_noMP_rep.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=GeneLocAnno_fc_TN_noMP_rep
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=GeneLocAnno_fc_TN_noMP_rep.out
#SBATCH --error=GeneLocAnno_fc_TN_noMP_rep.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

featureCounts -O -a /project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.SAF -F SAF -o /project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fc /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam_NoMP/*T*.sort.bam -s 1

featureCounts -O -a /project2/gilad/briana/threeprimeseq/data/mergedPeaks_noMP_GeneLoc/Filtered_APApeaks_merged_allchrom_noMP.sort.named.noCHR_geneLocParsed.SAF -F SAF -o /project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fc /project2/gilad/briana/threeprimeseq/data/Replicates_mergedBam_NoMP/*N*.sort.bam -s 1
</code></pre>
<p>Fix header:</p>
<p>fix_head_fc_geneLoc_tot_noMP_rep.py</p>
<pre class="bash"><code>infile= open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fc&quot;, &quot;r&quot;)
fout = open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;,&#39;w&#39;)
for line, i in enumerate(infile):
    if line == 1:
        i_list=i.split()
        libraries=i_list[:6]
        for sample in i_list[6:]:
            full = sample.split(&quot;/&quot;)[7]
            samp= full.split(&quot;-&quot;)[2:4]
            lim=&quot;_&quot;
            samp_st=lim.join(samp)
            libraries.append(samp_st)
        first_line= &quot;\t&quot;.join(libraries)
        fout.write(first_line + &#39;\n&#39;)
    else :
        fout.write(i)
fout.close()</code></pre>
<p>fix_head_fc_geneLoc_nuc_noMP_rep.py</p>
<pre class="bash"><code>infile= open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fc&quot;, &quot;r&quot;)
fout = open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;,&#39;w&#39;)
for line, i in enumerate(infile):
    if line == 1:
        i_list=i.split()
        libraries=i_list[:6]
        for sample in i_list[6:]:
            full = sample.split(&quot;/&quot;)[7]
            samp= full.split(&quot;-&quot;)[2:4]
            lim=&quot;_&quot;
            samp_st=lim.join(samp)
            libraries.append(samp_st)
        first_line= &quot;\t&quot;.join(libraries)
        fout.write(first_line + &#39;\n&#39;)
    else :
        fout.write(i)
fout.close()</code></pre>
<p>Make phenotype:<br />
create_fileid_geneLocAnno_nuclear_rep.py</p>
<pre class="bash"><code>fout = open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_file_id_mapping_nuclear_Transcript_head.txt&quot;,&#39;w&#39;)
infile= open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;, &quot;r&quot;)
for line, i in enumerate(infile):
    if line == 0:
        i_list=i.split()
        files= i_list[10:-2]
        for each in files:
            full = each.split(&quot;/&quot;)[7]
            samp= full.split(&quot;-&quot;)[2:4]
            lim=&quot;_&quot;
            samp_st=lim.join(samp)
            outLine= full[:-1] + &quot;\t&quot; + samp_st
            fout.write(outLine + &quot;\n&quot;)
fout.close()</code></pre>
<p>create_fileid_geneLocAnno_Total_rep.py</p>
<pre class="bash"><code>fout = open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_file_id_mapping_total_Transcript_head.txt&quot;,&#39;w&#39;)
infile= open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;, &quot;r&quot;)
for line, i in enumerate(infile):
    if line == 0:
        i_list=i.split()
        files= i_list[10:-2]
        for each in files:
            full = each.split(&quot;/&quot;)[7]
            samp= full.split(&quot;-&quot;)[2:4]
            lim=&quot;_&quot;
            samp_st=lim.join(samp)
            outLine= full[:-1] + &quot;\t&quot; + samp_st
            fout.write(outLine + &quot;\n&quot;)
fout.close()</code></pre>
<p>makePhenoRefSeqPeaks_GeneLoc_Total_noMP_rep.py</p>
<pre class="bash"><code>#PYTHON 3

dic_IND = {}
dic_BAM = {}

for ln in open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_file_id_mapping_total_Transcript_head.txt&quot;):
    bam, IND = ln.split(&quot;\t&quot;)
    IND = IND.strip()
    dic_IND[bam] = IND
    if IND not in dic_BAM:
        dic_BAM[IND] = []
    dic_BAM[IND].append(bam)


#now I have ind dic with keys as the bam and ind as the values
#I also have a bam dic with ind as the keys and bam as the values  
    
inds=list(dic_BAM.keys()) #list of ind libraries  

#gene start and end dictionaries: 
dic_geneS = {}
dic_geneE = {}
for ln in open(&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_endAllGenes.sort.bed&quot;):
    chrom, start, end, geneID, score, strand = ln.split(&#39;\t&#39;)
    gene= geneID.split(&quot;:&quot;)[1]
#    if &quot;-&quot; in gene:
 #       gene=gene.split(&quot;-&quot;)[0]
    if gene not in dic_geneS:
        dic_geneS[gene]=int(start)
        dic_geneE[gene]=int(end)
        


#list of genes   

count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;, &quot;r&quot;)
genes=[]
for line , i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        gene=id_list[5]
        if gene not in genes:
            genes.append(gene)
            
#make the ind and gene dic  
dic_dub={}
for g in genes:
    dic_dub[g]={}
    for i in inds:
        dic_dub[g][i]=0


#populate the dictionary  
count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;, &quot;r&quot;)
for line, i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        g= id_list[5]
        values=list(i_list[6:])
        list_list=[]
        for ind,val in zip(inds, values):
            list_list.append([ind, val])
        for num, name in enumerate(list_list):
            dic_dub[g][list_list[num][0]] += int(list_list[num][1])
        

#write the file by acessing the dictionary and putting values in the table ver the value in the dic 
        

fout=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.fc&quot;,&quot;w&quot;)
peak=[&quot;chrom&quot;]
inds_noL=[]
for each in inds:
    indsNA= &quot;NA&quot; + each[:-2]
    inds_noL.append(indsNA) 
fout.write(&quot; &quot;.join(peak + inds_noL) + &#39;\n&#39; )


count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;, &quot;r&quot;)
for line , i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        gene=id_list[5]
        #start=dic_geneS[id_list[5]]
        start=int(id_list[2])
        #end=dic_geneE[id_list[5]]
        end=int(id_list[3])
        buff=[]
        buff.append(&quot;chr%s:%d:%d:%s_%s_%s&quot;%(id_list[1], start, end, id_list[5], id_list[4], id_list[0]))
        for x,y in zip(i_list[6:], inds):
            b=int(dic_dub[gene][y])
            t=int(x)
            buff.append(&quot;%d/%d&quot;%(t,b))
        fout.write(&quot; &quot;.join(buff)+ &#39;\n&#39;)
        
fout.close()</code></pre>
<p>makePhenoRefSeqPeaks_GeneLoc_Nuclear_noMP_rep.py</p>
<pre class="bash"><code>#PYTHON 3

dic_IND = {}
dic_BAM = {}

for ln in open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_file_id_mapping_nuclear_Transcript_head.txt&quot;):
    bam, IND = ln.split(&quot;\t&quot;)
    IND = IND.strip()
    dic_IND[bam] = IND
    if IND not in dic_BAM:
        dic_BAM[IND] = []
    dic_BAM[IND].append(bam)


#now I have ind dic with keys as the bam and ind as the values
#I also have a bam dic with ind as the keys and bam as the values  
    
inds=list(dic_BAM.keys()) #list of ind libraries  

#gene start and end dictionaries: 
dic_geneS = {}
dic_geneE = {}
for ln in open(&quot;/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/ncbiRefSeq_endAllGenes.sort.bed&quot;):
    chrom, start, end, geneID, score, strand = ln.split(&#39;\t&#39;)
    gene= geneID.split(&quot;:&quot;)[1]
#    if &quot;-&quot; in gene:
 #       gene=gene.split(&quot;-&quot;)[0]
    if gene not in dic_geneS:
        dic_geneS[gene]=int(start)
        dic_geneE[gene]=int(end)
        


#list of genes   

count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;, &quot;r&quot;)
genes=[]
for line , i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        gene=id_list[5]
        if gene not in genes:
            genes.append(gene)
            
#make the ind and gene dic  
dic_dub={}
for g in genes:
    dic_dub[g]={}
    for i in inds:
        dic_dub[g][i]=0


#populate the dictionary  
count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;, &quot;r&quot;)
for line, i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        g= id_list[5]
        values=list(i_list[6:])
        list_list=[]
        for ind,val in zip(inds, values):
            list_list.append([ind, val])
        for num, name in enumerate(list_list):
            dic_dub[g][list_list[num][0]] += int(list_list[num][1])
        

#write the file by acessing the dictionary and putting values in the table ver the value in the dic 
        

fout=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.fc&quot;,&quot;w&quot;)
peak=[&quot;chrom&quot;]
inds_noL=[]
for each in inds:
    indsNA= &quot;NA&quot; + each[:-2]
    inds_noL.append(indsNA) 
fout.write(&quot; &quot;.join(peak + inds_noL) + &#39;\n&#39; )


count_file=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;, &quot;r&quot;)
for line , i in enumerate(count_file):
    if line &gt; 1:
        i_list=i.split()
        id=i_list[0]
        id_list=id.split(&quot;:&quot;)
        gene=id_list[5]
        #start=dic_geneS[id_list[5]]
        start=int(id_list[2])
        #end=dic_geneE[id_list[5]]
        end=int(id_list[3])
        buff=[]
        buff.append(&quot;chr%s:%d:%d:%s_%s_%s&quot;%(id_list[1], start, end, id_list[5], id_list[4], id_list[0]))
        for x,y in zip(i_list[6:], inds):
            b=int(dic_dub[gene][y])
            t=int(x)
            buff.append(&quot;%d/%d&quot;%(t,b))
        fout.write(&quot; &quot;.join(buff)+ &#39;\n&#39;)
        
fout.close()</code></pre>
<p>run_makePhen_sep_GeneLocAnno_noMP_rep.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_makePhen_sep_GeneLocAnno_noMP_rep
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=run_makePhen_sep_GeneLocAnno_noMP_rep.out
#SBATCH --error=run_makePhen_sep_GeneLocAnno_noMP_rep.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

python makePhenoRefSeqPeaks_GeneLoc_Total_noMP_rep.py  

python makePhenoRefSeqPeaks_GeneLoc_Nuclear_noMP_rep.py  </code></pre>
<p>Next filter 5% peaks (5% from full sample analysis)</p>
<p>filterPheno_bothFraction_GeneLocAnno_5perc_rep.py</p>
<pre class="bash"><code>#python  

totalokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt&quot;

totalokPeaks5perc={}
for ln in open(totalokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    totalokPeaks5perc[peakname]=&quot;&quot;


nuclearokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt&quot;
nuclearokPeaks5perc={}
for ln in open(nuclearokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    nuclearokPeaks5perc[peakname]=&quot;&quot;
    
    
totalPhenoBefore=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.fc&quot;,&quot;r&quot;)
totalPhenoAfter=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc&quot;, &quot;w&quot;)
for num, ln in enumerate(totalPhenoBefore):
    if num ==0:
        totalPhenoAfter.write(ln)
    else:  
        id=ln.split()[0].split(&quot;:&quot;)[3].split(&quot;_&quot;)[2]
        if id in totalokPeaks5perc.keys():
            totalPhenoAfter.write(ln)
totalPhenoAfter.close()  

nuclearPhenoBefore=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.fc&quot;,&quot;r&quot;)
nuclearPhenoAfter=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc&quot;, &quot;w&quot;)
for num, ln in enumerate(nuclearPhenoBefore):
    if num ==0:
        nuclearPhenoAfter.write(ln)
    else:  
        id=ln.split()[0].split(&quot;:&quot;)[3].split(&quot;_&quot;)[2]
        if id in nuclearokPeaks5perc.keys():
            nuclearPhenoAfter.write(ln)
nuclearPhenoAfter.close() </code></pre>
<p>Filter the count file for good peaks:</p>
<p>filterCountsReplicates.py</p>
<pre class="bash"><code>nucCountIn=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc&quot;,&quot;r&quot;)
nucCountOut=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.5perc.fc&quot;,&quot;w&quot;)
totCountIn=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc&quot;,&quot;r&quot;)
totCountOut=open(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_filtPeakOppstrand_cov_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.5perc.fc&quot;,&quot;w&quot;)

totalokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt&quot;

totalokPeaks5perc={}
for ln in open(totalokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    totalokPeaks5perc[peakname]=&quot;&quot;

nuclearokPeaks5perc_file=&quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt&quot;
nuclearokPeaks5perc={}
for ln in open(nuclearokPeaks5perc_file,&quot;r&quot;):
    peakname=ln.split()[5]
    nuclearokPeaks5perc[peakname]=&quot;&quot;
    
for num, ln in enumerate(totCountIn):
    if num ==0:
        continue
    elif num==1:
        totCountOut.write(ln)
    else:  
        id=ln.split()[0].split(&quot;:&quot;)[0]
        if id in totalokPeaks5perc.keys():
            totCountOut.write(ln)
totCountOut.close()     

for num, ln in enumerate(nucCountIn):
    if num ==0:
        continue
    elif num==1:
        nucCountOut.write(ln)
    else:  
        id=ln.split()[0].split(&quot;:&quot;)[0]
        if id in totalokPeaks5perc.keys():
            nucCountOut.write(ln)
nucCountOut.close()     
</code></pre>
<p>Convert to usage:</p>
<p>This takes just counts:</p>
<pre class="bash"><code>less /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc | cut -f1 -d&quot; &quot; --complement | sed &#39;1d&#39; &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc_counts

less /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc | cut -f1 -d&quot; &quot; --complement | sed &#39;1d&#39; &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc_counts</code></pre>
<p>convertReplicates2Usage.py</p>
<pre class="bash"><code>def convert(infile, outfile):
  final=open(outfile, &quot;w&quot;)
  for ln in open(infile, &quot;r&quot;):
    line_list=ln.split()
    new_list=[]
    for i in line_list:
      num, dem = i.split(&quot;/&quot;)
      if dem == &quot;0&quot;:
        perc = &quot;0.00&quot;
      else:
        perc = int(num)/int(dem)
        perc=round(perc,2)
        perc= str(perc)
      new_list.append(perc)
    final.write(&quot;\t&quot;.join(new_list)+ &#39;\n&#39;)
  final.close()
  
convert(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc_counts&quot;,&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc_numeric.fc&quot;)

  
convert(&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc_counts&quot;,&quot;/project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc_numeric.fc&quot;)
</code></pre>
<p>Add back names:</p>
<pre class="bash"><code>less /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc | cut -f1 -d&quot; &quot; | sed &#39;1d&#39; &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/PeakIDs_Nuc.txt

less /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc | cut -f1 -d&quot; &quot; | sed &#39;1d&#39; &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/PeakIDs_Tot.txt
</code></pre>
<p>Merge back to numeri:</p>
<pre class="bash"><code>paste /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/PeakIDs_Nuc.txt /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc_numeric.fc &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc_numeric.named.fc


paste /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/PeakIDs_Tot.txt /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc_numeric.fc &gt; /project2/gilad/briana/threeprimeseq/data/Replicates_phenotypes_filtPeakTranscript_noMP_GeneLocAnno/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc_numeric.named.fc</code></pre>
<div id="compare-usage-accross-individual" class="section level2">
<h2>Compare usage accross individual</h2>
<p>This analysis will mirror the analysis I did in <a href="IndPeakUsageDiff.html">this analysis</a></p>
<pre class="r"><code>usageTot=read.table(&quot;../data/Replicates_usage/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc_numeric.named.fc&quot;, header=F, stringsAsFactors = F, col.names = c(&quot;chrom&quot;, &quot;NA18499&quot;, &quot;NA18912&quot;, &quot;NA19093&quot;, &quot;NA19140&quot;))

usageTot_2peak= usageTot  %&gt;% separate(chrom, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;geneInf&quot;), sep =&quot;:&quot;) %&gt;% separate(geneInf, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;), sep=&quot;_&quot;) %&gt;% group_by(gene) %&gt;% mutate(nPeak=n()) %&gt;% filter(nPeak==2) %&gt;% select(-chr, -start, -end, -strand, -peak, -nPeak) %&gt;% ungroup()</code></pre>
<pre><code>Warning: Expected 3 pieces. Additional pieces discarded in 3 rows [4917,
4918, 4919].</code></pre>
<pre class="r"><code>TotalCounts_allInd=read.table(&quot;../data/Replicates_usage/replicates_filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.5perc.fc&quot;, header=T, stringsAsFactors = F) %&gt;% separate(Geneid, into =c(&#39;peak&#39;, &#39;chr&#39;, &#39;start&#39;, &#39;end&#39;, &#39;strand&#39;, &#39;gene&#39;), sep = &quot;:&quot;) %&gt;% select(-peak, -chr, -start, -end, -strand, -Chr, -Start, -End, -Strand, -Length) %&gt;% group_by(gene) %&gt;% mutate(PeakCount=n()) %&gt;% filter(PeakCount==2) %&gt;% select(-PeakCount) %&gt;% ungroup()
colnames(TotalCounts_allInd)=colnames(usageTot_2peak)</code></pre>
<p>Function for difference based on counts:</p>
<pre class="r"><code>perIndDiffUsage=function(ind, counts=TotalCounts_allInd, usage=usageTot_2peak){
  ind=enquo(ind)
  #compute usage stats
  #seperate usage
  usage_ind=usage %&gt;% select(gene, !!ind) 
  usage_other = usage %&gt;% select(-gene,-!!ind) %&gt;% rowMeans()
  usage_indVal=as.data.frame(cbind(usage_ind,usage_other))
  usage_indVal$val=abs(usage_indVal[,2] - usage_indVal[,3])
  usage_indVal2= usage_indVal%&gt;% group_by(gene) %&gt;% select(gene, val) %&gt;% distinct(gene, .keep_all=T) 
  #seperate genes by percentile for this ind
  count_ind= counts %&gt;% select(gene, !!ind)
  colnames(count_ind)=c(&quot;gene&quot;, &quot;count&quot;)
  count_ind = count_ind %&gt;%  group_by(gene) %&gt;% summarize(Exp=sum(count)) %&gt;% filter(Exp &gt;0)  %&gt;% mutate(Percentile = percent_rank(Exp)) 
  count_ind_perc10= count_ind %&gt;% filter(Percentile&lt;.1)
  count_ind_perc20= count_ind %&gt;% filter(Percentile&lt;.2, Percentile&gt;.1)
  count_ind_perc30= count_ind %&gt;% filter(Percentile&lt;.3, Percentile&gt;.2)
  count_ind_perc40= count_ind %&gt;% filter(Percentile&lt;.4, Percentile&gt;.3)
  count_ind_perc50= count_ind %&gt;% filter(Percentile&lt;.5, Percentile&gt;.4)
  count_ind_perc60= count_ind %&gt;% filter(Percentile&lt;.6, Percentile&gt;.5)
  count_ind_perc70= count_ind %&gt;% filter(Percentile&lt;.7, Percentile&gt;.6)
  count_ind_perc80= count_ind %&gt;% filter(Percentile&lt;.8, Percentile&gt;.7)
  count_ind_perc90= count_ind %&gt;% filter(Percentile&lt;.9, Percentile&gt;.8)
  count_ind_perc100= count_ind %&gt;% filter(Percentile&lt;1, Percentile&gt;.9)
  #subset and sum usage 
  out10_df= usage_indVal2 %&gt;% inner_join(count_ind_perc10, by=&quot;gene&quot;)
  out20_df= usage_indVal2 %&gt;% inner_join(count_ind_perc20, by=&quot;gene&quot;)
  out30_df= usage_indVal2 %&gt;% inner_join(count_ind_perc30, by=&quot;gene&quot;)
  out40_df= usage_indVal2 %&gt;% inner_join(count_ind_perc40, by=&quot;gene&quot;)
  out50_df= usage_indVal2 %&gt;% inner_join(count_ind_perc50, by=&quot;gene&quot;)
  out60_df= usage_indVal2 %&gt;% inner_join(count_ind_perc60, by=&quot;gene&quot;)
  out70_df= usage_indVal2 %&gt;% inner_join(count_ind_perc70, by=&quot;gene&quot;)
  out80_df= usage_indVal2 %&gt;% inner_join(count_ind_perc80, by=&quot;gene&quot;)
  out90_df= usage_indVal2 %&gt;% inner_join(count_ind_perc90, by=&quot;gene&quot;)
  out100_df= usage_indVal2 %&gt;% inner_join(count_ind_perc100, by=&quot;gene&quot;)
  #output list of 10 values
  out= c((sum(out10_df$val)/nrow(out10_df)), (sum(out20_df$val)/nrow(out20_df)), (sum(out30_df$val)/nrow(out30_df)), (sum(out40_df$val)/nrow(out40_df)), (sum(out50_df$val)/nrow(out50_df)), (sum(out60_df$val)/nrow(out60_df)), (sum(out70_df$val)/nrow(out70_df)), (sum(out80_df$val)/nrow(out80_df)), (sum(out90_df$val)/nrow(out90_df)), (sum(out100_df$val)/nrow(out100_df)))
  return(out)
}</code></pre>
<p>Run this:</p>
<pre class="r"><code>Inds=colnames(TotalCounts_allInd)[2:ncol(TotalCounts_allInd)]
Percentile=c(10,20,30,40,50,60,70,80,90,100)
for (i in Inds){
  x= perIndDiffUsage(i)
  Percentile=cbind(Percentile, x)
}
colnames(Percentile)=c(&quot;Percentile&quot;, Inds)</code></pre>
<pre class="r"><code>#order the 
Percentile_df=as.data.frame(Percentile) 
Percentile_melt=melt(Percentile_df, id.vars=c(&quot;Percentile&quot;))
colnames(Percentile_melt)=c(&quot;Percentile&quot;, &quot;Individual&quot;, &quot;AvgUsageDiff&quot;)</code></pre>
<pre class="r"><code>ggplot(Percentile_melt, aes(x=Percentile, y=Individual, fill=AvgUsageDiff)) + geom_tile() + labs(title=&quot;Average peak usage difference for individaul vs. others&quot;) + scale_fill_gradientn(colours = c(&quot;white&quot;, &quot;red&quot;, &quot;black&quot;))</code></pre>
<p><img src="figure/mergeReplicates.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] bindrcpp_0.2.2  reshape2_1.4.3  forcats_0.3.0   stringr_1.4.0  
 [5] dplyr_0.7.6     purrr_0.2.5     readr_1.1.1     tidyr_0.8.1    
 [9] tibble_1.4.2    ggplot2_3.0.0   tidyverse_1.2.1 workflowr_1.2.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.19     cellranger_1.1.0 plyr_1.8.4       compiler_3.5.1  
 [5] pillar_1.3.0     git2r_0.24.0     bindr_0.1.1      tools_3.5.1     
 [9] digest_0.6.17    lubridate_1.7.4  jsonlite_1.6     evaluate_0.13   
[13] nlme_3.1-137     gtable_0.2.0     lattice_0.20-35  pkgconfig_2.0.2 
[17] rlang_0.2.2      cli_1.0.1        rstudioapi_0.9.0 yaml_2.2.0      
[21] haven_1.1.2      withr_2.1.2      xml2_1.2.0       httr_1.3.1      
[25] knitr_1.20       hms_0.4.2        fs_1.2.6         rprojroot_1.3-2 
[29] grid_3.5.1       tidyselect_0.2.4 glue_1.3.0       R6_2.3.0        
[33] readxl_1.1.0     rmarkdown_1.11   modelr_0.1.2     magrittr_1.5    
[37] whisker_0.3-2    backports_1.1.2  scales_1.0.0     htmltools_0.3.6 
[41] rvest_0.3.2      assertthat_0.2.0 colorspace_1.3-2 labeling_0.3    
[45] stringi_1.2.4    lazyeval_0.2.1   munsell_0.5.0    broom_0.5.0     
[49] crayon_1.3.4    </code></pre>
</div>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
