<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Mispriming Approach</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Mispriming Approach</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>1/11/2019</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-01-14</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/6bc9243ded950fc7e317fd6ef901657801c66b46" target="_blank">6bc9243</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/PreAshExplore.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeaksUsed/
    Untracked:  data/RNAkalisto/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_trans/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/InvestigatePeak2GeneAssignment.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   code/Snakefile

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/6bc9243ded950fc7e317fd6ef901657801c66b46/analysis/mispriming_approach.Rmd" target="_blank">6bc9243</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-14
</td>
<td style="text-align:left;">
evaluate clean reads, make new file for misprime filter
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/49ad9e16ad5dd53e98e627a47d2a98bac08852c4/docs/mispriming_approach.html" target="_blank">49ad9e1</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-12
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/7a08009e4051cbe4c76d285282747de6158f7275/analysis/mispriming_approach.Rmd" target="_blank">7a08009</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-12
</td>
<td style="text-align:left;">
analyze 1 line
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/4b31426a6ca15af3dd0967caedae26d97847853e/docs/mispriming_approach.html" target="_blank">4b31426</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-11
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/ec05274615efa5e97d1cb448731ad09d9b229750/analysis/mispriming_approach.Rmd" target="_blank">ec05274</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-11
</td>
<td style="text-align:left;">
approach to extract bases
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/580e24472c52b98efb062d7a15c0b436b2079a78/docs/mispriming_approach.html" target="_blank">580e244</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-11
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/42fcbdd1ff21ebf0e08b3df26501f37430f2290c/analysis/mispriming_approach.Rmd" target="_blank">42fcbdd</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-11
</td>
<td style="text-align:left;">
initialize mispriming approach file
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<p>In this analysis I am gonig to explore the ways to handle mispriming in the 3’ seq data. Some people call this internal priming. This is when the polyDt primer attached to an RNA molecule that has a long stretch of A’s rather than to the tail. You can identify when this is happening because polyA tails are not in the genome but mispriming As are. In my data I need to look for Ts upstream of the read. This is because our reads are on the opposite strand.</p>
<p>Sheppard et al. cited 2 other papers, Beaudoing et al 2000 and Tian et al 2005. Thet excluded reads with 6 consequitive upstream As or those with 7 in a 10nt window. They did this at the read level.</p>
<p>I started thinking about this in <a href="https://brimittleman.github.io/threeprimeseq/filter_As.html" class="uri">https://brimittleman.github.io/threeprimeseq/filter_As.html</a>. I did not have it mapped out correctly because I was looking for A’s on one strand and T’s on the other.</p>
<p>I will assess the problem then will create a blacklist to get rid of the reads. I should do this in the snakefile before we create BW for the peak calling.</p>
<p>I can start by updating 6up_bed.sh. To make a new script that grabs the upstream 10 bases. I will look for7 of 10 T’s in this region. I am going to do this in python because it is more straight forward to read then an awk script. I can also wrap it easier this way. I can also account for negative values and values larger than the chromosome this way.</p>
<div id="retreive-10-upstream-bases-for-each-read" class="section level2">
<h2>Retreive 10 upstream bases for each read</h2>
<p>Upstream10Bases.py</p>
<pre class="bash"><code>#python  
def main(Fin, Fout):
  outBed=open(Fout, &quot;w&quot;)
  chrom_lengths=open(&quot;/project2/gilad/briana/genome_anotation_data/chrom_lengths2.sort.bed&quot;,&quot;r&quot;)
  #make a dictionary with chrom lengths
  length_dic={}
  for i in chrom_lengths:
    chrom, start, end = i.split()
    length_dic[str(chrom)]=int(end)  

#write file 
  for ln in open(Fin):
    chrom, start, end, name, score, strand = ln.split()
    chrom=str(chrom)
    if strand==&quot;+&quot;:
      start_new=int(start)-10
      if start_new &lt;= 1:
        start_new = 0 
      end_new= int(start)
      if end_new == 0:
        end_new=1
      outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new, end_new, name, score, strand))
    if strand == &quot;-&quot;:
      start_new=int(end)
      end_new=int(end) + 10
      if end_new &gt;= length_dic[chrom]:
        end_new = length_dic[chrom]
        start_new=end_new-1
      outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new, end_new, name, score, strand))
  outBed.close()  

if __name__ == &quot;__main__&quot;:
    import sys
    inFile = sys.argv[1]
    fileNoPath=inFile.split(&quot;/&quot;)[-1]
    fileshort=fileNoPath[:-4]
    outFile=&quot;/project2/gilad/briana/threeprimeseq/data/bed_10up/&quot; + fileshort + &quot;10up.bed&quot;
    main(inFile, outFile)
</code></pre>
<p>I can wrap this for all of the files.<br />
wrap_Upstream10Bases.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=w_Upstream10Bases
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=w_Upstream10Bases.out
#SBATCH --error=w_Upstream10Bases.err
#SBATCH --partition=broadwl
#SBATCH --mem=8G
#SBATCH --mail-type=END


module load Anaconda3  

source activate three-prime-env


for i in $(ls /project2/gilad/briana/threeprimeseq/data/bed_sort/*-combined-sort.bed); do
            python  Upstream10Bases.py  $i 
        done
</code></pre>
<p>I need to sort the files:</p>
<p>Next step is running the nuc function to get the sequences of the positions I just put in the bed files.</p>
<p>bedtools nuc</p>
<ul>
<li><p>-fi (fasta file) /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa</p></li>
<li><p>-bed results from 10up stream</p></li>
<li><p>-s strand specific</p></li>
<li><p>-seq print exracted sequence</p></li>
<li><blockquote>
<p>output</p>
</blockquote></li>
</ul>
<p>Nuc10BasesUp.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=Nuc10BasesUp
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=Nuc10BasesUp.out
#SBATCH --error=Nuc10BasesUp.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

for i in $(ls /project2/gilad/briana/threeprimeseq/data/bed_10up/*);do
   describer=$(echo ${i} | sed -e &#39;s/.*YL-SP-//&#39; | sed -e &quot;s/-combined-sort10up.bed$//&quot;)
   bedtools nuc -s -seq -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed $i &gt; /project2/gilad/briana/threeprimeseq/data/nuc_10up/TenBaseUP.${describer}.txt   
done</code></pre>
</div>
<div id="evaluate-problem-in-1-line" class="section level2">
<h2>Evaluate problem in 1 line</h2>
<pre class="r"><code>library(data.table)
require(ggseqlogo)</code></pre>
<pre><code>Loading required package: ggseqlogo</code></pre>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.1.1
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ───────────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.0.0     ✔ purrr   0.2.5
✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.3.1
✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>── Conflicts ──────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::between()   masks data.table::between()
✖ dplyr::filter()    masks stats::filter()
✖ dplyr::first()     masks data.table::first()
✖ dplyr::lag()       masks stats::lag()
✖ dplyr::last()      masks data.table::last()
✖ purrr::transpose() masks data.table::transpose()</code></pre>
<div id="nuclear" class="section level3">
<h3>Nuclear</h3>
<p>Goals for this section:</p>
<ul>
<li>make a seq logo plot</li>
</ul>
<p>I made logo plot in <a href="https://brimittleman.github.io/Net-seq/explore_umi_usage.html" class="uri">https://brimittleman.github.io/Net-seq/explore_umi_usage.html</a> with ggseq logo.</p>
<pre class="r"><code>res_colNames=c(&quot;chrom&quot;,&quot;start&quot;, &quot;end&quot;, &quot;name&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;pctAT&quot;, &quot;pctGC&quot;, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, &quot;T&quot;, &quot;N&quot;, &quot;Other&quot;, &quot;Length&quot;, &quot;Seq&quot;)</code></pre>
<pre class="r"><code>nuc_18486_N= fread(&quot;../data/nuc_10up/TenBaseUP.18486-N.txt&quot;, col.names = res_colNames)</code></pre>
<p>Extract seq for seq logo plot:</p>
<pre class="r"><code>#filter for full 10 bp  - removes 422 reads (too close to ends)
nuc_18486_N=nuc_18486_N %&gt;% filter(Length==10)
seqs_18486N= nuc_18486_N$Seq</code></pre>
<p>Scheme for logo plot:</p>
<pre class="r"><code>cs1 = make_col_scheme(chars=c(&#39;A&#39;, &#39;T&#39;, &#39;C&#39;, &#39;G&#39;, &#39;N&#39;), groups=c(&#39;A&#39;, &#39;T&#39;, &#39;C&#39;, &#39;G&#39;, &#39;N&#39;), cols=c(&#39;red&#39;, &#39;blue&#39;, &#39;green&#39;, &#39;yellow&#39;, &#39;pink&#39;))</code></pre>
<p>Create plot:</p>
<pre class="r"><code>ggseqlogo(seqs_18486N, col_scheme=cs1,  method = &#39;prob&#39;)</code></pre>
<p><img src="figure/mispriming_approach.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-9-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/49ad9e16ad5dd53e98e627a47d2a98bac08852c4/docs/figure/mispriming_approach.Rmd/unnamed-chunk-9-1.png" target="_blank">49ad9e1</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2019-01-12
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>This is not overwhelming:</p>
<ul>
<li>count number of non passing reads with 6 T’s in a row</li>
</ul>
<pre class="r"><code>SixT=&quot;TTTTTT&quot;
nuc_18486_N_6Ts=nuc_18486_N %&gt;% filter(grepl(SixT, Seq)) 

perc_Bad6T= nrow(nuc_18486_N_6Ts)/nrow(nuc_18486_N)
perc_Bad6T</code></pre>
<pre><code>[1] 0.01797875</code></pre>
<ul>
<li>count number of non passing with 7 of 10 T’s</li>
</ul>
<pre class="r"><code>nuc_18486_N_70perc= nuc_18486_N%&gt;% mutate(percT=T/Length) %&gt;% filter(percT&gt;=.7) 

perc_Bad70= nrow(nuc_18486_N_70perc)/nrow(nuc_18486_N)

perc_Bad70</code></pre>
<pre><code>[1] 0.460071</code></pre>
<ul>
<li>count number of total non passing reads</li>
</ul>
<p>For this I need to use an or statement.</p>
<pre class="r"><code>nuc_18486_N_bad=  nuc_18486_N%&gt;% mutate(percT=T/Length) %&gt;% filter(percT&gt;=.7 | grepl(SixT, Seq) )

perc_Bad=nrow(nuc_18486_N_bad)/nrow(nuc_18486_N)

perc_Bad</code></pre>
<pre><code>[1] 0.4622981</code></pre>
<p>This shows us that 46% of reads pass these filters.</p>
<p>Make a logo plot for clean reads.</p>
<pre class="r"><code>nuc_18486_N_good=nuc_18486_N%&gt;% mutate(percT=T/Length) %&gt;% filter(percT&lt;.7,  !grepl(SixT, Seq) )


ggseqlogo(nuc_18486_N_good$Seq, col_scheme=cs1,  method = &#39;prob&#39;)</code></pre>
<p><img src="figure/mispriming_approach.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="total" class="section level3">
<h3>Total</h3>
<pre class="r"><code>nuc_18486_T= fread(&quot;../data/nuc_10up/TenBaseUP.18486-T.txt&quot;, col.names = res_colNames)</code></pre>
<ul>
<li>Seqlogo plot</li>
</ul>
<p>Filter less than 10 base pair in length for seqlogo</p>
<pre class="r"><code>nuc_18486_T=nuc_18486_T %&gt;% filter(Length==10)
seqs_18486T= nuc_18486_T$Seq</code></pre>
<p>Create plot:</p>
<pre class="r"><code>ggseqlogo(seqs_18486T, col_scheme=cs1,  method = &#39;prob&#39;)</code></pre>
<p><img src="figure/mispriming_approach.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>count number of non passing reads with 6 T’s in a row</li>
</ul>
<pre class="r"><code>nuc_18486_T_6Ts=nuc_18486_T %&gt;% filter(grepl(SixT, Seq)) 

perc_Bad6T_tot= nrow(nuc_18486_T_6Ts)/nrow(nuc_18486_T)
perc_Bad6T_tot</code></pre>
<pre><code>[1] 0.01999222</code></pre>
<ul>
<li>count number of non passing with 7 of 10 T’s</li>
</ul>
<pre class="r"><code>nuc_18486_T_70perc= nuc_18486_T%&gt;% mutate(percT=T/Length) %&gt;% filter(percT&gt;=.7) 

perc_Bad70_tot= nrow(nuc_18486_T_70perc)/nrow(nuc_18486_T)

perc_Bad70_tot</code></pre>
<pre><code>[1] 0.2460797</code></pre>
<ul>
<li>count number of total non passing reads</li>
</ul>
<p>For this I need to use an or statement.</p>
<pre class="r"><code>nuc_18486_T_bad=  nuc_18486_T%&gt;% mutate(percT=T/Length) %&gt;% filter(percT&gt;=.7, grepl(SixT, Seq) )

perc_Bad_tot=nrow(nuc_18486_T_bad)/nrow(nuc_18486_T)

perc_Bad_tot</code></pre>
<pre><code>[1] 0.01466245</code></pre>
<p>This shows us that 25% of reads pass these filters</p>
<p>Make a logo plot for clean reads.</p>
<pre class="r"><code>nuc_18486_T_good=nuc_18486_T%&gt;% mutate(percT=T/Length) %&gt;% filter(percT&lt;.7 | !grepl(SixT, Seq) )


ggseqlogo(nuc_18486_T_good$Seq, col_scheme=cs1,  method = &#39;prob&#39;)</code></pre>
<p><img src="figure/mispriming_approach.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>These dont look super different.</p>
</div>
</div>
<div id="for-all" class="section level2">
<h2>For all</h2>
<p>I may have to use python when i look at all beacuse this is not fast.</p>
<p>I will look at each read in a file and check if for 70% Ts or 6Ts in a row.</p>
<p>filterMissprimingInNuc10.py</p>
<pre class="bash"><code>#python  
def main(Fin, Fout):
  outBed=open(Fout, &quot;w&quot;)
  inBed=open(Fin, &quot;r&quot;)
  for ind, ln in enumerate(inBed):
    if ind &gt;=1:
      chrom,start, end, name, score, strand, pctAT, pctGC, A, C, G, T, N, Other, Length, Sequence = ln.split()
      Tperc= float(T) / float(Length)
      if Tperc &lt; .7:
          if &quot;TTTTTT&quot; not in Sequence:
              start_new=int(start)
              end_new=int(end)
              outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new , end_new , name, score, strand))
  outBed.close()

if __name__ == &quot;__main__&quot;:
    import sys
    inFile = sys.argv[1]
    fileNoPath=inFile.split(&quot;/&quot;)[-1]
    sampleName=fileNoPath.split(&quot;.&quot;)[1]
    outFile=&quot;/project2/gilad/briana/threeprimeseq/data/nuc_10up_CleanReads/TenBaseUP.&quot; + sampleName + &quot;.CleanReads.bed&quot;
    main(inFile, outFile)</code></pre>
<p>run_filterMissprimingInNuc10.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=Nrun_filterMissprimingInNuc10
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=run_filterMissprimingInNuc10.out
#SBATCH --error=run_filterMissprimingInNuc10.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

for i in $(ls /project2/gilad/briana/threeprimeseq/data/nuc_10up/*);do
    python filterMissprimingInNuc10.py $i  
done </code></pre>
<p>I will look at these stats then move to getting rid ofthe peaks from these reads.</p>
<pre class="r"><code>CleanStats=read.csv(&quot;../data/nuc_10up/CleanCount_stats.csv&quot;, header = T) %&gt;% separate(Sample_ID, into=c(&quot;Sample&quot;, &quot;Fraction&quot;), by=&quot;_&quot;) %&gt;% mutate(Perc_PostFilter=PostMPFilter/mappedReads)


cleanStatPlot=ggplot(CleanStats, aes(x=Sample, by=Fraction, fill=Fraction, y=Perc_PostFilter)) + geom_bar(stat=&quot;identity&quot;, position = &quot;Dodge&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1),axis.text.y = element_text(size=12),axis.title.y=element_text(size=10,face=&quot;bold&quot;), axis.title.x=element_text(size=12,face=&quot;bold&quot;))+ scale_fill_manual(values=c(&quot;deepskyblue3&quot;,&quot;darkviolet&quot;)) + labs(y=&quot;Percent Reads Passing Misprime Filter&quot;, title=&quot;Accounting for mispriming in 3&#39; Seq Data&quot;)

ggsave(filename = &quot;../output/plots/CleanStatsPlot.png&quot;, plot = cleanStatPlot)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<p>Plot number of clean reads per ind:</p>
<pre class="r"><code>ggplot(CleanStats, aes(x=Sample, by=Fraction, fill=Fraction, y=PostMPFilter)) + geom_bar(stat=&quot;identity&quot;, position = &quot;Dodge&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1),axis.text.y = element_text(size=12),axis.title.y=element_text(size=10,face=&quot;bold&quot;), axis.title.x=element_text(size=12,face=&quot;bold&quot;))+scale_fill_manual(values=c(&quot;deepskyblue3&quot;,&quot;darkviolet&quot;)) + labs(y=&quot;Reads Passing Misprime Filter&quot;, title=&quot;Accounting for mispriming in 3&#39; Seq Data&quot;) + scale_y_log10()</code></pre>
<p><img src="figure/mispriming_approach.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CleanStatsMelt= melt(CleanStats, id.vars=c(&quot;Sample&quot;, &quot;Fraction&quot;)) %&gt;% filter(variable==&quot;PostMPFilter&quot;) %&gt;%  group_by(Fraction) %&gt;% summarise(mean=mean(value), sd=sd(value))

ggplot(CleanStatsMelt,aes(x=Fraction, y=mean, fill=Fraction)) + geom_col()+ geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.2)+ theme(axis.text.y = element_text(size=12),axis.title.y=element_text(size=10,face=&quot;bold&quot;), axis.title.x=element_text(size=12,face=&quot;bold&quot;))+ scale_fill_manual(values=c(&quot;deepskyblue3&quot;,&quot;darkviolet&quot;))+ labs(title=&quot;Clean Reads by Fraction&quot;, y=&quot;Clean Reads&quot;)</code></pre>
<p><img src="figure/mispriming_approach.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="extra-not-using" class="section level2">
<h2>Extra not using</h2>
<p>sort_10upbedFile.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=sort_10upbedFile
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=sort_10upbedFile.out
#SBATCH --error=sort_10upbedFile.err
#SBATCH --partition=broadwl
#SBATCH --mem=8G
#SBATCH --mail-type=END

for i in $( ls /project2/gilad/briana/threeprimeseq/data/bed_10up/*);do
  describer=$(echo ${i} | sed -e &#39;s/.*YL-SP-//&#39; | sed -e &quot;s/-combined-sort10up.bed$//&quot;)
  sort -k 1,1 -k2,2n $i &gt; /project2/gilad/briana/threeprimeseq/data/bed_10up_sort/YL-SP-${describer}-combined-sort10up.sort.bed
  done</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] bindrcpp_0.2.2    forcats_0.3.0     stringr_1.3.1    
 [4] dplyr_0.7.6       purrr_0.2.5       readr_1.1.1      
 [7] tidyr_0.8.1       tibble_1.4.2      ggplot2_3.0.0    
[10] tidyverse_1.2.1   workflowr_1.1.1   ggseqlogo_0.1    
[13] data.table_1.11.8

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4  reshape2_1.4.3    haven_1.1.2      
 [4] lattice_0.20-35   colorspace_1.3-2  htmltools_0.3.6  
 [7] yaml_2.2.0        rlang_0.2.2       R.oo_1.22.0      
[10] pillar_1.3.0      withr_2.1.2       glue_1.3.0       
[13] R.utils_2.7.0     modelr_0.1.2      readxl_1.1.0     
[16] bindr_0.1.1       plyr_1.8.4        munsell_0.5.0    
[19] gtable_0.2.0      cellranger_1.1.0  rvest_0.3.2      
[22] R.methodsS3_1.7.1 evaluate_0.11     labeling_0.3     
[25] knitr_1.20        broom_0.5.0       Rcpp_0.12.19     
[28] scales_1.0.0      backports_1.1.2   jsonlite_1.5     
[31] hms_0.4.2         digest_0.6.17     stringi_1.2.4    
[34] grid_3.5.1        rprojroot_1.3-2   cli_1.0.1        
[37] tools_3.5.1       magrittr_1.5      lazyeval_0.2.1   
[40] crayon_1.3.4      whisker_0.3-2     pkgconfig_2.0.2  
[43] xml2_1.2.0        lubridate_1.7.4   rstudioapi_0.8   
[46] assertthat_0.2.0  rmarkdown_1.10    httr_1.3.1       
[49] R6_2.3.0          nlme_3.1-137      git2r_0.23.0     
[52] compiler_3.5.1   </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
