<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Nuclear vs Total nominal associations</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Nuclear vs Total nominal associations</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>3/9/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-03-12
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>threeprimeseq/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree586c5919db97a5e7e70d4a486e01d1dc2e89ba0ftargetblank586c591a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/586c5919db97a5e7e70d4a486e01d1dc2e89ba0f" target="_blank">586c591</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree586c5919db97a5e7e70d4a486e01d1dc2e89ba0ftargetblank586c591a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/perm_QTL_trans_noMP_5percov/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/4suDataIGV.Rmd
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/EmpDistforOverlaps.Rmd
    Untracked:  analysis/EvaleQTLs.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/groSeqAnalysis.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  analysis/verifybam_dubs.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/AllPeak_counts/
    Untracked:  data/ApaQTLs/
    Untracked:  data/ApaQTLs_otherPhen/
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/DistTXN2Peak_genelocAnno/
    Untracked:  data/FeatureoverlapPeaks/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/GWAS_overlap/
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/LocusZoom_Unexp/
    Untracked:  data/LocusZoom_proc/
    Untracked:  data/MatchedSnps/
    Untracked:  data/NucSpecQTL/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeakCounts_noMP_5perc/
    Untracked:  data/PeakCounts_noMP_genelocanno/
    Untracked:  data/PeakUsage/
    Untracked:  data/PeakUsage_noMP/
    Untracked:  data/PeakUsage_noMP_GeneLocAnno/
    Untracked:  data/PeaksUsed/
    Untracked:  data/PeaksUsed_noMP_5percCov/
    Untracked:  data/PolyA_DB/
    Untracked:  data/QTL_overlap/
    Untracked:  data/RNAkalisto/
    Untracked:  data/RefSeq_annotations/
    Untracked:  data/Replicates_usage/
    Untracked:  data/Signal_Loc/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/WASP_STAT/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/apaExamp_proc/
    Untracked:  data/apaQTL_examp_noMP/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_GeneLocAnno/
    Untracked:  data/diff_iso_proc/
    Untracked:  data/diff_iso_trans/
    Untracked:  data/eQTLs_Lietal/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/molPheno_noMP/
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/pacbio_cov/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/peaks4DT/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov_3UTR/
    Untracked:  data/perm_QTL_diffWindow/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/protAndAPAAndExplmRes.Rda
    Untracked:  data/protAndAPAlmRes.Rda
    Untracked:  data/protAndExpressionlmRes.Rda
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  data/threePrimeSeqMetaData55Ind.txt
    Untracked:  data/threePrimeSeqMetaData55Ind.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  output/LZ/
    Untracked:  output/deeptools_plots/
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/NewPeakPostMP.Rmd
    Modified:   analysis/PeakToXper.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/characterize_apaQTLs.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/fixBWChromNames.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/initialPacBioQuant.Rmd
    Modified:   analysis/mispriming_approach.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlapMolQTL.opposite.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/peakQCPPlots.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/pipeline_55Ind.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/test.smash.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   analysis/unexplainedeQTL_analysis.Rmd
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/0537a728cd63a6a825f3615ca5e161a3de7a2be2/docs/NuclearSpecQTL.html" target="_blank">0537a72</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/88e8b7d96bb75122ca7b8d9fffbda245da0dd517/analysis/NuclearSpecQTL.Rmd" target="_blank">88e8b7d</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-11
</td>
<td>
try with permuted values
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Look for the snps that are QTLs in the nuclear fraction but not the total fraction. To do this I want to plot the association in the total fraction vs. the association in the nuclear fraction. I can look for snps that fall off the 1:1 line towards the nuclear fraction.</p>
<p>I need to look for the snps tested in both fractions. The nominal results are in:</p>
<ul>
<li>/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt<br />
</li>
<li>/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt</li>
</ul>
<p>This will not include the 18982867 more assocaitions in nucelar. I can only look at those that are tested in both analyses.</p>
<p>Format of file: * peakID * snp * dist * pval * slope (effect size)</p>
<p>I can make a dictionary with gene:snp:peak as the keys and a dictionary for the values- the inner dictionary will have the fraction as the key and the pvalue as the value</p>
<p>totalandnuclear_commonassociation.py</p>
<pre class="bash"><code>totRes=open(&quot;/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt&quot;,&quot;r&quot;)
nucRes=open(&quot;/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt&quot;, &quot;r&quot;)  


outfile=open(&quot;/project2/gilad/briana/threeprimeseq/data/NucSpecQTL/TotNucRes_overlapassociations.txt&quot;, &quot;w&quot;)

resDict={}
for ln in totRes:
    gene=ln.split()[0].split(&quot;:&quot;)[-1].split(&quot;_&quot;)[0]
    peak=ln.split()[0].split(&quot;:&quot;)[-1].split(&quot;_&quot;)[-1]
    snp=ln.split()[1]
    id=gene + &quot;:&quot; + peak + &quot;:&quot; + snp
    pval=ln.split()[3]
    resDict[id]={}
    resDict[id][&quot;Total&quot;]=pval

for ln in nucRes:
    gene=ln.split()[0].split(&quot;:&quot;)[-1].split(&quot;_&quot;)[0]
    peak=ln.split()[0].split(&quot;:&quot;)[-1].split(&quot;_&quot;)[-1]
    snp=ln.split()[1]
    id=gene + &quot;:&quot; + peak + &quot;:&quot; + snp
    pval=ln.split()[3]
    if id in resDict.keys():
        resDict[id][&quot;Nuclear&quot;]=pval
    else:
        continue   
        
#now i have a double dictionary. i need to write it out. i want id, the total pval, then the nuc pval  
for outer, inner in resDict.items():
    id=outer
    outPval=[]
    for key in inner:
        outPval.append(inner[key])
    print(outPval)
    if(len(outPval))==2:
         outfile.write(&quot;%s\t%s\t%s\n&quot;%(id, outPval[0], outPval[1]))


outfile.close()        
        
</code></pre>
<p>run it: run_totalandnuclear_commonassociation.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_totalandnuclear_commonassociation
#SBATCH --account=pi-yangili1
#SBATCH --time=5:00:00
#SBATCH --output=run_totalandnuclear_commonassociation.out
#SBATCH --error=run_totalandnuclear_commonassociation.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env


python totalandnuclear_commonassociation.py
</code></pre>
<p>Tot assoc: 105456802 nuclear assoc: 124439669 common: 94176434 This means 41 percent are common. This means in the make phenotpye step, different genes and peaks are removed. Results: /project2/gilad/briana/threeprimeseq/data/NucSpecQTL/TotNucRes_overlapassociations.txt</p>
<p>makeTotalNucAssocPlot.R</p>
<pre class="r"><code>library(tidyverse)
library(data.table)

file=fread(&quot;../data/NucSpecQTL/TotNucRes_overlapassociations.txt&quot;, col.names =c(&quot;ID&quot;,&quot;Total&quot;, &quot;Nuclear&quot;))

cor_plot=ggplot(file, aes(x=-log10(Total), y=-log10(Nuclear))) + geom_point() + geom_density2d(na.rm = TRUE, size = 1, colour = &#39;red&#39;) + labs(title=&quot;Common Peak/Snp/ID associations Total and Nuclear&quot;, x=&quot;-log10(Total Pval)&quot;, y=&quot;-log10(Nuclear Pval)&quot;) + geom_smooth(aes(x=-log10(Total),y=-log10(Nuclear)),method = &quot;lm&quot;)


#print(summary(lm(data=file, -log10(Total) ~ -log10(Nuclear))))

#cor_plot

#ggsave(plot, &quot;/project2/gilad/briana/threeprimeseq/output/TotalvNucPavalCommonAssoc.png&quot;)  </code></pre>
<p>in python with pyplotlib:</p>
<pre class="bash"><code>
import matplotlib.pyplot as plt
import numpy as np


inF=open(&quot;/project2/gilad/briana/threeprimeseq/data/NucSpecQTL/TotNucRes_overlapassociations.txt&quot;, &quot;r&quot;)


total=[]
nuclear=[]
for ln in inF:
  loca, tot, nuc = ln.split()
  total.append(float(tot))
  nuclear.append(float(nuc))

  
np.asarray(total)
np.asarray(nuclear)
diff =np.subtract(total, nuclear)

plt.plot(total,nuclear)
plt.savefig(&quot;/project2/gilad/briana/threeprimeseq/output/TotalvNucPavalCommonAssoc.png&quot;)</code></pre>
<p>run_makeTotalNucAssocPlot.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_makeTotalNucAssocPlot
#SBATCH --account=pi-yangili1
#SBATCH --time=5:00:00
#SBATCH --output=run_makeTotalNucAssocPlot.out
#SBATCH --error=run_makeTotalNucAssocPlot.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env

Rscript makeTotalNucAssocPlot.R</code></pre>
<p>These are too many values: I need to use the permuted. this will jsut be the pvalue for the same peak:gene combo</p>
<div id="try-with-perm" class="section level2">
<h2>try with perm</h2>
<p>totalandnuclear_commonassociationPerm.py</p>
<pre class="bash"><code>totRes=open(&quot;/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total.fixed.pheno_5perc_permRes.txt&quot;,&quot;r&quot;)
nucRes=open(&quot;/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear.fixed.pheno_5perc_permRes.txt&quot;, &quot;r&quot;)  


outfile=open(&quot;/project2/gilad/briana/threeprimeseq/data/NucSpecQTL/TotNucRes_Permoverlapassociations.txt&quot;, &quot;w&quot;)

resDict={}
for ln in totRes:
    gene=ln.split()[0].split(&quot;:&quot;)[-1].split(&quot;_&quot;)[0]
    peak=ln.split()[0].split(&quot;:&quot;)[-1].split(&quot;_&quot;)[-1]
    id=gene + &quot;:&quot; + peak 
    pval=ln.split()[-2]
    resDict[id]={}
    resDict[id][&quot;Total&quot;]=pval

for ln in nucRes:
    gene=ln.split()[0].split(&quot;:&quot;)[-1].split(&quot;_&quot;)[0]
    peak=ln.split()[0].split(&quot;:&quot;)[-1].split(&quot;_&quot;)[-1]
    id=gene + &quot;:&quot; + peak
    pval=ln.split()[-2]
    if id in resDict.keys():
        resDict[id][&quot;Nuclear&quot;]=pval
    else:
        continue   
        
#now i have a double dictionary. i need to write it out. i want id, the total pval, then the nuc pval  
for outer, inner in resDict.items():
    id=outer
    outPval=[]
    for key in inner:
        outPval.append(inner[key])
    print(outPval)
    if(len(outPval))==2:
         outfile.write(&quot;%s\t%s\t%s\n&quot;%(id, outPval[0], outPval[1]))


outfile.close()        
        
</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ───────────────────────────────────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.1.0       ✔ purrr   0.3.1  
✔ tibble  2.0.1       ✔ dplyr   0.8.0.1
✔ tidyr   0.8.3       ✔ stringr 1.4.0  
✔ readr   1.3.1       ✔ forcats 0.4.0  </code></pre>
<pre><code>Warning: package &#39;tibble&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;purrr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;forcats&#39; was built under R version 3.5.2</code></pre>
<pre><code>── Conflicts ──────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>Warning: package &#39;data.table&#39; was built under R version 3.5.2</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>Warning: package &#39;cowplot&#39; was built under R version 3.5.2</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre class="r"><code>file=fread(&quot;../data/NucSpecQTL/TotNucRes_Permoverlapassociations.txt&quot;, col.names =c(&quot;ID&quot;,&quot;Total&quot;, &quot;Nuclear&quot;))

cor_plot=ggplot(file, aes(x=Total, y=Nuclear)) + geom_point() + geom_density2d(na.rm = TRUE, size = 1, colour = &#39;red&#39;) + labs(title=&quot;Common Peak associations Total and Nuclear&quot;) 

cor_plot</code></pre>
<pre><code>Warning: Removed 20 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/NuclearSpecQTL.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/0537a728cd63a6a825f3615ca5e161a3de7a2be2/docs/figure/NuclearSpecQTL.Rmd/unnamed-chunk-7-1.png" target="_blank">0537a72</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This is similar to what was happening in the nominal case.</p>
<p>I think I need to plot just the total in nuclaer and nuclear in total. this way i can see the outliers.</p>
<pre class="r"><code>names=c(&quot;SNP&quot;, &quot;peak&quot;, &quot;pval&quot;)
NucQTLinTot=read.table(&quot;../data/QTL_overlap/NucQTLs_inTotFractionRes.txt&quot;, stringsAsFactors = F, col.names = names)
TotQTLinNuc=read.table(&quot;../data/QTL_overlap/TotQTLs_inNucFractionRes.txt&quot;, stringsAsFactors = F, col.names = names)

QTL_names= c(&quot;pid&quot;, &quot;nvar&quot;, &quot;shape1&quot;, &quot;shape2&quot;, &quot;dummy&quot;, &quot;SNP&quot;, &quot;dist&quot;, &quot;npval&quot;, &quot;slope&quot;, &quot;ppval&quot;, &quot;bpval&quot;, &quot;bh&quot;)
NucQTL=read.table(&quot;../data/ApaQTLs/NuclearapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt&quot;, col.names = QTL_names, stringsAsFactors = F) %&gt;% separate(pid, into=c(&quot;chr&quot;, &quot;start&quot;,&quot;end&quot;, &quot;peakID&quot;), sep=&quot;:&quot;) %&gt;% separate(peakID, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;), sep=&quot;_&quot;) %&gt;% select(SNP, peak, ppval)

TotQTL=read.table(&quot;../data/ApaQTLs/TotalapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt&quot;, col.names = QTL_names, stringsAsFactors = F) %&gt;% separate(pid, into=c(&quot;chr&quot;, &quot;start&quot;,&quot;end&quot;, &quot;peakID&quot;), sep=&quot;:&quot;) %&gt;% separate(peakID, into=c(&quot;gene&quot;, &quot;strand&quot;, &quot;peak&quot;), sep=&quot;_&quot;) %&gt;% select(SNP, peak, ppval)</code></pre>
<p>Join these:</p>
<pre class="r"><code>TotQTL_bothfrac= TotQTL %&gt;% inner_join(TotQTLinNuc, by=c(&quot;SNP&quot;, &quot;peak&quot;))
colnames(TotQTL_bothfrac)= c(&quot;SNP&quot;, &quot;peak&quot;, &quot;Total&quot;, &quot;Nuclear&quot;)</code></pre>
<pre class="r"><code>NucQTL_bothfrac= NucQTL %&gt;% inner_join(NucQTLinTot, by=c(&quot;SNP&quot;, &quot;peak&quot;))
colnames(NucQTL_bothfrac)= c(&quot;SNP&quot;, &quot;peak&quot;, &quot;Nuclear&quot;, &quot;Total&quot;)</code></pre>
<p>All qtls together:</p>
<pre class="r"><code>allQTL=TotQTL_bothfrac %&gt;% bind_rows(NucQTL_bothfrac)</code></pre>
<pre class="r"><code>ggplot(allQTL, aes(x=Nuclear, y=Total)) + geom_point()</code></pre>
<p><img src="figure/NuclearSpecQTL.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The pvalues are not consistent. I need to get the original pvalues from the nominal files.</p>
<p>nominal associations for each QTL:</p>
<p>getNominal4QTLs.py</p>
<pre class="bash"><code>
nucQTLs=&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Nuclear.apaQTLs.sort.bed&quot;  
totQTLs=&quot;/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Total.apaQTLs.sort.bed&quot;  

nucNom=&quot;/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt&quot;
totNom=&quot;/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt&quot;

outtot=&quot;/project2/gilad/briana/threeprimeseq/data/NucSpecQTL/TotQTLs_inTotFractionRes.txt&quot;
outnuc=&quot;/project2/gilad/briana/threeprimeseq/data/NucSpecQTL/NucQTLs_inNucFractionRes.txt&quot;


def sameFract(inRes, inQTL, out):
    fout=open(out, &quot;w&quot;)
    qtl_dic={}
    #SNP is key, peak is value
    for ln in open(inQTL,&quot;r&quot;):
        snp=ln.split()[2]
        chrom=ln.split()[0]
        peak=ln.split()[3].split(&quot;:&quot;)[0]
        qtl=str(chrom) + &quot;:&quot; + str(snp)
        if qtl not in qtl_dic.keys():
            qtl_dic[qtl]=[peak]
        else: 
            qtl_dic[qtl].append(peak)
    #print(qtl_dic)
    for ln in open(inRes, &quot;r&quot;):
        pval=ln.split()[3]
        snp=ln.split()[1]
        peak=ln.split()[0].split(&quot;:&quot;)[3].split(&quot;_&quot;)[-1]
        if snp in qtl_dic.keys():
            if peak in qtl_dic[snp]:
                fout.write(&quot;%s\t%s\t%s\n&quot;%(snp, peak, pval))   
    fout.close()
    

sameFract(nucNom, nucQTLs,outnuc)  
sameFract(totNom, totQTLs, outtot)  
                </code></pre>
<p>run_getNominal4QTLs.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_getNominal4QTLs
#SBATCH --account=pi-yangili1
#SBATCH --time=5:00:00
#SBATCH --output=run_getNominal4QTLs.out
#SBATCH --error=run_getNominal4QTLs.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env

python getNominal4QTLs.py</code></pre>
<p>I now want to plot the total vs nuclear nominal pvalues for the nuclear QTLs.</p>
<pre class="r"><code>colnames(NucQTLinTot)=c(&quot;SNP&quot;, &quot;peak&quot;, &quot;totalPval&quot;)

NucQTLinNuc=read.table(&quot;../data/NucSpecQTL/NucQTLs_inNucFractionRes.txt&quot;, col.names = c(&quot;SNP&quot;, &quot;peak&quot;,&quot;nuclearPval&quot;), stringsAsFactors = F)


NucQTL_all= NucQTLinTot %&gt;% inner_join(NucQTLinNuc, by=c(&quot;SNP&quot;, &quot;peak&quot;))

summary(lm(data=NucQTL_all,-log10(totalPval) ~ -log10(nuclearPval)))</code></pre>
<pre><code>
Call:
lm(formula = -log10(totalPval) ~ -log10(nuclearPval), data = NucQTL_all)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.1638 -2.7988 -0.9846  1.5394 17.9462 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   4.1767     0.1719    24.3   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 3.824 on 494 degrees of freedom</code></pre>
<pre class="r"><code>bothPval_nucqtl=ggplot(NucQTL_all, aes(x=-log10(totalPval), y=-log10(nuclearPval))) + geom_point() + geom_density2d(na.rm = TRUE, size = 1, colour = &#39;red&#39;) + geom_smooth(aes(x=-log10(totalPval),y=-log10(nuclearPval)),method = &quot;lm&quot;) + labs(title=&quot;Nominal Pvalues for Nuclear APA qtls&quot;)
#+geom_text(aes(label=SNP),hjust=0, vjust=0)

bothPval_nucqtl</code></pre>
<p><img src="figure/NuclearSpecQTL.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(NucQTL_all, aes(x=totalPval, y=nuclearPval)) + geom_point() + labs(title=&quot;Nominal Pval results for Nuclear QTLs&quot;) + geom_density2d(na.rm = TRUE, size = 1, colour = &#39;red&#39;)  +geom_vline(xintercept=.05)</code></pre>
<p><img src="figure/NuclearSpecQTL.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>nNucQTLNotSigTot= NucQTL_all %&gt;% filter(totalPval&gt;.05) %&gt;% nrow()
nNucQTLNotSigTot</code></pre>
<pre><code>[1] 118</code></pre>
<p>For Total QTLs:</p>
<pre class="r"><code>colnames(TotQTLinNuc)=c(&quot;SNP&quot;, &quot;peak&quot;, &quot;nuclearPval&quot;)

TotQTLinTot=read.table(&quot;../data/NucSpecQTL/TotQTLs_inTotFractionRes.txt&quot;, col.names = c(&quot;SNP&quot;, &quot;peak&quot;,&quot;totalPval&quot;), stringsAsFactors = F)


TotQTL_all= TotQTLinNuc %&gt;% inner_join(TotQTLinTot, by=c(&quot;SNP&quot;, &quot;peak&quot;))

summary(lm(data=NucQTL_all,-log10(nuclearPval) ~ -log10(totalPval)))</code></pre>
<pre><code>
Call:
lm(formula = -log10(nuclearPval) ~ -log10(totalPval), data = NucQTL_all)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.4270 -1.9985 -1.1978  0.9371 15.8122 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   7.1208     0.1385   51.41   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 3.082 on 494 degrees of freedom</code></pre>
<pre class="r"><code>bothPval_totqtl=ggplot(TotQTL_all, aes(x=-log10(nuclearPval), y=-log10(totalPval))) + geom_point() + geom_density2d(na.rm = TRUE, size = 1, colour = &#39;red&#39;) + geom_smooth(aes(x=-log10(nuclearPval),y=-log10(totalPval)),method = &quot;lm&quot;) + labs(title=&quot;Nominal Pvalues for Total APA qtls&quot;)
#+geom_text(aes(label=SNP),hjust=0, vjust=0)

bothPval_totqtl</code></pre>
<p><img src="figure/NuclearSpecQTL.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>bothFracPlots=plot_grid(bothPval_nucqtl,bothPval_totqtl)
bothFracPlots</code></pre>
<p><img src="figure/NuclearSpecQTL.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(bothFracPlots, file=&quot;../output/plots/NominalPvalcondQTL.png&quot;, height=7, width=12 )</code></pre>
<pre class="r"><code>ggplot(TotQTL_all, aes(x=nuclearPval, y=totalPval)) + geom_point() + labs(title=&quot;Nominal Pval results for Total QTLs&quot;) + geom_density2d(na.rm = TRUE, size = 1, colour = &#39;red&#39;) </code></pre>
<p><img src="figure/NuclearSpecQTL.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>nTotQTLNotSigNuc= TotQTL_all %&gt;% filter(nuclearPval&gt;.05) %&gt;% nrow()
nTotQTLNotSigNuc</code></pre>
<pre><code>[1] 39</code></pre>
<p>Is the difference significant. More likely to have nuclear specific qlts?</p>
<pre class="r"><code>phyper(118, 495, 272, 157, lower.tail = F)</code></pre>
<pre><code>[1] 0.0005207115</code></pre>
<p>Plot for this:</p>
<pre class="r"><code>compareSpecQTL=as.data.frame(cbind(fraction=c(&quot;Total QTL&quot;, &quot;Total QTL&quot;,&quot;Nuclear QTL&quot;,&quot;Nuclear QTL&quot;), SigInOther=c(&quot;No&quot;,&quot;Yes&quot;,&quot;No&quot;,&quot;Yes&quot;),Value=c(.14,.86 ,.24,.76)))

compareSpecQTL$Value=as.numeric(as.character(compareSpecQTL$Value))</code></pre>
<pre class="r"><code>qtls_oppfrac=ggplot(compareSpecQTL,aes(x=fraction, fill=SigInOther, y=Value))  + geom_bar(stat=&quot;identity&quot;) +scale_fill_manual(values=c(&quot;red&quot;,&quot;grey&quot;)) + labs(y=&quot;Proportion&quot;, title=&quot;QTLs by nominal significance in other fraction&quot;)  + annotate(&quot;text&quot;, label=&quot;p &lt; .0005&quot;, y=1.2, x=1.5) + geom_segment(aes(x=1.1, xend=1.9, y=1.1, yend=1.1))

qtls_oppfrac</code></pre>
<p><img src="figure/NuclearSpecQTL.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(qtls_oppfrac, file=&quot;../output/plots/QTLsSiginOppFraction.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] cowplot_0.9.4     data.table_1.12.0 forcats_0.4.0    
 [4] stringr_1.4.0     dplyr_0.8.0.1     purrr_0.3.1      
 [7] readr_1.3.1       tidyr_0.8.3       tibble_2.0.1     
[10] ggplot2_3.1.0     tidyverse_1.2.1  

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5 xfun_0.5         haven_2.1.0      lattice_0.20-38 
 [5] colorspace_1.4-0 generics_0.0.2   htmltools_0.3.6  yaml_2.2.0      
 [9] rlang_0.3.1      pillar_1.3.1     glue_1.3.0       withr_2.1.2     
[13] modelr_0.1.4     readxl_1.3.0     plyr_1.8.4       munsell_0.5.0   
[17] gtable_0.2.0     workflowr_1.2.0  cellranger_1.1.0 rvest_0.3.2     
[21] evaluate_0.13    labeling_0.3     knitr_1.21       broom_0.5.1     
[25] Rcpp_1.0.0       scales_1.0.0     backports_1.1.3  jsonlite_1.6    
[29] fs_1.2.6         hms_0.4.2        digest_0.6.18    stringi_1.3.1   
[33] grid_3.5.1       rprojroot_1.3-2  cli_1.0.1        tools_3.5.1     
[37] magrittr_1.5     lazyeval_0.2.1   crayon_1.3.4     whisker_0.3-2   
[41] pkgconfig_2.0.2  MASS_7.3-51.1    xml2_1.2.0       lubridate_1.7.4 
[45] assertthat_0.2.0 rmarkdown_1.11   httr_1.4.0       rstudioapi_0.9.0
[49] R6_2.4.0         nlme_3.1-137     git2r_0.24.0     compiler_3.5.1  </code></pre>
</div>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
