<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Location of Signal Site</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<div class="container-fluid main-container">

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Location of Signal Site</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>3/6/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-03-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>threeprimeseq/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree2d022d64d775a0c67e61626558a51aa47bd24a37targetblank2d022d6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/2d022d64d775a0c67e61626558a51aa47bd24a37" target="_blank">2d022d6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanthreeprimeseqtree2d022d64d775a0c67e61626558a51aa47bd24a37targetblank2d022d6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/perm_QTL_trans_noMP_5percov/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/4suDataIGV.Rmd
    Untracked:  analysis/DirectionapaQTL.Rmd
    Untracked:  analysis/EmpDistforOverlaps.Rmd
    Untracked:  analysis/EvaleQTLs.Rmd
    Untracked:  analysis/OverlapFullGWASCatelog.Rmd
    Untracked:  analysis/YL_QTL_test.Rmd
    Untracked:  analysis/groSeqAnalysis.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  analysis/verifybam_dubs.Rmd
    Untracked:  code/PeaksToCoverPerReads.py
    Untracked:  code/strober_pc_pve_heatmap_func.R
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/AllPeak_counts/
    Untracked:  data/ApaQTLs/
    Untracked:  data/ApaQTLs_otherPhen/
    Untracked:  data/ChromHmmOverlap/
    Untracked:  data/DistTXN2Peak_genelocAnno/
    Untracked:  data/FeatureoverlapPeaks/
    Untracked:  data/GM12878.chromHMM.bed
    Untracked:  data/GM12878.chromHMM.txt
    Untracked:  data/LianoglouLCL/
    Untracked:  data/LocusZoom/
    Untracked:  data/LocusZoom_Unexp/
    Untracked:  data/LocusZoom_proc/
    Untracked:  data/MatchedSnps/
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/PeakCounts/
    Untracked:  data/PeakCounts_noMP_5perc/
    Untracked:  data/PeakCounts_noMP_genelocanno/
    Untracked:  data/PeakUsage/
    Untracked:  data/PeakUsage_noMP/
    Untracked:  data/PeakUsage_noMP_GeneLocAnno/
    Untracked:  data/PeaksUsed/
    Untracked:  data/PeaksUsed_noMP_5percCov/
    Untracked:  data/PolyA_DB/
    Untracked:  data/QTL_overlap/
    Untracked:  data/RNAkalisto/
    Untracked:  data/RefSeq_annotations/
    Untracked:  data/Replicates_usage/
    Untracked:  data/Signal_Loc/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/UnderstandPeaksQC/
    Untracked:  data/WASP_STAT/
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/YL_QTL_test/
    Untracked:  data/apaExamp/
    Untracked:  data/apaExamp_proc/
    Untracked:  data/apaQTL_examp_noMP/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/diff_iso_GeneLocAnno/
    Untracked:  data/diff_iso_proc/
    Untracked:  data/diff_iso_trans/
    Untracked:  data/eQTLs_Lietal/
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/example_gene_peakQuant/
    Untracked:  data/explainProtVar/
    Untracked:  data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/molPheno_noMP/
    Untracked:  data/mol_overlap/
    Untracked:  data/mol_pheno/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/nuc_10up/
    Untracked:  data/other_qtls/
    Untracked:  data/pQTL_otherphen/
    Untracked:  data/pacbio_cov/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/peaks4DT/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov/
    Untracked:  data/perm_QTL_GeneLocAnno_noMP_5percov_3UTR/
    Untracked:  data/perm_QTL_diffWindow/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/perm_QTL_trans_filt/
    Untracked:  data/protAndAPAAndExplmRes.Rda
    Untracked:  data/protAndAPAlmRes.Rda
    Untracked:  data/protAndExpressionlmRes.Rda
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  data/threePrimeSeqMetaData.csv
    Untracked:  data/threePrimeSeqMetaData55Ind.txt
    Untracked:  data/threePrimeSeqMetaData55Ind.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup.xlsx
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt
    Untracked:  data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.xlsx
    Untracked:  output/LZ/
    Untracked:  output/deeptools_plots/
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/CompareLianoglouData.Rmd
    Modified:   analysis/NewPeakPostMP.Rmd
    Modified:   analysis/apaQTLoverlapGWAS.Rmd
    Modified:   analysis/characterize_apaQTLs.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/coloc_apaQTLs_protQTLs.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explainpQTLs.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/fixBWChromNames.Rmd
    Modified:   analysis/flash2mash.Rmd
    Modified:   analysis/mispriming_approach.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlapMolQTL.opposite.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/peakQCPPlots.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/pipeline_55Ind.Rmd
    Modified:   analysis/swarmPlots_QTLs.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   analysis/test.smash.Rmd
    Modified:   analysis/understandPeaks.Rmd
    Modified:   analysis/unexplainedeQTL_analysis.Rmd
    Modified:   code/Snakefile

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/2d022d64d775a0c67e61626558a51aa47bd24a37/analysis/SignalSiteLoc.Rmd" target="_blank">2d022d6</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-09
</td>
<td>
add location proportion plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/ca1a9f465d3821e9ca286c5f5b2feba428df179f/docs/SignalSiteLoc.html" target="_blank">ca1a9f4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/64fe413faa3780f1108a322dde565e73f98d56b1/analysis/SignalSiteLoc.Rmd" target="_blank">64fe413</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-08
</td>
<td>
add signal site loc hist
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/ffb0e845598e99c4ef5309787ec42ef86ee7652b/docs/SignalSiteLoc.html" target="_blank">ffb0e84</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/52e7514c85b1d53cf84c23529b70b2d0f22b3fd7/analysis/SignalSiteLoc.Rmd" target="_blank">52e7514</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-07
</td>
<td>
start new analysis - 3’ side of peak
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/638d12e0fc0bd1d866d78479cbcf2aa027e85c37/docs/SignalSiteLoc.html" target="_blank">638d12e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/72c34ce2e9c8ea98431a947dc7616740d920f495/analysis/SignalSiteLoc.Rmd" target="_blank">72c34ce</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-07
</td>
<td>
signal site loc based on front side
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/4bf5d094c2774322c3ee6f6424a6980232eab5c6/docs/SignalSiteLoc.html" target="_blank">4bf5d09</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/8717550230a201751961b12e5deaeafcc9527a24/analysis/SignalSiteLoc.Rmd" target="_blank">8717550</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
<td>
res for AATAAA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/4023fe07cdb3542ab8420cd638f2ae9dc1432660/docs/SignalSiteLoc.html" target="_blank">4023fe0</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/d5611909a1aae47b7eaffbaf7ad81da4947fdf88/analysis/SignalSiteLoc.Rmd" target="_blank">d561190</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
<td>
analysis up to getting seqs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/threeprimeseq/ba63ea22968208956eeb9dff3f6fb3573863fed1/docs/SignalSiteLoc.html" target="_blank">ba63ea2</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/c200503cbf5505dd25e7f30321019a5758f6e9b4/analysis/SignalSiteLoc.Rmd" target="_blank">c200503</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
<td>
add signal site loc analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>In the <a href="SignalSiteEnrich.html">Signal Site enrichment analysis</a> I looked at the peaks to see if signal sites are enriched upstream of my peaks. I found this is true but now I want to see where the signal sites are in comparison to my peaks. I am going to use the biostrings package tool matchPWM for this analysis.</p>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.2.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ───────────────────────────────────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.1.0       ✔ purrr   0.3.1  
✔ tibble  2.0.1       ✔ dplyr   0.8.0.1
✔ tidyr   0.8.3       ✔ stringr 1.4.0  
✔ readr   1.3.1       ✔ forcats 0.4.0  </code></pre>
<pre><code>Warning: package &#39;tibble&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;purrr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;stringr&#39; was built under R version 3.5.2</code></pre>
<pre><code>Warning: package &#39;forcats&#39; was built under R version 3.5.2</code></pre>
<pre><code>── Conflicts ──────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(Biostrings)</code></pre>
<pre><code>Warning: package &#39;Biostrings&#39; was built under R version 3.5.2</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind,
    colMeans, colnames, colSums, dirname, do.call, duplicated,
    eval, evalq, Filter, Find, get, grep, grepl, intersect,
    is.unsorted, lapply, lengths, Map, mapply, match, mget, order,
    paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,
    Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which, which.max,
    which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    first, rename</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    expand</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    collapse, desc, slice</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    reduce</code></pre>
<pre><code>Loading required package: XVector</code></pre>
<pre><code>
Attaching package: &#39;XVector&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    compact</code></pre>
<pre><code>
Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    strsplit</code></pre>
<pre class="r"><code>library(BSgenome)</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Warning: package &#39;GenomeInfoDb&#39; was built under R version 3.5.2</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: rtracklayer</code></pre>
<pre><code>Warning: package &#39;rtracklayer&#39; was built under R version 3.5.2</code></pre>
<pre class="r"><code>library(genomation)</code></pre>
<pre><code>Loading required package: grid</code></pre>
<div id="make-location-file" class="section level2">
<h2>Make location file</h2>
<p>I need to get the coordinates for the regions I care about. I want to look at the peak and 150bp upstream. This is probably larger than I will need to look at but it will be good to have an inclusive look first.</p>
<p>I want to use the peak file and make a file that is the peak and upstream 150:</p>
<p>Upstream150Bases.py</p>
<pre class="python"><code>#python  
def main(Fin, Fout):
  outBed=open(Fout, &quot;w&quot;)
  chrom_lengths=open(&quot;/project2/gilad/briana/genome_anotation_data/chrom_lengths2.sort.bed&quot;,&quot;r&quot;)
  #make a dictionary with chrom lengths
  length_dic={}
  for i in chrom_lengths:
    chrom, start, end = i.split()
    length_dic[str(chrom)]=int(end)  

#write file 
  for ln in open(Fin):
    chrom, start, end, name, score, strand = ln.split()
    chrom=str(chrom)
    if strand==&quot;+&quot;:
      start_new=int(start)-150
      if start_new &lt;= 1:
        start_new = 0 
      end_new= int(end)
      if end_new == 0:
        end_new=1
      outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new, end_new, name, score, strand))
    if strand == &quot;-&quot;:
      start_new=int(start)
      end_new=int(end) + 150
      outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new, end_new, name, score, strand))
  outBed.close()  

if __name__ == &quot;__main__&quot;:
    import sys
    inFile = sys.argv[1]
    outFile=sys.argv[2] 
    main(inFile, outFile)</code></pre>
<p>run_get150up.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_get150up
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=run_get150upt.out
#SBATCH --error=run_get150up.err
#SBATCH --partition=broadwl
#SBATCH --mem=16G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

python Upstream150Bases.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand.bed  /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_peakand150up.bed </code></pre>
</div>
<div id="get-subject-reads" class="section level2">
<h2>Get subject (reads)</h2>
<p>Input the regions:</p>
<p>Fix chromosomes:</p>
<pre class="r"><code>PeakRegions=read.table(&quot;../data/Signal_Loc/APAPeaks_5percCov_fixedStrand_peakand150up.bed&quot;, header=F,col.names = c(&quot;chr&quot;,&quot;start&quot;, &quot;end&quot;, &quot;peak&quot;, &quot;score&quot;, &quot;strand&quot;)) %&gt;% mutate(Chrom=paste(&quot;chr&quot;, chr, sep=&quot;&quot;)) %&gt;% select(Chrom, start,end,peak,score,strand)
write.table(PeakRegions, file=&quot;../data/Signal_Loc/APAPeaks_5percCov_fixedStrand_peakand150up_fixedChr.bed&quot;, quote=F, col.names = F, row.names = F, sep=&quot;\t&quot;)</code></pre>
<pre class="r"><code>#convert to reads 

reads.GR= readGeneric(file=&quot;../data/Signal_Loc/APAPeaks_5percCov_fixedStrand_peakand150up_fixedChr.bed&quot;,chr =1, start = 2, end =3,  meta.cols =4, header=F, zero.based=TRUE,strand=6)</code></pre>
<p>I need to overlap these positions with the genome</p>
<div id="make-motifs" class="section level3">
<h3>Make motifs</h3>
<pre class="r"><code>AATAAA= PWM(&quot;AATAAA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
ATTAAA= PWM(&quot;ATTAAA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
AGTAAA= PWM(&quot;AGTAAA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
TATAAA= PWM(&quot;TATAAA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
CATAAA= PWM(&quot;CATAAA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
GATAAA= PWM(&quot;GATAAA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
AATATA= PWM(&quot;AATATA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
AATACA= PWM(&quot;AATACA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
AATAGA= PWM(&quot;AATAGA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
AAAAAG= PWM(&quot;AAAAAG&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))
ACTAAA= PWM(&quot;ACTAAA&quot;, type = c(&quot;log2probratio&quot;, &quot;prob&quot;), prior.params = c(A=0.25, C=0.25, G=0.25, T=0.25))</code></pre>
</div>
<div id="find-the-mathes" class="section level3">
<h3>find the mathes</h3>
<pre class="r"><code>genome.hg19 &lt;- getBSgenome(&quot;BSgenome.Hsapiens.UCSC.hg19&quot;)

#matches &lt;- matchPWM(pwm=AATAAA, subject = genome.hg19) %&gt;% keepStandardChromosomes(., species= &quot;Homo sapiens&quot;)


DNAstringSetPeaks=data.frame(seq=getSeq(genome.hg19, reads.GR))

x=DNAString(DNAstringSetPeaks[1,1])

hits &lt;- matchPWM(AATAAA,x,with.score=T) 


start(hits)</code></pre>
<pre><code>[1] 215 219 255</code></pre>
<p>Look over and make hits file for all</p>
<pre class="r"><code>list_AATAAA_res=c()
for (i in 1:nrow(DNAstringSetPeaks)){
  x=DNAString(DNAstringSetPeaks[i,1])
  list_AATAAA_res=c(list_AATAAA_res,matchPWM(AATAAA,x,with.score=T))
}</code></pre>
<p>Get out the start positions:</p>
<pre class="r"><code>starts_AATAAA=c()
nsig=c()
last_oc_AATAAA=c()
for (i in list_AATAAA_res){
  nsig=c(nsig, length(start(i)))
  starts_AATAAA=c(starts_AATAAA, start(i))
  #print(length(start(i)))
  if (length(start(i)) != 0 ){
    last_oc_AATAAA=c(last_oc_AATAAA, max(start(i),na.rm =T))
  }
}</code></pre>
<p>Histogram of results:</p>
<pre class="r"><code>summary(starts_AATAAA)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    1.0    79.0   175.0   156.7   216.0  1612.0 </code></pre>
<pre class="r"><code>hist(starts_AATAAA,breaks=10000)</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/638d12e0fc0bd1d866d78479cbcf2aa027e85c37/docs/figure/SignalSiteLoc.Rmd/unnamed-chunk-10-1.png" target="_blank">638d12e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/4bf5d094c2774322c3ee6f6424a6980232eab5c6/docs/figure/SignalSiteLoc.Rmd/unnamed-chunk-10-1.png" target="_blank">4bf5d09</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>summary(nsig)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   1.000   3.000   3.857   5.000  74.000 </code></pre>
<pre class="r"><code>sum(nsig==0)</code></pre>
<pre><code>[1] 3584</code></pre>
<pre class="r"><code>sum(nsig==1)</code></pre>
<pre><code>[1] 8358</code></pre>
<pre class="r"><code>hist(nsig,breaks=100)</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/638d12e0fc0bd1d866d78479cbcf2aa027e85c37/docs/figure/SignalSiteLoc.Rmd/unnamed-chunk-11-1.png" target="_blank">638d12e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/4bf5d094c2774322c3ee6f6424a6980232eab5c6/docs/figure/SignalSiteLoc.Rmd/unnamed-chunk-11-1.png" target="_blank">4bf5d09</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Look at the first occurence:</p>
<pre class="r"><code>summary(last_oc_AATAAA)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    1.0   209.0   215.0   213.1   231.0  1612.0 </code></pre>
<pre class="r"><code>hist(last_oc_AATAAA,breaks=1000)</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/638d12e0fc0bd1d866d78479cbcf2aa027e85c37/docs/figure/SignalSiteLoc.Rmd/unnamed-chunk-12-1.png" target="_blank">638d12e</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-07
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/4bf5d094c2774322c3ee6f6424a6980232eab5c6/docs/figure/SignalSiteLoc.Rmd/unnamed-chunk-12-1.png" target="_blank">4bf5d09</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I want to get the closest occurance of the</p>
</div>
</div>
<div id="change-analysis" class="section level2">
<h2>Change analysis:</h2>
<p>I want to start at the end of the peak (most downstream) and look for the signal sites. For confidence a peak is one PAS, I will look only at peaks less than 100bp long. I will extend the peak upstream 100 basepairs. I will look from the downstream end for the sites.</p>
<div id="subset-peaks-less-than-200bp" class="section level3">
<h3>Subset peaks less than 200bp</h3>
<p>filterPeaks100length.py</p>
<pre class="bash"><code>peaks=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand.bed&quot;, &quot;r&quot;)

outPeaks=open(&quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_filter4length.bed&quot;, &quot;w&quot;)

nNotOk=0
for ln in peaks:
    start= int(ln.split()[1])
    end=int(ln.split()[2])
    length=end - start 
    if length &lt;= 100:
        outPeaks.write(ln)
    else:
        nNotOk +=1
        
print(nNotOk)
outPeaks.close()
      </code></pre>
<p>This filters 12105 peaks.</p>
</div>
<div id="get-sequence-for-peaks-and-100-upstream" class="section level3">
<h3>Get sequence for peaks and 100 upstream</h3>
<p>Upstream100Bases_filteredpeaks.py</p>
<pre class="bash"><code>#python  
def main(Fin, Fout):
  outBed=open(Fout, &quot;w&quot;)
  chrom_lengths=open(&quot;/project2/gilad/briana/genome_anotation_data/chrom_lengths2.sort.bed&quot;,&quot;r&quot;)
  #make a dictionary with chrom lengths
  length_dic={}
  for i in chrom_lengths:
    chrom, start, end = i.split()
    length_dic[str(chrom)]=int(end)  

#write file 
  for ln in open(Fin):
    chrom, start, end, name, score, strand = ln.split()
    chrom=str(chrom)
    if strand==&quot;+&quot;:
      start_new=int(start)-100
      if start_new &lt;= 1:
        start_new = 0 
      end_new= int(end)
      if end_new == 0:
        end_new=1
      outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new, end_new, name, score, strand))
    if strand == &quot;-&quot;:
      start_new=int(start)
      end_new=int(end) + 100
      outBed.write(&quot;%s\t%d\t%d\t%s\t%s\t%s\n&quot;%(chrom, start_new, end_new, name, score, strand))
  outBed.close()  

if __name__ == &quot;__main__&quot;:
    import sys
    inFile = sys.argv[1]
    outFile=sys.argv[2] 
    main(inFile, outFile)</code></pre>
<p>Run this:</p>
<pre class="bash"><code>python Upstream100Bases_filteredpeaks.py /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_filter4length.bed  /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_filter4length_upstream100.bed </code></pre>
<p>Run bedtools nuc for this to get the sequences:</p>
<p>nucpeaksand100up.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=nucpeaksand100up
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=nucpeaksand100up.out
#SBATCH --error=nucpeaksand100up.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

bedtools nuc -seq -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_filter4length_upstream100.bed  &gt; /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_filter4length_upstream100_Seq.bed  </code></pre>
<p>This is looking at the positive strand left to right always. I need to go from the right and look at the reverse signal sites. I can look into ways to flip a string in python</p>
<p>‘a string’[::-1]</p>
</div>
<div id="find-distance-to-peaks" class="section level3">
<h3>find distance to peaks</h3>
<p><strong>change region I am looking at before I do this - move to new analysis and try a new method for this analysis</strong></p>
<p>DistPAS2Sig.py</p>
<pre class="bash"><code>def main(Insite, out):
  sigsite=[Insite]
  
  inBed=open(&quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_filter4length_upstream100_Seq.bed&quot;, &quot;r&quot;)
  outRes=open(out, &quot;w&quot;)
  
  #function for reverse compliments  
  
  def ReverseComplement1(seq):
      seq_dict = {&#39;A&#39;:&#39;T&#39;,&#39;T&#39;:&#39;A&#39;,&#39;G&#39;:&#39;C&#39;,&#39;C&#39;:&#39;G&#39;, &#39;a&#39;:&#39;t&#39;, &#39;t&#39;:&#39;a&#39;, &#39;g&#39;:&#39;c&#39;, &#39;c&#39;:&#39;g&#39;}
      bases=[seq_dict[base] for base in seq]
      bases=reversed(bases)
      return(&quot;&quot;.join(bases))
  
  
  #reverse comp each signal site 
  sigsite_revComp=[]
  for i in sigsite:
      sigsite_revComp.append(ReverseComplement1(i))
      
  #want a dictionary for each of the sites and its reverse compliment:
  sigsites_dic={}
  for i in range(len(sigsite)):
      sigsites_dic[sigsite[i]]=sigsite_revComp[i]    
      
  
  #function to get occurance: takes in sig site and sequence (give it the correct stranded stuff)
  
  #make 2 of these, this is for the pos strand
  def getOccurance(sigsite, seq):
      if sigsite in seq:
          length=len(seq)
          pos= seq.rfind(sigsite)
          posF=length-pos
          return(posF)
      else:
          return(-9)
          
  #negative strand occurance function:
  
  def getOccurance_neg(sigsite, seq):
      sigsite=sigsites_dic[sigsite]
      if sigsite in seq:
          pos= seq.find(sigsite)
          return(pos + 6)
      else:
          return(-9)
  
  #i can only addpend the value if the function does not return -9
      
  
  #function i can run on each signal site  
  
  
  #loop through peaks and check for every site, first ask stand and do the rev  
  def loop41site(site):
      resList=[]
      for ln in inBed:
          strand=ln.split()[5]
          seq= ln.split()[15]
          if strand == &quot;+&quot;:
              loc= getOccurance(site, seq)
              if loc !=-9:
                  resList.append(loc)
          else: 
              loc=getOccurance_neg(site,seq)
              if loc !=-9:
                  resList.append(loc)
      return(resList)
          
  
  #run this for each sig site
  res_dic={}
  for i in sigsite:
      res_dic[i]=[]
  for i in sigsite:
      reslist=loop41site(i)
      res_dic[i]=reslist
  
  
  outRes.write(&quot;%s\n&quot;%(sigsite[0]))
  for i in reslist:
      outRes.write(&quot;%d\n&quot;%(i))
      
  
  
  outRes.close()
      
    
if __name__ == &quot;__main__&quot;:
    import sys
    Site_in = sys.argv[1]
    outFile= &quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/Loc_%s_Distance2end.txt&quot;%(Site_in)
    main(Site_in, outFile)
</code></pre>
<p>make a test with just 1 site:</p>
<p>test_DistPAS2Sig.py</p>
<pre class="bash"><code>
sigsite=[&#39;ATTAAA&#39;] 

inBed=open(&quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_filter4length_upstream100_Seq.bed&quot;, &quot;r&quot;)
outRes=open(&#39;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/test.text&#39;, &quot;w&quot;)

#function for reverse compliments  

def ReverseComplement1(seq):
    seq_dict = {&#39;A&#39;:&#39;T&#39;,&#39;T&#39;:&#39;A&#39;,&#39;G&#39;:&#39;C&#39;,&#39;C&#39;:&#39;G&#39;, &#39;a&#39;:&#39;t&#39;, &#39;t&#39;:&#39;a&#39;, &#39;g&#39;:&#39;c&#39;, &#39;c&#39;:&#39;g&#39;}
    bases=[seq_dict[base] for base in seq]
    bases=reversed(bases)
    return(&quot;&quot;.join(bases))


#reverse comp each signal site 
sigsite_revComp=[]
for i in sigsite:
    sigsite_revComp.append(ReverseComplement1(i))
    
#want a dictionary for each of the sites and its reverse compliment:
sigsites_dic={}
for i in range(len(sigsite)):
    sigsites_dic[sigsite[i]]=sigsite_revComp[i]    
    

#function to get occurance: takes in sig site and sequence (give it the correct stranded stuff)

#make 2 of these, this is for the pos strand
def getOccurance(sigsite, seq):
    if sigsite in seq:
        print(sigsite)
        print(seq)
        pos= seq.rfind(sigsite)
        return(pos)
    else:
        return(-9)
        
#negative strand occurance function:

def getOccurance_neg(sigsite, seq):
    sigsite=sigsites_dic[sigsite]
    if sigsite in seq:
        pos= seq.find(sigsite)
        return(pos + 6)
    else:
        return(-9)

#i can only addpend the value if the function does not return -9
    

#function i can run on each signal site  


#loop through peaks and check for every site, first ask stand and do the rev  
def loop41site(site):
    resList=[]
    for ln in inBed:
        strand=ln.split()[5]
        seq= ln.split()[15]
        if strand == &quot;+&quot;:
            loc= getOccurance(site, seq)
            print(loc)
            if loc !=-9:
                resList.append(str(loc))
        else: 
            loc=getOccurance_neg(site,seq)
            if loc !=-9:
                resList.append(str(loc))
    return(resList)
        

#run this for each sig site
res_dic={}
for i in sigsite:
    res_dic[i]=[]
for i in sigsite:
    reslist=loop41site(i)
    res_dic[i]=reslist
    
for key, value in res_dic.items():
    valString=&quot;:&quot;.join(value)
    outRes.write(&quot;%s\t%s\n&quot;%(key, valString))


outRes.close()
   
   
   </code></pre>
<p>run_DistPAS2Sig.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_DistPAS2Sig
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=run_DistPAS2Sig.out
#SBATCH --error=run_DistPAS2Sig.err
#SBATCH --partition=broadwl
#SBATCH --mem=16G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env


python DistPAS2Sig.py AATAAA
python DistPAS2Sig.py ATTAAA
python DistPAS2Sig.py AGTAAA
python DistPAS2Sig.py TATAAA
python DistPAS2Sig.py CATAAA
python DistPAS2Sig.py GATAAA
python DistPAS2Sig.py AATATA
python DistPAS2Sig.py AATACA
python DistPAS2Sig.py AATAGA
python DistPAS2Sig.py AAAAAG
python DistPAS2Sig.py ACTAAA</code></pre>
</div>
<div id="make-histograms" class="section level3">
<h3>Make histograms:</h3>
<pre class="r"><code>Loc_AATAAA= read.table(&quot;../data/Signal_Loc/Loc_AATAAA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AATAAA&quot;)
nrow(Loc_AATAAA)</code></pre>
<pre><code>[1] 11809</code></pre>
<pre class="r"><code>Loc_AAAAAG= read.table(&quot;../data/Signal_Loc/Loc_AAAAAG_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AAAAAG&quot;)


Loc_AATACA= read.table(&quot;../data/Signal_Loc/Loc_AATACA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AATACA&quot;)


Loc_AATAGA= read.table(&quot;../data/Signal_Loc/Loc_AATAGA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AATAGA&quot;)


Loc_AATATA= read.table(&quot;../data/Signal_Loc/Loc_AATATA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AATATA&quot;)


Loc_ACTAAA= read.table(&quot;../data/Signal_Loc/Loc_ACTAAA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;ACTAAA&quot;)


Loc_AGTAAA= read.table(&quot;../data/Signal_Loc/Loc_AGTAAA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AGTAAA&quot;)



Loc_ATTAAA= read.table(&quot;../data/Signal_Loc/Loc_ATTAAA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;ATTAAA&quot;)

Loc_CATAAA= read.table(&quot;../data/Signal_Loc/Loc_CATAAA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;CATAAA&quot;)

Loc_GATAAA= read.table(&quot;../data/Signal_Loc/Loc_GATAAA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;GATAAA&quot;)

Loc_TATAAA= read.table(&quot;../data/Signal_Loc/Loc_TATAAA_Distance2end.txt&quot;, header=T) %&gt;% mutate(Site=&quot;TATAAA&quot;)</code></pre>
<pre class="r"><code>dist_Loc= ggplot(Loc_AATAAA, aes(x=AATAAA)) + 
  geom_histogram(bins=100, fill=&quot;red&quot;, alpha=.3) + 
  labs(title=&quot;Distance from PAS&quot;, x=&quot;Number of PAS&quot;, y=&quot;N basepair from PAS&quot;) +
  geom_histogram(bins=100,data=Loc_AAAAAG, aes(x=AAAAAG), fill=&quot;orange&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATACA, aes(x=AATACA), fill=&quot;yellow&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATAGA, aes(x=AATAGA), fill=&quot;green&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATATA, aes(x=AATATA), fill=&quot;blue&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_ACTAAA, aes(x=ACTAAA), fill=&quot;purple&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AGTAAA, aes(x=AGTAAA), fill=&quot;firebrick3&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AGTAAA, aes(x=AGTAAA), fill=&quot;darksalmon&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_CATAAA, aes(x=CATAAA), fill=&quot;darkslategray&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_GATAAA, aes(x=GATAAA), fill=&quot;deeppink1&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_TATAAA, aes(x=TATAAA), fill=&quot;lightcyan1&quot;, alpha=.3) 
  
  
dist_Loc</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ca1a9f465d3821e9ca286c5f5b2feba428df179f/docs/figure/SignalSiteLoc.Rmd/unnamed-chunk-21-1.png" target="_blank">ca1a9f4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Make a long dataframe for all of this to make it easier to manipulate.</p>
<pre class="r"><code>colnames(Loc_AATAAA)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AAAAAG)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AATACA)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AATAGA)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AATATA)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_ACTAAA)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AGTAAA)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_CATAAA)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_ATTAAA)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_GATAAA)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_TATAAA)=c(&quot;Count&quot;, &quot;Site&quot;)



AllsiteDF=as.data.frame(rbind(Loc_AATAAA,Loc_AAAAAG,Loc_AATACA,Loc_AATAGA,Loc_AATATA,Loc_ACTAAA,Loc_AGTAAA,Loc_ATTAAA, Loc_GATAAA,Loc_TATAAA,Loc_CATAAA)) %&gt;% mutate(NegCount=-1*as.integer(as.character(Count)), Cononical=ifelse(Site==&quot;AATAAA&quot;, &quot;Yes&quot;,&quot;No&quot;))


 AllsiteDF_to100= AllsiteDF %&gt;% filter(Count &lt; 100)</code></pre>
<p>plot:</p>
<pre class="r"><code>ggplot(AllsiteDF_to100, aes(group=Site, x=NegCount, fill=Site)) + geom_histogram(position=&quot;stack&quot;,bins=50 ) + labs(x=&quot;Distance from PAS&quot;, y=&quot;N annotated Sites&quot;, title=&quot;Location of annotated signal sites&quot;)  </code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Do this as proportion:</p>
<pre class="r"><code>AllsiteDF_to100_prop=AllsiteDF_to100 %&gt;% group_by(Site,NegCount) %&gt;% summarise(CountperPos=n()) %&gt;% mutate(TotCount=sum(CountperPos),prop=CountperPos/TotCount)

#%&gt;% ungroup() %&gt;% group_by(Site) %&gt;% mutate(nType=sum(Count), prop=CountperPos/nType)</code></pre>
<pre class="r"><code>annoationPAS_allpeak=ggplot(AllsiteDF_to100_prop, aes(fill=Site, y=prop, x=NegCount)) + geom_density(stat=&quot;identity&quot;) + facet_wrap(~Site) + labs(x=&quot;Distance from PAS&quot;, y=&quot;Proportion of Signal Site&quot;, title=&quot;Location of annotated signal sites&quot;)

annoationPAS_allpeak</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(annoationPAS_allpeak, file=&quot;../output/plots/annoationPAS_allpeakPropLocFacet.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<pre class="r"><code>annoationPAS_allpeakProphist=ggplot(AllsiteDF_to100_prop, aes(fill=Site, by=Site, y=prop, x=NegCount)) + geom_histogram(stat=&quot;identity&quot;, position=&quot;stack&quot;) + labs(x=&quot;Distance from PAS&quot;, y=&quot;Proportion of Signal Site&quot;, title=&quot;Location of annotated signal sites&quot;)</code></pre>
<pre><code>Warning: Ignoring unknown parameters: binwidth, bins, pad</code></pre>
<pre class="r"><code>annoationPAS_allpeakProphist</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(annoationPAS_allpeakProphist, file=&quot;../output/plots/annoationPAS_allpeakPropLocStackHist.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div id="run-for-nuclear-specific-pas" class="section level3">
<h3>Run for nuclear specific PAS:</h3>
<p><em>Used more in NUclear</em> * /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc.bed</p>
<p><em>Used more in nucelar, in intron</em> * /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron.bed</p>
<p>Filter out ones that are too long:</p>
<p>filterPeaks100length_nuc.py</p>
<pre class="bash"><code>peaks=open(&quot; /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc.bed&quot;, &quot;r&quot;)

outPeaks=open(&quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_filter4length.bed&quot;, &quot;w&quot;)

nNotOk=0
for ln in peaks:
    start= int(ln.split()[1])
    end=int(ln.split()[2])
    length=end - start 
    if length &lt;= 100:
        outPeaks.write(ln)
    else:
        nNotOk +=1
        
print(nNotOk)
outPeaks.close()
      </code></pre>
<p>filterPeaks100length_nucintron.py</p>
<pre class="bash"><code>peaks=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron.bed&quot;, &quot;r&quot;)

outPeaks=open(&quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron_filter4length.bed&quot;, &quot;w&quot;)

nNotOk=0
for ln in peaks:
    start= int(ln.split()[1])
    end=int(ln.split()[2])
    length=end - start 
    if length &lt;= 100:
        outPeaks.write(ln)
    else:
        nNotOk +=1
        
print(nNotOk)
outPeaks.close()
      </code></pre>
<pre class="bash"><code>python Upstream100Bases_filteredpeaks.py /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron_filter4length.bed /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron_filter4length_upstream100.bed

python Upstream100Bases_filteredpeaks.py /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_filter4length.bed /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_filter4length_upstream100.bed
</code></pre>
<p>nucpeaksand100up_nuc.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=nucpeaksand100up
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=nucpeaksand100up.out
#SBATCH --error=nucpeaksand100up.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

bedtools nuc -seq -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron_filter4length_upstream100.bed  &gt; /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron_filter4length_upstream100_seq.bed

bedtools nuc -seq -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_filter4length_upstream100.bed  &gt; /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_filter4length_upstream100_seq.bed


</code></pre>
<p>DistPAS2Sig_nuclear.py</p>
<pre class="bash"><code>def main(Insite, out):
  sigsite=[Insite]
  
  inBed=open(&quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_filter4length_upstream100_seq.bed&quot;, &quot;r&quot;)
  outRes=open(out, &quot;w&quot;)
  
  #function for reverse compliments  
  
  def ReverseComplement1(seq):
      seq_dict = {&#39;A&#39;:&#39;T&#39;,&#39;T&#39;:&#39;A&#39;,&#39;G&#39;:&#39;C&#39;,&#39;C&#39;:&#39;G&#39;, &#39;a&#39;:&#39;t&#39;, &#39;t&#39;:&#39;a&#39;, &#39;g&#39;:&#39;c&#39;, &#39;c&#39;:&#39;g&#39;}
      bases=[seq_dict[base] for base in seq]
      bases=reversed(bases)
      return(&quot;&quot;.join(bases))
  
  
  #reverse comp each signal site 
  sigsite_revComp=[]
  for i in sigsite:
      sigsite_revComp.append(ReverseComplement1(i))
      
  #want a dictionary for each of the sites and its reverse compliment:
  sigsites_dic={}
  for i in range(len(sigsite)):
      sigsites_dic[sigsite[i]]=sigsite_revComp[i]    
      
  
  #function to get occurance: takes in sig site and sequence (give it the correct stranded stuff)
  
  #make 2 of these, this is for the pos strand
  def getOccurance(sigsite, seq):
      if sigsite in seq:
          length=len(seq)
          pos= seq.rfind(sigsite)
          posF=length-pos
          return(posF)
      else:
          return(-9)
          
  #negative strand occurance function:
  
  def getOccurance_neg(sigsite, seq):
      sigsite=sigsites_dic[sigsite]
      if sigsite in seq:
          pos= seq.find(sigsite)
          return(pos + 6)
      else:
          return(-9)
  
  #i can only addpend the value if the function does not return -9
      
  
  #function i can run on each signal site  
  
  
  #loop through peaks and check for every site, first ask stand and do the rev  
  def loop41site(site):
      resList=[]
      for ln in inBed:
          strand=ln.split()[5]
          seq= ln.split()[15]
          if strand == &quot;+&quot;:
              loc= getOccurance(site, seq)
              if loc !=-9:
                  resList.append(loc)
          else: 
              loc=getOccurance_neg(site,seq)
              if loc !=-9:
                  resList.append(loc)
      return(resList)
          
  
  #run this for each sig site
  res_dic={}
  for i in sigsite:
      res_dic[i]=[]
  for i in sigsite:
      reslist=loop41site(i)
      res_dic[i]=reslist
  
  
  outRes.write(&quot;%s\n&quot;%(sigsite[0]))
  for i in reslist:
      outRes.write(&quot;%d\n&quot;%(i))
      
  
  
  outRes.close()
      
    
if __name__ == &quot;__main__&quot;:
    import sys
    Site_in = sys.argv[1]
    outFile= &quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/Loc_%s_Distance2end_nuclear.txt&quot;%(Site_in)
    main(Site_in, outFile)
</code></pre>
<p>DistPAS2Sig_nucIntron.py</p>
<pre class="bash"><code>def main(Insite, out):
  sigsite=[Insite]
  
  inBed=open(&quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron_filter4length_upstream100_seq.bed&quot;, &quot;r&quot;)
  outRes=open(out, &quot;w&quot;)
  
  #function for reverse compliments  
  
  def ReverseComplement1(seq):
      seq_dict = {&#39;A&#39;:&#39;T&#39;,&#39;T&#39;:&#39;A&#39;,&#39;G&#39;:&#39;C&#39;,&#39;C&#39;:&#39;G&#39;, &#39;a&#39;:&#39;t&#39;, &#39;t&#39;:&#39;a&#39;, &#39;g&#39;:&#39;c&#39;, &#39;c&#39;:&#39;g&#39;}
      bases=[seq_dict[base] for base in seq]
      bases=reversed(bases)
      return(&quot;&quot;.join(bases))
  
  
  #reverse comp each signal site 
  sigsite_revComp=[]
  for i in sigsite:
      sigsite_revComp.append(ReverseComplement1(i))
      
  #want a dictionary for each of the sites and its reverse compliment:
  sigsites_dic={}
  for i in range(len(sigsite)):
      sigsites_dic[sigsite[i]]=sigsite_revComp[i]    
      
  
  #function to get occurance: takes in sig site and sequence (give it the correct stranded stuff)
  
  #make 2 of these, this is for the pos strand
  def getOccurance(sigsite, seq):
      if sigsite in seq:
          length=len(seq)
          pos= seq.rfind(sigsite)
          posF=length-pos
          return(posF)
      else:
          return(-9)
          
  #negative strand occurance function:
  
  def getOccurance_neg(sigsite, seq):
      sigsite=sigsites_dic[sigsite]
      if sigsite in seq:
          pos= seq.find(sigsite)
          return(pos + 6)
      else:
          return(-9)
  
  #i can only addpend the value if the function does not return -9
      
  
  #function i can run on each signal site  
  
  
  #loop through peaks and check for every site, first ask stand and do the rev  
  def loop41site(site):
      resList=[]
      for ln in inBed:
          strand=ln.split()[5]
          seq= ln.split()[15]
          if strand == &quot;+&quot;:
              loc= getOccurance(site, seq)
              if loc !=-9:
                  resList.append(loc)
          else: 
              loc=getOccurance_neg(site,seq)
              if loc !=-9:
                  resList.append(loc)
      return(resList)
          
  
  #run this for each sig site
  res_dic={}
  for i in sigsite:
      res_dic[i]=[]
  for i in sigsite:
      reslist=loop41site(i)
      res_dic[i]=reslist
  
  
  outRes.write(&quot;%s\n&quot;%(sigsite[0]))
  for i in reslist:
      outRes.write(&quot;%d\n&quot;%(i))
      
  
  
  outRes.close()
      
    
if __name__ == &quot;__main__&quot;:
    import sys
    Site_in = sys.argv[1]
    outFile= &quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/Loc_%s_Distance2end_nuclearIntron.txt&quot;%(Site_in)
    main(Site_in, outFile)
</code></pre>
<p>Run both of these:<br />
run_DistPAS2Sig_nuc.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_DistPAS2Sig_nuc
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=un_DistPAS2Sig_nuc.out
#SBATCH --error=un_DistPAS2Sig_nuc.err
#SBATCH --partition=broadwl
#SBATCH --mem=16G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env


python DistPAS2Sig_nucIntron.py AATAAA
python DistPAS2Sig_nucIntron.py ATTAAA
python DistPAS2Sig_nucIntron.py AGTAAA
python DistPAS2Sig_nucIntron.py TATAAA
python DistPAS2Sig_nucIntron.py CATAAA
python DistPAS2Sig_nucIntron.py GATAAA
python DistPAS2Sig_nucIntron.py AATATA
python DistPAS2Sig_nucIntron.py AATACA
python DistPAS2Sig_nucIntron.py AATAGA
python DistPAS2Sig_nucIntron.py AAAAAG
python DistPAS2Sig_nucIntron.py ACTAAA


python DistPAS2Sig_nuclear.py AATAAA
python DistPAS2Sig_nuclear.py ATTAAA
python DistPAS2Sig_nuclear.py AGTAAA
python DistPAS2Sig_nuclear.py TATAAA
python DistPAS2Sig_nuclear.py CATAAA
python DistPAS2Sig_nuclear.py GATAAA
python DistPAS2Sig_nuclear.py AATATA
python DistPAS2Sig_nuclear.py AATACA
python DistPAS2Sig_nuclear.py AATAGA
python DistPAS2Sig_nuclear.py AAAAAG
python DistPAS2Sig_nuclear.py ACTAAA
</code></pre>
<pre class="r"><code>Loc_AATAAA_Nuc= read.table(&quot;../data/Signal_Loc/Loc_AATAAA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AATAAA&quot;)


Loc_AAAAAG_Nuc= read.table(&quot;../data/Signal_Loc/Loc_AAAAAG_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AAAAAG&quot;)


Loc_AATACA_Nuc= read.table(&quot;../data/Signal_Loc/Loc_AATACA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AATACA&quot;)


Loc_AATAGA_Nuc= read.table(&quot;../data/Signal_Loc/Loc_AATAGA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AATAGA&quot;)


Loc_AATATA_Nuc =read.table(&quot;../data/Signal_Loc/Loc_AATATA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AATATA&quot;)


Loc_ACTAAA_Nuc= read.table(&quot;../data/Signal_Loc/Loc_ACTAAA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;ACTAAA&quot;)


Loc_AGTAAA_Nuc= read.table(&quot;../data/Signal_Loc/Loc_AGTAAA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;AGTAAA&quot;)



Loc_ATTAAA_Nuc=read.table(&quot;../data/Signal_Loc/Loc_ATTAAA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;ATTAAA&quot;)

Loc_CATAAA_Nuc= read.table(&quot;../data/Signal_Loc/Loc_CATAAA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;CATAAA&quot;)

Loc_GATAAA_Nuc= read.table(&quot;../data/Signal_Loc/Loc_GATAAA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;GATAAA&quot;)

Loc_TATAAA_Nuc= read.table(&quot;../data/Signal_Loc/Loc_TATAAA_Distance2end_nuclear.txt&quot;, header=T) %&gt;% mutate(Site=&quot;TATAAA&quot;)



dist_Loc_Nuc= ggplot(Loc_AATAAA_Nuc, aes(x=AATAAA)) + 
  geom_histogram(bins=100, fill=&quot;red&quot;, alpha=.3) + 
  labs(title=&quot;Distance from PAS in Nuclear Used peaks&quot;, x=&quot;Number of PAS&quot;, y=&quot;N basepair from PAS&quot;) +
  geom_histogram(bins=100,data=Loc_AAAAAG_Nuc, aes(x=AAAAAG), fill=&quot;orange&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATACA_Nuc, aes(x=AATACA), fill=&quot;yellow&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATAGA_Nuc, aes(x=AATAGA), fill=&quot;green&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATATA_Nuc, aes(x=AATATA), fill=&quot;blue&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_ACTAAA_Nuc, aes(x=ACTAAA), fill=&quot;purple&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AGTAAA_Nuc, aes(x=AGTAAA), fill=&quot;firebrick3&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AGTAAA_Nuc, aes(x=AGTAAA), fill=&quot;darksalmon&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_CATAAA_Nuc, aes(x=CATAAA), fill=&quot;darkslategray&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_GATAAA_Nuc, aes(x=GATAAA), fill=&quot;deeppink1&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_TATAAA_Nuc, aes(x=TATAAA), fill=&quot;lightcyan1&quot;, alpha=.3) 
  
  
dist_Loc_Nuc</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Make a long dataframe for all of this to make it easier to manipulate.</p>
<pre class="r"><code>colnames(Loc_AATAAA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AAAAAG_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AATACA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AATAGA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AATATA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_ACTAAA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_AGTAAA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_CATAAA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_ATTAAA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_GATAAA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)
colnames(Loc_TATAAA_Nuc)=c(&quot;Count&quot;, &quot;Site&quot;)



AllsiteDF_Nuc=as.data.frame(rbind(Loc_AATAAA_Nuc,Loc_AAAAAG_Nuc,Loc_AATACA_Nuc,Loc_AATAGA_Nuc,Loc_AATATA_Nuc,Loc_ACTAAA_Nuc,Loc_AGTAAA_Nuc,Loc_ATTAAA_Nuc, Loc_GATAAA_Nuc,Loc_TATAAA_Nuc,Loc_CATAAA_Nuc)) %&gt;% mutate(NegCount=-1*as.integer(as.character(Count)), Cononical=ifelse(Site==&quot;AATAAA&quot;, &quot;Yes&quot;,&quot;No&quot;))


 AllsiteDF_to100_Nuc= AllsiteDF_Nuc %&gt;% filter(Count &lt; 100)</code></pre>
<p>plot:</p>
<pre class="r"><code>ggplot(AllsiteDF_to100_Nuc, aes(group=Site, x=NegCount, fill=Site)) + geom_histogram(position=&quot;stack&quot;,bins=50 ) + labs(x=&quot;Distance from PAS&quot;, y=&quot;N annotated Sites&quot;, title=&quot;Location of annotated signal sites- Used more in Nuclear&quot;)  </code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/threeprimeseq/blob/ca1a9f465d3821e9ca286c5f5b2feba428df179f/docs/figure/SignalSiteLoc.Rmd/unnamed-chunk-36-1.png" target="_blank">ca1a9f4</a>
</td>
<td>
Briana Mittleman
</td>
<td>
2019-03-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Do this as proportion:</p>
<pre class="r"><code>AllsiteDF_to100_prop_Nuc=AllsiteDF_to100_Nuc %&gt;% group_by(Site,NegCount) %&gt;% summarise(CountperPos=n()) %&gt;% mutate(TotCount=sum(CountperPos),prop=CountperPos/TotCount)

#%&gt;% ungroup() %&gt;% group_by(Site) %&gt;% mutate(nType=sum(Count), prop=CountperPos/nType)</code></pre>
<pre class="r"><code>annoationPAS_Nucpeak=ggplot(AllsiteDF_to100_prop_Nuc, aes(fill=Site, y=prop, x=NegCount)) + geom_histogram(stat=&quot;identity&quot;) + facet_wrap(~Site) + labs(x=&quot;Distance from PAS&quot;, y=&quot;Proportion of Signal Site&quot;, title=&quot;Location of annotated signal sites for PAS used more in Nuclear&quot;)</code></pre>
<pre><code>Warning: Ignoring unknown parameters: binwidth, bins, pad</code></pre>
<pre class="r"><code>annoationPAS_Nucpeak</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(annoationPAS_Nucpeak, file=&quot;../output/plots/annoationPAS_NuclearpeaksPropLocFacet.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<pre class="r"><code>annoationPAS_NucpeakProphist=ggplot(AllsiteDF_to100_prop_Nuc, aes(fill=Site, by=Site, y=prop, x=NegCount)) + geom_histogram(stat=&quot;identity&quot;, position=&quot;stack&quot;) + labs(x=&quot;Distance from PAS&quot;, y=&quot;Proportion of Signal Site&quot;, title=&quot;Location of annotated signal sites used more in Nuc&quot;)</code></pre>
<pre><code>Warning: Ignoring unknown parameters: binwidth, bins, pad</code></pre>
<pre class="r"><code>annoationPAS_NucpeakProphist</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(annoationPAS_NucpeakProphist, file=&quot;../output/plots/annoationPAS_NucpeakPropLocStackHist.png&quot;)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<pre class="r"><code>Loc_AATAAA_Int= read.table(&quot;../data/Signal_Loc/Loc_AATAAA_Distance2end_nuclearIntron.txt&quot;, header=T)


Loc_AAAAAG_Int= read.table(&quot;../data/Signal_Loc/Loc_AAAAAG_Distance2end_nuclearIntron.txt&quot;, header=T)


Loc_AATACA_Int= read.table(&quot;../data/Signal_Loc/Loc_AATACA_Distance2end_nuclearIntron.txt&quot;, header=T)


Loc_AATAGA_Int= read.table(&quot;../data/Signal_Loc/Loc_AATAGA_Distance2end_nuclearIntron.txt&quot;, header=T)


Loc_AATATA_Int =read.table(&quot;../data/Signal_Loc/Loc_AATATA_Distance2end_nuclearIntron.txt&quot;, header=T)


Loc_ACTAAA_Int= read.table(&quot;../data/Signal_Loc/Loc_ACTAAA_Distance2end_nuclearIntron.txt&quot;, header=T)


Loc_AGTAAA_Int= read.table(&quot;../data/Signal_Loc/Loc_AGTAAA_Distance2end_nuclearIntron.txt&quot;, header=T)



Loc_ATTAAA_Int=read.table(&quot;../data/Signal_Loc/Loc_ATTAAA_Distance2end_nuclearIntron.txt&quot;, header=T)

Loc_CATAAA_Int= read.table(&quot;../data/Signal_Loc/Loc_CATAAA_Distance2end_nuclearIntron.txt&quot;, header=T)

Loc_GATAAA_Int= read.table(&quot;../data/Signal_Loc/Loc_GATAAA_Distance2end_nuclearIntron.txt&quot;, header=T)

Loc_TATAAA_Int= read.table(&quot;../data/Signal_Loc/Loc_TATAAA_Distance2end_nuclearIntron.txt&quot;, header=T)



dist_Loc_Int= ggplot(Loc_AATAAA_Int, aes(x=AATAAA)) + 
  geom_histogram(bins=100, fill=&quot;red&quot;, alpha=.3) + 
  labs(title=&quot;Distance from PAS in Intronic Nuclear Used peaks&quot;, x=&quot;Number of PAS&quot;, y=&quot;N basepair from PAS&quot;) +
  geom_histogram(bins=100,data=Loc_AAAAAG_Int, aes(x=AAAAAG), fill=&quot;orange&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATACA_Int, aes(x=AATACA), fill=&quot;yellow&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATAGA_Int, aes(x=AATAGA), fill=&quot;green&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATATA_Int, aes(x=AATATA), fill=&quot;blue&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_ACTAAA_Int, aes(x=ACTAAA), fill=&quot;purple&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AGTAAA_Int, aes(x=AGTAAA), fill=&quot;firebrick3&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AGTAAA_Int, aes(x=AGTAAA), fill=&quot;darksalmon&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_CATAAA_Int, aes(x=CATAAA), fill=&quot;darkslategray&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_GATAAA_Int, aes(x=GATAAA), fill=&quot;deeppink1&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_TATAAA_Int, aes(x=TATAAA), fill=&quot;lightcyan1&quot;, alpha=.3) 
  
  
dist_Loc_Int</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="all-intronic-peaks" class="section level3">
<h3>All intronic peaks:</h3>
<p>*/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_INTRON.bed (17854)</p>
<p>filterPeaks100length_Intron.py</p>
<pre class="bash"><code>peaks=open(&quot;/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_INTRON.bed&quot;, &quot;r&quot;)

outPeaks=open(&quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_INTRON_filter4length.bed&quot;, &quot;w&quot;)

nNotOk=0
for ln in peaks:
    start= int(ln.split()[1])
    end=int(ln.split()[2])
    length=end - start 
    if length &lt;= 100:
        outPeaks.write(ln)
    else:
        nNotOk +=1
        
print(nNotOk)
outPeaks.close()
      </code></pre>
<p>12250 peaks</p>
<pre class="bash"><code>python Upstream100Bases_filteredpeaks.py /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_INTRON_filter4length.bed /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_INTRON_filter4length_upstream100.bed</code></pre>
<p>nucpeaksand100up_intron.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=nucpeaksand100up_intron
#SBATCH --account=pi-yangili1
#SBATCH --time=8:00:00
#SBATCH --output=nucpeaksand100up_intron.out
#SBATCH --error=nucpeaksand100up_intron.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

bedtools nuc -seq -fi /project2/gilad/briana/genome_anotation_data/genome/Homo_sapiens.GRCh37.75.dna_sm.all.fa -bed /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_INTRON_filter4length_upstream100.bed &gt; /project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_INTRON_filter4length_upstream100_seq.bed

</code></pre>
<p>DistPAS2Sig_Intron.py</p>
<pre class="bash"><code>def main(Insite, out):
  sigsite=[Insite]
  
  inBed=open(&quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/APAPeaks_5percCov_fixedStrand_INTRON_filter4length_upstream100_seq.bed&quot;, &quot;r&quot;)
  outRes=open(out, &quot;w&quot;)
  
  #function for reverse compliments  
  
  def ReverseComplement1(seq):
      seq_dict = {&#39;A&#39;:&#39;T&#39;,&#39;T&#39;:&#39;A&#39;,&#39;G&#39;:&#39;C&#39;,&#39;C&#39;:&#39;G&#39;, &#39;a&#39;:&#39;t&#39;, &#39;t&#39;:&#39;a&#39;, &#39;g&#39;:&#39;c&#39;, &#39;c&#39;:&#39;g&#39;}
      bases=[seq_dict[base] for base in seq]
      bases=reversed(bases)
      return(&quot;&quot;.join(bases))
  
  
  #reverse comp each signal site 
  sigsite_revComp=[]
  for i in sigsite:
      sigsite_revComp.append(ReverseComplement1(i))
      
  #want a dictionary for each of the sites and its reverse compliment:
  sigsites_dic={}
  for i in range(len(sigsite)):
      sigsites_dic[sigsite[i]]=sigsite_revComp[i]    
      
  
  #function to get occurance: takes in sig site and sequence (give it the correct stranded stuff)
  
  #make 2 of these, this is for the pos strand
  def getOccurance(sigsite, seq):
      if sigsite in seq:
          length=len(seq)
          pos= seq.rfind(sigsite)
          posF=length-pos
          return(posF)
      else:
          return(-9)
          
  #negative strand occurance function:
  
  def getOccurance_neg(sigsite, seq):
      sigsite=sigsites_dic[sigsite]
      if sigsite in seq:
          pos= seq.find(sigsite)
          return(pos + 6)
      else:
          return(-9)
  
  #i can only addpend the value if the function does not return -9
      
  
  #function i can run on each signal site  
  
  
  #loop through peaks and check for every site, first ask stand and do the rev  
  def loop41site(site):
      resList=[]
      for ln in inBed:
          strand=ln.split()[5]
          seq= ln.split()[15]
          if strand == &quot;+&quot;:
              loc= getOccurance(site, seq)
              if loc !=-9:
                  resList.append(loc)
          else: 
              loc=getOccurance_neg(site,seq)
              if loc !=-9:
                  resList.append(loc)
      return(resList)
          
  
  #run this for each sig site
  res_dic={}
  for i in sigsite:
      res_dic[i]=[]
  for i in sigsite:
      reslist=loop41site(i)
      res_dic[i]=reslist
  
  
  outRes.write(&quot;%s\n&quot;%(sigsite[0]))
  for i in reslist:
      outRes.write(&quot;%d\n&quot;%(i))
      
  
  
  outRes.close()
      
    
if __name__ == &quot;__main__&quot;:
    import sys
    Site_in = sys.argv[1]
    outFile= &quot;/project2/gilad/briana/threeprimeseq/data/Signal_Loc/Loc_%s_Distance2end_Intronic.txt&quot;%(Site_in)
    main(Site_in, outFile)
</code></pre>
<p>run_DistPAS2Sig_Intron.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_DistPAS2Sig_Intronc
#SBATCH --account=pi-yangili1
#SBATCH --time=36:00:00
#SBATCH --output=run_DistPAS2Sig_Intron.out
#SBATCH --error=run_DistPAS2Sig_Intron.err
#SBATCH --partition=broadwl
#SBATCH --mem=16G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env


python DistPAS2Sig_Intron.py AATAAA
python DistPAS2Sig_Intron.py ATTAAA
python DistPAS2Sig_Intron.py AGTAAA
python DistPAS2Sig_Intron.py TATAAA
python DistPAS2Sig_Intron.py CATAAA
python DistPAS2Sig_Intron.py GATAAA
python DistPAS2Sig_Intron.py AATATA
python DistPAS2Sig_Intron.py AATACA
python DistPAS2Sig_Intron.py AATAGA
python DistPAS2Sig_Intron.py AAAAAG
python DistPAS2Sig_Intron.py ACTAAA</code></pre>
<pre class="r"><code>Loc_AATAAA_AllInt= read.table(&quot;../data/Signal_Loc/Loc_AATAAA_Distance2end_Intronic.txt&quot;, header=T)


Loc_AAAAAG_AllInt= read.table(&quot;../data/Signal_Loc/Loc_AAAAAG_Distance2end_Intronic.txt&quot;, header=T)


Loc_AATACA_AllInt= read.table(&quot;../data/Signal_Loc/Loc_AATACA_Distance2end_Intronic.txt&quot;, header=T)


Loc_AATAGA_AllInt= read.table(&quot;../data/Signal_Loc/Loc_AATAGA_Distance2end_Intronic.txt&quot;, header=T)


Loc_AATATA_AllInt =read.table(&quot;../data/Signal_Loc/Loc_AATATA_Distance2end_Intronic.txt&quot;, header=T)


Loc_ACTAAA_AllInt= read.table(&quot;../data/Signal_Loc/Loc_ACTAAA_Distance2end_Intronic.txt&quot;, header=T)


Loc_AGTAAA_AllInt= read.table(&quot;../data/Signal_Loc/Loc_AGTAAA_Distance2end_Intronic.txt&quot;, header=T)



Loc_ATTAAA_AllInt=read.table(&quot;../data/Signal_Loc/Loc_ATTAAA_Distance2end_Intronic.txt&quot;, header=T)

Loc_CATAAA_AllInt= read.table(&quot;../data/Signal_Loc/Loc_CATAAA_Distance2end_Intronic.txt&quot;, header=T)

Loc_GATAAA_AllInt= read.table(&quot;../data/Signal_Loc/Loc_GATAAA_Distance2end_Intronic.txt&quot;, header=T)

Loc_TATAAA_AllInt= read.table(&quot;../data/Signal_Loc/Loc_TATAAA_Distance2end_Intronic.txt&quot;, header=T)



dist_Loc_AllInt= ggplot(Loc_AATAAA_AllInt, aes(x=AATAAA)) + 
  geom_histogram(bins=100, fill=&quot;red&quot;, alpha=.3) + 
  labs(title=&quot;Distance from PAS in Intronic peaks&quot;, x=&quot;Number of PAS&quot;, y=&quot;N basepair from PAS&quot;) +
  geom_histogram(bins=100,data=Loc_AAAAAG_AllInt, aes(x=AAAAAG), fill=&quot;orange&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATACA_AllInt, aes(x=AATACA), fill=&quot;yellow&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATAGA_AllInt, aes(x=AATAGA), fill=&quot;green&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AATATA_AllInt, aes(x=AATATA), fill=&quot;blue&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_ACTAAA_AllInt, aes(x=ACTAAA), fill=&quot;purple&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AGTAAA_AllInt, aes(x=AGTAAA), fill=&quot;firebrick3&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_AGTAAA_AllInt, aes(x=AGTAAA), fill=&quot;darksalmon&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_CATAAA_AllInt, aes(x=CATAAA), fill=&quot;darkslategray&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_GATAAA_AllInt, aes(x=GATAAA), fill=&quot;deeppink1&quot;, alpha=.3) +
  geom_histogram(bins=100,data=Loc_TATAAA_AllInt, aes(x=TATAAA), fill=&quot;lightcyan1&quot;, alpha=.3)
  
  
dist_Loc_AllInt</code></pre>
<p><img src="figure/SignalSiteLoc.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      stats4    parallel  stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] BSgenome.Hsapiens.UCSC.hg19_1.4.0 genomation_1.14.0                
 [3] BSgenome_1.50.0                   rtracklayer_1.42.2               
 [5] GenomicRanges_1.34.0              GenomeInfoDb_1.18.2              
 [7] Biostrings_2.50.2                 XVector_0.22.0                   
 [9] IRanges_2.16.0                    S4Vectors_0.20.1                 
[11] BiocGenerics_0.28.0               forcats_0.4.0                    
[13] stringr_1.4.0                     dplyr_0.8.0.1                    
[15] purrr_0.3.1                       readr_1.3.1                      
[17] tidyr_0.8.3                       tibble_2.0.1                     
[19] ggplot2_3.1.0                     tidyverse_1.2.1                  
[21] workflowr_1.2.0                  

loaded via a namespace (and not attached):
 [1] nlme_3.1-137                bitops_1.0-6               
 [3] matrixStats_0.54.0          fs_1.2.6                   
 [5] lubridate_1.7.4             httr_1.4.0                 
 [7] rprojroot_1.3-2             tools_3.5.1                
 [9] backports_1.1.3             R6_2.4.0                   
[11] KernSmooth_2.23-15          lazyeval_0.2.1             
[13] colorspace_1.4-0            seqPattern_1.14.0          
[15] withr_2.1.2                 tidyselect_0.2.5           
[17] compiler_3.5.1              git2r_0.24.0               
[19] cli_1.0.1                   rvest_0.3.2                
[21] Biobase_2.42.0              xml2_1.2.0                 
[23] DelayedArray_0.8.0          labeling_0.3               
[25] scales_1.0.0                digest_0.6.18              
[27] Rsamtools_1.34.1            rmarkdown_1.11             
[29] pkgconfig_2.0.2             htmltools_0.3.6            
[31] plotrix_3.7-4               rlang_0.3.1                
[33] readxl_1.3.0                rstudioapi_0.9.0           
[35] impute_1.56.0               generics_0.0.2             
[37] jsonlite_1.6                BiocParallel_1.16.6        
[39] RCurl_1.95-4.12             magrittr_1.5               
[41] GenomeInfoDbData_1.2.0      Matrix_1.2-15              
[43] Rcpp_1.0.0                  munsell_0.5.0              
[45] reticulate_1.11.1           stringi_1.3.1              
[47] whisker_0.3-2               yaml_2.2.0                 
[49] SummarizedExperiment_1.12.0 zlibbioc_1.28.0            
[51] plyr_1.8.4                  crayon_1.3.4               
[53] lattice_0.20-38             haven_2.1.0                
[55] hms_0.4.2                   knitr_1.21                 
[57] pillar_1.3.1                reshape2_1.4.3             
[59] XML_3.98-1.19               glue_1.3.0                 
[61] evaluate_0.13               data.table_1.12.0          
[63] modelr_0.1.4                cellranger_1.1.0           
[65] gtable_0.2.0                assertthat_0.2.0           
[67] xfun_0.5                    gridBase_0.4-7             
[69] broom_0.5.1                 GenomicAlignments_1.18.1   </code></pre>
</div>
</div>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
