<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Top APA QTL in other phenotypes</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Top APA QTL in other phenotypes</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>10/8/2018</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-10-11</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/1d97af50fc177f96c533cb8ae1db8ac552622794" target="_blank">1d97af5</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  KalistoAbundance18486.txt
    Untracked:  analysis/genometrack_figs.Rmd
    Untracked:  analysis/ncbiRefSeq_sm.sort.mRNA.bed
    Untracked:  analysis/snake.config.notes.Rmd
    Untracked:  analysis/verifyBAM.Rmd
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/NuclearApaQTLs.txt
    Untracked:  data/RNAkalisto/
    Untracked:  data/TotalApaQTLs.txt
    Untracked:  data/Totalpeaks_filtered_clean.bed
    Untracked:  data/YL-SP-18486-T-combined-genecov.txt
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/apaExamp/
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/clean_peaks/
    Untracked:  data/comb_map_stats.csv
    Untracked:  data/comb_map_stats.xlsx
    Untracked:  data/comb_map_stats_39ind.csv
    Untracked:  data/combined_reads_mapped_three_prime_seq.csv
    Untracked:  data/ensemble_to_genename.txt
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.bed
    Untracked:  data/filtered_APApeaks_merged_allchrom_refseqTrans.closest2End.noties.bed
    Untracked:  data/first50lines_closest.txt
    Untracked:  data/gencov.test.csv
    Untracked:  data/gencov.test.txt
    Untracked:  data/gencov_zero.test.csv
    Untracked:  data/gencov_zero.test.txt
    Untracked:  data/gene_cov/
    Untracked:  data/joined
    Untracked:  data/leafcutter/
    Untracked:  data/merged_combined_YL-SP-threeprimeseq.bg
    Untracked:  data/mol_overlap/
    Untracked:  data/nom_QTL/
    Untracked:  data/nom_QTL_opp/
    Untracked:  data/nom_QTL_trans/
    Untracked:  data/nuc6up/
    Untracked:  data/other_qtls/
    Untracked:  data/peakPerRefSeqGene/
    Untracked:  data/perm_QTL/
    Untracked:  data/perm_QTL_opp/
    Untracked:  data/perm_QTL_trans/
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  data/temp.file1
    Untracked:  data/temp.file2
    Untracked:  data/temp.gencov.test.txt
    Untracked:  data/temp.gencov_zero.test.txt
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/28ind.peak.explore.Rmd
    Modified:   analysis/39indQC.Rmd
    Modified:   analysis/PeakToGeneAssignment.Rmd
    Modified:   analysis/cleanupdtseq.internalpriming.Rmd
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/diff_iso_pipeline.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/overlapMolQTL.Rmd
    Modified:   analysis/overlap_qtls.Rmd
    Modified:   analysis/peakOverlap_oppstrand.Rmd
    Modified:   analysis/pheno.leaf.comb.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   code/Snakefile

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/1d97af50fc177f96c533cb8ae1db8ac552622794/analysis/swarmPlots_QTLs.Rmd" target="_blank">1d97af5</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-11
</td>
<td style="text-align:left;">
add examples
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/8211a07a3567d273d979b7fc94d4f362d6e217b4/docs/swarmPlots_QTLs.html" target="_blank">8211a07</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-11
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/9b23ee6968213faab18a363e8a9dea9039145ab3/analysis/swarmPlots_QTLs.Rmd" target="_blank">9b23ee6</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-11
</td>
<td style="text-align:left;">
add example plots
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/ad6a5bf0c314ec96c059415f1c6199ba1006f788/docs/swarmPlots_QTLs.html" target="_blank">ad6a5bf</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-11
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/373d3514820c8ca0a51e1445875468cf452f0f35/analysis/swarmPlots_QTLs.Rmd" target="_blank">373d351</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-11
</td>
<td style="text-align:left;">
add pheno code- working
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/e73be701f5c8832eb0aaec4c5a4d0fe3bb508b0a/docs/swarmPlots_QTLs.html" target="_blank">e73be70</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-09
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/2f5f071473a18c27ff4556ea72dd9356459e6cdf/analysis/swarmPlots_QTLs.Rmd" target="_blank">2f5f071</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-09
</td>
<td style="text-align:left;">
add pheno code- not working yet
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/b6d5c1987a8865f38762287c614c28a1453c438a/docs/swarmPlots_QTLs.html" target="_blank">b6d5c19</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-08
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/cdec3c139e3a8ff6a3868ec8acfa546212b5b176/analysis/swarmPlots_QTLs.Rmd" target="_blank">cdec3c1</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-08
</td>
<td style="text-align:left;">
change colors
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/077ed603f855c9fd95c9e77bb1008695e8ba3891/docs/swarmPlots_QTLs.html" target="_blank">077ed60</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-08
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/50c8b7688449816b06aaa21cc7dea3c1d5e36a2e/analysis/swarmPlots_QTLs.Rmd" target="_blank">50c8b76</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-08
</td>
<td style="text-align:left;">
plots for EIF2A in mult phenos
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="upload-data" class="section level2">
<h2>Upload Data:</h2>
<p>Library</p>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.1.1
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(reshape2)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────────────────────────────────────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.0.0     ✔ purrr   0.2.5
✔ tibble  1.4.2     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.3.1
✔ readr   1.1.1     ✔ forcats 0.3.0</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(VennDiagram)</code></pre>
<pre><code>Loading required package: grid</code></pre>
<pre><code>Loading required package: futile.logger</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre><code>The following objects are masked from &#39;package:reshape2&#39;:

    dcast, melt</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<p>Permuted Results from APA:</p>
<pre class="r"><code>nuclearAPA=read.table(&quot;../data/perm_QTL_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_transcript_permResBH.txt&quot;, stringsAsFactors = F, header = T)
totalAPA=read.table(&quot;../data/perm_QTL_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_transcript_permResBH.txt&quot;, stringsAsFactors = F, header=T)  </code></pre>
<p>I want to use a buzz swarm plot to plot peak usage for some of the top QTLs. I can use the examples I gave Tony.</p>
<p>Nuclear:<br />
* <strong>peak305794, sid: 7:128635754</strong></p>
<ul>
<li><strong>peak: 164036, sid: 2:3502035</strong></li>
</ul>
<p>Total:</p>
<ul>
<li><p><strong>Peak: peak228606, SID 3:150302010</strong></p></li>
<li><p>Peak: peak152751, SID 19:4236475</p></li>
</ul>
<p>I need to pull out the genotypes for each snp and the corresponding phenotype. I want to make a python script that I can give a snp and a peak and it will make a table with the genotypes and phenotypes for the necessary gene snp pair.</p>
</div>
<div id="example-peak-peak228606-sid-3150302010" class="section level2">
<h2>Example Peak: peak228606, SID 3:150302010</h2>
<pre class="r"><code>geno3_m=fread(&quot;../data/apaExamp/geno3_150302010.txt&quot;, header = T) %&gt;% dplyr::select(starts_with(&quot;NA&quot;)) %&gt;% t
geno3df= data.frame(geno3_m) %&gt;% separate(geno3_m, into=c(&quot;geno&quot;, &quot;dose&quot;, &quot;extra&quot;), sep=&quot;:&quot;) %&gt;% dplyr::select(dose) %&gt;% rownames_to_column(var=&quot;ind&quot;)
apaphen228606_m= fread(&quot;../data/apaExamp/Total.peak228606.txt&quot;, header = T) %&gt;% dplyr::select(starts_with(&quot;NA&quot;)) %&gt;% t
apaphen228606_df=data.frame(apaphen228606_m) %&gt;% rownames_to_column(var=&quot;ind&quot;)</code></pre>
<pre class="r"><code>toplotAPA=geno3df %&gt;% inner_join(apaphen228606_df, by=&quot;ind&quot;)
toplotAPA$dose= as.factor(toplotAPA$dose)
colnames(toplotAPA)= c(&quot;ind&quot;, &quot;Genotype&quot;, &quot;APA&quot;)
EIF2A_APAex=ggplot(toplotAPA, aes(y=APA, x=Genotype, by=Genotype, fill=Genotype)) + geom_boxplot() + geom_jitter() + labs(y=&quot;APA phenotype&quot;, title=&quot;Total APA: Peak 228606, EIF2A&quot;) + scale_fill_brewer(palette=&quot;YlOrRd&quot;)
ggsave(&quot;../output/plots/EIF2a_APA.png&quot;, EIF2A_APAex)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<p>This is in the gene EIF2A, I need to find this in the eQTL data. The ensg id for this gene is ENSG00000144895.</p>
<pre class="r"><code>RNAseqEIF2A_m=read.table(&quot;../data/apaExamp/RNAseq.phenoEIF2A.txt&quot;, header=T) %&gt;% dplyr::select(starts_with(&quot;NA&quot;)) %&gt;% t 
RNAseqEIF2A_df= data.frame(RNAseqEIF2A_m) %&gt;% rownames_to_column(&quot;ind&quot;)

plotRNA=geno3df %&gt;% inner_join(RNAseqEIF2A_df, by=&quot;ind&quot;)
plotRNA$dose= as.factor(plotRNA$dose)
colnames(plotRNA)= c(&quot;ind&quot;, &quot;Genotype&quot;, &quot;Expression&quot;)

EIF2A_RNAex=ggplot(plotRNA, aes(y=Expression, x=Genotype, by=Genotype, fill=Genotype)) + geom_boxplot() + geom_jitter() + labs(y=&quot;Normalized Expression&quot;, title=&quot;Gene Expression: EIF2A&quot;) + scale_fill_brewer(palette=&quot;YlGn&quot;)

ggsave(&quot;../output/plots/EIF2a_RNA.png&quot;, EIF2A_RNAex)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<p>Try this in protein:</p>
<pre class="r"><code>ProtEIF2A_m=read.table(&quot;../data/apaExamp/ProtEIF2A.txt&quot;, header=T) %&gt;% dplyr::select(starts_with(&quot;NA&quot;)) %&gt;% t 
ProtEIF2A_df= data.frame(ProtEIF2A_m) %&gt;% rownames_to_column(&quot;ind&quot;)

plotProt=geno3df %&gt;% inner_join(ProtEIF2A_df, by=&quot;ind&quot;)
plotProt$dose= as.factor(plotProt$dose)
colnames(plotProt)= c(&quot;ind&quot;, &quot;Genotype&quot;, &quot;Prot_level&quot;)

IF2A_Protex= ggplot(plotProt, aes(y=Prot_level, x=Genotype, by=Genotype, fill=Genotype)) + geom_boxplot() + geom_jitter() + labs(y=&quot;Normalized Protein Level&quot;, title=&quot;Protein Level: EIF2A&quot;) +scale_fill_brewer(palette=&quot;PuBu&quot;)

ggsave(&quot;../output/plots/EIF2a_Prot.png&quot;, IF2A_Protex)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<pre class="r"><code>multphenoEIF2a=plot_grid(EIF2A_APAex,IF2A_Protex,EIF2A_RNAex,nrow=1)
ggsave(&quot;../output/plots/EIF2a_multpheno.png&quot;, multphenoEIF2a, width=15, height=5)</code></pre>
<p>Do this with 4su 60:</p>
<p>have to remove the #</p>
<pre class="r"><code>su60_EIF2A_m=read.table(&quot;../data/apaExamp/Foursu60EIF2A.txt&quot;, header=T) %&gt;% dplyr::select(starts_with(&quot;NA&quot;)) %&gt;% t 
su60_EIF2A_df= data.frame(su60_EIF2A_m) %&gt;% rownames_to_column(&quot;ind&quot;)

plot4su60=geno3df %&gt;% inner_join(su60_EIF2A_df, by=&quot;ind&quot;)
plot4su60$dose= as.factor(plot4su60$dose)
colnames(plot4su60)= c(&quot;ind&quot;, &quot;Genotype&quot;, &quot;su60&quot;)

EIF2A_4su60ex=ggplot(plot4su60, aes(y=su60, x=Genotype, by=Genotype, fill=Genotype)) + geom_boxplot() + geom_jitter() + labs(y=&quot;4su60&quot;, title=&quot;4su 60min Value: EIF2A&quot;) + scale_fill_brewer(palette=&quot;RdPu&quot;) +  theme_classic()

ggsave(&quot;../output/plots/EIF2a_4su60.png&quot;, EIF2A_4su60ex)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<p>Geuvadis RNA</p>
<pre class="r"><code>rnaG_EIF2A_m=read.table(&quot;../data/apaExamp/RNA_GEU_EIF2A.txt&quot;, header=T) %&gt;% dplyr::select(starts_with(&quot;NA&quot;)) %&gt;% t 
rnaG_EIF2A_df= data.frame(rnaG_EIF2A_m) %&gt;% rownames_to_column(&quot;ind&quot;)

plotRNAg=geno3df %&gt;% inner_join(rnaG_EIF2A_df, by=&quot;ind&quot;)
plotRNAg$dose= as.factor(plotRNAg$dose)
colnames(plotRNAg)= c(&quot;ind&quot;, &quot;Genotype&quot;, &quot;RNAg&quot;)

EIF2A_RNAgex=ggplot(plotRNAg, aes(y=RNAg, x=Genotype, by=Genotype, fill=Genotype)) + geom_boxplot() + geom_jitter() + labs(y=&quot;RNA expression&quot;, title=&quot;RNA Expression Geuvadis: EIF2A&quot;) + scale_fill_brewer(palette=&quot;RdPu&quot;)

ggsave(&quot;../output/plots/EIF2a_RNAg.png&quot;, EIF2A_RNAgex)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
<p>Ribo:</p>
<pre class="r"><code>ribo_EIF2A_m=read.table(&quot;../data/apaExamp/Ribo_EIF2A.txt&quot;, header=T) %&gt;% dplyr::select(starts_with(&quot;NA&quot;)) %&gt;% t 
ribo_EIF2A_df= data.frame(ribo_EIF2A_m) %&gt;% rownames_to_column(&quot;ind&quot;)

plotrib=geno3df %&gt;% inner_join(ribo_EIF2A_df, by=&quot;ind&quot;)
plotrib$dose= as.factor(plotrib$dose)
colnames(plotrib)= c(&quot;ind&quot;, &quot;Genotype&quot;, &quot;Ribo&quot;)

EIF2A_riboex=ggplot(plotrib, aes(y=Ribo, x=Genotype, by=Genotype, fill=Genotype)) + geom_boxplot() + geom_jitter() + labs(y=&quot;RNA expression&quot;, title=&quot;Ribo Geuvadis: EIF2A&quot;) + scale_fill_brewer(palette=&quot;RdPu&quot;)

ggsave(&quot;../output/plots/EIF2a_Ribo.png&quot;, EIF2A_riboex)</code></pre>
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div id="create-a-script-to-make-the-relevent-files" class="section level2">
<h2>Create a script to make the relevent files</h2>
<p>Python script that take a chromosome, snp, peak#, fraction</p>
<p>createQTLsnpAPAPhenTable.py</p>
<pre class="bash"><code>def main(PhenFile, GenFile, outFile, snp, peak):
    fout=open(outFile, &quot;w&quot;)
    #Phen=open(PhenFile, &quot;r&quot;)
    Gen=open(GenFile, &quot;r&quot;)
    #get ind and pheno info
    def get_pheno():
      Phen=open(PhenFile, &quot;r&quot;)
      for num, ln in enumerate(Phen):
          if num == 0:
              indiv= ln.split()[4:]
          else:
              id=ln.split()[3].split(&quot;:&quot;)[3]
              peakID=id.split(&quot;_&quot;)[2]
              if peakID == peak:
                  pheno_list=ln.split()[4:]
                  pheno_data=list(zip(indiv,pheno_list))
                  print(pheno_data)
                  return(pheno_data)
    def get_geno():
      for num, lnG in enumerate(Gen):
          if num == 13:
              Ind_geno=lnG.split()[9:]
          if num &gt;= 14: 
              sid= lnG.split()[2]
              if sid == snp: 
                  gen_list=lnG.split()[9:]
                  allele1=[]
                  allele2=[]
                  for i in gen_list:
                      genotype=i.split(&quot;:&quot;)[0]
                      allele1.append(genotype.split(&quot;|&quot;)[0])
                      allele2.append(genotype.split(&quot;|&quot;)[1])
            #now i have my indiv., phen, allele 1, alle 2     
                  geno_data=list(zip(Ind_geno, allele1, allele2))
                  print(geno_data)
                  return(geno_data)

    phenotype=get_pheno()
    pheno_df=pd.DataFrame(data=phenotype,columns=[&quot;Ind&quot;, &quot;Pheno&quot;])
    genotype=get_geno()
    geno_df=pd.DataFrame(data=genotype, columns=[&quot;Ind&quot;, &quot;Allele1&quot;, &quot;Allele2&quot;])
    full_df=pd.merge(geno_df, pheno_df, how=&quot;inner&quot;, on=&quot;Ind&quot;)
    full_df.to_csv(fout, sep=&quot;\t&quot;, encoding=&#39;utf-8&#39;, index=False)
    fout.close()
    

if __name__ == &quot;__main__&quot;:
    import sys
    import pandas as pd
    chrom=sys.argv[1]
    snp = sys.argv[2]
    peak = sys.argv[3]
    fraction=sys.argv[4]
    
    PhenFile = &quot;/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript/filtered_APApeaks_merged_allchrom_refseqGenes.Transcript_sm_quant.%s.pheno_fixed.txt.gz.phen_chr%s&quot;%(fraction, chrom)
    GenFile= &quot;/project2/gilad/briana/YRI_geno_hg19/chr%s.dose.filt.vcf&quot;%(chrom)
    outFile = &quot;/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/qtlSNP_PeakAPA%s.%s%s.txt&quot;%(fraction, snp, peak)
    main(PhenFile, GenFile, outFile, snp, peak)
    </code></pre>
<p>Use the results to plot the nuclear pheno:</p>
<pre class="r"><code>EIF2a_APAnuc=read.table(&quot;../data/apaExamp/qtlSNP_PeakAPANuclear.3:150302010peak228606.txt&quot;, header=T, stringsAsFactors = F) %&gt;% mutate(Geno=Allele1 + Allele2)

EIF2a_APAnuc$Geno= as.factor(as.character(EIF2a_APAnuc$Geno))


ggplot(EIF2a_APAnuc, aes(y=Pheno, x=Geno, by=Geno, fill=Geno)) + geom_boxplot() + geom_jitter() + labs(y=&quot;APA Nuc Usage&quot;, title=&quot;APA nuc: EIF2A&quot;) + scale_fill_brewer(palette=&quot;RdPu&quot;)</code></pre>
<p><img src="figure/swarmPlots_QTLs.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-12-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/e73be701f5c8832eb0aaec4c5a4d0fe3bb508b0a/docs/figure/swarmPlots_QTLs.Rmd/unnamed-chunk-12-1.png" target="_blank">e73be70</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-09
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>This does the total and nuclear fraction of APA. I will do this for a snp and gene and get all of the other phenotypes. This will be similar other than changing the names of the genes and seperating the name for all but protein.</p>
<p>createQTLsnpMolPhenTable.py</p>
<pre class="bash"><code>def main(PhenFile, GenFile, outFile, snp, gene, molPhen):
    #genenames=open(&quot;/project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt&quot;, &quot;r&quot; )
    #for ln in genenames:
     #   geneName=ln.split()[1]
      #  if geneName == gene:
       #gene_ensg=ln.split()[0]
    gene_ensg=gene  
    fout=open(outFile, &quot;w&quot;)
    #Phen=open(PhenFile, &quot;r&quot;)
    Gen=open(GenFile, &quot;r&quot;)
    def getPheno(geneE=gene_ensg , mp=molPhen):
      pheno=open(PhenFile, &quot;r&quot;)
      #get ind and pheno info
      mp_use=mp[1:-1]
      if mp_use==&quot;prot&quot;:
        for num,ln in enumerate(pheno):
            if num == 0:
                indiv= ln.split()[4:]
            else:
                gene=ln.split()[3]
                if gene == str(geneE):
                    print(&quot;x&quot;)
                    pheno_list=ln.split()[4:]
                    pheno_data= list(zip(indiv, pheno_list))
                    return(pheno_data)
      else:
        for num,ln in enumerate(pheno):
            if num == 0:
                indiv= ln.split()[4:]
            else: 
                full_gene=ln.split()[3]
                gene= full_gene.split(&quot;.&quot;)[0]
                if gene == geneE:
                    print(gene)
                    pheno_list=ln.split()[4:]
                    pheno_data= list(zip(indiv, pheno_list))
                    return(pheno_data)
    def getGeno(geno, SNP):
      for num, lnG in enumerate(geno):
          if num == 13:
              Ind_geno=lnG.split()[9:]
          if num &gt;= 14: 
              sid= lnG.split()[2]
              if sid == SNP: 
                  gen_list=lnG.split()[9:]
                  allele1=[]
                  allele2=[]
                  for i in gen_list:
                      genotype=i.split(&quot;:&quot;)[0]
                      allele1.append(genotype.split(&quot;|&quot;)[0])
                      allele2.append(genotype.split(&quot;|&quot;)[1])
            #now i have my indiv., phen, allele 1, alle 2     
                  geno_data=list(zip(Ind_geno, allele1, allele2))
                  return(geno_data)
    
                      
                  
    phenotype_data=getPheno()
    print(phenotype_data)
    pheno_df=pd.DataFrame(data=phenotype_data,columns=[&quot;Ind&quot;, &quot;Pheno&quot;])  
  
    genotype_data=getGeno(Gen, snp)
    print(genotype_data)
    geno_df=pd.DataFrame(data=genotype_data, columns=[&quot;Ind&quot;, &quot;Allele1&quot;, &quot;Allele2&quot;])
    
    full_df=pd.merge(geno_df, pheno_df, how=&quot;inner&quot;, on=&quot;Ind&quot;)
    full_df.to_csv(fout, sep=&quot;\t&quot;, encoding=&#39;utf-8&#39;, index=False)
    fout.close()
    

if __name__ == &quot;__main__&quot;:
    import sys
    import pandas as pd
    chrom=sys.argv[1]
    snp = sys.argv[2]
    gene = sys.argv[3]
    molPhen=sys.argv[4]
    
    PhenFile = &quot;/project2/gilad/briana/threeprimeseq/data/molecular_phenos/fastqtl_qqnorm%sphase2.fixed.noChr.txt&quot;%(molPhen)
    GenFile= &quot;/project2/gilad/briana/YRI_geno_hg19/chr%s.dose.filt.vcf&quot;%(chrom)
    outFile = &quot;/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/qtlSNP_Peak%s%s%s.txt&quot;%(molPhen, snp, gene)
    main(PhenFile, GenFile, outFile, snp, gene,molPhen)
    </code></pre>
<p>test this:</p>
<pre class="bash"><code>python createQTLsnpMolPhenTable.py 3 3:150302010 EIF2A _RNAseq_</code></pre>
<p>list for phenos:</p>
<ul>
<li><p>4su_30</p></li>
<li><p>4su_60</p></li>
<li><p>RNAseqGeuvadis</p></li>
<li><p>RNAseq</p></li>
<li><p>prot</p></li>
<li><p>ribo</p></li>
</ul>
<p>Create a bash script that will use a for loop to run the python script on a all of the phenotypes</p>
<p>run_createQTLsnpMolPhenTable.sh</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=run_createQTLsnpMolPhenTable
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=run_createQTLsnpMolPhenTable.out
#SBATCH --error=run_createQTLsnpMolPhenTable.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load python 


chrom=$1
snp=$2
gene=$3

for i in &quot;_4su_30_&quot; &quot;_4su_60_&quot; &quot;_RNAseqGeuvadis_&quot; &quot;_RNAseq_&quot; &quot;_prot.&quot; &quot;_ribo_&quot;
do
python createQTLsnpMolPhenTable.py ${chrom} ${snp} ${gene} ${i}
done
</code></pre>
</div>
<div id="function-to-create-plots" class="section level2">
<h2>Function to create plots:</h2>
<p>I want to imput the files with the following structure:</p>
<p>/Users/bmittleman1/Documents/Gilad_lab/threeprimeseq/data/apaExamp/qtlSNP_Peak<em>molpheno</em>.snp.peak/gene.txt</p>
<p>I will use these to make cowplot with ggplot boxplots for each phenotypes. To do this I will create a function that takes in a snp, peak, and gene and creates each phenotype plot. It will then return the cowplot plot grid.</p>
<pre class="r"><code>plotQTL_func= function(SNP, peak, gene){
  apaN_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_PeakAPANuclear.&quot;, SNP, peak, &quot;.txt&quot;, sep = &quot;&quot; ), header=T)
  apaT_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_PeakAPATotal.&quot;, SNP, peak, &quot;.txt&quot;, sep = &quot;&quot; ), header=T)
  su30_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_4su_30_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header = T)
  su60_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_4su_60_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header=T)
  RNA_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_RNAseq_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;),header=T)
  RNAg_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_RNAseqGeuvadis_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header = T)
  ribo_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_ribo_&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;),header=T)
  prot_file=read.table(paste(&quot;../data/apaExamp/qtlSNP_Peak_prot.&quot;, SNP, gene, &quot;.txt&quot;, sep=&quot;&quot;), header=T)
  
  ggplot_func= function(file, molPhen,GENE){
    file = file %&gt;% mutate(genotype=Allele1 + Allele2)
    file$genotype= as.factor(as.character(file$genotype))
    plot=ggplot(file, aes(y=Pheno, x=genotype, by=genotype, fill=genotype)) + geom_boxplot(width=.25) + geom_jitter() + labs(y=&quot;Phenotpye&quot;,title=paste(molPhen, GENE, sep=&quot;: &quot;)) + scale_fill_brewer(palette=&quot;Paired&quot;)
    return(plot)
  }
  
  apaNplot=ggplot_func(apaN_file, &quot;Apa Nuclear&quot;, gene)
  apaTplot=ggplot_func(apaT_file, &quot;Apa Total&quot;, gene)
  su30plot=ggplot_func(su30_file, &quot;4su30&quot;,gene)
  su60plot=ggplot_func(su60_file, &quot;4su60&quot;,gene)
  RNAplot=ggplot_func(RNA_file, &quot;RNA Seq&quot;,gene)
  RNAgPlot=ggplot_func(RNAg_file, &quot;RNA Seq Geuvadis&quot;,gene)
  riboPlot= ggplot_func(ribo_file, &quot;Ribo Seq&quot;,gene)
  protplot=ggplot_func(prot_file, &quot;Protein&quot;,gene)
  
  full_plot= plot_grid(apaNplot,apaTplot, su30plot, su60plot, RNAplot, RNAgPlot, riboPlot, protplot,nrow=2)
  return (full_plot)
}</code></pre>
<p>Try this with the EIF2A QTL:</p>
<pre class="r"><code>eif2a_allplots=plotQTL_func(SNP=&quot;3:150302010&quot;, peak=&quot;peak228606&quot;, gene=&quot;EIF2A&quot;)

ggsave(&quot;../output/plots/EIF2A_allplots.png&quot;, eif2a_allplots, height=5, width=12)</code></pre>
</div>
<div id="try-with-another-snp" class="section level2">
<h2>Try with another snp:</h2>
<ul>
<li>peak164036, sid: 2:3502035</li>
</ul>
<p>Step 1: Figure out what gene the peak is in.</p>
<pre class="bash"><code>grep peak164036 /project2/gilad/briana/threeprimeseq/data/perm_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_transcript_permResBH.txt</code></pre>
<p>This peak is in ADI1</p>
<p>Step2: Get the total and nuclear APA values by genotype with createQTLsnpAPAPhenTable.py</p>
<pre class="bash"><code>
python createQTLsnpAPAPhenTable.py 2 2:3502035 peak164036 Total

python createQTLsnpAPAPhenTable.py 2 2:3502035 peak164036 Nuclear</code></pre>
<p>Step 3: Get the ensg gene name:</p>
<pre class="bash"><code>grep ADI1 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt</code></pre>
<p>Step 4: Run this on the other phenotypes with : run_createQTLsnpMolPhenTable.sh</p>
<pre class="bash"><code>
sbatch run_createQTLsnpMolPhenTable.sh &quot;2&quot; &quot;2:3502035&quot; &quot;ENSG00000182551&quot;
</code></pre>
<p>Step 4: copy files to computer:</p>
<pre class="bash"><code>scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/qtlSNP_Peak*2:*  . </code></pre>
<p>Step 5: plot</p>
<pre class="r"><code>plotQTL_func(SNP=&quot;2:3502035&quot;, peak=&quot;peak164036&quot;, gene=&quot;ENSG00000182551&quot;)</code></pre>
<p><img src="figure/swarmPlots_QTLs.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-23-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/8211a07a3567d273d979b7fc94d4f362d6e217b4/docs/figure/swarmPlots_QTLs.Rmd/unnamed-chunk-23-1.png" target="_blank">8211a07</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<ul>
<li>peak305794, sid: 7:128635754</li>
</ul>
<pre class="bash"><code>grep peak305794 /project2/gilad/briana/threeprimeseq/data/perm_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_transcript_permResBH.txt
#gene=IRF5
grep IRF5 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ensg= ENSG00000128604

python createQTLsnpAPAPhenTable.py 7 7:128635754  peak305794 Total
python createQTLsnpAPAPhenTable.py 7 7:128635754  peak305794 Nuclear


sbatch run_createQTLsnpMolPhenTable.sh &quot;7&quot; &quot;7:128635754&quot; &quot;ENSG00000128604&quot;

scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/qtlSNP_Peak_*7:* .
scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/qtlSNP_PeakAPA*.7* .</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;7:128635754&quot;, peak=&quot;peak305794&quot;, gene=&quot;ENSG00000128604&quot;)</code></pre>
<p><img src="figure/swarmPlots_QTLs.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-25-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/8211a07a3567d273d979b7fc94d4f362d6e217b4/docs/figure/swarmPlots_QTLs.Rmd/unnamed-chunk-25-1.png" target="_blank">8211a07</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<ul>
<li>Peak: peak152751, SID 19:4236475</li>
</ul>
<pre class="bash"><code>grep peak152751 /project2/gilad/briana/threeprimeseq/data/perm_APAqtl_trans/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_transcript_permResBH.txt
#gene=EBI3
grep EBI3 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ensg= ENSG00000105246

python createQTLsnpAPAPhenTable.py 19 19:4236475  peak152751 Total
python createQTLsnpAPAPhenTable.py 19 19:4236475  peak152751 Nuclear


sbatch run_createQTLsnpMolPhenTable.sh &quot;19&quot; &quot;19:4236475 &quot; &quot;ENSG00000105246&quot;

scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*19:4236475* .</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;19:4236475&quot;, peak=&quot;peak152751&quot;, gene=&quot;ENSG00000105246&quot;)</code></pre>
<p><img src="figure/swarmPlots_QTLs.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-27-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/8211a07a3567d273d979b7fc94d4f362d6e217b4/docs/figure/swarmPlots_QTLs.Rmd/unnamed-chunk-27-1.png" target="_blank">8211a07</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<ul>
<li>4:84382181:84382182:MRPS18C_+_peak241853, snp4:84382264</li>
</ul>
<pre class="bash"><code>
#gene=MRPS18C
grep MRPS18C /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ensg= ENSG00000163319

python createQTLsnpAPAPhenTable.py 4 4:84382264  peak241853 Total
python createQTLsnpAPAPhenTable.py 4 4:84382264  peak241853 Nuclear


sbatch run_createQTLsnpMolPhenTable.sh &quot;4&quot; &quot;4:84382264 &quot; &quot;ENSG00000163319&quot;

scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*4:84382264* .</code></pre>
<p>We dont have protein information for this gene</p>
<pre class="r"><code>plotQTL_func(SNP=&quot;4:84382264&quot;, peak=&quot;peak241853&quot;, gene=&quot;ENSG00000163319&quot;)</code></pre>
<p><img src="figure/swarmPlots_QTLs.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-29-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/8211a07a3567d273d979b7fc94d4f362d6e217b4/docs/figure/swarmPlots_QTLs.Rmd/unnamed-chunk-29-1.png" target="_blank">8211a07</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-10-11
</td>
</tr>
</tbody>
</table>
<p></details></p>
<ul>
<li>7:66703515:66703516:TYW1_+_peak298097 7:66595366</li>
</ul>
<pre class="bash"><code>
#gene=TYW1
grep TYW1 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ensg= ENSG00000198874

python createQTLsnpAPAPhenTable.py 7 7:66595366 peak298097 Total
python createQTLsnpAPAPhenTable.py 7 7:66595366  peak298097 Nuclear


sbatch run_createQTLsnpMolPhenTable.sh &quot;7&quot; &quot;7:66595366&quot; &quot;ENSG00000198874&quot;

scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*7:66595366* .</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;7:66595366&quot;, peak=&quot;peak298097&quot;, gene=&quot;ENSG00000198874&quot;)</code></pre>
<p><img src="figure/swarmPlots_QTLs.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>8:2792874:2792875:CSMD1_-_peak310334 8:3037787</li>
</ul>
<pre class="bash"><code>
#gene=CSMD1
grep CSMD1 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ensg= ENSG00000183117

python createQTLsnpAPAPhenTable.py 8 8:3037787   peak310334 Total
python createQTLsnpAPAPhenTable.py 8 8:3037787    peak310334 Nuclear


sbatch run_createQTLsnpMolPhenTable.sh &quot;8&quot; &quot;8:3037787&quot; &quot;ENSG00000183117&quot;

scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*8:3037787* .</code></pre>
<p>We do not have molecular phenotypes for this gene.</p>
<ul>
<li>6:11183530:11183531:NEDD9_-_peak272002 6:11212754</li>
</ul>
<pre class="bash"><code>
#gene=NEDD9
grep NEDD9 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ensg= ENSG00000111859

python createQTLsnpAPAPhenTable.py 6 6:11212754  peak272002 Total
python createQTLsnpAPAPhenTable.py 6 6:11212754  peak272002 Nuclear


sbatch run_createQTLsnpMolPhenTable.sh &quot;6&quot; &quot;6:11212754&quot; &quot;ENSG00000111859&quot;

scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*6:11212754* .</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;6:11212754&quot;, peak=&quot;peak272002&quot;, gene=&quot;ENSG00000111859&quot;)</code></pre>
<pre><code>Warning: Removed 9 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 9 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/swarmPlots_QTLs.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<ul>
<li>12:51453213:51453214:LETMD1_+_peak71110 12:51405335</li>
</ul>
<pre class="bash"><code>
#gene=LETMD1
grep LETMD1 /project2/gilad/briana/genome_anotation_data/ensemble_to_genename.txt
#ensg= ENSG00000050426

python createQTLsnpAPAPhenTable.py 12 12:51405335  peak71110 Total
python createQTLsnpAPAPhenTable.py 12 12:51405335  peak71110 Nuclear


sbatch run_createQTLsnpMolPhenTable.sh &quot;12&quot; &quot;12:51405335&quot; &quot;ENSG00000050426&quot;

scp brimittleman@midway2.rcc.uchicago.edu:/project2/gilad/briana/threeprimeseq/data/ApaQTL_examples/*12:51405335* .</code></pre>
<pre class="r"><code>plotQTL_func(SNP=&quot;12:51405335&quot;, peak=&quot;peak71110&quot;, gene=&quot;ENSG00000050426&quot;)</code></pre>
<pre><code>Warning: Removed 7 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 7 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/swarmPlots_QTLs.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] bindrcpp_0.2.2      cowplot_0.9.3       data.table_1.11.8  
 [4] VennDiagram_1.6.20  futile.logger_1.4.3 forcats_0.3.0      
 [7] stringr_1.3.1       dplyr_0.7.6         purrr_0.2.5        
[10] readr_1.1.1         tidyr_0.8.1         tibble_1.4.2       
[13] ggplot2_3.0.0       tidyverse_1.2.1     reshape2_1.4.3     
[16] workflowr_1.1.1    

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.4     haven_1.1.2          lattice_0.20-35     
 [4] colorspace_1.3-2     htmltools_0.3.6      yaml_2.2.0          
 [7] rlang_0.2.2          R.oo_1.22.0          pillar_1.3.0        
[10] glue_1.3.0           withr_2.1.2          R.utils_2.7.0       
[13] RColorBrewer_1.1-2   lambda.r_1.2.3       modelr_0.1.2        
[16] readxl_1.1.0         bindr_0.1.1          plyr_1.8.4          
[19] munsell_0.5.0        gtable_0.2.0         cellranger_1.1.0    
[22] rvest_0.3.2          R.methodsS3_1.7.1    evaluate_0.11       
[25] labeling_0.3         knitr_1.20           broom_0.5.0         
[28] Rcpp_0.12.19         formatR_1.5          backports_1.1.2     
[31] scales_1.0.0         jsonlite_1.5         hms_0.4.2           
[34] digest_0.6.17        stringi_1.2.4        rprojroot_1.3-2     
[37] cli_1.0.1            tools_3.5.1          magrittr_1.5        
[40] lazyeval_0.2.1       futile.options_1.0.1 crayon_1.3.4        
[43] whisker_0.3-2        pkgconfig_2.0.2      xml2_1.2.0          
[46] lubridate_1.7.4      assertthat_0.2.0     rmarkdown_1.10      
[49] httr_1.3.1           rstudioapi_0.8       R6_2.3.0            
[52] nlme_3.1-137         git2r_0.23.0         compiler_3.5.1      </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
