<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Test SMASH</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Test SMASH</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>7/16/2018</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-07-19</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/threeprimeseq/tree/31482e2756b96a67021ff4f25923469e45a99453" target="_blank">31482e2</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  data/18486.genecov.txt
    Untracked:  data/APApeaksYL.total.inbrain.bed
    Untracked:  data/YL-SP-18486-T_S9_R1_001-genecov.txt
    Untracked:  data/bedgraph_peaks/
    Untracked:  data/bin200.5.T.nuccov.bed
    Untracked:  data/bin200.Anuccov.bed
    Untracked:  data/bin200.nuccov.bed
    Untracked:  data/gene_cov/
    Untracked:  data/leafcutter/
    Untracked:  data/nuc6up/
    Untracked:  data/reads_mapped_three_prime_seq.csv
    Untracked:  data/smash.cov.results.bed
    Untracked:  data/smash.cov.results.csv
    Untracked:  data/smash.cov.results.txt
    Untracked:  data/smash_testregion/
    Untracked:  data/ssFC200.cov.bed
    Untracked:  output/picard/
    Untracked:  output/plots/
    Untracked:  output/qual.fig2.pdf

Unstaged changes:
    Modified:   analysis/dif.iso.usage.leafcutter.Rmd
    Modified:   analysis/explore.filters.Rmd
    Modified:   analysis/test.max2.Rmd
    Modified:   code/Snakefile

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/31482e2756b96a67021ff4f25923469e45a99453/analysis/test.smash.Rmd" target="_blank">31482e2</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-19
</td>
<td style="text-align:left;">
gapdh
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/319322373c1d7d83d53b6eb868a5fefae3dd682d/docs/test.smash.html" target="_blank">3193223</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-18
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/9d320762cc20d555c01ef24c55f9b9c5253631f7/analysis/test.smash.Rmd" target="_blank">9d32076</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-18
</td>
<td style="text-align:left;">
add actb example
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/0464829eb7efe6e59a1e4c9b119aacf011b4bdaa/docs/test.smash.html" target="_blank">0464829</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-17
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/cc5cc50c1522e65db9c5347be32b84b8a3cc8701/analysis/test.smash.Rmd" target="_blank">cc5cc50</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-17
</td>
<td style="text-align:left;">
test region smash results
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/brimittleman/threeprimeseq/d61f59030ac60a83efbe121b621608644b6a3203/docs/test.smash.html" target="_blank">d61f590</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-17
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/89ebcac67144a2affcfc48336fd37621f7c74833/analysis/test.smash.Rmd" target="_blank">89ebcac</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-17
</td>
<td style="text-align:left;">
add smash test
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<p>In this analysis I will use the tutorial I did for the SMASH package on chip seq data to test it on the three prime seq data. In order to complete this I need to make a matrix with genome location counts for where reads start for positions 880001:1011072 on chr1, I am using this region because I already know it fits the <span class="math inline">\(2^{x}\)</span> criterion. I need the matrix to be individual by basepair. I can use genome cov in all of the total fractions then merge the results together to make a matrix.</p>
<div id="create-coverage-files" class="section level2">
<h2>Create Coverage files</h2>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=5gencov
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=5gencov.out
#SBATCH --error=5gencov.err
#SBATCH --partition=broadwl
#SBATCH --mem=40G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env 

#imput sorted bam file 
bam=$1

describer=$(echo ${bam} | sed -e &#39;s/.*\YL-SP-//&#39; | sed -e &quot;s/-sort.bam$//&quot;)


bedtools genomecov-ibam $1 -d  -5 &gt; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.${describer}.bed</code></pre>
<p>run on /project2/gilad/briana/threeprimeseq/data/sort/YL-SP-18486-N_S10_R1_001-sort.bam</p>
<p>wrap this function:</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=w_5gencov
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=w_5gencov.out
#SBATCH --error=w_5gencov.err
#SBATCH --partition=broadwl
#SBATCH --mem=16G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env 

for i in $(ls /project2/gilad/briana/threeprimeseq/data/sort/*.bam); do
        sbatch 5primegencov.sh $i 
    done</code></pre>
</div>
<div id="test-example-region" class="section level2">
<h2>Test example region</h2>
<p>First I will get ch1 880001:1011072 for each individual.</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=test.reg
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=test.reg.out
#SBATCH --error=test.reg.err
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --mail-type=END

awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18486-T_S9_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18486-T_S9_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18497-T_S11_R1_001.bed   &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18497-T_S11_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;5&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18500-T_S19_R1_001.bed   &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18500-T_S19_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18505-T_S1_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18505-T_S1_R1_001.testregion.bed

awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; gencov5prime.18508-T_S5_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18508-T_S5_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39;/project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18853-T_S31_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18853-T_S31_R1_001.testregion.bed



awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18870-T_S23_R1_001.bed   &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18870-T_S23_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19128-T_S29_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19128-T_S29_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19141-T_S17_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19141-T_S17_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19193-T_S21_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19193-T_S21_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19209-T_S15_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19209-T_S15_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19233-T_S7_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19223-T_S7_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19225-T_S27_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19225-T_S27_R1_001.testregion.bed

awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19238-T_S3_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19238-T_S3_R1_001.testregion.bed

awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19239-T_S13_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19239-T_S13_R1_001.testregion.bed


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19257-T_S25_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19257-T_S25_R1_001.testregion.bed
</code></pre>
<p>3 didnt work. Try these again.</p>
<pre class="bash"><code>
#!/bin/bash

#SBATCH --job-name=test.reg2
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=test.reg2.out
#SBATCH --error=test.reg2.err
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --mail-type=END


awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18508-T_S5_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18508-T_S5_R1_001.testregion.bed

awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18853-T_S31_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18853-T_S31_R1_001.testregion.bed

awk &#39;$1 == 1 &amp;&amp; $2 &gt;= 880001 &amp;&amp; $2 &lt;= 1011072 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18500-T_S19_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18500-T_S19_R1_001.testregion.bed
</code></pre>
<p>Now I will pull in these regions and merge them to make a matrix I can put into smashr.</p>
<pre class="r"><code>test_18468=read.table(&quot;../data/smash_testregion/gencov5prime.18486-T_S9_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18486&quot;))

test_18497=read.table(&quot;../data/smash_testregion/gencov5prime.18497-T_S11_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18497&quot;))

test_18500=read.table(&quot;../data/smash_testregion/gencov5prime.18500-T_S19_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18500&quot;))  

test_18505=read.table(&quot;../data/smash_testregion/gencov5prime.18505-T_S1_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18505&quot;))

test_18508=read.table(&quot;../data/smash_testregion/gencov5prime.18508-T_S5_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18508&quot;))

test_18853=read.table(&quot;../data/smash_testregion/gencov5prime.18853-T_S31_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18853&quot;))

test_18870=read.table(&quot;../data/smash_testregion/gencov5prime.18870-T_S23_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18870&quot;))

test_19128=read.table(&quot;../data/smash_testregion/gencov5prime.19128-T_S29_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19128&quot;))

test_19239=read.table(&quot;../data/smash_testregion/gencov5prime.19239-T_S13_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19239&quot;))

test_19257=read.table(&quot;../data/smash_testregion/gencov5prime.19257-T_S25_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19257&quot;))

test_19141=read.table(&quot;../data/smash_testregion/gencov5prime.19141-T_S17_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19141&quot;))

test_19193=read.table(&quot;../data/smash_testregion/gencov5prime.19193-T_S21_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19193&quot;))

test_19209=read.table(&quot;../data/smash_testregion/gencov5prime.19209-T_S15_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19209&quot;))

test_19223=read.table(&quot;../data/smash_testregion/gencov5prime.19223-T_S7_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19223&quot;))

test_19225=read.table(&quot;../data/smash_testregion/gencov5prime.19225-T_S27_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19225&quot;))

test_19238=read.table(&quot;../data/smash_testregion/gencov5prime.19238-T_S3_R1_001.testregion.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19238&quot;))</code></pre>
<p>Load Packages:</p>
<pre class="r"><code>library(devtools)</code></pre>
<pre><code>Warning: package &#39;devtools&#39; was built under R version 3.4.4</code></pre>
<pre class="r"><code>library(scales)
library(smashr)
library(tidyr)
library(workflowr)</code></pre>
<pre><code>Loading required package: rmarkdown</code></pre>
<pre><code>This is workflowr version 1.0.1
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 3.4.4</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<p>Bind all of the count</p>
<pre class="r"><code>test_matrix=cbind(test_18468$T18486, test_18497$T18497, test_18500$T18500, test_18505$T18505, test_18508$T18508, test_18853$T18853, test_18870$T18870, test_19128$T19128, test_19141$T19141, test_19193$T19193, test_19209$T19209, test_19223$T19223, test_19225$T19225, test_19238$T19238, test_19239$T19239, test_19257$T19257) %&gt;% t</code></pre>
<p>Run smash:</p>
<pre class="r"><code>res = smash.poiss(test_matrix[1,]+test_matrix[2,],post.var=TRUE)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre><code>Squarem-1
Objective fn: 316440
Objective fn: 75069.3  Extrapolation: 0  Steplength: 1
Objective fn: 20919.7  Extrapolation: 1  Steplength: 1.7761
Objective fn: 17547  Extrapolation: 1  Steplength: 4
Objective fn: 17138.8  Extrapolation: 1  Steplength: 4.55255
Objective fn: 16836.6  Extrapolation: 1  Steplength: 1.86253
Objective fn: 16818.8  Extrapolation: 1  Steplength: 6.42316
Objective fn: 16816  Extrapolation: 1  Steplength: 1.66778
Objective fn: 16815.9  Extrapolation: 1  Steplength: 7.08232</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre><code>Squarem-1
Objective fn: 547800
Objective fn: 101309  Extrapolation: 0  Steplength: 1
Objective fn: 37390.2  Extrapolation: 1  Steplength: 1.89191
Objective fn: 32343.6  Extrapolation: 1  Steplength: 4
Objective fn: 31416.8  Extrapolation: 1  Steplength: 5.23485
Objective fn: 31019  Extrapolation: 1  Steplength: 2.24046
Objective fn: 31004.3  Extrapolation: 1  Steplength: 5.95071
Objective fn: 31003.4  Extrapolation: 1  Steplength: 2.25357
Objective fn: 31003.3  Extrapolation: 1  Steplength: 6.79538</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre><code>Squarem-1
Objective fn: 882921
Objective fn: 118772  Extrapolation: 0  Steplength: 1
Objective fn: 49764.5  Extrapolation: 1  Steplength: 1.87244
Objective fn: 44341.9  Extrapolation: 1  Steplength: 4
Objective fn: 43183.7  Extrapolation: 1  Steplength: 4.56042
Objective fn: 41597.4  Extrapolation: 1  Steplength: 1.93991
Objective fn: 41582.5  Extrapolation: 1  Steplength: 7.15247
Objective fn: 41416.1  Extrapolation: 1  Steplength: 1.69043
Objective fn: 41408.8  Extrapolation: 1  Steplength: 9.74108
Objective fn: 41406.9  Extrapolation: 1  Steplength: 1.59935
Objective fn: 41406.9  Extrapolation: 1  Steplength: 5.74521
Objective fn: 41406.8  Extrapolation: 1  Steplength: 3.48951</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre><code>Squarem-1
Objective fn: 1.2791e+06
Objective fn: 88217.7  Extrapolation: 0  Steplength: 1
Objective fn: 48098  Extrapolation: 1  Steplength: 3.21173
Objective fn: 41534.8  Extrapolation: 1  Steplength: 3.31029
Objective fn: 39112.5  Extrapolation: 1  Steplength: 1.9387
Objective fn: 38929.7  Extrapolation: 1  Steplength: 4
Objective fn: 38852.8  Extrapolation: 1  Steplength: 1.89245
Objective fn: 38845.6  Extrapolation: 1  Steplength: 5.02972
Objective fn: 38844.1  Extrapolation: 1  Steplength: 1.53262
Objective fn: 38843.9  Extrapolation: 1  Steplength: 6.32374</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre><code>Squarem-1
Objective fn: 1.19992e+06
Objective fn: 127113  Extrapolation: 0  Steplength: 1
Objective fn: 103768  Extrapolation: 1  Steplength: 2.87597
Objective fn: 100999  Extrapolation: 1  Steplength: 4
Objective fn: 99983.8  Extrapolation: 1  Steplength: 3.04432
Objective fn: 99154  Extrapolation: 1  Steplength: 4.6448
Objective fn: 98661.4  Extrapolation: 1  Steplength: 2.95052
Objective fn: 98391.1  Extrapolation: 1  Steplength: 5.09439
Objective fn: 98302.4  Extrapolation: 1  Steplength: 2.05767
Objective fn: 98288.2  Extrapolation: 1  Steplength: 13.5878
Objective fn: 98224.6  Extrapolation: 1  Steplength: 1.53983
Objective fn: 98212.4  Extrapolation: 1  Steplength: 16
Objective fn: 98207  Extrapolation: 1  Steplength: 2.54441
Objective fn: 98203.1  Extrapolation: 1  Steplength: 8.20699
Objective fn: 98202  Extrapolation: 1  Steplength: 2.00162
Objective fn: 98199.8  Extrapolation: 1  Steplength: 16.5155
Objective fn: 98198.5  Extrapolation: 1  Steplength: 1.62338
Objective fn: 98199  Extrapolation: 1  Steplength: 44.8518
Objective fn: 98195.9  Extrapolation: 1  Steplength: 1.4749
Objective fn: 98195.6  Extrapolation: 0  Steplength: 1
Objective fn: 98195.4  Extrapolation: 0  Steplength: 1
Objective fn: 98195.2  Extrapolation: 0  Steplength: 1
Objective fn: 98195.1  Extrapolation: 0  Steplength: 1
Objective fn: 98194.8  Extrapolation: 1  Steplength: 16
Objective fn: 98194.6  Extrapolation: 0  Steplength: 1
Objective fn: 98194.5  Extrapolation: 0  Steplength: 1
Objective fn: 98194.4  Extrapolation: 0  Steplength: 1
Objective fn: 98194.2  Extrapolation: 1  Steplength: 16
Objective fn: 98194  Extrapolation: 0  Steplength: 1
Objective fn: 98193.9  Extrapolation: 0  Steplength: 1
Objective fn: 98193.7  Extrapolation: 1  Steplength: 16
Objective fn: 98193.6  Extrapolation: 0  Steplength: 1
Objective fn: 98193.5  Extrapolation: 0  Steplength: 1
Objective fn: 98193.4  Extrapolation: 0  Steplength: 1
Objective fn: 98193  Extrapolation: 1  Steplength: 16
Objective fn: 98192.9  Extrapolation: 0  Steplength: 1
Objective fn: 98192.8  Extrapolation: 0  Steplength: 1
Objective fn: 98192.7  Extrapolation: 0  Steplength: 1
Objective fn: 98192.1  Extrapolation: 1  Steplength: 16
Objective fn: 98192  Extrapolation: 0  Steplength: 1
Objective fn: 98191.9  Extrapolation: 0  Steplength: 1
Objective fn: 98190.9  Extrapolation: 1  Steplength: 16
Objective fn: 98190.8  Extrapolation: 0  Steplength: 1
Objective fn: 98190.6  Extrapolation: 0  Steplength: 1
Objective fn: 98189.2  Extrapolation: 1  Steplength: 16
Objective fn: 98189  Extrapolation: 0  Steplength: 1
Objective fn: 98186.8  Extrapolation: 1  Steplength: 16
Objective fn: 98186.6  Extrapolation: 0  Steplength: 1
Objective fn: 98183.6  Extrapolation: 1  Steplength: 16
Objective fn: 98183.3  Extrapolation: 0  Steplength: 1
Objective fn: 98179.6  Extrapolation: 1  Steplength: 16
Objective fn: 98179.1  Extrapolation: 0  Steplength: 1
Objective fn: 98175  Extrapolation: 1  Steplength: 16
Objective fn: 98174.4  Extrapolation: 0  Steplength: 1
Objective fn: 98171  Extrapolation: 1  Steplength: 16
Objective fn: 98170.2  Extrapolation: 0  Steplength: 1
Objective fn: 98169.6  Extrapolation: 0  Steplength: 1
Objective fn: 98167.5  Extrapolation: 1  Steplength: 16
Objective fn: 98166.5  Extrapolation: 0  Steplength: 1
Objective fn: 98165.9  Extrapolation: 0  Steplength: 1
Objective fn: 98165.4  Extrapolation: 0  Steplength: 1
Objective fn: 98164.7  Extrapolation: 1  Steplength: 16
Objective fn: 98163.9  Extrapolation: 0  Steplength: 1
Objective fn: 98163.3  Extrapolation: 0  Steplength: 1
Objective fn: 98163  Extrapolation: 0  Steplength: 1
Objective fn: 98162.7  Extrapolation: 0  Steplength: 1
Objective fn: 98162.6  Extrapolation: 1  Steplength: 16
Objective fn: 98162.1  Extrapolation: 0  Steplength: 1
Objective fn: 98161.8  Extrapolation: 0  Steplength: 1
Objective fn: 98161.6  Extrapolation: 0  Steplength: 1
Objective fn: 98161.5  Extrapolation: 0  Steplength: 1
Objective fn: 98161.4  Extrapolation: 0  Steplength: 1
Objective fn: 98161.3  Extrapolation: 0  Steplength: 1
Objective fn: 98161.2  Extrapolation: 1  Steplength: 16
Objective fn: 98161.1  Extrapolation: 0  Steplength: 1
Objective fn: 98161  Extrapolation: 0  Steplength: 1
Objective fn: 98160.9  Extrapolation: 0  Steplength: 1
Objective fn: 98160.9  Extrapolation: 0  Steplength: 1
Objective fn: 98160.9  Extrapolation: 0  Steplength: 1
Objective fn: 98160.9  Extrapolation: 1  Steplength: 16
Objective fn: 98160.8  Extrapolation: 0  Steplength: 1
Objective fn: 98160.8  Extrapolation: 0  Steplength: 1
Objective fn: 98160.8  Extrapolation: 0  Steplength: 1
Objective fn: 98160.8  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 1  Steplength: 16
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 1  Steplength: 16
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 1  Steplength: 16
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 1  Steplength: 16
Objective fn: 98161.6  Extrapolation: 1  Steplength: 25.5964
Objective fn: 98160.7  Extrapolation: 1  Steplength: 4.9659
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 1  Steplength: 16
Objective fn: 98160.7  Extrapolation: 0  Steplength: 1
Objective fn: 98160.6  Extrapolation: 0  Steplength: 1
Objective fn: 98160.6  Extrapolation: 0  Steplength: 1
Objective fn: 98160.7  Extrapolation: 1  Steplength: 16
Objective fn: 98161.4  Extrapolation: 1  Steplength: 28.4418
Objective fn: 98160.6  Extrapolation: 1  Steplength: 4.88053
Objective fn: 98161.5  Extrapolation: 1  Steplength: 64
Objective fn: 98160.9  Extrapolation: 1  Steplength: 1.58458
Objective fn: 98160.6  Extrapolation: 1  Steplength: 5.74946
Objective fn: 98160.7  Extrapolation: 1  Steplength: 18.0007
Objective fn: 98160.6  Extrapolation: 1  Steplength: 1.31963
Objective fn: 98160.8  Extrapolation: 1  Steplength: 14.5938
Objective fn: 98160.6  Extrapolation: 1  Steplength: 5.87148
Objective fn: 98160.6  Extrapolation: 1  Steplength: 2.79445
Objective fn: 98160.6  Extrapolation: 1  Steplength: 10.536
Objective fn: 98160.6  Extrapolation: 1  Steplength: 1.26176
Objective fn: 98160.6  Extrapolation: 0  Steplength: 1
Objective fn: 98160.6  Extrapolation: 0  Steplength: 1
Objective fn: 98160.6  Extrapolation: 1  Steplength: 64
Objective fn: 98160.7  Extrapolation: 1  Steplength: 11.4375
Objective fn: 98160.6  Extrapolation: 1  Steplength: 3.26847
Objective fn: 98160.6  Extrapolation: 1  Steplength: 4.39932
Objective fn: 98160.6  Extrapolation: 1  Steplength: 3.21355
Objective fn: 98160.6  Extrapolation: 1  Steplength: 7.1878
Objective fn: 98160.6  Extrapolation: 1  Steplength: 1.51531
Objective fn: 98160.5  Extrapolation: 1  Steplength: 256
Objective fn: 98160.5  Extrapolation: 1  Steplength: 1.39451
Objective fn: 98160.7  Extrapolation: 1  Steplength: 479.942
Objective fn: 98160.5  Extrapolation: 1  Steplength: 1.41467
Objective fn: 98160.6  Extrapolation: 1  Steplength: 14.2088
Objective fn: 98160.5  Extrapolation: 1  Steplength: 2.52095
Objective fn: 98160.5  Extrapolation: 1  Steplength: 9.92198
Objective fn: 98160.5  Extrapolation: 1  Steplength: 2.79295</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm</code></pre>
<pre><code>Squarem-1
Objective fn: 687462
Objective fn: 233579  Extrapolation: 0  Steplength: 1
Objective fn: 163301  Extrapolation: 1  Steplength: 2.96537
Objective fn: 101350  Extrapolation: 0  Steplength: 1
Objective fn: 87205.9  Extrapolation: 1  Steplength: 3.31731
Objective fn: 83310  Extrapolation: 1  Steplength: 1.41235
Objective fn: 83078.4  Extrapolation: 1  Steplength: 4
Objective fn: 83067.1  Extrapolation: 1  Steplength: 2.82442
Objective fn: 83065.1  Extrapolation: 1  Steplength: 2.48778
Objective fn: 83064.6  Extrapolation: 1  Steplength: 2.94018
Objective fn: 83064.5  Extrapolation: 1  Steplength: 2.28751
Objective fn: 83064.5  Extrapolation: 1  Steplength: 3.24189</code></pre>
<pre class="r"><code>bppos = 880001:1011072

plot(bppos,test_matrix[1,]+test_matrix[2,],xlab=&quot;position&quot;,ylab=&quot;counts&quot;,pch=16,cex=0.5, col=alpha(&quot;black&quot;,0.04))</code></pre>
<p><img src="figure/test.smash.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-9-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/0464829eb7efe6e59a1e4c9b119aacf011b4bdaa/docs/figure/test.smash.Rmd/unnamed-chunk-9-1.png" target="_blank">0464829</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-17
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>plot(bppos,res$est,type=&#39;l&#39;,xlab=&quot;position&quot;,ylab=&quot;intensity&quot;)</code></pre>
<p><img src="figure/test.smash.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-9-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/0464829eb7efe6e59a1e4c9b119aacf011b4bdaa/docs/figure/test.smash.Rmd/unnamed-chunk-9-2.png" target="_blank">0464829</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-17
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Create a coverage file with the results.</p>
<pre class="r"><code>cov=cbind(test_18468$chr, test_18468$base + 1, test_18468$base, res$est)</code></pre>
</div>
<div id="test-actb" class="section level2">
<h2>Test actb</h2>
<p>I want to try this on a region with higher background for example where actb is. I can run a smaller region of <span class="math inline">\(2^{10}\)</span> bases. chr7:5,566,662-5,567,686. The following script to extract the region is called test.actbregion.sh.</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=test.regactb
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=test.regact.out
#SBATCH --error=test.regact.err
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --mail-type=END

awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18486-T_S9_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18486-T_S9_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18497-T_S11_R1_001.bed   &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18497-T_S11_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18500-T_S19_R1_001.bed   &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18500-T_S19_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18505-T_S1_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18505-T_S1_R1_001.testregion.actb.bed

awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18508-T_S5_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18508-T_S5_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18853-T_S31_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18853-T_S31_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18870-T_S23_R1_001.bed   &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18870-T_S23_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19128-T_S29_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19128-T_S29_R1_001.testregion.actb.bed

awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19141-T_S17_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19141-T_S17_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19193-T_S21_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19193-T_S21_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19209-T_S15_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19209-T_S15_R1_001.testregion.actb.bed

awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19233-T_S7_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19223-T_S7_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19225-T_S27_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19225-T_S27_R1_001.testregion.actb.bed

awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19238-T_S3_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19238-T_S3_R1_001.testregion.actb.bed

awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19239-T_S13_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19239-T_S13_R1_001.testregion.actb.bed


awk &#39;$1 == 7 &amp;&amp; $2 &gt;= 5566662 &amp;&amp; $2 &lt;= 5567686 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19257-T_S25_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19257-T_S25_R1_001.testregion.actb.bed
</code></pre>
<p>18853 did not work, running seperatly.</p>
<pre class="r"><code>actb_test_18468=read.table(&quot;../data/smash_testregion/gencov5prime.18486-T_S9_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18486&quot;))

actb_test_18497=read.table(&quot;../data/smash_testregion/gencov5prime.18497-T_S11_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18497&quot;))

actb_test_18500=read.table(&quot;../data/smash_testregion/gencov5prime.18500-T_S19_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18500&quot;))  

actb_test_18505=read.table(&quot;../data/smash_testregion/gencov5prime.18505-T_S1_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18505&quot;))

actb_test_18508=read.table(&quot;../data/smash_testregion/gencov5prime.18508-T_S5_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18508&quot;))

actb_test_18853=read.table(&quot;../data/smash_testregion/gencov5prime.18853-T_S31_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18853&quot;))

actb_test_18870=read.table(&quot;../data/smash_testregion/gencov5prime.18870-T_S23_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18870&quot;))

actb_test_19128=read.table(&quot;../data/smash_testregion/gencov5prime.19128-T_S29_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19128&quot;))

actb_test_19239=read.table(&quot;../data/smash_testregion/gencov5prime.19239-T_S13_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19239&quot;))

actb_test_19257=read.table(&quot;../data/smash_testregion/gencov5prime.19257-T_S25_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19257&quot;))

actb_test_19141=read.table(&quot;../data/smash_testregion/gencov5prime.19141-T_S17_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19141&quot;))

actb_test_19193=read.table(&quot;../data/smash_testregion/gencov5prime.19193-T_S21_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19193&quot;))

actb_test_19209=read.table(&quot;../data/smash_testregion/gencov5prime.19209-T_S15_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19209&quot;))

actb_test_19223=read.table(&quot;../data/smash_testregion/gencov5prime.19223-T_S7_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19223&quot;))

actb_test_19225=read.table(&quot;../data/smash_testregion/gencov5prime.19225-T_S27_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19225&quot;))

actb_test_19238=read.table(&quot;../data/smash_testregion/gencov5prime.19238-T_S3_R1_001.testregion.actb.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19238&quot;))</code></pre>
<p>Make matrix</p>
<pre class="r"><code>actb_test_matrix=cbind(actb_test_18468$T18486, actb_test_18497$T18497, actb_test_18500$T18500, actb_test_18505$T18505, actb_test_18508$T18508, actb_test_18853$T18853, actb_test_18870$T18870, actb_test_19128$T19128, actb_test_19141$T19141, actb_test_19193$T19193, actb_test_19209$T19209, actb_test_19223$T19223, actb_test_19225$T19225, actb_test_19238$T19238, actb_test_19239$T19239, actb_test_19257$T19257) %&gt;% t</code></pre>
<p>Run smash:</p>
<pre class="r"><code>actb_res = smash.poiss(actb_test_matrix[1,]+actb_test_matrix[2,],post.var=TRUE)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm</code></pre>
<p>Make plots:</p>
<pre class="r"><code>actb_bppos = 5566662:5567686

plot(actb_bppos,actb_test_matrix[1,]+actb_test_matrix[2,],xlab=&quot;position&quot;,ylab=&quot;counts&quot;,pch=16,cex=0.5, col=alpha(&quot;black&quot;,1), main=&quot;Raw data ACTB&quot;)</code></pre>
<p><img src="figure/test.smash.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-15-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/319322373c1d7d83d53b6eb868a5fefae3dd682d/docs/figure/test.smash.Rmd/unnamed-chunk-15-1.png" target="_blank">3193223</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-18
</td>
</tr>
</tbody>
</table>
<p></details></p>
<pre class="r"><code>plot(actb_bppos,actb_res$est,type=&#39;l&#39;,xlab=&quot;position&quot;,ylab=&quot;intensity&quot;, main=&quot;SMASH results ACTB&quot;)</code></pre>
<p><img src="figure/test.smash.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of unnamed-chunk-15-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/brimittleman/threeprimeseq/blob/319322373c1d7d83d53b6eb868a5fefae3dd682d/docs/figure/test.smash.Rmd/unnamed-chunk-15-2.png" target="_blank">3193223</a>
</td>
<td style="text-align:left;">
Briana Mittleman
</td>
<td style="text-align:left;">
2018-07-18
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
<div id="test-gapdh" class="section level2">
<h2>Test Gapdh</h2>
<p>Check on another highly expressed gene to see if this dual peak pattern appears again. I will look at GAPDH. Chr12 6,646,755-6,647,779</p>
<pre class="bash"><code>#!/bin/bash

#SBATCH --job-name=test.reggap
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=test.reggap.out
#SBATCH --error=test.reggap.err
#SBATCH --partition=broadwl
#SBATCH --mem=20G
#SBATCH --mail-type=END

awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18486-T_S9_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18486-T_S9_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18497-T_S11_R1_001.bed   &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18497-T_S11_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18500-T_S19_R1_001.bed   &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18500-T_S19_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18505-T_S1_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18505-T_S1_R1_001.testregion.gap.bed

awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18508-T_S5_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18508-T_S5_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18853-T_S31_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18853-T_S31_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.18870-T_S23_R1_001.bed   &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.18870-T_S23_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19128-T_S29_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19128-T_S29_R1_001.testregion.gap.bed

awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19141-T_S17_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19141-T_S17_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19193-T_S21_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19193-T_S21_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19209-T_S15_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19209-T_S15_R1_001.testregion.gap.bed

awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19233-T_S7_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19223-T_S7_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19225-T_S27_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19225-T_S27_R1_001.testregion.gap.bed

awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19238-T_S3_R1_001.bed &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19238-T_S3_R1_001.testregion.gap.bed

awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19239-T_S13_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19239-T_S13_R1_001.testregion.gap.bed


awk &#39;$1 == 12 &amp;&amp; $2 &gt;= 6646755 &amp;&amp; $2 &lt;= 6647779 {print}&#39; /project2/gilad/briana/threeprimeseq/data/test.smash/gencov5prime.19257-T_S25_R1_001.bed  &gt; /project2/gilad/briana/threeprimeseq/data/test.region/gencov5prime.19257-T_S25_R1_001.testregion.gap.bed
</code></pre>
<p>Pull in the data:</p>
<pre class="r"><code>gap_test_18468=read.table(&quot;../data/smash_testregion/gencov5prime.18486-T_S9_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18486&quot;))

gap_test_18497=read.table(&quot;../data/smash_testregion/gencov5prime.18497-T_S11_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18497&quot;))

gap_test_18500=read.table(&quot;../data/smash_testregion/gencov5prime.18500-T_S19_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18500&quot;))  

gap_test_18505=read.table(&quot;../data/smash_testregion/gencov5prime.18505-T_S1_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18505&quot;))

gap_test_18508=read.table(&quot;../data/smash_testregion/gencov5prime.18508-T_S5_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18508&quot;))

gap_test_18853=read.table(&quot;../data/smash_testregion/gencov5prime.18853-T_S31_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18853&quot;))

gap_test_18870=read.table(&quot;../data/smash_testregion/gencov5prime.18870-T_S23_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T18870&quot;))

gap_test_19128=read.table(&quot;../data/smash_testregion/gencov5prime.19128-T_S29_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19128&quot;))

gap_test_19239=read.table(&quot;../data/smash_testregion/gencov5prime.19239-T_S13_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19239&quot;))

gap_test_19257=read.table(&quot;../data/smash_testregion/gencov5prime.19257-T_S25_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19257&quot;))

gap_test_19141=read.table(&quot;../data/smash_testregion/gencov5prime.19141-T_S17_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19141&quot;))

gap_test_19193=read.table(&quot;../data/smash_testregion/gencov5prime.19193-T_S21_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19193&quot;))

gap_test_19209=read.table(&quot;../data/smash_testregion/gencov5prime.19209-T_S15_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19209&quot;))

gap_test_19223=read.table(&quot;../data/smash_testregion/gencov5prime.19223-T_S7_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19223&quot;))

gap_test_19225=read.table(&quot;../data/smash_testregion/gencov5prime.19225-T_S27_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19225&quot;))

gap_test_19238=read.table(&quot;../data/smash_testregion/gencov5prime.19238-T_S3_R1_001.testregion.gap.bed&quot;, col.names=c(&quot;chr&quot;, &quot;base&quot;, &quot;T19238&quot;))</code></pre>
<p>Make matrix</p>
<pre class="r"><code>gap_test_matrix=cbind(gap_test_18468$T18486, gap_test_18497$T18497, gap_test_18500$T18500, gap_test_18505$T18505, gap_test_18508$T18508, gap_test_18853$T18853, gap_test_18870$T18870, gap_test_19128$T19128, gap_test_19141$T19141, gap_test_19193$T19193, gap_test_19209$T19209, gap_test_19223$T19223, gap_test_19225$T19225, gap_test_19238$T19238, gap_test_19239$T19239, gap_test_19257$T19257) %&gt;% t</code></pre>
<p>Run smash:</p>
<pre class="r"><code>gap_res = smash.poiss(gap_test_matrix[1,]+gap_test_matrix[2,],post.var=TRUE)</code></pre>
<pre><code>Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm
Due to absence of package REBayes, switching to EM algorithm</code></pre>
<p>Make plots:</p>
<pre class="r"><code>gap_bppos =6646755:6647779

plot(gap_bppos,gap_test_matrix[1,]+gap_test_matrix[2,],xlab=&quot;position&quot;,ylab=&quot;counts&quot;,pch=16,cex=0.5, col=alpha(&quot;black&quot;,1), main=&quot;Raw data GAPDH&quot;)</code></pre>
<p><img src="figure/test.smash.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(gap_bppos,gap_res$est,type=&#39;l&#39;,xlab=&quot;position&quot;,ylab=&quot;intensity&quot;, main=&quot;SMASH results GAPDH&quot;)</code></pre>
<p><img src="figure/test.smash.Rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.2 (2017-09-28)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_0.7.5     workflowr_1.0.1 rmarkdown_1.8.5 tidyr_0.7.2    
[5] smashr_1.2-0    scales_0.5.0    devtools_1.13.6

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.17      bindr_0.1.1       pillar_1.1.0     
 [4] compiler_3.4.2    git2r_0.21.0      plyr_1.8.4       
 [7] R.methodsS3_1.7.1 R.utils_2.6.0     bitops_1.0-6     
[10] iterators_1.0.10  tools_3.4.2       digest_0.6.15    
[13] tibble_1.4.2      evaluate_0.10.1   memoise_1.1.0    
[16] lattice_0.20-35   pkgconfig_2.0.1   rlang_0.2.1      
[19] Matrix_1.2-12     foreach_1.4.4     yaml_2.1.19      
[22] parallel_3.4.2    bindrcpp_0.2.2    withr_2.1.1      
[25] stringr_1.3.1     knitr_1.18        caTools_1.17.1   
[28] tidyselect_0.2.4  rprojroot_1.3-2   grid_3.4.2       
[31] glue_1.2.0        data.table_1.11.4 R6_2.2.2         
[34] purrr_0.2.5       ashr_2.2-7        magrittr_1.5     
[37] whisker_0.3-2     backports_1.1.2   codetools_0.2-15 
[40] htmltools_0.3.6   MASS_7.3-48       assertthat_0.2.0 
[43] colorspace_1.3-2  wavethresh_4.6.8  stringi_1.2.2    
[46] munsell_0.4.3     doParallel_1.0.11 pscl_1.5.2       
[49] truncnorm_1.0-8   SQUAREM_2017.10-1 R.oo_1.22.0      </code></pre>
</div>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.0.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
