<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Total and Nuclear fraction RNA seq</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Three Prime Sequencing in Human LCLs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brimittleman/threeprimeseq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Total and Nuclear fraction RNA seq</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>5/8/2018</em></h4>

</div>


<p>I will use this analysis to look at general QC measurements for the RNA seq on the total and nuclear fraction with respect to my three-prime-seq snakemake pipeline.</p>
<p>There are 2 lines with files for chromatin, nuclear and cytoplasmic. There is a sample switch in this anaylsis chrom.1.S3 and Cyt.1.S1 are switched. We know this from a heatmap analysis.</p>
<div id="library" class="section level3">
<h3>Library</h3>
<pre class="r"><code>library(ggplot2)
library(dplyr)</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>## Warning: package &#39;edgeR&#39; was built under R version 3.4.3</code></pre>
<pre><code>## Loading required package: limma</code></pre>
<pre><code>## Warning: package &#39;limma&#39; was built under R version 3.4.3</code></pre>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>## Loading required package: rmarkdown</code></pre>
<pre><code>## This is workflowr version 1.0.1
## Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(reshape2)</code></pre>
<pre><code>## Warning: package &#39;reshape2&#39; was built under R version 3.4.3</code></pre>
</div>
<div id="mapped-reads" class="section level3">
<h3>Mapped reads</h3>
<p>First I want to look at the number of reads that map. I will flip the sample swap for this analysis so it it correct.</p>
<pre class="r"><code>#wc -l file / 4

read_chrom1=63641411
read_chrom2= 61957677

read_nuc1= 73481177
read_nuc2= 60386733

read_cyt1= 74784589
read_cyt2= 56100500</code></pre>
<p>Now look at how many reads mapped:</p>
<pre class="r"><code>#samtools view -c -F 4 file 
map_chrom1=53856948
map_chrom2= 52630632

map_nuc1= 65023318
map_nuc2= 53445556

map_cyt1=67829422
map_cyt2=49619272</code></pre>
<p>Percent mapped per line:</p>
<pre class="r"><code>perc_map_chrom1= map_chrom1/read_chrom1 *100
perc_map_chrom2=map_chrom2/read_chrom2 * 100

perc_map_nuc1=map_nuc1/read_nuc1 * 100
perc_map_nuc2= map_nuc2/read_nuc2 * 100

perc_map_cyt1=map_cyt1/read_cyt1 * 100
perc_map_cyt2=map_cyt2/read_cyt2 * 100</code></pre>
<p>Put this information in a dataframe to plot it:</p>
<pre class="r"><code>percmap=c(perc_map_chrom1,perc_map_chrom2,perc_map_nuc1,perc_map_nuc2,perc_map_cyt1, perc_map_cyt2)
fraction=c(&quot;chrom&quot;, &quot;chrom&quot;, &quot;nuc&quot;, &quot;nuc&quot;, &quot;cyt&quot;, &quot;cyt&quot;)
sample=c(&quot;chrom1&quot;, &quot;chrom2&quot;, &quot;nuc1&quot;, &quot;nuc2&quot;, &quot;cyt1&quot;, &quot;cyt2&quot;)
percmap_df=as.data.frame(cbind(sample,fraction, percmap), stringsAsFactors = F)
percmap_df$percmap= as.numeric(percmap_df$percmap)

ggplot(percmap_df, aes(y=percmap, x=sample, fill=fraction)) + geom_col() + scale_y_continuous(limits=c(0,100)) + labs(y=&quot;Percent mapped&quot;, title=&quot;Percent of reads mapped&quot;)</code></pre>
<p><img src="total.nuc.rnaseq_files/figure-html/percmap%20df-1.png" width="672" /> This shows that mapping is a little bit higher for the cytoplasmic and nuclear fractions than for the chromatin fraction.</p>
</div>
<div id="gene-count-analysis" class="section level3">
<h3>Gene count analysis:</h3>
<p>Import gene count data. I will fix the sample switch at this point.</p>
<pre class="r"><code>cov_chrom1=read.table(&quot;../data/total.nuc.cyt.genecounts/gene_covYG-SP20-Cyt-1_S1_L005_R1_001-genecov.txt&quot;, stringsAsFactors = FALSE)
names(cov_chrom1)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;count&quot; )
cov_chrom2=read.table(&quot;../data/total.nuc.cyt.genecounts/gene_covYG-SP20-Chrom-2_S6_L005_R1_001-genecov.txt&quot;, stringsAsFactors = FALSE)
names(cov_chrom2)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;count&quot; )

cov_nuc1=read.table(&quot;../data/total.nuc.cyt.genecounts/gene_covYG-SP20-Nuc-1_S2_L005_R1_001-genecov.txt&quot;, stringsAsFactors = FALSE)
names(cov_nuc1)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;count&quot; )
cov_nuc2=read.table(&quot;../data/total.nuc.cyt.genecounts/gene_covYG-SP20-Nuc-2_S5_L005_R1_001-genecov.txt&quot;, stringsAsFactors = FALSE)
names(cov_nuc2)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;count&quot; )


cov_cyt1= read.table(&quot;../data/total.nuc.cyt.genecounts/gene_covYG-SP20-Chrom-1_S3_L005_R1_001-genecov.txt&quot;, stringsAsFactors = FALSE)
names(cov_cyt1)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;count&quot; )
cov_cyt2=read.table(&quot;../data/total.nuc.cyt.genecounts/gene_covYG-SP20-Cyt-2_S4_L005_R1_001-genecov.txt&quot;, stringsAsFactors = FALSE)
names(cov_cyt2)=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;gene&quot;, &quot;score&quot;, &quot;strand&quot;, &quot;count&quot; )</code></pre>
<p>Look at the distribution of the gene counts. To do this I need to make a matrix with gene by sample to give the cpm command.</p>
<pre class="r"><code>count_matrix=cbind(cov_chrom1$count, cov_chrom2$count, cov_nuc1$count, cov_nuc2$count, cov_cyt1$count, cov_cyt2$count)

gene_length=cov_chrom1 %&gt;% mutate(genelength=end-start) %&gt;% select(genelength) 
gene_length_vec=as.vector(gene_length$genelength)
count_matrix_cpm=cpm(count_matrix, log=T, gene.length=gene_length_vec )</code></pre>
<p>Plot the distribution of the log cpm counts.</p>
<pre class="r"><code>plotDensities(count_matrix_cpm, legend = &quot;bottomright&quot;, main=&quot;Pre-filtering&quot;)
abline(v = 0, lty = 3)</code></pre>
<p><img src="total.nuc.rnaseq_files/figure-html/plotdensity-1.png" width="672" /></p>
<p>I will only keep genes that have a log cpm greater than 1 in at least 3 samples.</p>
<pre class="r"><code>keep.exprs=rowSums(count_matrix_cpm&gt;1) &gt;=3
count_matrix_cpm_filt= count_matrix_cpm[keep.exprs,]

plotDensities(count_matrix_cpm_filt, legend = &quot;bottomright&quot;, main=&quot;Post-filtering&quot;)</code></pre>
<p><img src="total.nuc.rnaseq_files/figure-html/filt-1.png" width="672" /></p>
<p>Create boxplots:</p>
<pre class="r"><code>#pre standardization  
colnames(count_matrix)=sample
boxplot(count_matrix, main=&quot;Raw Counts by library&quot;)</code></pre>
<p><img src="total.nuc.rnaseq_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre class="r"><code>#prefilt
colnames(count_matrix_cpm)=sample
boxplot(count_matrix_cpm, main=&quot;Log CPM Counts by library prefilter&quot;)</code></pre>
<p><img src="total.nuc.rnaseq_files/figure-html/unnamed-chunk-2-2.png" width="672" /></p>
<pre class="r"><code>#filtered
colnames(count_matrix_cpm_filt)=sample
boxplot(count_matrix_cpm_filt, las=2,main=&quot;Log CPM Counts by library filtered&quot;)</code></pre>
<p><img src="total.nuc.rnaseq_files/figure-html/unnamed-chunk-2-3.png" width="672" /> This shows that we do not need to do any extreme normalization. I will skip it for this initial analysis.</p>
<p>Now I can look at the number of genes detected in each sample according to this filtering practice. I will use &gt;1 log cpm for “detected”.</p>
<pre class="r"><code>detected_chrom1=sum(count_matrix_cpm_filt[,1] &gt; 1)
detected_chrom2=sum(count_matrix_cpm_filt[,2] &gt; 1)

detected_nuc1=sum(count_matrix_cpm_filt[,3] &gt; 1)
detected_nuc2=sum(count_matrix_cpm_filt[,4] &gt; 1)

detected_cyt1=sum(count_matrix_cpm_filt[,5] &gt; 1)
detected_cyt2=sum(count_matrix_cpm_filt[,6] &gt; 1)</code></pre>
<p>Plot these values like the percent mapped.</p>
<pre class="r"><code>detected_vec=c(detected_chrom1, detected_chrom2, detected_nuc1, detected_nuc2, detected_cyt1, detected_cyt2)

det_df=as.data.frame(cbind(sample,fraction, detected_vec), stringsAsFactors = F)
det_df$detected_vec= as.numeric(det_df$detected_vec)

ggplot(det_df, aes(y=detected_vec, x=sample, fill=fraction)) + geom_col() + labs(x=&quot;Sample&quot;, y=&quot;Number of detected genes&quot;, title=&quot;Number Detected genes &gt;1 log-cpm&quot;) + scale_y_continuous(limits=c(0,20345))</code></pre>
<p><img src="total.nuc.rnaseq_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<pre class="r"><code>total_genes=20345
det_df_prop= det_df %&gt;% mutate(prop_det=detected_vec/total_genes)

ggplot(det_df_prop, aes(y=prop_det, x=sample, fill=fraction)) + geom_col() + labs(x=&quot;Sample&quot;, y=&quot;Proportion detected genes&quot;, title=&quot;Proportion Detected Genes &gt;1 log-cpm&quot;) + scale_y_continuous(limits=c(0,1))</code></pre>
<p><img src="total.nuc.rnaseq_files/figure-html/unnamed-chunk-3-2.png" width="672" /></p>
</div>
<div id="cluster-analysis" class="section level3">
<h3>Cluster analysis</h3>
<p>Create a correlation matrix between the counts.</p>
<pre class="r"><code>cor_function=function(data){
  corr_matrix= matrix(0,ncol(data),ncol(data))
  for (i in seq(1,ncol(data))){
    for (j in seq(1,ncol(data))){
      x=cor.test(data[,i],  data[,j], method=&#39;pearson&#39;)
      cor_ij=as.numeric(x$estimate)
      corr_matrix[i,j]=cor_ij
    }
  }
  return(corr_matrix)
}


count_cor=cor_function(count_matrix_cpm_filt)
rownames(count_cor)=sample
colnames(count_cor)=sample</code></pre>
<p>Reshape correlation matrix and plot it.</p>
<pre class="r"><code>melted_count_corr=melt(count_cor)
ggheatmap=ggplot(data = melted_count_corr, aes(x=Var1, y=Var2, fill=value)) +
  geom_tile() +
  labs(title=&quot;Correlation Heatplot&quot;)


ggheatmap</code></pre>
<p><img src="total.nuc.rnaseq_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>This is the expected plot. The data clusters by fraction and the total and nuclear are more similar than the chromatin fraction.</p>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
