---
title: "ApaQTLs with all individuals"
author: "Briana Mittleman"
date: "9/25/2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(reshape2)
library(workflowr)
```


I am using the code from peakOverlap_oppstrand.Rmd analysis to call QTLs on the full set of individuals. (still missing 4 due to genotype issues- Remove 18500, 19092 and 19193, 18497 - at 35).   

Scripts:  
* APAqtl_nominal_oppstrand.sh   

* APAqtl_perm_Opp.sh    

```{bash, eval=F}
cat /project2/gilad/briana/threeprimeseq/data/perm_APAqtl_Opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total* > /project2/gilad/briana/threeprimeseq/data/perm_APAqtl_Opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_permRes.txt

cat /project2/gilad/briana/threeprimeseq/data/perm_APAqtl_Opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear* > /project2/gilad/briana/threeprimeseq/data/perm_APAqtl_Opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_permRes.txt

```





Write a script to ad the BH correction of the permuted QTL pvalues. I will write the plots to 


APAqtlpermCorrectQQplot.R


```{r, eval=F}

library(dplyr)


##total results
tot.perm= read.table("/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_Opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_permRes.txt",head=F, stringsAsFactors=F, col.names = c("pid", "nvar", "shape1", "shape2", "dummy", "sid", "dist", "npval", "slope", "ppval", "bpval"))

#BH correction
tot.perm$bh=p.adjust(tot.perm$bpval, method="fdr")

#plot qqplot
pdf("/project2/gilad/briana/threeprimeseq/output/plots/qqplot_total_APAperm.pdf") 
qqplot_total= qqplot(-log10(runif(nrow(tot.perm))), -log10(tot.perm$bpval),ylab="-log10 Total permuted pvalue", xlab="Uniform expectation", main="Total permuted pvalues for all snps")
abline(0,1)
dev.off()

#write df with BH  

write.table(tot.perm, file = "/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_Opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_permResBH.txt", col.names = T, row.names = F, quote = F)

##nuclear results  


nuc.perm= read.table("/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_Opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_permRes.txt",head=F, stringsAsFactors=F, col.names = c("pid", "nvar", "shape1", "shape2", "dummy", "sid", "dist", "npval", "slope", "ppval", "bpval"))
nuc.perm$bh=p.adjust(nuc.perm$bpval, method="fdr")


#plot qqplot
pdf("/project2/gilad/briana/threeprimeseq/output/plots/qqplot_nuclear_APAperm.pdf") 
qqplot(-log10(runif(nrow(nuc.perm))), -log10(nuc.perm$bpval),ylab="-log10 Nuclear permuted pvalue", xlab="Uniform expectation", main="Nuclear permuted pvalues for all snps")
abline(0,1)
dev.off()

# write df with BH
write.table(nuc.perm, file = "/project2/gilad/briana/threeprimeseq/data/perm_APAqtl_Opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_permResBH.txt", col.names = T, row.names = F, quote = F)



```


Write a script to run this:  

run_APAqtlpermCorrectQQplot.sh

```{bash, eval=F}
#!/bin/bash


#SBATCH --job-name=run_APAqtlpermCorrectQQplot
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=run_APAqtlpermCorrectQQplot.out
#SBATCH --error=run_APAqtlpermCorrectQQplot.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env


Rscript APAqtlpermCorrectQQplot.R
```


###Total results  


```{r}
tot_permBH=read.table("../data/perm_QTL_opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Total_permResBH.txt", header=T, stringsAsFactors = F)
```


Check to quality of the tests:  

```{r}
plot(tot_permBH$ppval, tot_permBH$bpval, xlab="Direct method", ylab="Beta approximation", main="Total Check plot")
abline(0, 1, col="red")
```

```{r}
plot(-log10(tot_permBH$bh), main="Total BH corrected pval")
abline(h=1,col="Red")
```


I am going to look how many variants pass the 10% FDR.


```{r}
tot_qtl_10= tot_permBH %>% filter(-log10(bh) > 1)
nrow(tot_qtl_10)
```

This is not accounting for the same peak in multiple genes. I want to look at the number of unique snps that are significant.  


```{r}
tot_qtl_10uniq= tot_permBH %>% filter(-log10(bh) > 1)  %>% summarise(n_distinct(sid)) 
tot_qtl_10uniq
```
###Nuclear results  


```{r}
nuc_permBH=read.table("../data/perm_QTL_opp/filtered_APApeaks_merged_allchrom_refseqGenes_pheno_Nuclear_permResBH.txt", header=T, stringsAsFactors = F)
```


Check to quality of the tests:  

```{r}
plot(nuc_permBH$ppval, nuc_permBH$bpval, xlab="Direct method", ylab="Beta approximation", main="Nuclear Check plot")
abline(0, 1, col="red")
```

```{r}
plot(-log10(nuc_permBH$bh), main="Nuclear BH corrected pval")
abline(h=1,col="Red")
```


I am going to look how many variants pass the 10% FDR.


```{r}
nuc_qtl_10= nuc_permBH %>% filter(-log10(bh) > 1)
nrow(nuc_qtl_10)
```

This is not accounting for the same peak in multiple genes. I want to look at the number of unique snps that are significant.  


```{r}
nuc_qtl_10uniq= nuc_permBH %>% filter(-log10(bh) > 1)  %>% summarise(n_distinct(sid)) 
nuc_qtl_10uniq
```


###Compare number of sig QTLs by FDR cuttoff

```{r}
nQTL_tot=c()
FDR=seq(.05, .5, .01)
for (i in FDR){
  x=tot_permBH %>% filter(bh < i ) %>% nrow()
  nQTL_tot=c(nQTL_tot, x)
}

FDR=seq(.05, .5, .01)
nQTL_nuc=c()
for (i in FDR){
  x=nuc_permBH %>% filter(bh < i ) %>% nrow()
  nQTL_nuc=c(nQTL_nuc, x)
}

nQTL=as.data.frame(cbind(FDR, Total=nQTL_tot, Nuclear=nQTL_nuc))
nQTL_long=melt(nQTL, id.vars = "FDR")

ggplot(nQTL_long, aes(x=FDR, y=value, by=variable, col=variable)) + geom_line(size=1.5) + labs(y="Number of Significant QTLs", title="APAqtls detected by FDR cuttoff", color="Fraction")
```

###Explore QTLs  
Look at distribution of SNP to peak in each fraction:   

```{r}
ggplot(nuc_qtl_10, aes(x=log10(abs(dist) + 1)) )+ geom_histogram(binwidth=.15, alpha=.5 ) + geom_histogram(data=tot_qtl_10, aes(x=log10(abs(dist) + 1)),fill="Red", alpha=.5,binwidth=.15)  +  annotate("text", x=1, y=950, col="Red", label="Total") + annotate("text", x=1, y=900, col="Black", label="Nuclear") + geom_rect(linetype=1, xmin=.5, xmax=1.5, ymin=850, ymax=1000, color="Black", alpha=0)
```
```{r}
ggplot(nuc_qtl_10, aes(x=log10(abs(dist) + 1)) )+ geom_density( alpha=.25 ,fill="Black") + geom_density(data=tot_qtl_10, aes(x=log10(abs(dist) + 1)),fill="Red", alpha=.25)  + annotate("text", x=1, y=.77, col="Red", label="Total") + annotate("text", x=1, y=.72, col="Black", label="Nuclear") + geom_rect(linetype=1, xmin=.5, xmax=1.5, ymin=.69, ymax=.8, color="Black", alpha=0)

```
