---
title: "Overlap APAqtls with CTCF qtls"
author: "Briana Mittleman"
date: "3/19/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

I will look at ctcf data too see if this insulator element could act as a mechanism for apa qtls. This is in line with the kinetic model. We know CTCF binding slows polymerase. We are testing if this slow down is associated with APA as well.  

The ctcf data I will use can be found at https://www.ncbi.nlm.nih.gov/pubmed/27010758  

I will download the normalized phenotype file each row a binding region and each column a sample.  

The regions are defined as chromosome, start, end in hg19. I will format this file so I have an ID like i do for the APA analysis.  

```{r}
library(tidyverse)
library(workflowr)
library(data.table)
library(cowplot)
```


##PAS Enrichment at CTCF
First, I will look at overlap between the PAS and these CTCF sites. I can do this with deep tools by making a bed file.  


CTCF2bed.py
```{bash,eval=F}
CTCF=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFbinding.csv", "r")
bedFile=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFbindingLoc.bed", "w")

for i,ln in enumerate(CTCF):
    if i >0: 
        chrm=ln.split(",")[0]
        start=ln.split(",")[1]
        end=ln.split(",")[2]
        bedFile.write("%s\t%s\t%s\n"%(chrm, start, end))
bedFile.close()
```

Deeptools plots:  

TotandNucAtCTCF_DTPlot_noMPFilt.sh

```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=TotandNucAtCTCF_DTPlot_noMPFilt
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=TotandNucAtCTCF_DTPlot_noMPFilt.out
#SBATCH --error=TotandNucAtCTCF_DTPlot_noMPFilt.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env


computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/mergedBW/Total_MergedBamCoverage.bw /project2/gilad/briana/threeprimeseq/data/mergedBW/Nuclear_MergedBamCoverage.bw -R /project2/gilad/briana/threeprimeseq/data/CTCF/CTCFbindingLoc.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/CTCF/TotalandNucAtCTCF.gz

plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/CTCF/TotalandNucAtCTCF.gz --refPointLabel "CTCF" --plotTitle "Combined 3' at CTCF" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/CTCF/TotalandNucAtCTCF.png
```

No enrichemnt  

##Call CTCF QTL

I want to reformat the phenotypes, this is easiest in R.


```{r}
CTCF=read.csv("../data/CTCF/CTCFbinding.csv", header=T) %>% mutate(ID= paste(chrm,start, end, sep=":")) %>% dplyr::select(chrm, start, end, ID, contains("NA"))


write.table(CTCF, file="../data/CTCF/CTCFbinding.pheno.bed",col.names = T, row.names = F, quote = F, sep="\t" )
```

put on midway  

```{bash,eval=F}
#remove header
sort -k1,1 -k2,2n CTCFbinding.pheno.bed > CTCFbinding.pheno.sort.bed
#add header

bgzip CTCFbinding.pheno.sort.bed
tabix CTCFbinding.pheno.sort.bed.gz  

#get the PCs 
#midway1
#export PATH=/project/gilad/software/midway1/qtltools-1.0:$PATH


QTLtools pca --bed /project2/gilad/briana/threeprimeseq/data/CTCF/CTCFbinding.pheno.sort.bed.gz  --scale --center --out /project2/gilad/briana/threeprimeseq/data/CTCF/CTCFbinding.pheno.sort.bed.PC.out


head -n 6 CTCFbinding.pheno.sort.bed.PC.out.pca > CTCFbinding.pheno.sort.bed.5PCs.out.pca
```


Make samples file: 


smaplesCTCF.py
```{bash,eval=F}
ctcf=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFbinding.csv", "r")
sampleFile=open("/project2/gilad/briana/threeprimeseq/data/CTCF/samples.txt", "w")

samplesVCF=open("/project2/gilad/briana/YRI_geno_hg19/vcf.samples.txt", "r")

samplesoK={}
for ln in samplesVCF:
  samList=ln.split()
  for i in samList:
      samplesoK[i]=""
      
print(samplesoK)
for i, ln in enumerate(ctcf):
    if i >0:
        lnList=ln.split(",")
        for each in lnList:
            if each in samplesoK.keys():
                sampleFile.write("%s\n"%(each))
            else: 
                print("notInvcf")
                
sampleFile.close()

    
```

**VCF file does not have these samples. **


CTCFqtl_nom.sh

```{bash,eval=F}
#!/bin/bash


#SBATCH --job-name=CTCFqtl_nom
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=CTCFqtl_nom.out
#SBATCH --error=CTCFqtl_nom.err
#SBATCH --partition=broadwl
#SBATCH --mem=12G
#SBATCH --mail-type=END
  
for i in $(seq 1 30)
do
/home/brimittleman/software/bin/FastQTL/bin/fastQTL.static --vcf /project2/gilad/briana/YRI_geno_hg19/allChrom.dose.filt.vcf.gz  --cov /project2/gilad/briana/threeprimeseq/data/CTCF/CTCFbinding.pheno.sort.bed.5PCs.out.pca --bed /project2/gilad/briana/threeprimeseq/data/CTCF/CTCFbinding.pheno.sort.bed.gz --out /project2/gilad/briana/threeprimeseq/data/CTCF/nom/fastqtl_CTCFbinding.nominal.out --chunk $i 30  --window 5e5 --include-samples /project2/gilad/briana/threeprimeseq/data/CTCF/samples.txt
done
```



##Use QTLs they called   

Info from site:  
This is the called QTLs 1% FDR threshold (q value <= 0.01) and kept only cluster variants defined as having P value within one order of magnitude to the P value of the lead variant for the same binding region.


I can make a file with the snp positions and I will look for these in my nominal APA data  
```{r}
ctcfQTL=read.csv("../data/CTCF/CTCFQTLS.csv")

ctcfQTL_snponly=ctcfQTL %>% dplyr::select(VARIANT_CHRM, VARIANT_POS) %>% mutate(snp_loc=paste(VARIANT_CHRM,VARIANT_POS,sep= ":")) %>% dplyr::select(snp_loc)


write.table(ctcfQTL_snponly, file="../data/CTCF/CTCFqtl_snps.txt", col.names = F, row.names = F, quote = F)
```


Look for these snps in nominal data: 

CTCFqtlinAPA.py

```{bash,eval=F}
def main(apa, ctcfQTL, outFile):
    fout=open(outFile,"w")
    ctcfdic={}
    for ln in open(ctcfQTL,"r"):
        snp=ln.split()[0]
        ctcfdic[snp]=""
    for ln in open(apa, "r"):
            snpApa =ln.split()[1]   
            if snpApa in ctcfdic.keys():
                fout.write(ln)
    fout.close()
                

if __name__ == "__main__":
    import sys
    fraction=sys.argv[1]
    OutFile=sys.argv[2]
    ctcfQTL="/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFqtl_snps.txt"
    if fraction=="Total":
        nomFile="/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt"
    else:
        nomFile="/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt"
    main(nomFile, ctcfQTL, OutFile) 
```


Run:
run_CTCFqtlinAPA.sh
```{bash,eval=F}
#!/bin/bash


#SBATCH --job-name=run_CTCFqtlinAPA
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=run_CTCFqtlinAPA.out
#SBATCH --error=run_CTCFqtlinAPA.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env

python CTCFqtlinAPA.py "Total" "/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFQtlinTotalAPA.txt"
python CTCFqtlinAPA.py "Nuclear" "/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFQtlinNuclearAPA.txt"
```


Make empirical distribution: 

I can do empirical distribution based on genes not in this set. I will make a list of the genes with] an overlap in total and in nuclear.  

I can then find the matched peak numbers based on the genes that do have an overlap.


getCTCFgenes.py
```{bash,eval=F}


apaNuc=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFQtlinNuclearAPA.txt", "r")
apaTot=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFQtlinTotalAPA.txt","r")

nucGenes=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFQtlinNuclearAPA_Genes.txt", "w")
totGenes=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFQtlinTotalAPA_Genes.txt", "w")


def overlapGenes(inFile, outFile):
     #make dictionary with gene (this will have unique)
     geneDic={}
     for ln in inFile:
          gene=ln.split()[0].split(":")[-1].split("_")[0]
          if gene not in geneDic.keys():
              geneDic[gene]=""
     for k,v in geneDic.items():
         outFile.write("%s\n"%(k))
     outFile.close()
     
overlapGenes(apaTot, totGenes)
overlapGenes(apaNuc,nucGenes)

```


Plot these compared to the actuall apa QTLs.  
```{r}
ctcfinTot=read.table("../data/CTCF/CTCFQtlinTotalAPA.txt",stringsAsFactors = F,col.names = c("ID", "snp", "dist", "pval", "slope"))

apaTot=read.table("../data/perm_QTL_GeneLocAnno_noMP_5percov/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total.fixed.pheno_5perc_permResBH.txt", header=T, stringsAsFactors=F)%>% drop_na()


ctcfinNuc=read.table("../data/CTCF/CTCFQtlinNuclearAPA.txt",stringsAsFactors = F,col.names = c("ID", "snp", "dist", "pval", "slope"))

apaNuc=read.table("../data/perm_QTL_GeneLocAnno_noMP_5percov/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear.fixed.pheno_5perc_permResBH.txt", header=T, stringsAsFactors=F) %>% drop_na()


```


I need the nominal association so i am making a fair comparison  


nom4apaQTLSnps.py
```{bash,eval=F}
totQTL=open("/project2/gilad/briana/threeprimeseq/data/ApaQTLs/TotalapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt","r")
totNom=open("/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt", "r")
totNomQTL=open("/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Totalapa.NomPvalAssoc4allQTL.txt", "w")

nucQTL=open("/project2/gilad/briana/threeprimeseq/data/ApaQTLs/NuclearapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt","r")
nucNom=open("/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt","r")
nucNomQTL=open("/project2/gilad/briana/threeprimeseq/data/ApaQTLs/Nuclearapa.NomPvalAssoc4allQTL.txt","w")


def allAssocAPAqtl(QTL, nom, outFile):
    snpdic={}
    for ln in QTL:
        snp =ln.split()[5]
        if snp not in snpdic.keys():
            snpdic[snp]=""
    for ln in nom:
        snp=ln.split()[1]
        if snp in snpdic.keys():
            outFile.write(ln)
    outFile.close()
    
    
allAssocAPAqtl(totQTL, totNom, totNomQTL)
allAssocAPAqtl(nucQTL, nucNom, nucNomQTL)


```

```{r}
totQTLnom=read.table("../data/CTCF/Totalapa.NomPvalAssoc4allQTL.txt", stringsAsFactors = F,  col.names = c("ID", "snp", "dist", "pval", "slope"))
nucQTLnom=read.table("../data/CTCF/Nuclearapa.NomPvalAssoc4allQTL.txt", stringsAsFactors = F,  col.names = c("ID", "snp", "dist", "pval", "slope"))
```


```{r}



qqplot(-log10(runif(nrow(totQTLnom))),-log10(totQTLnom$pval), ylab="-log10 Tota pval", xlab="Uniform expectation", main="Total pvalue association for CTCF qtls") 
points(sort(-log10(runif(nrow(ctcfinTot)))), sort(-log10(ctcfinTot$pval)),col="blue")
abline(0,1)
legend("bottomright", legend=c("ApaQTLs", "CTCF qtls"),col=c("black", "blue"), pch=16,bty = 'n')
```


```{r}

qqplot(-log10(runif(nrow(nucQTLnom))),-log10(nucQTLnom$pval), ylab="-log10 Nuclear pval", xlab="Uniform expectation", main="Nuclear pvalue association for CTCF qtls") 
points(sort(-log10(runif(nrow(ctcfinNuc)))), sort(-log10(ctcfinNuc$pval)),col="blue")
abline(0,1)
legend("bottomright", legend=c("ApaQTLs", "CTCF qtls"),col=c("black", "blue"), pch=16,bty = 'n')
```

Use different black line, dont condition on apaQTLs look at every tested snp. I can sort the pvalues in the nominal file then sample every 100th value to deal wilth how large the vector is:


Python interactively 
```{bash,eval=F}
totNom="/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt"
nucNom="/project2/gilad/briana/threeprimeseq/data/nominal_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz.qqnorm_allNomRes.txt"

import numpy as np
import pandas as pd

Nomnames=["ID", "snp", "dist", "pval", "slope"]
nomDF=pd.read_table(totNom, sep=" ", names=Nomnames, header=None)
a=nomDF.iloc[:,3]
sorted=np.sort(a)
final=sorted[0::100]

np.savetxt('/project2/gilad/briana/threeprimeseq/data/CTCF/TotNomPvalsOneHund.out', final) 


nomNucDF=pd.read_table(nucNom, sep=" ", names=Nomnames, header=None)
b=nomNucDF.iloc[:,3]
sortedNuc=np.sort(b)
finalNuc=sortedNuc[0::100]
np.savetxt('/project2/gilad/briana/threeprimeseq/data/CTCF/NucNomPvalsOneHund.out', finalNuc) 
```


use this:

```{r}
totnomPvalsmall=read.table("../data/CTCF/TotNomPvalsOneHund.out", header = F, stringsAsFactors = F, col.names = c("pval"))


qqplot(-log10(runif(nrow(totnomPvalsmall))),-log10(totnomPvalsmall$pval), ylab="-log10 Total pval", xlab="Uniform expectation", main="Total pvalue association for CTCF qtls") 
points(sort(-log10(runif(nrow(ctcfinTot)))), sort(-log10(ctcfinTot$pval)),col="blue")
abline(0,1)
legend("bottomright", legend=c("1/100 Apa SNPs", "CTCF qtls"),col=c("black", "blue"), pch=16,bty = 'n')
```

```{r}
nucnomPvalsmall=read.table("../data/CTCF/NucNomPvalsOneHund.out", header = F, stringsAsFactors = F, col.names = c("pval"))


qqplot(-log10(runif(nrow(nucnomPvalsmall))),-log10(nucnomPvalsmall$pval), ylab="-log10 Nuclear pval", xlab="Uniform expectation", main="Nuclear pvalue association for CTCF qtls") 
points(sort(-log10(runif(nrow(ctcfinNuc)))), sort(-log10(ctcfinNuc$pval)),col="blue")
abline(0,1)
legend("bottomright", legend=c("1/100 Apa SNPs", "CTCF qtls"),col=c("black", "blue"), pch=16,bty = 'n')
```


###Look at some examples  

Look at the points in the nuclear fraction that are -log10(pval)>5.

```{r}
ctcfinNuc %>% filter(-log10(pval)>5) %>% arrange(pval)
```
Top genes with effects:

* CHD1L

* DHRS7  

* DDI2

* TAB1

* CCDC18  (CCDC18-AS1)

* B4GALt7  

* TMSF4

* ERVK13-1

* DEK 

* RGS14

* KLF12

I will see if any of these have a called apaQTL  

```{r}
qtlNames=c("ID")
apaQTLsNuc=read.table("../data/ApaQTLs/NuclearapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt", stringsAsFactors = F,  col.names = c("pid", "nvar", "shape1", "shape2", "dummy", "sid", "dist", "npval", "slope", "ppval", "bpval","bh"))
apaQTLsTot=read.table("../data/ApaQTLs/TotalapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt", stringsAsFactors = F,  col.names = c("pid", "nvar", "shape1", "shape2", "dummy", "sid", "dist", "npval", "slope", "ppval", "bpval","bh"))
```

CHD1l:

1:146735002:146735080:CHD1L_-_peak9190- 1:146728013


1:146767359:146767443:CHD1L_-_peak9202 1:146748962 

DHRS7:  

14:60611650:60611732:DHRS7_+_peak48511



I should add the ctcfQTLs as a bedfile:


ctcfQTL2bed.py
```{bash,eval=F}
inQTL=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFQTLS.csv"."r")
outBed=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFQTL.bed","w")

for i, ln in enumerate(inQTL):
    if i >0:
        chrom=ln.split(",")[0]
        start=ln.split(",")[1]
        end=ln.split(",")[2]
        id=chrom + ":" + start + ":" + end  
        varEnd=int(ln.split(",")[4])
        varStart=varEnd-1
        varChrom=ln.split(",")[3]
        outBed.write("%s\t%d\t%d\t%s\n"%(varChrom, varStart, varEnd, id))
    
outBed.close()  
```



Intersect this with out QTLs and all of those in LD with ours:  

interactively


```{bash,eval=F}


allOverlapQTL='/project2/gilad/briana/threeprimeseq/data/GWAS_overlap_processed/AllOverlapSnps.bed'


import pybedtools as pybedtools

ctcf_file=open("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFQTL.bed","r")
ctcf=pybedtools.BedTool(ctcf_file)
snps_file=open(allOverlapQTL, "r")
snps=pybedtools.BedTool(snps_file)
snpOverctcf=snps.intersect(ctcf, wa=True,wb=True)
snpOverctcf.saveas("/project2/gilad/briana/threeprimeseq/data/CTCF/CTCFqtlOverlapAPAqtls.txt")
    
    
```



One interesting exaple 

ERVK13-1 chr16:2,709,806-2,712,090

CTCF site between 2 peaks. overlap with CTCF qtl and nuclear qtl associated with downstream QTL  


16	2710797	2710798

the qtl is snp_16_2710771

TAB1  
chr22:39,792,017-39,826,801
22,39795500,39795832,22,39822116,rs5757667,A,G,AA=G;AN=102;AC=73;VT=SNP,0.871868708751,3.54445306103e-09,-1.2411307501,0.715823195148,0.121500443991,2.055844e-06



Overall does not look to be a big driver  
