---
title: "APA QTL example Heatmap"
author: "Briana Mittleman"
date: "3/26/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(tidyverse)
library(reshape2)
library(cowplot)
```


Start with EIF2a example:

3	150302009	150302010	peak114357:EIF2A	5.39078186842105e-07	+


Get the phenotype

```{bash,eval=F}
less /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz | grep EIF2A_ > /project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/EIF2a_TotalPeaksPheno.txt

less chr3.dose.filt.vcf.gz | grep 3:150302010 > /project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/EIF2a_TotalPeaksGenotype.txt

less chr3.dose.filt.vcf.gz | head -n14 | tail -n1 > /project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/genotypeHeader.txt
```


```{r}
phenohead=read.table("../data/ExampleQTLplot2/Phenotypeheader.txt", header = T,stringsAsFactors = F)
phenoEIF=read.table("../data/ExampleQTLplot2/EIF2a_TotalPeaksPheno.txt", col.names =colnames(phenohead),stringsAsFactors = F)


meltpheno=melt(phenoEIF, id.vars = "chrom", value.name = "Ratio", variable.name = "Individual") %>% separate(Ratio, into=c("num", "denom"), sep="/") 

meltpheno$Individual= as.character(meltpheno$Individual)
meltpheno$num= as.numeric(meltpheno$num)
meltpheno$denom=as.numeric(meltpheno$denom)
```

I want to join the genotype.  

```{r}
genoHead=read.table("../data/ExampleQTLplot2/genotypeHeader.txt", header = T,stringsAsFactors = F)
genoEIF=read.table("../data/ExampleQTLplot2/EIF2a_TotalPeaksGenotype.txt", col.names =colnames(genoHead),stringsAsFactors = F ) %>% select(ID,contains("NA")) 

lettersGeno=read.table("../data/ExampleQTLplot2/EIF2a_TotalPeaksGenotype.txt", col.names =colnames(genoHead),stringsAsFactors = F, colClasses = c("character") ) %>% select(REF, ALT)

refAllele=as.character(lettersGeno$REF)
altAllele=as.character(lettersGeno$ALT)

genoMelt=melt(genoEIF, id.vars = "ID", value.name = "FullGeno", variable.name = "Individual" ) %>% separate(FullGeno, into=c("geno","dose","extra1"), sep=":") %>% select(Individual, dose) %>% mutate(genotype=ifelse(round(as.integer(dose))==0, paste(refAllele, refAllele, sep=""), ifelse(round(as.integer(dose))==1, paste(refAllele,altAllele, sep=""), paste(altAllele,altAllele,sep=""))))
genoMelt$Individual= as.character(genoMelt$Individual)



```


Join these:
```{r}
PhenandGene= meltpheno %>% inner_join(genoMelt, by="Individual")  %>% group_by(chrom, genotype) %>% summarise(SumNum=sum(num), SumDenom=sum(denom)) %>% mutate(PAU=SumNum/SumDenom) %>% separate(chrom, into=c("chr","start", "end", "id"), sep=":") %>% separate(id, into=c("gene", "strand", "peak"), sep="_")
```

Check sums

```{r}
Groupsumscheck = PhenandGene %>% group_by(genotype) %>% summarise(SUM=sum(PAU))

```


```{r}

my_palette <- colorRampPalette(c("white", "khaki1", "orange", "red", "darkred", "black"))



eif2aplot=ggplot(PhenandGene, aes(peak, genotype)) + geom_tile(aes(fill = PAU))+ scale_fill_gradientn(colors =my_palette(100)) + labs(title="EIF2A", y="Genotype",x="PAS")

eif2aplot


ggsave(eif2aplot, file="../output/plots/testEIF2A.png")

```



###Make plot reproducible on midway  

I want the script to take a fraction, gene, chr, a snp (chr:loc) 


steps: 
* get phenotypes from /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz and /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno_5perc.fc.gz 
* get genotype from /project2/gilad/briana/YRI_geno_hg19/chrX.dose.filt.vcf.gz
*rscript for making plot  


I will write the rscript first:

makeQTLheatmap.R

```{r,eval=F}
library(tidyverse)
library(reshape2)
library(optparse)
library(cowplot)


option_list = list(
  make_option(c("-P", "--pheno"), action="store", default=NA, type='character',
              help="input pheno file"),
  make_option(c("-G", "--geno"), action="store", default=NA, type='character',
              help="input genotype"),
  make_option(c("-g", "--gene"), action="store", default=NA, type='character',
              help="gene"),
  make_option(c("-o", "--output"), action="store", default=NA, type='character',
              help="output file for plot")
)

opt_parser <- OptionParser(option_list=option_list)
opt <- parse_args(opt_parser)


phenohead=read.table("/project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/Phenotypeheader.txt", header = T,stringsAsFactors = F)
pheno=read.table(opt$pheno, col.names =colnames(phenohead),stringsAsFactors = F)


meltpheno=melt(pheno, id.vars = "chrom", value.name = "Ratio", variable.name = "Individual") %>% separate(Ratio, into=c("num", "denom"), sep="/") 

meltpheno$Individual= as.character(meltpheno$Individual)
meltpheno$num= as.numeric(meltpheno$num)
meltpheno$denom=as.numeric(meltpheno$denom)


genoHead=read.table("/project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/genotypeHeader.txt", header = T,stringsAsFactors = F)
geno=read.table(opt$geno, col.names =colnames(genoHead),stringsAsFactors = F ) %>% select(ID,contains("NA")) 


lettersGeno=read.table(opt$geno, col.names =colnames(genoHead),stringsAsFactors = F,colClasses = c("character")) %>% select(REF, ALT)

refAllele=lettersGeno$REF
altAllele=lettersGeno$ALT


genoMelt=melt(geno, id.vars = "ID", value.name = "FullGeno", variable.name = "Individual" ) %>% separate(FullGeno, into=c("geno","dose","extra1"), sep=":") %>% select(Individual, dose) %>% mutate(genotype=ifelse(round(as.integer(dose))==0, paste(refAllele, refAllele, sep=""), ifelse(round(as.integer(dose))==1, paste(refAllele,altAllele, sep=""), paste(altAllele,altAllele,sep=""))))
genoMelt$Individual= as.character(genoMelt$Individual)

PhenandGene= meltpheno %>% inner_join(genoMelt, by="Individual")  %>% group_by(chrom, genotype) %>% summarise(SumNum=sum(num), SumDenom=sum(denom)) %>% mutate(PAU=SumNum/SumDenom) %>% separate(chrom, into=c("chr","start", "end", "id"), sep=":") %>% separate(id, into=c("gene", "strand", "peak"), sep="_")



my_palette <- colorRampPalette(c("white", "khaki1", "orange", "red", "darkred", "black"))



heatplot=ggplot(PhenandGene, aes(peak, genotype)) + geom_tile(aes(fill = PAU))+ scale_fill_gradientn(colors =my_palette(100)) + labs(title=opt$gene, y="Genotype" , x= "PAS")



ggsave(heatplot, filename=opt$output, device=png, height=10, width=10)
```


qtlHeatmap.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=qtlHeatmap
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=qtlHeatmap.out
#SBATCH --error=qtlHeatmap.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env

Fraction=$1
gene=$2
chrom=$3
snp=$4


less /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.${Fraction}.fixed.pheno_5perc.fc.gz | grep ${gene}_ > /project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/${gene}_${Fraction}PeaksPheno.txt


less /project2/gilad/briana/YRI_geno_hg19/chr${chrom}.dose.filt.vcf.gz | grep ${snp} > /project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/${gene}_${Fraction}PeaksGenotype.txt


Rscript makeQTLheatmap.R -P /project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/${gene}_${Fraction}PeaksPheno.txt -G  /project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/${gene}_${Fraction}PeaksGenotype.txt -g ${gene} -o /project2/gilad/briana/threeprimeseq/data/ExampleQTLPlots2/${gene}_${Fraction}_${SNP}.png
```

```{r}
totalQTLs=read.table("../data/ApaQTLs/TotalapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt", header=F)
```
14:65401627:65401711:CHURC1_-_peak48994

14:65389250
```{bash,eval=F}
sbatch qtlHeatmap.sh "Total" "CHURC1" "14" "14:65389250"
```
12:57489617:57489715:STAT6_+_peak36983 12:57489648
```{bash,eval=F}
sbatch qtlHeatmap.sh "Total" "STAT6" "12" "12:57489648"
```


Try a nuclear one: 

```{r}
nucQTLs=read.table("../data/ApaQTLs/NuclearapaQTLs.GeneLocAnno.noMP.5perc.10FDR.txt", header=F)

```
19:4688114:4688228:DPP9_+_peak77244

19:4680128

```{bash,eval=F}
sbatch qtlHeatmap.sh "Nuclear" "DPP9" "19" "19:4680128"
```

4:83355978:83356052:ENOPH1_-_peak121076 4:83352186
```{bash,eval=F}
sbatch qtlHeatmap.sh "Nuclear" "ENOPH1" "4" "4:83352186"
```


