---
title: "Chromatin marks and PAS"
author: "Briana Mittleman"
date: "3/20/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

http://science.sciencemag.org/content/352/6291/aad9926.full?ijkey=fkp/DIzVNS9RY&keytype=ref&siteid=sci


This article talks about chromatin modifications for heterochromatin downstream of PAS. I will look at enrichment for repressive histone marks downstream of my called PAS.  


Repressive marks H3K27me3, H3K9me3


http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeUwHistone/

H3k27me3
H3k36me3
H3k4me3


Deeptools plot  


h3k27me3DTmypeaks.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=h3k27me3DTmypeaks
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=h3k27me3DTmypeaks.out
#SBATCH --error=h3k27me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep1.bigWig -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3K27me3_myPeaksNompfilt.gz


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3K27me3_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "H3K27me3 at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3K27me3_myPeaksNompfilt.png
```


Put all of the marks on one plot:  

I also want to just use the last base of the peak  APAPAS_5percCov_fixedStrand.bed
histonemarksDTmypeaks.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=histonemarksDTmypeaks
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=histonemarksDTmypeaks.out
#SBATCH --error=histonemarksDTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k36me3StdRawRep1.bigWig  /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k4me3StdRawRep1.bigWig -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/HistMarks_myPeaksNompfilt.gz


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/HistMarks_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "Histone marks at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/HistMarks_myPeaksNompfilt.png
```


Scales are too different to put these on the same spot:  

H3k27me3DTmyPAS.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=H3k27me3DTmypeaks.sh
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=H3k27me3DTmypeaks.out
#SBATCH --error=H3k27me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bw  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 5000 -a 5000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt.gz --outFileNameMatrix /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt_matrix.txt --outFileSortedRegions /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt_SortedRegions.txt 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "H3k27me3 at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt.png
```


H3k36me3DTmyPAS.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=H3k27me3DTmypeaks.sh
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=H3k36me3DTmypeaks.out
#SBATCH --error=H3k36me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bw   -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 5000 -a 5000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt.gz --outFileNameMatrix /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt_matrix.txt --outFileSortedRegions /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt_SortedRegions.txt 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "H3k36me3 at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt.png
```



H3k4me3DTmyPAS.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=H3k4me3DTmypeaks.sh
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=H3k36me3DTmypeaks.out
#SBATCH --error=H3k36me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bw  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 5000 -a 5000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt.gz --outFileNameMatrix project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt_matrix.txt --outFileSortedRegions /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt_SortedRegions.txt 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "H3k4me3 at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt.png
```

##Download both replicates for these and merge:
mergeH3k27me3.sh
```{bash,eval=F}

#!/bin/bash

#SBATCH --job-name=mergeH3k27me3
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeH3k27me3.out
#SBATCH --error=mergeH3k27me3.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env


bigWigMerge /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep2.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bedGraph


sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bedGraph /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.sort.bedGraph

bedGraphToBigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.sort.bedGraph /project2/gilad/briana/genome_anotation_data/chrom.length.chr.txt  /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bw



```

mergeH3k36me3.sh
```{bash,eval=F}

#!/bin/bash

#SBATCH --job-name=mergeH3k36me3
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeH3k36me3.out
#SBATCH --error=mergeH3k36me3.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env


bigWigMerge /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k36me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k36me3StdRawRep2.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bedGraph


sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bedGraph > /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.sort.bedGraph

bedGraphToBigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.sort.bedGraph /project2/gilad/briana/genome_anotation_data/chrom.length.chr.txt  /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bw



```

mergeH3k4me3.sh
```{bash,eval=F}

#!/bin/bash

#SBATCH --job-name=mergeH3k4me3
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeH3k4me3.out
#SBATCH --error=mergeH3k4me3.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env


bigWigMerge /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k4me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k4me3StdRawRep2.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bedGraph


sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bedGraph >/project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.sort.bedGraph

bedGraphToBigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.sort.bedGraph /project2/gilad/briana/genome_anotation_data/chrom.length.chr.txt  /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bw



```



##MNASE:


MNASEmyPAS.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPAS
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAS.out
#SBATCH --error=MNASEmyPAS.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNompfilt.png
```

Try second mnase track.  

MNASEmyPAS_secondfile.sh
```{bash,eval=F}

#!/bin/bash

#SBATCH --job-name=MNASEmyPAS_secondfile
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAS_secondfile.out
#SBATCH --error=MNASEmyPAS_secondfile.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeSydhNsomeGm12878Sig.bigWig  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase2_myPeaksNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase2_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase2_myPeaksNompfilt.png

```




Convert to PAS ratehr than peak:
APAPeaks_5percCov_fixedStrand_INTRON.bed

```{bash,eval=F}
python Peak2PAS.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_INTRON.bed /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_INTRON.bed
```


Run this with intronic vs utr  
MNASEmyPASIntron.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPASIntron
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASIntron.out
#SBATCH --error=MNASEmyPASIntron.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_INTRON.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksIntronNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksIntronNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Intronic Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksIntronNompfilt.png
```

Nuclear specific:  
```{bash,eval=F}
python Peak2PAS.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc.bed /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc.bed

```


MNASEmyPASNuclear.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPASNuclear
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASNuclear.out
#SBATCH --error=MNASEmyPASNuclear.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Nuclear specific PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearNompfilt.png
```


Nuclear Intronic:  

APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron.bed

```{bash,eval=F}
python Peak2PAS.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron.bed /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc_Intron.bed

```


MNASEmyPASNuclearIntronic.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPASNuclearIntronic
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASNuclearIntronic.out
#SBATCH --error=MNASEmyPASNuclearIntronic.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc_Intron.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearIntronNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearIntronNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Nuclear specific PAS in Intron" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearIntronNompfilt.png
```

