---
title: "Chromatin marks and PAS"
author: "Briana Mittleman"
date: "3/20/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(tidyverse)
library(workflowr)
library(data.table)
library(cowplot)
```

http://science.sciencemag.org/content/352/6291/aad9926.full?ijkey=fkp/DIzVNS9RY&keytype=ref&siteid=sci


This article talks about chromatin modifications for heterochromatin downstream of PAS. I will look at enrichment for repressive histone marks downstream of my called PAS.  


Repressive marks H3K27me3, H3K9me3


http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeUwHistone/

H3k27me3
H3k36me3
H3k4me3


Deeptools plot  


h3k27me3DTmypeaks.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=h3k27me3DTmypeaks
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=h3k27me3DTmypeaks.out
#SBATCH --error=h3k27me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep1.bigWig -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3K27me3_myPeaksNompfilt.gz


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3K27me3_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "H3K27me3 at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3K27me3_myPeaksNompfilt.png
```


Put all of the marks on one plot:  

I also want to just use the last base of the peak  APAPAS_5percCov_fixedStrand.bed
histonemarksDTmypeaks.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=histonemarksDTmypeaks
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=histonemarksDTmypeaks.out
#SBATCH --error=histonemarksDTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k36me3StdRawRep1.bigWig  /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k4me3StdRawRep1.bigWig -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/HistMarks_myPeaksNompfilt.gz


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/HistMarks_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "Histone marks at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/HistMarks_myPeaksNompfilt.png
```


Scales are too different to put these on the same spot:  

H3k27me3DTmyPAS.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=H3k27me3DTmypeaks.sh
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=H3k27me3DTmypeaks.out
#SBATCH --error=H3k27me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bw  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 5000 -a 5000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt.gz --outFileNameMatrix /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt_matrix.txt --outFileSortedRegions /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt_SortedRegions.txt 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "H3k27me3 at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k27me3_myPeaksNompfilt.png
```


H3k36me3DTmyPAS.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=H3k27me3DTmypeaks.sh
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=H3k36me3DTmypeaks.out
#SBATCH --error=H3k36me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bw   -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 5000 -a 5000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt.gz --outFileNameMatrix /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt_matrix.txt --outFileSortedRegions /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt_SortedRegions.txt 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "H3k36me3 at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k36me3_myPeaksNompfilt.png
```



H3k4me3DTmyPAS.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=H3k4me3DTmypeaks.sh
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=H3k36me3DTmypeaks.out
#SBATCH --error=H3k36me3DTmypeaks.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bw  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 5000 -a 5000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt.gz --outFileNameMatrix project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt_matrix.txt --outFileSortedRegions /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt_SortedRegions.txt 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "H3k4me3 at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/H3k4me3_myPeaksNompfilt.png
```

##Download both replicates for these and merge:
mergeH3k27me3.sh
```{bash,eval=F}

#!/bin/bash

#SBATCH --job-name=mergeH3k27me3
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeH3k27me3.out
#SBATCH --error=mergeH3k27me3.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env


bigWigMerge /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k27me3StdRawRep2.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bedGraph


sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bedGraph /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.sort.bedGraph

bedGraphToBigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.sort.bedGraph /project2/gilad/briana/genome_anotation_data/chrom.length.chr.txt  /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k27me3.bw



```

mergeH3k36me3.sh
```{bash,eval=F}

#!/bin/bash

#SBATCH --job-name=mergeH3k36me3
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeH3k36me3.out
#SBATCH --error=mergeH3k36me3.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env


bigWigMerge /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k36me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k36me3StdRawRep2.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bedGraph


sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bedGraph > /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.sort.bedGraph

bedGraphToBigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.sort.bedGraph /project2/gilad/briana/genome_anotation_data/chrom.length.chr.txt  /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k36me3.bw



```

mergeH3k4me3.sh
```{bash,eval=F}

#!/bin/bash

#SBATCH --job-name=mergeH3k4me3
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=mergeH3k4me3.out
#SBATCH --error=mergeH3k4me3.err
#SBATCH --partition=broadwl
#SBATCH --mem=36G
#SBATCH --mail-type=END
module load Anaconda3
source activate three-prime-env


bigWigMerge /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k4me3StdRawRep1.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeUwHistoneGm06990H3k4me3StdRawRep2.bigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bedGraph


sort -k1,1 -k2,2n /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bedGraph >/project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.sort.bedGraph

bedGraphToBigWig /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.sort.bedGraph /project2/gilad/briana/genome_anotation_data/chrom.length.chr.txt  /project2/gilad/briana/threeprimeseq/data/ChipSeq/MergedGm06990H3k4me3.bw



```



##MNASE:


MNASEmyPAS.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPAS
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAS.out
#SBATCH --error=MNASEmyPAS.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNompfilt.png
```

Try second mnase track.  

MNASEmyPAS_secondfile.sh
```{bash,eval=F}

#!/bin/bash

#SBATCH --job-name=MNASEmyPAS_secondfile
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAS_secondfile.out
#SBATCH --error=MNASEmyPAS_secondfile.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/wgEncodeSydhNsomeGm12878Sig.bigWig  -R /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase2_myPeaksNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase2_myPeaksNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at All Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase2_myPeaksNompfilt.png

```




Convert to PAS ratehr than peak:
APAPeaks_5percCov_fixedStrand_INTRON.bed

```{bash,eval=F}
python Peak2PAS.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_INTRON.bed /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_INTRON.bed
```


Run this with intronic vs utr  
MNASEmyPASIntron.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPASIntron
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASIntron.out
#SBATCH --error=MNASEmyPASIntron.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_INTRON.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksIntronNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksIntronNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Intronic Called PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksIntronNompfilt.png
```

Nuclear specific:  
```{bash,eval=F}
python Peak2PAS.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc.bed /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc.bed

```


MNASEmyPASNuclear.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPASNuclear
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASNuclear.out
#SBATCH --error=MNASEmyPASNuclear.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Nuclear specific PAS" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearNompfilt.png
```


Nuclear Intronic:  

APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron.bed

```{bash,eval=F}
python Peak2PAS.py /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPeaks_5percCov_fixedStrand_SigUsageNuc_Intron.bed /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc_Intron.bed

```


MNASEmyPASNuclearIntronic.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPASNuclearIntronic
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASNuclearIntronic.out
#SBATCH --error=MNASEmyPASNuclearIntronic.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_SigUsageNuc_Intron.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearIntronNompfilt.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearIntronNompfilt.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Nuclear specific PAS in Intron" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksNuclearIntronNompfilt.png
```


###Sepatate by mean usage (strong/weak)

####Total  
Mean usage: 

top 20% and bottom 20%. by mean usage  

```{r}
meanUsageTot=read.table("../data/PeaksUsed_noMP_5percCov/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt", col.names = c("chr", "start", "end","gene", "strand", "name", "meanUsage")) %>% mutate(perc=ntile(meanUsage,n=100)) 


meanUsageTot_bot20=meanUsageTot %>% filter(perc <20) %>% dplyr::select(name)
meanUsageTot_top20=meanUsageTot %>% filter(perc >80)%>% dplyr::select(name)
```


Write out the peaks:

```{r}
write.table(meanUsageTot_bot20, file="../data/PeaksUsed_noMP_5percCov/TotalPeaksBottom20Usage.txt", quote=F, row.names = F, col.names = F, sep="\t")

write.table(meanUsageTot_top20, file="../data/PeaksUsed_noMP_5percCov/TotalPeaksTop20Usage.txt", quote=F, row.names = F, col.names = F, sep="\t")
```

Copy to 
/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/


Subset PAS file: 

subsetPAStottop20perc.py

```{bash,eval=F}
top20=open("/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/TotalPeaksTop20Usage.txt", "r")


AllPas=open("/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed", "r")

Top20PAS=open("/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_top20inTotal.bed", "w")





def subsetPAS(use, outpas, PAS):
    okPAS={}
    for ln in use:
        peak=ln.strip()
        okPAS[peak]=""
    for ln in PAS:
        peaknum=ln.split()[3].split(":")[-1]
        print
        if peaknum in okPAS.keys():
            outpas.write(ln)
    outpas.close()


subsetPAS(top20, Top20PAS, AllPas)
```

subsetPAStotbottom20perc.py

```{bash,eval=F}

bottom20=open("/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/TotalPeaksBottom20Usage.txt", "r")

AllPas=open("/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed", "r")


Bottom20PAS=open("/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_bottom20inTotal.bed", "w")




def subsetPAS(use, outpas, PAS):
    okPAS={}
    for ln in use:
        peak=ln.strip()
        okPAS[peak]=""
    for ln in PAS:
        peaknum=ln.split()[3].split(":")[-1]
        print
        if peaknum in okPAS.keys():
            outpas.write(ln)
    outpas.close()


subsetPAS(bottom20, Bottom20PAS, AllPas)

```


Deeptools plots for these:

MNASEmyPAStop20tot.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPAStop20tot
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAStop20tot.out
#SBATCH --error=MNASEmyPAStop20tot.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_top20inTotal.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20tot.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20tot.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Top 20% Total Usage" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20tot.png
```




MNASEmyPASbottom20tot.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPASbottom20tot
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASbottom20tot.out
#SBATCH --error=MNASEmyPASbottom20tot.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_bottom20inTotal.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20tot.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20tot.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Bottom 20% Total Usage" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20tot.png
```


####Nuclear  


Mean usage: 

top 20% and bottom 20%. by mean usage  

```{r}
meanUsageNuc=read.table("../data/PeaksUsed_noMP_5percCov/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt", col.names = c("chr", "start", "end","gene", "strand", "name", "meanUsage")) %>% mutate(perc=ntile(meanUsage,n=100)) 


meanUsageNuc_bot20=meanUsageNuc %>% filter(perc <20) %>% dplyr::select(name)
meanUsageNuc_top20=meanUsageNuc %>% filter(perc >80)%>% dplyr::select(name)
```


Write out the peaks:

```{r}
write.table(meanUsageNuc_bot20, file="../data/PeaksUsed_noMP_5percCov/NuclearPeaksBottom20Usage.txt", quote=F, row.names = F, col.names = F, sep="\t")

write.table(meanUsageNuc_top20, file="../data/PeaksUsed_noMP_5percCov/NuclearPeaksTop20Usage.txt", quote=F, row.names = F, col.names = F, sep="\t")
```

Copy to 
/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/


Subset PAS file: 

subsetPASnuctop20perc.py

```{bash,eval=F}
top20=open("/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/NuclearPeaksTop20Usage.txt", "r")


AllPas=open("/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed", "r")

Top20PAS=open("/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_top20inNuclear.bed", "w")





def subsetPAS(use, outpas, PAS):
    okPAS={}
    for ln in use:
        peak=ln.strip()
        okPAS[peak]=""
    for ln in PAS:
        peaknum=ln.split()[3].split(":")[-1]
        print
        if peaknum in okPAS.keys():
            outpas.write(ln)
    outpas.close()


subsetPAS(top20, Top20PAS, AllPas)
```



subsetPASnucbottom20perc.py

```{bash,eval=F}

bottom20=open("/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/NuclearPeaksBottom20Usage.txt", "r")

AllPas=open("/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed", "r")


Bottom20PAS=open("/project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_bottom20inNuclear.bed", "w")




def subsetPAS(use, outpas, PAS):
    okPAS={}
    for ln in use:
        peak=ln.strip()
        okPAS[peak]=""
    for ln in PAS:
        peaknum=ln.split()[3].split(":")[-1]
        print
        if peaknum in okPAS.keys():
            outpas.write(ln)
    outpas.close()


subsetPAS(bottom20, Bottom20PAS, AllPas)

```



Deeptools plots for these:

MNASEmyPAStop20nuc.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPAStop20nuc
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPAStop20nuc.out
#SBATCH --error=MNASEmyPAStop20nuc.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_top20inNuclear.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20nuc.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20nuc.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Top 20% Nuclear Usage" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeakstop20nuc.png
```




MNASEmyPASbottom20nuc.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=MNASEmyPASbottom20nuc
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=MNASEmyPASbottom20nuc.out
#SBATCH --error=MNASEmyPASbottom20nuc.err
#SBATCH --partition=bigmem2
#SBATCH --mem=100G
#SBATCH --mail-type=END

module load Anaconda3
source activate three-prime-env

computeMatrix reference-point -S /project2/gilad/briana/threeprimeseq/data/ChipSeq/ENCFF000VME.bigWig  -R  /project2/gilad/briana/threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand_bottom20inNuclear.bed -b 1000 -a 1000  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20nuc.gz 


plotHeatmap --sortRegions descend -m /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20nuc.gz --refPointLabel "Called PAS" --plotTitle "MNASE at Bottom 20% Nuclear Usage" --heatmapHeight 7 --colorMap YlGnBu  -out /project2/gilad/briana/threeprimeseq/data/ChipSeq/mnase_myPeaksbottom20nuc.png
```
