---
site: workflowr::wflow_site
title: "Home"
output:
  workflowr::wflow_html:
    toc: false
---
Briana Mittleman  


This is the site I will use for my human 3' seq project. I will have my workflow and analysis here. The three prime seq protocol I am using is Lexogen, QuantSeq 3' mRNA-Seq library kit. The first analysis will be on the total and nucleic fractions of the LCL used in the net-seq analysis found at https://brimittleman.github.io/Net-seq/  


Initial analysis on RNA seq of nuclear and total fractions: 

* [Initial data QC](total.nuc.rnaseq.html)  

Initial analysis on first 32 3' seq libraries:  
  
* [Initial mapping QC](initial_map_qc.html) 
* [Initial gene leve QC](initial_gene_qc.html)  
* [Compare coverage in Three prime seq to RNAseq](cov.analysis.threeprime1.html)  
* [Coverage in 200bp bins](cov200bpwind.html)  
* [Differential isoform usage between fractions](dif.iso.usage.leafcutter.html)  
* [Explore filter cuttoffs](explore.filters.html)  
* [Enrichment at PAS](enrichmentPAS.html)  


Explore and filter mis-primming:  
  
* [Look at number of As in bins](a.content.html)  
* [Create script to filter mapped A's](filter_As.html)  
* [Test cleanUpdTSeq for calling internal priming](cleanupdtseq.internalpriming.html)


Peak calling:  

* [Testing macs2 on 3'seq data](test.max2.html)  
* [Call peak regions de-novo](deNovopeakcalling.html)
* [Compare peaks in Derti Brain set](eval.brain.html)  
* [Test SMASH smoothing](test.smash.html)  
* [Expand SMASH analysis for use on midway](expand.smash.html)

