---
site: workflowr::wflow_site
title: "Home"
output:
  workflowr::wflow_html:
    toc: false
---
Briana Mittleman  


This is the site I will use for my human 3' seq project. I will have my workflow and analysis here. The three prime seq protocol I am using is Lexogen, QuantSeq 3' mRNA-Seq library kit. The first analysis will be on the total and nucleic fractions of the LCL used in the net-seq analysis found at https://brimittleman.github.io/Net-seq/  


##Initial analysis on RNA seq of nuclear and total fractions: 

* [Initial data QC](total.nuc.rnaseq.html)  

###Initial analysis on first 32 3' seq libraries:  
  
* [Initial mapping QC](initial_map_qc.html) 
* [Initial gene leve QC](initial_gene_qc.html)  
* [Compare coverage in Three prime seq to RNAseq](cov.analysis.threeprime1.html)  
* [Coverage in 200bp bins](cov200bpwind.html)  
* [Differential isoform usage between fractions](dif.iso.usage.leafcutter.html)  
* [Explore filter cuttoffs](explore.filters.html)  
* [Enrichment at PAS](enrichmentPAS.html)    
* [Second Lane data](comb.lanes.qc.html)  


##Explore and filter mis-primming:  
  
* [Look at number of As in bins](a.content.html)  
* [Create script to filter mapped A's](filter_As.html)  
* [Test cleanUpdTSeq for calling internal priming](cleanupdtseq.internalpriming.html)  
* [Pipeline to create peaks and clean them](peak.cov.pipeline.html)  


##Peak calling:  

* [Testing macs2 on 3'seq data](test.max2.html)  
* [Call peak regions de-novo](deNovopeakcalling.html)
* [Compare peaks in Derti Brain set](eval.brain.html)  
* [Test SMASH smoothing](test.smash.html)  
* [Expand SMASH analysis for use on midway](expand.smash.html)  
* [Data Processing Figure design](dataprocfigures.html)  


###Peak coverage/Nominal QTLs/Diff iso:  

* [Start evaluating peak quantificaiton](28ind.peak.explore.html)  
* [Prepare the data for leafcutter QTL calling](pheno.leaf.comb.html)
* [Call nominal QTLs](apaQTLwLeafcutter.html)  
* [Differential Isoform pipeline](diff_iso_pipeline.html)  


###Rerun analysis opp strand:  

* [Rerun QTL analysis with off strand](peakOverlap_oppstrand.html)  

###Evaluate and learn from QTLs:  

* [Overlap QTLs with other molecular QTLs](overlap_qtls.html)


##Analysis for all 39 indivduals:  

* [Read and peak QC plots](39indQC.html)

* [Genome tracks to visualize reads](genometrack_figs.html)

* [APAqtl for full ind set](apaQTLsAllInd.html)

* [QTL and Diff iso wih correct peak to gene assignment](PeakToGeneAssignment.html)

* [Condition Total and Nuclear QTLs on each other](conditionAPAqtlFraction.html)

* [Characterize Total Apa QTLS](characterizeTotalApaQtls.html)  

* [Characterize Nuclear Apa QTLS](characterizeNuclearApaQtls.html)  

* [ChromHMM enrichment analysis](chromHmm_enrichment.html)

* [Overlap with GWAS](apaQTLoverlapGWAS.html)  

* [Visualize peaks in significant genes](peakGeneCovEx.html)

* [Look  at filtering low peaks and modify leafcutter](modifyLeafcutter.html)  


###Molecular QTL analysis and Overlap:

* [Rerun molecular QTLs](callMolQTLS.html)

* [Make overlap QTL plots](overlapMolQTL.html)  

* [Plot top apaQTLs in other phenotypes](swarmPlots_QTLs.html)  

* [LocusZoom plot for example QTLs](locusZoom.html)

* [Overlap QTLs starting with APAqtls](overlapMolQTL.opposite.html)  

* [Investigate APA protein RNA relationship](apa_prot_overlap.html)  

* [Use flash and mash to look at sharing](flash2mash.html)

* [Linear models to explain pQTLS](explainpQTLs.html)

Until now I have been focusing on way to develop the story for the paper. The next steps will be to systematically understand the data in order to filter to high quality data to use in the final analysis. 


* [Investigate Peak quality](understandPeaks.html)  

* [Investigate Effect of included PCs on QTL number](QTLsbyPCnum.html)  

* [Peak to gene assignment](InvestigatePeak2GeneAssignment.html)  

* [Look at our peaks compared to peaks from Lianogluo et al](CompareLianoglouData.html)
  
* [Explore cofactors agaisnt PCs for 39ind](CovariateOnPCs.html)  

* [Deal with Mispriming problem](mispriming_approach.html)

* [Create new peaks after mispriming filter](NewPeakPostMP.html)

* [QC plots peaks >5% usage](peakQCPPlots.html)

* [Deep Tools plots for internal most](proxDeepToolsPlot.html)


##Data analysis post MP cleaning and filtering  

In the previous sections (specifically [here](NewPeakPostMP.html)) I reran some QC and the QTL analysis with the filtered and clean data. Here I want to look at some of the other analysis with the processed data.  

* [Processed data - Different between fractions](FractionDiffProcessed.html)

* [Pre ash processing for Stephens lab](PreAshExplore.html)  

* [New approach for assisgning peaks to a gene](ApproachForGeneAssignment.html)  

##55 Individuals  

I now have 55 individuals. I am going to call peaks and QTLs in this data.  

* [Full QTL pipeline in 55 ind (cleaning, peaks, assign genes, filter usage, QTL, differential isoform analysis)](pipeline_55Ind.html)  

* [Individual differences in peak usage](IndPeakUsageDiff.html)  

* [QC plots for understanding apaQTLs](QConApaQTLs.html)

* [Test if mapping bias leads to false positive QTLs](accountMapBias.html)

* [Number of QTLs by window size](TestQTLbySizeRegion.html)  
