---
site: workflowr::wflow_site
title: "Home"
output:
  workflowr::wflow_html:
    toc: false
---
Briana Mittleman  


This is the site I will use for my human 3' seq project. I will have my workflow and analysis here. The three prime seq protocol I am using is Lexogen, QuantSeq 3' mRNA-Seq library kit. The first analysis will be on the total and nucleic fractions of the LCL used in the net-seq analysis found at https://brimittleman.github.io/Net-seq/  


##Initial analysis on RNA seq of nuclear and total fractions: 

* [Initial data QC](total.nuc.rnaseq.html)  

###Initial analysis on first 32 3' seq libraries:  
  
* [Initial mapping QC](initial_map_qc.html) 
* [Initial gene leve QC](initial_gene_qc.html)  
* [Compare coverage in Three prime seq to RNAseq](cov.analysis.threeprime1.html)  
* [Coverage in 200bp bins](cov200bpwind.html)  
* [Differential isoform usage between fractions](dif.iso.usage.leafcutter.html)  
* [Explore filter cuttoffs](explore.filters.html)  
* [Enrichment at PAS](enrichmentPAS.html)    
* [Second Lane data](comb.lanes.qc.html)  


##Explore and filter mis-primming:  
  
* [Look at number of As in bins](a.content.html)  
* [Create script to filter mapped A's](filter_As.html)  
* [Test cleanUpdTSeq for calling internal priming](cleanupdtseq.internalpriming.html)  
* [Pipeline to create peaks and clean them](peak.cov.pipeline.html)  


##Peak calling:  

* [Testing macs2 on 3'seq data](test.max2.html)  
* [Call peak regions de-novo](deNovopeakcalling.html)
* [Compare peaks in Derti Brain set](eval.brain.html)  
* [Test SMASH smoothing](test.smash.html)  
* [Expand SMASH analysis for use on midway](expand.smash.html)  
* [Data Processing Figure design](dataprocfigures.html)  


###Peak coverage/Nominal QTLs/Diff iso:  

* [Start evaluating peak quantificaiton](28ind.peak.explore.html)  
* [Prepare the data for leafcutter QTL calling](pheno.leaf.comb.html)
* [Call nominal QTLs](apaQTLwLeafcutter.html)  
* [Differential Isoform pipeline](diff_iso_pipeline.html)  


###Rerun analysis opp strand:  

* [Rerun QTL analysis with off strand](peakOverlap_oppstrand.html)  

###Evaluate and learn from QTLs:  

* [Overlap QTLs with other molecular QTLs](overlap_qtls.html)


##Analysis for all 39 indivduals:  

* [Read and peak QC plots](39indQC.html)

* [Genome tracks to visualize reads](genometrack_figs.html)

* [APAqtl for full ind set](apaQTLsAllInd.html)

* [QTL and Diff iso wih correct peak to gene assignment](PeakToGeneAssignment.html)

* [Condition Total and Nuclear QTLs on each other](conditionAPAqtlFraction.html)

* [Characterize Total Apa QTLS](characterizeTotalApaQtls.html)  

* [Characterize Nuclear Apa QTLS](characterizeNuclearApaQtls.html)  

* [ChromHMM enrichment analysis](chromHmm_enrichment.html)

* [Overlap with GWAS](apaQTLoverlapGWAS.html)  

* [Visualize peaks in significant genes](peakGeneCovEx.html)


###Molecular QTL analysis and Overlap:

* [Rerun molecular QTLs](callMolQTLS.html)

* [Make overlap QTL plots](overlapMolQTL.html)  

* [Plot top apaQTLs in other phenotypes](swarmPlots_QTLs.html)  

* [LocusZoom plot for example QTLs](locusZoom.html)

* [Overlap QTLs starting with APAqtls](overlapMolQTL.opposite.html)  

* [Investigate APA protein RNA relationship](apa_prot_overlap.html)  

* [Use flash and mash to look at sharing](flash2mash.html)
