---
title: "Individaul differences in PAS Usage"
author: "Briana Mittleman"
date: "2/5/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(workflowr)
library(reshape2)
library(matrixStats)
library(tidyverse)
```

So far i have been looking at mean peak usage for my filters. As a QC metric, I want to look at the variance in this measurement. I want to understand the reproducibility of the data at a usage percent level. I also want to see if this value is dependent on coverage. I will look at the peaks used in the [QTL analysis with 55 individuals](pipeline_55Ind.html) and comopute an RNSD value for each gene. This value is computed as $\sqrt{\sum_{n=1}^N (X-Y)^2}$. Here n is the number of peaks in the gene up to N. X and Y are different individuals. I will plot this value for each gene. I can do this for 2 individuals with low depth and 2 with high depth.   

I can start with just the total individuals.  

First step is to convert /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs_3UTR/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.3UTR.fixed.pheno_5perc.fc.gz to numeric.  

First I will cut the first column to just get the counts:

```{bash,eval=F}
less /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz | cut -f1 -d" " --complement | sed '1d' > /project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc_counts 
```


5percCovUsageToNumeric.py
```{bash,eval=F}
def convert(infile, outfile):
  final=open(outfile, "w")
  for ln in open(infile, "r"):
    line_list=ln.split()
    new_list=[]
    for i in line_list:
      num, dem = i.split("/")
      if dem == "0":
        perc = "0.00"
      else:
        perc = int(num)/int(dem)
        perc=round(perc,2)
        perc= str(perc)
      new_list.append(perc)
    final.write("\t".join(new_list)+ '\n')
  final.close()
  
convert("/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc_counts","/project2/gilad/briana/threeprimeseq/data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.numeric.txt")

```

Get the gene names from the first file: 

```{bash ,eval=F}
less /project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.fc.gz | cut -f1 -d" " | sed '1d' > PeakIDs.txt
```


Merge the files: PeakIDs.txt and the numeric version 

```{bash,eval=F}
paste PeakIDs.txt filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.numeric.txt > filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.numeric.named.txt
```

```{r}
names=read.table("../data/PeakUsage_noMP_GeneLocAnno/PeakUsageHeader.txt",stringsAsFactors = F) %>% t %>% as_data_frame()
usageTot=read.table("../data/PeakUsage_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno_5perc.numeric.named.txt", header=F, stringsAsFactors = F)
colnames(usageTot)= names$V1
```

I want to use ind based on coverage 

```{r}
metadataTotal=read.table("../data/threePrimeSeqMetaData55Ind_noDup.txt", header=T) %>% filter(fraction=="total")

#top
metadataTotal %>% arrange(desc(reads)) %>% slice(1:2)

#bottom
metadataTotal %>% arrange(reads) %>% slice(1:2)
```

2 Top read ind: NA18504, NA18855  
2 bottom read ind: NA19160, NA19101  


```{r}
topInd=usageTot %>% select(chrom, NA18504, NA18855) %>% separate(chrom, into=c("chr", "start", "end", "geneInf"), sep =":") %>% separate(geneInf, into=c("gene", "strand", "peak"), sep="_") %>% mutate(val=(NA18504-NA18855)^2) %>% group_by(gene) %>% summarise(sumPeaks=sum(val)) %>% mutate(RNSD=sqrt(sumPeaks)) %>% mutate(Sample="Top") %>% select(Sample, RNSD)
bottomInd=usageTot %>% select(chrom, NA19160, NA19101) %>% separate(chrom, into=c("chr", "start", "end", "geneInf"), sep =":") %>% separate(geneInf, into=c("gene", "strand", "peak"), sep="_") %>% mutate(val=(NA19160-NA19101)^2) %>% group_by(gene) %>% summarise(sumPeaks=sum(val)) %>% mutate(RNSD=sqrt(sumPeaks)) %>% mutate(Sample="Bottom") %>% select(Sample, RNSD)

bothInd= data.frame(rbind(topInd, bottomInd))

```

```{r}
ggplot(bothInd, aes(x=RNSD, by=Sample, fill=Sample))+geom_density(alpha=.4) +labs(title="RNSD for peak usage in top 2 and bottom 2 individuals by reads")
```

```{r}
ggplot(bothInd, aes(y=RNSD, x=Sample, fill=Sample))+geom_violin() + labs(title="RNSD for peak usage in top 2 and bottom 2 individuals by reads")
```

Change the statistic to increase the interpretability. 

I can look at just genes with 2 peaks. I can then sum the absolute value of the individual differences.  

```{r}
TwoPeakGenes_top= usageTot %>% select(chrom, NA18504, NA18855) %>% separate(chrom, into=c("chr", "start", "end", "geneInf"), sep =":") %>% separate(geneInf, into=c("gene", "strand", "peak"), sep="_") %>% group_by(gene) %>% mutate(nPeak=n()) %>% filter(nPeak==2)
n2Peak=unique(TwoPeakGenes_top$gene) %>% length() 
```
This gives us 3448 genes.  

```{r}
TwoPeakGenes_top_stat= TwoPeakGenes_top %>% mutate(absDiff=abs(NA18504-NA18855)) %>% group_by(gene) %>% select(gene, absDiff) %>% distinct(gene, .keep_all=T) 
Avg2PeakGeneTop=sum(TwoPeakGenes_top_stat$absDiff)/n2Peak
Avg2PeakGeneTop
```

Now do this for the bottom 2 ind:  

```{r}
TwoPeakGenes_bottom= usageTot %>% select(chrom, NA19160, NA19101) %>% separate(chrom, into=c("chr", "start", "end", "geneInf"), sep =":") %>% separate(geneInf, into=c("gene", "strand", "peak"), sep="_") %>% group_by(gene) %>% mutate(nPeak=n()) %>% filter(nPeak==2)

TwoPeakGenes_bottom_stat= TwoPeakGenes_bottom %>% mutate(absDiff=abs(NA19101-NA19160)) %>% group_by(gene) %>% select(gene, absDiff) %>% distinct(gene, .keep_all=T) 
Avg2PeakGeneBottom=sum(TwoPeakGenes_bottom_stat$absDiff)/n2Peak
Avg2PeakGeneBottom
```

This demonstrates we may have some noise in the data and have not reached sequencing saturation. However, this could be driven by lowly expressed genes. I will fraction this analysis by the top expressed and bottom expressed genes. To do this I will pull in the count data (before we had usage parameters) and add to this table the mean counts. 


Count files:

* /project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc  
* /project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc  

I need to filter these for the peaks I kept after the 5% usage filter.  


filterCounts_5percCovPeaks.py
```{bash,eval=F}
#python  

totalokPeaks5perc_file="/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total_fixed.pheno.5percPeaks.txt"

totalokPeaks5perc={}
for ln in open(totalokPeaks5perc_file,"r"):
    peakname=ln.split()[5]
    totalokPeaks5perc[peakname]=""


nuclearokPeaks5perc_file="/project2/gilad/briana/threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear_fixed.pheno.5percPeaks.txt"
nuclearokPeaks5perc={}
for ln in open(nuclearokPeaks5perc_file,"r"):
    peakname=ln.split()[5]
    nuclearokPeaks5perc[peakname]=""
    
    
totalPhenoBefore=open("/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.fc","r")
totalPhenoAfter=open("/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.5perc.fc", "w")
for num, ln in enumerate(totalPhenoBefore):
    if num == 1:
        totalPhenoAfter.write(ln)
    if num >1:  
        id=ln.split()[0].split(":")[0]
        if id in totalokPeaks5perc.keys():
            totalPhenoAfter.write(ln)
totalPhenoAfter.close()  

nuclearPhenoBefore=open("/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.fc","r")
nuclearPhenoAfter=open("/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.5perc.fc", "w")
for num, ln in enumerate(nuclearPhenoBefore):
    if num ==1:
        nuclearPhenoAfter.write(ln)
    if num > 1:  
        id=ln.split()[0].split(":")[0]
        if id in nuclearokPeaks5perc.keys():
            nuclearPhenoAfter.write(ln)
nuclearPhenoAfter.close() 
```

Pull the filtered counts for the total here and get the genes in TwoPeakGenes_top_stat$gene. For simplicity I will just look at the ind. with the top coverage.  


```{r}
TotalCounts=read.table("../data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.5perc.fc", header=T, stringsAsFactors = F) %>% separate(Geneid, into =c('peak', 'chr', 'start', 'end', 'strand', 'gene'), sep = ":") %>% select(-peak, -chr, -start, -end, -strand, -Chr, -Start, -End, -Strand, -Length) %>% group_by(gene) %>% mutate(PeakCount=n()) %>% filter(PeakCount==2) %>% select(gene, X18504_T) %>% group_by(gene) %>% summarise(Exp=sum(X18504_T)) 
```

Top 1000 genes  

```{r}
TotalCounts_top1000= TotalCounts %>% arrange(desc(Exp)) %>% top_n(1000)
```

Join this with top ind:  


```{r}
TwoPeakGenes_top_stat_top100 = TwoPeakGenes_top_stat %>% inner_join(TotalCounts_top1000, by="gene")

Avg2PeakGeneTopExpHigh=sum(TwoPeakGenes_top_stat_top100$absDiff)/nrow(TwoPeakGenes_top_stat_top100)
Avg2PeakGeneTopExpHigh
```



Do the same for low expressed genes (not 0)  

```{r}
TotalCounts_bottom1000= TotalCounts %>% filter(Exp > 0) %>% top_n(-1000)
TwoPeakGenes_top_stat_bottom100 = TwoPeakGenes_top_stat %>% inner_join(TotalCounts_bottom1000, by="gene")

Avg2PeakGeneTopExpLow=sum(TwoPeakGenes_top_stat_bottom100$absDiff)/nrow(TwoPeakGenes_top_stat_bottom100)

Avg2PeakGeneTopExpLow

```



Do this for the other individuals as wel..  

```{r}
TwoPeakGenes_bottom_stat_top100 = TwoPeakGenes_bottom_stat %>% inner_join(TotalCounts_top1000, by="gene")

Avg2PeakGeneBottomExpHigh=sum(TwoPeakGenes_bottom_stat_top100$absDiff)/nrow(TwoPeakGenes_bottom_stat_top100)
Avg2PeakGeneBottomExpHigh


```
```{r}
TwoPeakGenes_bottom_stat_bottom100 = TwoPeakGenes_bottom_stat %>% inner_join(TotalCounts_bottom1000, by="gene")

Avg2PeakGeneBottomExpLow=sum(TwoPeakGenes_bottom_stat_bottom100$absDiff)/nrow(TwoPeakGenes_bottom_stat_bottom100)

Avg2PeakGeneBottomExpLow
```



Make a plot with these values:  



```{r}
avgPeakUsageTable=data.frame(rbind(c("Top", "High", .1), c("Top", "Low", .32), c("Bottom", "High", .19), c("Bottom", "Low", .35)))
colnames(avgPeakUsageTable)= c("Coverage", "Expression", "AverageUsageDiff")
```

```{r}
ggplot(avgPeakUsageTable,aes(y=AverageUsageDiff, x=Coverage, by=Expression, fill=Expression)) + geom_bar(stat="identity",position="dodge") + labs(title="Difference between individual usage \ndepends on expression and coverage")
```

I want to look at this based on every 10% of gene expression. First I will remove genes with 0 coverage in this ind.  
```{r}
TotalCounts_no0=TotalCounts %>% filter(Exp>0) 
nrow(TotalCounts_no0)
```

I can add a column that is the rank over the sum (or the percentile)  
```{r}
TotalCounts_no0_perc= TotalCounts_no0 %>% mutate(Percentile = percent_rank(Exp)) 
```


```{r}
TotalCounts_no0_perc10= TotalCounts_no0_perc %>% filter(Percentile<.1)

TotalCounts_no0_perc20= TotalCounts_no0_perc %>% filter(Percentile<.2, Percentile>.1)

TotalCounts_no0_perc30= TotalCounts_no0_perc %>% filter(Percentile<.3, Percentile>.2)


TotalCounts_no0_perc40= TotalCounts_no0_perc %>% filter(Percentile<.4, Percentile>.3)

TotalCounts_no0_perc50= TotalCounts_no0_perc %>% filter(Percentile<.5, Percentile>.4)

TotalCounts_no0_perc60= TotalCounts_no0_perc %>% filter(Percentile<.6, Percentile>.5)

TotalCounts_no0_perc70= TotalCounts_no0_perc %>% filter(Percentile<.7, Percentile>.6)

TotalCounts_no0_perc80= TotalCounts_no0_perc %>% filter(Percentile<8, Percentile>.7)
TotalCounts_no0_perc90= TotalCounts_no0_perc %>% filter(Percentile<.9, Percentile>.8)

TotalCounts_no0_perc100= TotalCounts_no0_perc %>% filter(Percentile<1, Percentile>.9)

```

Now I can make a function that takes one of these files, and computes the relevant stat. This takes the usage file and the expression file  


```{r}
getAvgDiffUsage=function(Usage, exp){
  df = Usage %>% inner_join(exp, by="gene")
  value=df=sum(df$absDiff)/nrow(df)
  return(value)
}
```


Run this for the top coverage at each exp:

```{r}
 
AvgUsageDiff_top10=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc10)
AvgUsageDiff_top20=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc20)
AvgUsageDiff_top30=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc30)
AvgUsageDiff_top40=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc40)
AvgUsageDiff_top50=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc50)
AvgUsageDiff_top60=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc60)
AvgUsageDiff_top70=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc70)
AvgUsageDiff_top80=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc80)
AvgUsageDiff_top90=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc90)
AvgUsageDiff_top100=getAvgDiffUsage(TwoPeakGenes_top_stat, exp=TotalCounts_no0_perc100)

AvgUsageTop=c(AvgUsageDiff_top10,AvgUsageDiff_top20,AvgUsageDiff_top30,AvgUsageDiff_top40,AvgUsageDiff_top50,AvgUsageDiff_top60,AvgUsageDiff_top70,AvgUsageDiff_top80,AvgUsageDiff_top90,AvgUsageDiff_top100)
```

Do this for Bottom coverage:  




```{r}
 
AvgUsageDiff_bottom10=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc10)
AvgUsageDiff_bottom20=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc20)
AvgUsageDiff_bottom30=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc30)
AvgUsageDiff_bottom40=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc40)
AvgUsageDiff_bottom50=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc50)
AvgUsageDiff_bottom60=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc60)
AvgUsageDiff_bottom70=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc70)
AvgUsageDiff_bottom80=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc80)
AvgUsageDiff_bottom90=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc90)
AvgUsageDiff_bottom100=getAvgDiffUsage(TwoPeakGenes_bottom_stat, exp=TotalCounts_no0_perc100)

AvgUsagebottom=c(AvgUsageDiff_bottom10,AvgUsageDiff_bottom20,AvgUsageDiff_bottom30,AvgUsageDiff_bottom40,AvgUsageDiff_bottom50,AvgUsageDiff_bottom60,AvgUsageDiff_bottom70,AvgUsageDiff_bottom80,AvgUsageDiff_bottom90,AvgUsageDiff_bottom100)
```


All together:  

```{r}
Percentile=c(10,20,30,40,50,60,70,80,90,100)
allAvgUsage=data.frame(cbind(Percentile,AvgUsageTop,AvgUsagebottom))
colnames(allAvgUsage)=c("Percentile","Top", "Bottom")
allAvgUsage$Top= as.numeric(as.character(allAvgUsage$Top))
allAvgUsage$Bottom= as.numeric(as.character(allAvgUsage$Bottom))
allAvgUsage_melt= melt(allAvgUsage, id.vars = c("Percentile"))
colnames(allAvgUsage_melt)=c("Percentile","Coverage", "AvgDifference")
```


```{r}
ggplot(allAvgUsage_melt, aes(x=Percentile, y=AvgDifference,by=Coverage, fill=Coverage)) + geom_bar(stat="identity", position = "dodge") + labs(title="Average Usage Difference between 2 individauls by 3' Seq \nCount percentile and coverage") + scale_x_discrete(limits = c(10,20,30,40,50,60,70,80,90,100))
```

Total counts for a low coverage individual:  

```{r}
TotalCounts_low=read.table("../data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.5perc.fc", header=T, stringsAsFactors = F) %>% separate(Geneid, into =c('peak', 'chr', 'start', 'end', 'strand', 'gene'), sep = ":") %>% select(-peak, -chr, -start, -end, -strand, -Chr, -Start, -End, -Strand, -Length) %>% group_by(gene) %>% mutate(PeakCount=n()) %>% filter(PeakCount==2) %>% select(gene, X19101_T) %>% group_by(gene) %>% summarise(Exp=sum(X19101_T))
TotalCounts_no0_low=TotalCounts_low %>% filter(Exp>0) 
TotalCounts_no0_perc_low= TotalCounts_no0_low %>% mutate(Percentile = percent_rank(Exp)) 
```



This will be more informative if we do 1 vs all. Now instead of Y being 1 individual it is the average of all other individuals at that peak. I will make a heatmap with y as the individual and x as the percentile. 

I need to create a function that can take in an individual, removes it from the matrix. computes the mean for the other individuals then merges the numbers back together, it will then compute the values for each percentile as I did before and returns a list of the averageUsageDifferences. I need to compute the percentile for the individual 

Input dataframe with usage values for genes with only 2 peaks.

```{r}
usageTot_2peak= usageTot  %>% separate(chrom, into=c("chr", "start", "end", "geneInf"), sep =":") %>% separate(geneInf, into=c("gene", "strand", "peak"), sep="_") %>% group_by(gene) %>% mutate(nPeak=n()) %>% filter(nPeak==2) %>% select(-chr, -start, -end, -strand, -peak, -nPeak) %>% ungroup()
```
```{r}
TotalCounts_AllInd=read.table("../data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.5perc.fc", header=T, stringsAsFactors = F) %>% separate(Geneid, into =c('peak', 'chr', 'start', 'end', 'strand', 'gene'), sep = ":") %>% select(-peak, -chr, -start, -end, -strand, -Chr, -Start, -End, -Strand, -Length) %>% group_by(gene) %>% mutate(PeakCount=n()) %>% filter(PeakCount==2) %>% select(-PeakCount) %>% ungroup()

colnames(TotalCounts_AllInd)=colnames(usageTot_2peak)
```


This will take tidy eval because my


```{r}

perIndDiffUsage=function(ind, counts=TotalCounts_AllInd, usage=usageTot_2peak){
  ind=enquo(ind)
  #compute usage stats
  #seperate usage
  usage_ind=usage %>% select(gene, !!ind) 
  usage_other = usage %>% select(-gene,-!!ind) %>% rowMeans()
  usage_indVal=as.data.frame(cbind(usage_ind,usage_other))
  usage_indVal$val=abs(usage_indVal[,2] - usage_indVal[,3])
  usage_indVal2= usage_indVal%>% group_by(gene) %>% select(gene, val) %>% distinct(gene, .keep_all=T) 
  #seperate genes by percentile for this ind
  count_ind= counts %>% select(gene, !!ind)
  colnames(count_ind)=c("gene", "count")
  count_ind = count_ind %>%  group_by(gene) %>% summarize(Exp=sum(count)) %>% filter(Exp >0)  %>% mutate(Percentile = percent_rank(Exp)) 
  count_ind_perc10= count_ind %>% filter(Percentile<.1)
  count_ind_perc20= count_ind %>% filter(Percentile<.2, Percentile>.1)
  count_ind_perc30= count_ind %>% filter(Percentile<.3, Percentile>.2)
  count_ind_perc40= count_ind %>% filter(Percentile<.4, Percentile>.3)
  count_ind_perc50= count_ind %>% filter(Percentile<.5, Percentile>.4)
  count_ind_perc60= count_ind %>% filter(Percentile<.6, Percentile>.5)
  count_ind_perc70= count_ind %>% filter(Percentile<.7, Percentile>.6)
  count_ind_perc80= count_ind %>% filter(Percentile<8, Percentile>.7)
  count_ind_perc90= count_ind %>% filter(Percentile<.9, Percentile>.8)
  count_ind_perc100= count_ind %>% filter(Percentile<1, Percentile>.9)
  #subset and sum usage 
  out10_df= usage_indVal2 %>% inner_join(count_ind_perc10, by="gene")
  out20_df= usage_indVal2 %>% inner_join(count_ind_perc20, by="gene")
  out30_df= usage_indVal2 %>% inner_join(count_ind_perc30, by="gene")
  out40_df= usage_indVal2 %>% inner_join(count_ind_perc40, by="gene")
  out50_df= usage_indVal2 %>% inner_join(count_ind_perc50, by="gene")
  out60_df= usage_indVal2 %>% inner_join(count_ind_perc60, by="gene")
  out70_df= usage_indVal2 %>% inner_join(count_ind_perc70, by="gene")
  out80_df= usage_indVal2 %>% inner_join(count_ind_perc80, by="gene")
  out90_df= usage_indVal2 %>% inner_join(count_ind_perc90, by="gene")
  out100_df= usage_indVal2 %>% inner_join(count_ind_perc100, by="gene")
  #output list of 10 values
  out= c((sum(out10_df$val)/nrow(out10_df)), (sum(out20_df$val)/nrow(out20_df)), (sum(out30_df$val)/nrow(out30_df)), (sum(out40_df$val)/nrow(out40_df)), (sum(out50_df$val)/nrow(out50_df)), (sum(out60_df$val)/nrow(out60_df)), (sum(out70_df$val)/nrow(out70_df)), (sum(out80_df$val)/nrow(out80_df)), (sum(out90_df$val)/nrow(out90_df)), (sum(out100_df$val)/nrow(out100_df)))
  return(out)
}


```

Run this function for each individual and append the result to a df:  


```{r}
Inds=colnames(TotalCounts_AllInd)[2:ncol(TotalCounts_AllInd)]
Percentile=c(10,20,30,40,50,60,70,80,90,100)
for (i in Inds){
  x= perIndDiffUsage(i)
  Percentile=cbind(Percentile, x)
}
colnames(Percentile)=c("Percentile", Inds)
```


```{r}


Lineorder=metadataTotal %>% arrange(desc(reads))  %>% select(line) %>% mutate(sample=paste("NA" , line, sep=""))
Lineorder=Lineorder$sample

Percentile_df=as.data.frame(Percentile) %>%   select(Percentile, Lineorder)

#order the 

Percentile_melt=melt(Percentile_df, id.vars=c("Percentile"))
colnames(Percentile_melt)=c("Percentile", "Individual", "AvgUsageDiff")
```


Plot this:  

```{r}
ggplot(Percentile_melt, aes(x=Percentile, y=Individual, fill=AvgUsageDiff)) + geom_tile() + labs(title="Average peak usage difference for individaul vs. others \n by percentile of gene counts arranged by reads counts") + scale_fill_gradientn(colours = c("white", "red", "black"))
```

```{r}


Lineorder_map=metadataTotal %>% arrange(desc(Mapped_noMP))  %>% select(line) %>% mutate(sample=paste("NA" , line, sep=""))
Lineorder_map=Lineorder_map$sample

Percentile_df2=as.data.frame(Percentile) %>%   select(Percentile, Lineorder_map)

#order the 

Percentile_melt2=melt(Percentile_df2, id.vars=c("Percentile"))
colnames(Percentile_melt2)=c("Percentile", "Individual", "AvgUsageDiff")

ggplot(Percentile_melt2, aes(x=Percentile, y=Individual, fill=AvgUsageDiff)) + geom_tile() + labs(title="Average peak usage difference for individaul vs. others \n by percentile of gene counts arranged by mapped reads counts") + scale_fill_gradientn(colours = c("white", "red", "black"))
```


Try this again but with Z scores. I will get the mean and SD for each peak and compute the |x-mean|/sd  


```{r}
perIndDiffUsage_Z=function(ind, counts=TotalCounts_AllInd, usage=usageTot_2peak){
  ind=enquo(ind)
  #compute usage stats
  #seperate usage
  usage_ind=usage %>% select(gene, !!ind) 
  usage_mean = usage %>% select(-gene) %>% rowMeans()
  usage_df=usage %>% select(-gene) 
  usage_sd= as.matrix(usage_df)%>% rowSds()
  usage_indVal=as.data.frame(cbind(usage_ind,usage_mean, usage_sd))
  usage_indVal$val=abs(usage_indVal[,2] - usage_indVal[,3])/usage_indVal[,4]
  usage_indVal2= usage_indVal%>% group_by(gene) %>% select(gene, val) %>% distinct(gene, .keep_all=T) 
  #seperate genes by percentile for this ind
  count_ind= counts %>% select(gene, !!ind)
  colnames(count_ind)=c("gene", "count")
  count_ind = count_ind %>%  group_by(gene) %>% summarize(Exp=sum(count)) %>% filter(Exp >0)  %>% mutate(Percentile = percent_rank(Exp)) 
  count_ind_perc10= count_ind %>% filter(Percentile<.1)
  count_ind_perc20= count_ind %>% filter(Percentile<.2, Percentile>.1)
  count_ind_perc30= count_ind %>% filter(Percentile<.3, Percentile>.2)
  count_ind_perc40= count_ind %>% filter(Percentile<.4, Percentile>.3)
  count_ind_perc50= count_ind %>% filter(Percentile<.5, Percentile>.4)
  count_ind_perc60= count_ind %>% filter(Percentile<.6, Percentile>.5)
  count_ind_perc70= count_ind %>% filter(Percentile<.7, Percentile>.6)
  count_ind_perc80= count_ind %>% filter(Percentile<8, Percentile>.7)
  count_ind_perc90= count_ind %>% filter(Percentile<.9, Percentile>.8)
  count_ind_perc100= count_ind %>% filter(Percentile<1, Percentile>.9)
  #subset and sum usage 
  out10_df= usage_indVal2 %>% inner_join(count_ind_perc10, by="gene")
  out20_df= usage_indVal2 %>% inner_join(count_ind_perc20, by="gene")
  out30_df= usage_indVal2 %>% inner_join(count_ind_perc30, by="gene")
  out40_df= usage_indVal2 %>% inner_join(count_ind_perc40, by="gene")
  out50_df= usage_indVal2 %>% inner_join(count_ind_perc50, by="gene")
  out60_df= usage_indVal2 %>% inner_join(count_ind_perc60, by="gene")
  out70_df= usage_indVal2 %>% inner_join(count_ind_perc70, by="gene")
  out80_df= usage_indVal2 %>% inner_join(count_ind_perc80, by="gene")
  out90_df= usage_indVal2 %>% inner_join(count_ind_perc90, by="gene")
  out100_df= usage_indVal2 %>% inner_join(count_ind_perc100, by="gene")
  #output list of 10 values
  out= c((sum(out10_df$val)/nrow(out10_df)), (sum(out20_df$val)/nrow(out20_df)), (sum(out30_df$val)/nrow(out30_df)), (sum(out40_df$val)/nrow(out40_df)), (sum(out50_df$val)/nrow(out50_df)), (sum(out60_df$val)/nrow(out60_df)), (sum(out70_df$val)/nrow(out70_df)), (sum(out80_df$val)/nrow(out80_df)), (sum(out90_df$val)/nrow(out90_df)), (sum(out100_df$val)/nrow(out100_df)))
  return(out)
}
```

```{r}
Inds=colnames(TotalCounts_AllInd)[2:ncol(TotalCounts_AllInd)]
Percentile_z=c(10,20,30,40,50,60,70,80,90,100)
for (i in Inds){
  x= perIndDiffUsage_Z(i)
  Percentile_z=cbind(Percentile_z, x)
}
colnames(Percentile_z)=c("Percentile", Inds)
```


```{r}
Percentile_z_df=as.data.frame(Percentile_z) %>%   select(Percentile, Lineorder)

#order the 

Percentile_z_melt=melt(Percentile_z_df, id.vars=c("Percentile"))
colnames(Percentile_z_melt)=c("Percentile", "Individual", "AvgUsageDiff")


ggplot(Percentile_z_melt, aes(x=Percentile, y=Individual, fill=AvgUsageDiff)) + geom_tile() + labs(title="Peak usage difference Z score for individaul vs. others \n by percentile of gene counts arranged by reads counts") + scale_fill_gradientn(colours = c("white", "red", "black"))
```
Do this by the mapped reads: 

```{r}
Percentile_z_df2=as.data.frame(Percentile_z) %>%   select(Percentile, Lineorder_map)

#order the 

Percentile_z_melt2=melt(Percentile_z_df2, id.vars=c("Percentile"))
colnames(Percentile_z_melt2)=c("Percentile", "Individual", "AvgUsageDiff")

ggplot(Percentile_z_melt2, aes(x=Percentile, y=Individual, fill=AvgUsageDiff)) + geom_tile() + labs(title="Peak usage difference Z score for individaul vs. others \n by percentile of gene counts arranged by mapped reads counts") + scale_fill_gradientn(colours = c("white", "red", "black"))
```

