---
title: "Connection between APA and RNA decay"
author: "Briana Mittleman"
date: "3/14/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

I want to ask if more nuclear specific transcripts compared to total is associated with RNA decay.  

```{r}
library(tidyverse)
library(reshape2)
library(MASS)
library(viridis)
```


```{r}
decay=read.table(file = "../data/RNAdecay/tr_decay_table_norm.txt", header=T, stringsAsFactors = F) %>% dplyr::select(gene_id,contains("RNAdecay"))
```

Change gene names:

```{r}
geneNames=read.table("../data/ensemble_to_genename.txt", sep="\t", col.names = c('gene_id', 'GeneName', 'source' ),stringsAsFactors = F)
```

```{r}
decay_geneNames=decay %>% inner_join(geneNames, by="gene_id") %>% dplyr::select(GeneName, contains("RNAdecay"))

decay_geneNames_long=melt(decay_geneNames,id.vars = "GeneName", value.name = "RNA_Decay", variable.name = "Decay_Ind") %>% separate(Decay_Ind, into=c("type", "ind"), sep="_") %>% mutate(Individual=paste("X" , ind, sep="")) %>% dplyr::select(GeneName, Individual, RNA_Decay)
```

##Prepare apa value:

For each gene I need to get nuclear counts/nuclear + counts  



I want to use the filtered 5% peak counts. 


/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.5perc.fc

/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.5perc.fc


Make a dictionary from the individuals in the first line. I want them to have NA##### format 

makepheno4decayComparison.py 
```{bash,eval=F}
nucCounts="/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.5perc.fc"

totCounts="/project2/gilad/briana/threeprimeseq/data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.5perc.fc"

#top key is individual
OutPutdic={}


#problem keeping ind connected to column

```


Try in R

Nuclear first:  
```{r}
NucAPA=read.table("../data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.5perc.fc", stringsAsFactors = F, header = T) %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length) %>% separate(Geneid, into=c("peak", "chrom", "start", "end", "strand", "GeneName"), sep=":") %>% dplyr::select(-chrom, -start, -end, -strand)

NucApaMelt=melt(NucAPA, id.vars =c( "peak", "GeneName"), value.name="count", variable.name="Ind") %>% separate(Ind, into=c('Individual', 'fraction') ,sep="_") %>% dplyr::select(peak, GeneName, Individual, count)


NucAPA_bygene= NucApaMelt %>% group_by(GeneName,Individual) %>% summarise(NuclearSum=sum(count))
```



Total first:  
```{r}
TotAPA=read.table("../data/filtPeakOppstrand_cov_noMP_GeneLocAnno_5perc/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.5perc.fc", stringsAsFactors = F, header = T) %>% dplyr::select(-Chr, -Start, -End, -Strand, -Length) %>% separate(Geneid, into=c("peak", "chrom", "start", "end", "strand", "GeneName"), sep=":") %>% dplyr::select(-chrom, -start, -end, -strand)

TotApaMelt=melt(TotAPA, id.vars =c( "peak", "GeneName"), value.name="count", variable.name="Ind")  %>% separate(Ind, into=c('Individual', 'fraction') ,sep="_") %>% dplyr::select(peak, GeneName, Individual, count)


TotAPA_bygene=TotApaMelt %>% group_by(GeneName,Individual) %>% summarise(TotalSum=sum(count))
```



Sum these together: 

```{r} 
Apa_all=TotAPA_bygene %>% inner_join(NucAPA_bygene, by=c("GeneName", "Individual")) %>% filter(NuclearSum>0 |TotalSum>0 )  %>% mutate(APAvalue=NuclearSum/(NuclearSum+TotalSum)) %>% dplyr::select(GeneName, Individual, APAvalue)
```



##Join ith decay

```{r}
APAandDecay=decay_geneNames_long %>% inner_join(Apa_all, by=c('GeneName', 'Individual'))


ngenes=APAandDecay %>% dplyr::select(GeneName) %>% unique() %>% nrow()
ngenes
```

plot it:

```{r}

summary(lm(data=APAandDecay, APAvalue~RNA_Decay))
APAdecalAllindplot=ggplot(APAandDecay, aes(y=APAvalue, x=RNA_Decay)) + geom_point(aes(col=Individual)) +geom_density2d(na.rm = TRUE, size = 1, colour = 'red') + geom_smooth(method="lm") + annotate("text", label="Estimated Slope= -.026", y=1, x=-1) + labs(title="Relationship between RNA decay \nand APA fraction counts", x=" mRNA decay rate/h", y= "Nuclear/(Nuclear + Total)")

APAdecalAllindplot


ggsave(APAdecalAllindplot, file="../output/plots/APAandRNADecay_allInd.png", height = 7, width=15)
```


1 individual:

```{r}
APAandDecay_18498= APAandDecay %>% filter(Individual=="X18498")

APAdecay_18498=ggplot(APAandDecay_18498, aes(y=APAvalue, x=RNA_Decay)) + geom_point() +geom_density2d(na.rm = TRUE, size = 1, colour = 'red') + annotate("text", label="Estimated Slope= -.133", y=0, x=-.8) + geom_smooth(method="lm")+ labs(title="Relationship between RNA decay \nand APA fraction counts", x=" mRNA decay rate/h", y= "Nuclear/(Nuclear + Total)")



APAdecay_18498


ggsave(APAdecay_18498, file="../output/plots/APAandRNADecay_18498.png")

summary(lm(data=APAandDecay_18498, APAvalue~RNA_Decay))
```

###Look at most variable decay values

Most of the genes have a similar decay rate. To se if there is a trend I need to  look at the genes with >1sd outside of the mean.  

```{r}
decay_zscore=decay_geneNames_long  %>% mutate(mean=mean(RNA_Decay), sd=sd(RNA_Decay)) %>%  group_by(GeneName) %>% mutate(geneMean=mean(RNA_Decay)) %>% mutate(Zscore=(geneMean-mean)/sd) %>% dplyr::select(GeneName, Zscore) %>% unique() 



decay_1sd= decay_zscore %>% filter(abs(Zscore)>1) %>% dplyr::select(GeneName)



```

Filter the apa and decay for these genes.

```{r}
APAandDecay_1sd= APAandDecay %>% filter(GeneName %in% decay_1sd$GeneName)

APAandDecay_1sd %>% dplyr::select(GeneName) %>% unique() %>% nrow()
```

```{r}
summary(lm(data=APAandDecay_1sd, APAvalue~RNA_Decay))

get_density <- function(x, y, ...) {
  dens <- MASS::kde2d(x, y, ...)
  ix <- findInterval(x, dens$x)
  iy <- findInterval(y, dens$y)
  ii <- cbind(ix, iy)
  return(dens$z[ii])
}

set.seed(1)
dat <- data.frame(
  x = c(
    rnorm(1e4, mean = 0, sd = 0.1),
    rnorm(1e3, mean = 0, sd = 0.1)
  ),
  y = c(
    rnorm(1e4, mean = 0, sd = 0.1),
    rnorm(1e3, mean = 0.1, sd = 0.2)
  )
)

APAandDecay_1sd$density <- get_density(APAandDecay_1sd$APAvalue, APAandDecay_1sd$RNA_Decay, n = 100)


APAdecalAllindplot_zgreat1=ggplot(APAandDecay_1sd, aes(y=APAvalue, x=RNA_Decay)) + geom_point(aes(color=density))+ geom_smooth(method="lm")+ labs(title="Relationship between RNA decay \nand APA fraction counts", x=" mRNA decay rate/h", y= "Nuclear/(Nuclear + Total)")  + scale_color_viridis()


APAdecalAllindplot_zgreat1



ggsave(APAdecalAllindplot_zgreat1, file="../output/plots/APAandRNADecay1SD_allInd.png", height = 7, width=7)
```





```{r}
APAandDecay1SD_18498= APAandDecay_1sd %>% filter(Individual=="X18498")



APAandDecay1SD_18498$density <- get_density(APAandDecay1SD_18498$APAvalue, APAandDecay1SD_18498$RNA_Decay, n = 100)

APAdecay1sqd_18498=ggplot(APAandDecay1SD_18498, aes(y=APAvalue, x=RNA_Decay)) + geom_point(aes(color=density)) +geom_smooth(method="lm")+ labs(title="Relationship between RNA decay \nand APA fraction counts", x=" mRNA decay rate/h", y= "Nuclear/(Nuclear + Total)") + scale_color_viridis()

summary(lm(data=APAandDecay1SD_18498, APAvalue~RNA_Decay))

APAdecay1sqd_18498

ggsave(APAdecay1sqd_18498, file="../output/plots/APAandRNADecay1SD_18498.png")
```

###Update plots, 1 per gene, standardize  

I need to full join the mapped read sizes to the APA data. 

```{r}
ApaBothFrac=TotAPA_bygene %>% inner_join(NucAPA_bygene, by=c("GeneName", "Individual"))

ApaBothFrac_melt=melt(ApaBothFrac, id.vars=c("GeneName", "Individual"),value.name="APA_val" ) %>% mutate(fraction=ifelse(variable=="TotalSum", "total", "nuclear"), line=substring(Individual, 2)) %>% dplyr::select(GeneName, fraction, line, APA_val)
```

I need the mapped read stats:  

```{r}
metadata=read.table("../data/threePrimeSeqMetaData55Ind_noDup_WASPMAP.txt", header = T,stringsAsFactors = F) %>% dplyr::select(line, fraction, Mapped_noMP)

metadata$line= as.character(metadata$line)
```


Join these: 


```{r}
ApaBothFracStand=ApaBothFrac_melt %>% full_join(metadata, by=c("line", "fraction")) %>% mutate(StandApa=APA_val/Mapped_noMP)
```


Group by the fraction, gene and get the mean of the standard. I can then divide the nuclear mean by the total mean for each gene

```{r}
ApaBothFracStand_geneMean=ApaBothFracStand %>% group_by(fraction, GeneName) %>% summarise(meanAPA=mean(StandApa, na.rm=T))
```
I want to spread this by fraction.


```{r}
ApaBothFracStand_geneMean_spread= spread(ApaBothFracStand_geneMean,fraction,meanAPA ) %>% mutate(APAVal=nuclear/(total+ nuclear)) 
```


###Join with decay  


```{r}
decay_byGene= decay_geneNames_long %>% group_by(GeneName) %>% summarise(MeanDecay=mean(RNA_Decay))



decay_byGene_1d= decay_byGene %>% mutate(Mean=mean(MeanDecay), SD=sd(MeanDecay), Zscore=(MeanDecay-Mean)/SD) %>% filter(abs(Zscore)>1) %>% dplyr::select(GeneName,MeanDecay)

```

```{r}
ApaAndDecaySt= ApaBothFracStand_geneMean_spread %>% inner_join(decay_byGene_1d, by="GeneName")

ApaAndDecaySt$density <- get_density(ApaAndDecaySt$APAVal, ApaAndDecaySt$MeanDecay, n = 100)
```


```{r}
apaanddecay1sd=ggplot(ApaAndDecaySt, aes(x=MeanDecay, y=APAVal)) + geom_point(aes(color=density)) + geom_smooth(method="lm") + labs(x="relative Decay", y="Nuclear/(Total+Nuclear)", title="Relationship between Nuclear proportion and RNA decay\n for decay gene 1SD outside mean")+ scale_color_viridis()

apaanddecay1sd

ggsave(apaanddecay1sd, file="../output/plots/ApaRationVDecy1SD.png")

```
```{r}
summary(lm(data=ApaAndDecaySt, APAVal~MeanDecay))
```



Try all of the value: 


```{r}
ApaAndDecayStall= ApaBothFracStand_geneMean_spread %>% inner_join(decay_byGene, by="GeneName")


ApaAndDecayStall$density <- get_density(ApaAndDecayStall$APAVal, ApaAndDecayStall$MeanDecay, n = 100)

apaanddecayallplot=ggplot(ApaAndDecayStall, aes(x=MeanDecay, y=APAVal)) + geom_point(aes(color=density)) + geom_smooth(method="lm") +labs(x="relative Decay", y="Nuclear/(Total+Nuclear)", title="Relationship between Nuclear proportion and RNA decay")+ scale_color_viridis()

apaanddecayallplot

ggsave(apaanddecayallplot, file="../output/plots/ApaRationVDecyAllGenes.png")
```

```{r}
summary(lm(data=ApaAndDecayStall, APAVal~MeanDecay))
```

##Make the same plot with transcription on the X (4su/4su+RNA)

4su data
```{r}
FourSU=read.table(file = "../data/RNAdecay/tr_decay_table_norm.txt", header=T, stringsAsFactors = F) %>%  dplyr::select(gene_id,contains("4su_30"))


FourSU_geneNames=FourSU %>% inner_join(geneNames, by="gene_id") %>% dplyr::select(GeneName, contains("4su_30"))

FourgeneNames_long=melt(FourSU_geneNames,id.vars = "GeneName", value.name = "FourSU", variable.name = "FourSU_ind") %>% separate(FourSU_ind, into=c("type","time", "1400", "MAf", "Individual"), sep="_") %>% dplyr::select(GeneName, Individual, FourSU) 

FourSU_geneMean=FourgeneNames_long %>% group_by(GeneName) %>%summarise(Mean_4su=mean(FourSU))



```


Gene expression data:

```{r}
RNA=read.table(file = "../data/RNAdecay/tr_decay_table_norm.txt", header=T, stringsAsFactors = F) %>%  dplyr::select(gene_id,contains("RNAseq_14000"))


RNA_geneNames=RNA %>% inner_join(geneNames, by="gene_id") %>% dplyr::select(GeneName, contains("RNA"))

RNAgeneNames_long=melt(RNA_geneNames,id.vars = "GeneName", value.name = "RNA", variable.name = "RNA_ind") %>%   separate(RNA_ind, into=c("type", "1400", "MAf", "Individual"), sep="_") %>% dplyr::select(GeneName, Individual, RNA) 

RNA_geneMean=RNAgeneNames_long %>% group_by(GeneName) %>%summarise(Mean_RNA=mean(RNA))


```

Join these and make the transcription phenotype

```{r}
Transcription=FourSU_geneMean %>% inner_join(RNA_geneMean, by="GeneName") %>% mutate(Transcription=Mean_4su/(Mean_4su + Mean_RNA)) %>% dplyr::select(GeneName, Transcription)
```


Join with APA:  

```{r}
APAandTranscrption= Transcription %>% inner_join(ApaBothFracStand_geneMean_spread, by="GeneName")
APAandTranscrption$density <- get_density(APAandTranscrption$APAVal, APAandTranscrption$Transcription, n = 100)

apaand4uplot=ggplot(APAandTranscrption, aes(x=Transcription, y=APAVal))+ geom_point(aes(color=density)) + geom_smooth(method = "lm") + labs(x="4su/4su+RNA", y="Nuclear/Nuclear+Total", title="Relationship between APA fraction and transcription") + scale_color_viridis()


summary(lm(data=APAandTranscrption, APAVal~Transcription))

```


```{r}

apaand4uplot
ggsave(apaand4uplot, file="../output/plots/NuclearTotalRatiov4suRNARatio.png")
```

